diff --git a/solenv/gbuild/CppunitTest.mk b/solenv/gbuild/CppunitTest.mk
index 45af5c9b25a7..f7ff4dc01431 100644
--- a/solenv/gbuild/CppunitTest.mk
+++ b/solenv/gbuild/CppunitTest.mk
@@ -76,7 +76,7 @@ gb_CppunitTest_RUNTIMEDEPS += \
 
 define gb_CppunitTest__make_args
 $(HEADLESS) \
-"-env:BRAND_BASE_DIR=$(call gb_Helper_make_url,$(INSTROOT))" \
+"-env:BRAND_BASE_DIR=$(call gb_Helper_make_url,/usr/lib/libreoffice)" \
 "-env:BRAND_SHARE_SUBDIR=$(LIBO_SHARE_FOLDER)" \
 "-env:BRAND_SHARE_RESOURCE_SUBDIR=$(LIBO_SHARE_RESOURCE_FOLDER)" \
 "-env:UserInstallation=$(call gb_Helper_make_url,$(call gb_CppunitTest_get_target,$*).user)" \
@@ -89,9 +89,9 @@ $(if $(URE),\
 	"-env:UNO_SERVICES=$(foreach item,$(UNO_SERVICES),$(call gb_Helper_make_url,$(item)))") \
 	$(if $(strip $(JAVA_URE)),\
 		-env:URE_MORE_JAVA_TYPES=$(call gb_Helper_make_url,$(call gb_Jar_get_target,unoil))) \
-	-env:URE_INTERNAL_LIB_DIR=$(call gb_Helper_make_url,$(INSTROOT)/$(LIBO_URE_LIB_FOLDER)) \
-	-env:LO_LIB_DIR=$(call gb_Helper_make_url,$(INSTROOT)/$(LIBO_LIB_FOLDER)) \
-	-env:LO_JAVA_DIR=$(call gb_Helper_make_url,$(INSTROOT)/$(LIBO_SHARE_JAVA_FOLDER)) \
+	-env:URE_INTERNAL_LIB_DIR=$(call gb_Helper_make_url,/usr/lib/libreoffice/$(LIBO_URE_LIB_FOLDER)) \
+	-env:LO_LIB_DIR=$(call gb_Helper_make_url,/usr/lib/libreoffice/$(LIBO_LIB_FOLDER)) \
+	-env:LO_JAVA_DIR=$(call gb_Helper_make_url,/usr/lib/libreoffice/$(LIBO_SHARE_JAVA_FOLDER)) \
 	--protector $(call gb_Library_get_target,unoexceptionprotector) unoexceptionprotector \
 	--protector $(call gb_Library_get_target,unobootstrapprotector) unobootstrapprotector \
  ) \
@@ -343,7 +343,7 @@ endef
 # Use instdir configuration
 define gb_CppunitTest_use_instdir_configuration
 $(call gb_CppunitTest_get_target,$(1)) : $(call gb_Package_get_target,postprocess_registry)
-$(call gb_CppunitTest__use_configuration,$(1),xcsxcu,$(INSTROOT)/$(LIBO_SHARE_FOLDER)/registry)
+$(call gb_CppunitTest__use_configuration,$(1),xcsxcu,/usr/lib/libreoffice/$(LIBO_SHARE_FOLDER)/registry)
 
 endef
 
