diff --git a/odk/Module_odk.mk b/odk/Module_odk.mk
index 410f8cf8ef1e..835a7c1ad559 100644
--- a/odk/Module_odk.mk
+++ b/odk/Module_odk.mk
@@ -57,13 +57,4 @@ $(eval $(call gb_Module_add_check_targets,odk,\
 	CustomTarget_check \
 ))
 
-ifneq ($(filter $(OS),LINUX MACOSX),)
-$(eval $(call gb_Module_add_subsequentcheck_targets,odk, \
-    CustomTarget_build-examples \
-    $(if $(ENABLE_JAVA),\
-        CustomTarget_build-examples_java \
-    ) \
-))
-endif
-
 # vim: set noet sw=4 ts=4:
