diff --git a/odk/CustomTarget_build-examples.mk b/odk/CustomTarget_build-examples.mk
index 89e427348087..1aa4d766823f 100644
--- a/odk/CustomTarget_build-examples.mk
+++ b/odk/CustomTarget_build-examples.mk
@@ -96,7 +95,7 @@ $(call gb_CustomTarget_get_target,odk/build-examples): \
         && export \
             UserInstallation=$(call gb_Helper_make_url,$(call gb_CustomTarget_get_workdir,odk/build-examples)/user) \
         $(foreach my_dir,$(my_example_dirs), \
-            && (cd $(INSTDIR)/$(SDKDIRNAME)/examples/$(my_dir) \
+            && (cd /usr/lib/libreoffice/sdk/examples/$(my_dir) \
                 && printf 'yes\n' | LC_ALL=C make))) \
             >$(call gb_CustomTarget_get_workdir,odk/build-examples)/log 2>&1 \
         || (RET=$$? \
@@ -104,17 +103,17 @@ $(call gb_CustomTarget_get_target,odk/build-examples): \
             && exit $$RET)
 
 $(call gb_CustomTarget_get_workdir,odk/build-examples)/setsdkenv: \
-        $(SRCDIR)/odk/config/setsdkenv_unix.sh.in \
+        /usr/lib/libreoffice/sdk/setsdkenv_unix.sh.in \
         $(BUILDDIR)/config_$(gb_Side).mk | \
         $(call gb_CustomTarget_get_workdir,odk/build-examples)/.dir
 	$(call gb_Output_announce,$(subst $(WORKDIR)/,,$@),$(true),SED,1)
 	sed -e 's!@OO_SDK_NAME@!sdk!' \
-        -e 's!@OO_SDK_HOME@!$(INSTDIR)/$(SDKDIRNAME)!' \
-        -e 's!@OFFICE_HOME@!$(INSTROOTBASE)!' -e 's!@OO_SDK_MAKE_HOME@!!' \
+        -e 's!@OO_SDK_HOME@!/usr/lib/libreoffice/sdk!' \
+        -e 's!@OFFICE_HOME@!/usr/lib/libreoffice!' -e 's!@OO_SDK_MAKE_HOME@!!' \
         -e 's!@OO_SDK_ZIP_HOME@!!' -e 's!@OO_SDK_CAT_HOME@!!' \
         -e 's!@OO_SDK_SED_HOME@!!' -e 's!@OO_SDK_CPP_HOME@!!' \
         -e 's!@OO_SDK_JAVA_HOME@!$(JAVA_HOME)!' \
-        -e 's!@OO_SDK_OUTPUT_DIR@!$(call gb_CustomTarget_get_workdir,odk/build-examples)/out!' \
+        -e 's!@OO_SDK_OUTPUT_DIR@!$(shell mktemp -q -d)/out!' \
         -e 's!@SDK_AUTO_DEPLOYMENT@!YES!' $< > $@
 
 # vim: set noet sw=4 ts=4:
diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index 4073dcfd9bf5..68c17ce0d1ed 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -339,10 +339,6 @@ define gb_Module_add_subsequentcheck_target
 $(call gb_Module__read_targetfile,$(1),$(2),subsequentcheck target)
 
 $(call gb_Module_get_subsequentcheck_target,$(1)) : $$(gb_Module_CURRENTTARGET)
-$$(gb_Module_CURRENTTARGET) :| \
-	$(call gb_Postprocess_get_target,AllModulesButInstsetNative) \
-    $(call gb_Package_get_target,instsetoo_native_setup) \
-    $(call gb_Package_get_target,instsetoo_native_setup_ure)
 $(call gb_Module_get_clean_target,$(1)) : $$(gb_Module_CURRENTCLEANTARGET)
 
 endef
