diff --git a/smoketest/CppunitTest_smoketest.mk b/smoketest/CppunitTest_smoketest.mk
index f7d64cae562f..9d47776ffb4b 100644
--- a/smoketest/CppunitTest_smoketest.mk
+++ b/smoketest/CppunitTest_smoketest.mk
@@ -34,10 +34,10 @@ userinstallation=$(WORKDIR)/CustomTarget/smoketest
 endif
 
 $(eval $(call gb_CppunitTest_add_arguments,smoketest,\
-	-env:arg-soffice=$(gb_JunitTest_SOFFICEARG) \
+	-env:arg-soffice=$(OOO_TEST_SOFFICE) \
 	-env:arg-user=$(userinstallation) \
 	-env:arg-env=$(gb_Helper_LIBRARY_PATH_VAR)"$$$${$(gb_Helper_LIBRARY_PATH_VAR)+=$$$$$(gb_Helper_LIBRARY_PATH_VAR)}" \
-	-env:arg-testarg.smoketest.doc=$(WORKDIR)/Zip/smoketestdoc.sxw \
+	-env:arg-testarg.smoketest.doc=/usr/share/libreoffice/smoketestdoc.sxw \
 ))
 
 $(call gb_CppunitTest_get_target,smoketest): \
diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index 0d87ccc55379..41c6564d3d7e 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -338,10 +338,6 @@ define gb_Module_add_subsequentcheck_target
 $(call gb_Module__read_targetfile,$(1),$(2),subsequentcheck target)
 
 $(call gb_Module_get_subsequentcheck_target,$(1)) : $$(gb_Module_CURRENTTARGET)
-$$(gb_Module_CURRENTTARGET) :| \
-	$(call gb_Postprocess_get_target,AllModulesButInstsetNative) \
-    $(call gb_Package_get_target,instsetoo_native_setup) \
-    $(call gb_Package_get_target,instsetoo_native_setup_ure)
 $(call gb_Module_get_clean_target,$(1)) : $$(gb_Module_CURRENTCLEANTARGET)
 
 endef
