diff --git a/solenv/gbuild/UITest.mk b/solenv/gbuild/UITest.mk
index 1edb3f769081..62d54917951b 100644
--- a/solenv/gbuild/UITest.mk
+++ b/solenv/gbuild/UITest.mk
@@ -51,15 +54,15 @@ $(call gb_UITest_get_target,%) :| $(gb_UITest_DEPS)
 		$(DEFS) \
 		SAL_LOG_FILE="$(dir $(call gb_UITest_get_target,$*))/soffice.out.log" \
 		TEST_LIB=$(call gb_Library_get_target,test) \
-		URE_BOOTSTRAP=vnd.sun.star.pathname:$(call gb_Helper_get_rcfile,$(INSTROOT)/$(LIBO_ETC_FOLDER)/fundamental) \
-		PYTHONPATH="$(PYPATH)" \
+		URE_BOOTSTRAP=vnd.sun.star.pathname:/usr/lib/libreoffice/program/fundamentalrc \
+		PYTHONPATH="$(filter-out $I/program,$(PYPATH):/usr/lib/libreoffice/program)" \
 		TestUserDir="$(call gb_Helper_make_url,$(dir $(call gb_UITest_get_target,$*)))" \
 		PYTHONDONTWRITEBYTECODE=0 \
 		$(if $(filter-out MACOSX WNT,$(OS_FOR_BUILD)),$(if $(ENABLE_HEADLESS),, \
 			SAL_USE_VCLPLUGIN=svp \
 		)) \
 		$(gb_UITest_COMMAND) \
-		--soffice=path:$(INSTROOT)/$(LIBO_BIN_FOLDER)/soffice \
+		--soffice=$(OOO_TEST_SOFFICE) \
 		--userdir=$(call gb_Helper_make_url,$(dir $(call gb_UITest_get_target,$*))user) \
 		--dir=$(strip $(MODULES)) \
 		$(gb_UITest_GDBTRACE) \
diff --git a/Makefile.in b/Makefile.in
index 037cb5dce95b..f1b7c28e3c59 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -13,7 +13,6 @@ gb_Top_MODULE_CHECK_TARGETS := slowcheck unitcheck subsequentcheck perfcheck uic
 
 MAKECMDGOALS?=all
 build_goal:=$(if $(filter build check,$(MAKECMDGOALS)),all)\
- $(if $(filter build-nocheck uicheck,$(MAKECMDGOALS)),build)\
  $(if $(filter check,$(MAKECMDGOALS)),subsequentcheck $(if $(filter Linux, $(shell uname)), uicheck))\
  $(filter all build-l10n-only build-non-l10n-only debugrun help showmodules translations $(gb_Top_MODULE_CHECK_TARGETS) check packageinfo gbuildtojson,$(MAKECMDGOALS))
 
diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index 4073dcfd9bf5..856f5efee988 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -201,7 +201,7 @@ perfcheck :
 	$(call gb_Output_announce_title,all perftests checked.)
 	$(call gb_Output_announce_bell)
 
-uicheck : build
+uicheck :
 	$(if $(gb_VERBOSE),$(call gb_Output_announce,loaded modules: $(sort $(gb_Module_ALLMODULES)),$(true),UIT,6))
 	$(call gb_Output_announce_title,all uicheck checked.)
 	$(call gb_Output_announce_bell)
@@ -372,11 +372,6 @@ define gb_Module_add_uicheck_target
 $(call gb_Module__read_targetfile,$(1),$(2),uicheck target)
 
 $(call gb_Module_get_uicheck_target,$(1)) : $$(gb_Module_CURRENTTARGET)
-$$(gb_Module_CURRENTTARGET) :| \
-	$(call gb_Postprocess_get_target,AllModulesButInstsetNative) \
-    $(call gb_Package_get_target,instsetoo_native_setup) \
-    $(call gb_Package_get_target,instsetoo_native_setup_ure)
-$(call gb_Module_get_uicheck_target,$(1)) : $$(gb_Module_CURRENTTARGET)
 $(call gb_Module_get_clean_target,$(1)) : $$(gb_Module_CURRENTCLEANTARGET)
 
 endef
@@ -479,7 +474,7 @@ endif
 subsequentcheck : $$(firstword $$(gb_Module_SUBSEQUENTCHECKTARGETSTACK))
 stagingcheck : $$(firstword $$(gb_Module_STAGINGCHECKTARGETSTACK))
 perfcheck : $$(firstword $$(gb_Module_PERFCHECKTARGETSTACK))
-uicheck : build $$(firstword $$(gb_Module_UICHECKTARGETSTACK))
+uicheck : $$(firstword $$(gb_Module_UICHECKTARGETSTACK))
 clean : $$(firstword $$(gb_Module_CLEANTARGETSTACK))
 
 ifneq ($$(words $$(gb_Module_TARGETSTACK)),1)
