OOo hangs if you print to non-default printer

Index: oo_cvs_641/vcl/source/gdi/print.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/print.cxx,v
retrieving revision 1.24.4.4
retrieving revision 1.24.4.6
diff -u -3 -p -r1.24.4.4 -r1.24.4.6
--- oo_cvs_641/vcl/source/gdi/print.cxx	19 Jun 2002 11:32:01 -0000	1.24.4.4
+++ oo_cvs_641/vcl/source/gdi/print.cxx	15 Apr 2003 14:07:46 -0000	1.24.4.6
@@ -532,7 +532,7 @@ void Printer::ImplInit( SalPrinterQueueI
 		if ( (pJobSetup->maPrinterName != pInfo->maPrinterName) ||
 			 (pJobSetup->maDriver != pInfo->maDriver) )
 		{
-			rtl_freeMemory( pJobSetup->mpDriverData );
+			delete[] pJobSetup->mpDriverData;
 			pJobSetup->mpDriverData = NULL;
 			pJobSetup->mnDriverDataLen = 0;
 		}
