--- oo_1.0.1_src/config_office/configure.in	2002-06-08 02:49:26.000000000 +0200
+++ oo_1.0.1_src/config_office/configure.in_new	2002-09-11 00:56:18.000000000 +0200
@@ -241,18 +241,16 @@
 if test "$_os" != "WINNT" && test -z "$with_gcc_home"; then
    AC_PROG_CC
 fi
-if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o \( \( "$_os" = "IRIX" -o "$_os" = "IRIX64" \) -a "$CC" = "gcc" \) ; then
+if test "$_os" = "Linux" -o "$_os" = "FreeBSD" -o "$_os" = "NetBSD" -o \( \( "$_os" = "IRIX" -o "$_os" = "IRIX64" \) -a "$GCC" = "gcc" \) ; then
    dnl ******************************************
    dnl Testing for GNU compiler and version...
    dnl ******************************************
    dnl Check whether the gnu gcc compiler is used.
    if test -z "$with_gcc_home"; then
-      if test "$CC" = "gcc"; then
-         AC_PATH_PROG(GCC, gcc)
-         COMPATH=`echo $GCC | $SED -n "s/\/gcc//p"`
-         _gcc_include_start=/usr/lib/gcc-lib
-         _gxx_include_start=/usr/include
-      fi
+      AC_PROG_CC
+      COMPATH=`echo $CC | $SED -n "s,/[^/]\+$,,p"`
+      _gcc_include_start=/usr/lib/gcc-lib
+      _gxx_include_start=/usr/include
    else
       dnl Compiler has been installed to its own directory and it's home is arg with_gcc_home
       _gcc_path="$with_gcc_home/bin/gcc"
@@ -268,7 +266,7 @@
 
 dnl parametrise the various gcc pathnames so we can reuse for gcc2.95 and gcc3
 if test -n "$enable_gcc3"; then
-   _gccincname1="g++-v3"
+   _gccincname1="c++/3.2"
 else
    _gccincname1="g++-3"
 fi
@@ -343,7 +341,7 @@
                               AC_MSG_WARN([found multiple g++-2 directories, please rename so you have one distinct]) >> warn
                            fi
                         else
-                           _gxx_include_path3=`find $_gxx_include_start -name "g++" -print 2> /dev/null`
+			   _gxx_include_path3=`find $_gxx_include_start -name "3.2" -print 2> /dev/null`
                            if test "$_gxx_include_path3"; then
                               _multiple=`echo $_gxx_include_path3 | $AWK '{ if ($2) print "true"; else print "false" }'`
                               if test "$_multiple" = "false"; then
@@ -821,6 +819,12 @@
    fi
    if test "$_os" = "Linux" -a "$_machine_type" = "ppc"; then
       _jdk_middle=`echo $_jdk | $AWK -F. '{ if ($2 != 2) print "false"; else print "true" }'`
+      _jdk_middle3=`echo $_jdk | $AWK -F. '{ if ($2 != 3) print "false"; else print "true" }'`
+
+      if test "$_jdk_middle" = "false" -a "$_jdk_middle3" = "true"; then
+         _jdk_minor="true"
+      fi
+
    fi
    if test "$_os" = "Linux" -a "$_machine_type" = "armv4l"; then
       _jdk_middle=`echo $_jdk | $AWK -F. '{ if ($2 != 2) print "false"; else print "true" }'`
@@ -975,10 +979,10 @@
 dnl ***************************************
 dnl testing for X libraries and includes...
 dnl ***************************************
-if test "$_os" = "Darwin" -a "x$x_includes" = "xno_x_includes"; then
-   dnl Mac OS X using Aqua graphics. Don't check for X11.
-   :
-elif test "$_os" != "WINNT" ; then
+dnl if test "$_os" = "Darwin" -a "x$x_includes" = "xno_x_includes"; then
+dnl    dnl Mac OS X using Aqua graphics. Don't check for X11.
+dnl    :
+dnl elif test "$_os" != "WINNT" ; then
    AC_PATH_X
    AC_PATH_XTRA
 
@@ -998,10 +1002,10 @@
    CFLAGS=$X_CFLAGS
    LDFLAGS="$X_LDFLAGS $X_LIBS"
    AC_CHECK_LIB(X11, XOpenDisplay, x_libs="-lX11 $X_EXTRA_LIBS", [AC_MSG_ERROR([X Development libraries not found])])
-else
-   x_includes="no_x_includes"
-   x_libraries="no_x_libraries"
-fi
+dnl else
+dnl    x_includes="no_x_includes"
+dnl    x_libraries="no_x_libraries"
+dnl fi
 dnl ***************************************
 dnl testing libc version for Linux...
 dnl ***************************************
@@ -1197,12 +1201,14 @@
 echo "\$MINGWIN32="$WITH_MINGWIN
 echo "\$with_use_shell="$with_use_shell
 echo "\$USE_GCC3="$USE_GCC3
+echo "\$CC="$CC
+echo "\$CXX="$CXX
 echo
 fi
 
 # Executing the set_soenv script to setup the environment variables.
 if test -z "$enable_check_only"; then
-   './set_soenv' $COMPATH $_gcc_include_path $_gxx_include_path $JAVA_HOME $TCSH $PERL $x_libraries $x_includes $_LOCAL_SOLENV $_LOCAL_SOLVER $upd $CYGWIN $STLPORT4 $enable_xprint $with_lang $with_asm_home $with_unzip_home $WITH_MINGWIN $with_use_shell $USE_GCC3
+   './set_soenv' $COMPATH $_gcc_include_path $_gxx_include_path $JAVA_HOME $TCSH $PERL $x_libraries $x_includes $_LOCAL_SOLENV $_LOCAL_SOLVER $upd $CYGWIN $STLPORT4 $enable_xprint $with_lang $with_asm_home $with_unzip_home $WITH_MINGWIN $with_use_shell $USE_GCC3 "$CC" "$CXX"
 else
    echo
    echo Test Complete
