Index: cvs1/product/inc/scp/udk_basefiles.scp
===================================================================
RCS file: /cvs/oo/udk/product/inc/scp/udk_basefiles.scp,v
retrieving revision 1.9.2.1.2.1
diff -u -3 -p -r1.9.2.1.2.1 udk_basefiles.scp
--- cvs1/product/inc/scp/udk_basefiles.scp	2002/06/07 18:15:12	1.9.2.1.2.1
+++ cvs1/product/inc/scp/udk_basefiles.scp	2002/07/31 14:55:02
@@ -1,50 +1,5 @@
 // all file objects that are relevant for the udk base
 
-// Temporary solution/hack: at the moment libstdc++.so and libgcc_s.so are needed for unxlngi4 environment
-// (setup is linked against it).
-#if ( defined( _gcc3 ) && defined( C300 ) )
-
-File gid_File_Lib_gcc
-    BIN_FILE_BODY;
-	Name		= STRING(libgcc_s.so.1);
-    Dir			= GCFG_BINARY_DIR;
-    Styles		= (PACKED, SETUPZIP);
-End
-
-#if ( CCNUMVER >= 000300010000 )
-
-File gid_File_Lib_Stdc
-    BIN_FILE_BODY;
-    Name               = STRING(libstdc++.so.4.0.0);
-    Dir                        = GCFG_BINARY_DIR;
-    Styles             = (PACKED, SETUPZIP);
-End
-
-Shortcut gid_Shortcut_Lib_Stdc
-    FileID      = gid_File_Lib_Stdc;
-    Dir         = GCFG_BINARY_DIR;
-    Name        = STRING(libstdc++.so.4);
-    Styles      = (NETWORK,RELATIVE);
-End
-
-#else
-
-File gid_File_Lib_Stdc
-    BIN_FILE_BODY;
-    Name		= STRING(libstdc++.so.3.0.1);
-    Dir			= GCFG_BINARY_DIR;
-    Styles		= (PACKED, SETUPZIP);
-End
-
-Shortcut gid_Shortcut_Lib_Stdc
-    FileID      = gid_File_Lib_Stdc;
-    Dir         = GCFG_BINARY_DIR;
-    Name        = STRING(libstdc++.so.3);
-    Styles      = (NETWORK,RELATIVE);
-End
-#endif
-#endif
-
 File gid_File_Lib_Sal
     BIN_FILE_BODY;
     #ifdef UNX
Index: cvs1/product/util/makefile.mk
===================================================================
RCS file: /cvs/oo/udk/product/util/makefile.mk,v
retrieving revision 1.64.2.1
diff -u -3 -p -r1.64.2.1 makefile.mk
--- cvs1/product/util/makefile.mk	2002/06/11 15:09:39	1.64.2.1
+++ cvs1/product/util/makefile.mk	2002/07/31 14:55:02
@@ -109,23 +109,18 @@ BINDINGDLL=sunpro5_uno
 DESTDIRBIN=$(DESTDIR)$/linux$/bin
 DESTDIRLIB=$(DESTDIR)$/linux$/lib
 DESTDIRDLL=$(DESTDIRLIB)
-STLPORTLIBNAME=libstlport_gcc.so
 BINDINGDLL=$(COMNAME)_uno
 
 .IF "$(COMID)"=="gcc3"
-.IF "$(CCNUMVER)">="000300010000"
-ADDITIONAL_DLLS= \
-       $(DESTDIRDLL)$/libgcc_s.so.1            \
-       $(DESTDIRDLL)$/libgcc_s.so              \
-       $(DESTDIRDLL)$/libstdc++.so.4           \
-       $(DESTDIRDLL)$/libstdc++.so.4.0.0
+.IF "$(CCNUMVER)">="000300020000"
+       STLPORTLIBNAME=libstlport_gcc_3.2.so
+.ELIF "$(CCNUMVER)">="000300010000"
+       STLPORTLIBNAME=libstlport_gcc_3.1.so
 .ELSE
-ADDITIONAL_DLLS= \
-	$(DESTDIRDLL)$/libgcc_s.so.1 		\
-	$(DESTDIRDLL)$/libgcc_s.so          \
-	$(DESTDIRDLL)$/libstdc++.so.3 		\
-	$(DESTDIRDLL)$/libstdc++.so.3.0.1
+       STLPORTLIBNAME=libstlport_gcc_3.0.so
 .ENDIF
+.ELSE
+	STLPORTLIBNAME=libstlport_gcc.so
 .ENDIF
 
 .ELIF "$(OS)"=="FREEBSD"
