--- oo_1.0_src/solenv/inc/unxlngi4.mk~	Tue Jun 25 22:46:38 2002
+++ oo_1.0_src/solenv/inc/unxlngi4.mk	Wed Jun 26 00:36:54 2002
@@ -195,8 +195,8 @@
 STDSHLGUIST=-lX11 -lXext -ldl -lm
 STDSHLCUIST=-ldl -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc3.2 -lstdc++
+LIBSTLPORTST=$(LIBSTLPORT)
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
--- oo_1.0_src/solenv/inc/unxlngppc4.mk~	Tue Jun 25 22:46:38 2002
+++ oo_1.0_src/solenv/inc/unxlngppc4.mk	Wed Jun 26 00:38:46 2002
@@ -198,8 +198,8 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc3.2 -lstdc++
+LIBSTLPORTST=$(LIBSTLPORT)
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
