--- oo_1.0_src/solenv/inc/unxlngi3.mk~	Wed Oct 24 19:21:47 2001
+++ oo_1.0_src/solenv/inc/unxlngi3.mk	Wed Jun 26 00:36:38 2002
@@ -185,8 +185,8 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_3.0
+LIBSTLPORTST=$(STATIC) -lstlport_gcc_3.0 $(DYNAMIC)
 
 
 # name of library manager
--- oo_1.0_src/solenv/inc/unxlngi4.mk~	Tue Jun 25 22:46:38 2002
+++ oo_1.0_src/solenv/inc/unxlngi4.mk	Wed Jun 26 00:36:54 2002
@@ -195,8 +195,8 @@
 STDSHLGUIST=-lX11 -lXext -ldl -lm
 STDSHLCUIST=-ldl -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_3.0 -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport_gcc_3.0 $(DYNAMIC)
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
--- oo_1.0_src/solenv/inc/unxlngppc4.mk~	Tue Jun 25 22:46:38 2002
+++ oo_1.0_src/solenv/inc/unxlngppc4.mk	Wed Jun 26 00:38:46 2002
@@ -198,8 +198,8 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc -lstdc++
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_3.0 -lstdc++
+LIBSTLPORTST=$(STATIC) -lstlport_gcc_3.0 $(DYNAMIC)
 
 #FILLUPARC=$(STATIC) -lsupc++ $(DYNAMIC)
 
--- oo_1.0_src/solenv/inc/unxlngppc.mk~	Thu Feb 28 21:39:42 2002
+++ oo_1.0_src/solenv/inc/unxlngppc.mk	Wed Jun 26 00:38:32 2002
@@ -127,8 +127,8 @@
 STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm
 STDSHLCUIMT=-ldl -lpthread -lm
 
-LIBSTLPORT=$(DYNAMIC) -lstlport_gcc
-LIBSTLPORTST=$(STATIC) -lstlport_gcc $(DYNAMIC)
+LIBSTLPORT=$(DYNAMIC) -lstlport_gcc_3.0
+LIBSTLPORTST=$(STATIC) -lstlport_gcc_3.0 $(DYNAMIC)
 
 # name of library manager
 LIBMGR=ar 
--- oo_1.0_src/solenv/inc/unxlngr.mk~	Wed Oct 24 19:21:47 2001
+++ oo_1.0_src/solenv/inc/unxlngr.mk	Wed Jun 26 00:43:44 2002
@@ -123,13 +123,13 @@
 STDSLOCUI=
 
 # libraries for linking applications
-STDLIBCUIST=-ldl -lm -lstlport_gcc
-STDLIBGUIMT=-ldl -lpthread -lm -lstlport_gcc
-STDLIBCUIMT=-ldl -lpthread -lm -lstlport_gcc
+STDLIBCUIST=-ldl -lm -lstlport_gcc_3.0
+STDLIBGUIMT=-ldl -lpthread -lm -lstlport_gcc_3.0
+STDLIBCUIMT=-ldl -lpthread -lm -lstlport_gcc_3.0
 STDLIBGUIST=-lX11 -ldl -lm
 # libraries for linking shared libraries
-STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm -lstlport_gcc
-STDSHLCUIMT=-ldl -lpthread -lm -lstlport_gcc
+STDSHLGUIMT=-lX11 -lXext -ldl -lpthread -lm -lstlport_gcc_3.0
+STDSHLCUIMT=-ldl -lpthread -lm -lstlport_gcc_3.0
 
 # name of library manager
 LIBMGR=ar
