--- oo_1.0_src/psprint/source/fontmanager/fontcache.cxx.fontcache-weight	2002-03-06 12:09:10.000000000 +0100
+++ oo_1.0_src/psprint/source/fontmanager/fontcache.cxx	2002-05-06 10:28:24.000000000 +0200
@@ -466,6 +466,7 @@ void FontCache::copyPrintFont( const Pri
     pTo->m_aAliases			= pFrom->m_aAliases;
     pTo->m_nPSName			= pFrom->m_nPSName;
     pTo->m_eItalic			= pFrom->m_eItalic;
+    pTo->m_eWeight			= pFrom->m_eWeight;
     pTo->m_eWidth			= pFrom->m_eWidth;
     pTo->m_ePitch			= pFrom->m_ePitch;
     pTo->m_aEncoding		= pFrom->m_aEncoding;
