Index: oo_cvs/sysui/oounix/office/scripts/soffice.sh
===================================================================
RCS file: /cvs/oo/gsl/sysui/oounix/office/scripts/Attic/soffice.sh,v
retrieving revision 1.1.2.1
diff -u -3 -p -r1.1.2.1 soffice.sh
--- oo_cvs/sysui/oounix/office/scripts/soffice.sh	2002/04/01 19:56:18	1.1.2.1
+++ oo_cvs/sysui/oounix/office/scripts/soffice.sh	2002/07/04 14:47:29
@@ -255,6 +255,10 @@ if [ "X${plugin_mode}" = "Xtrue" ]; then
 	export SAL_IGNOREXERRORS
 	exec "$sd_prog/$sd_binary" -plugin "$1" "$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9"
 else
-	exec "$sd_prog/$sd_binary" "$1" "$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9"
+        if [ "X${SAL_DEBUG}" = "X" ]; then
+            exec "$sd_prog/$sd_binary" "$1" "$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9"
+        else
+            ${SAL_DEBUG} "$sd_prog/$sd_binary"
+        fi
 fi
 
