--- oo_1.0.2_src/sysui/oounix/office/scripts/soffice.sh.orig	2003-03-27 11:18:14.000000000 +0100
+++ oo_1.0.2_src/sysui/oounix/office/scripts/soffice.sh	2003-03-27 11:29:57.000000000 +0100
@@ -208,7 +208,7 @@
 # parse command line arguments
 #
 
-plugin_mode=false
+plugin_flag=
 for DUMMY in ${1+"$@"}
 do
 	case $1 in
@@ -223,7 +223,9 @@
 		;;
 	?plugin)
 
-		plugin_mode=true
+		plugin_flag=-plugin
+		SAL_IGNOREXERRORS=true
+		export SAL_IGNOREXERRORS
 		shift
 		;;	
 	*)
@@ -255,14 +257,12 @@
 PATH="$sd_prog":$PATH
 export PATH
 
-# execute soffice binary
-# there is a bug in solaris sh that prevents the use of "$@"
-# because it cannot cope with spaces in directory names
-if [ "X${plugin_mode}" = "Xtrue" ]; then
-	SAL_IGNOREXERRORS=true
-	export SAL_IGNOREXERRORS
-	exec "$sd_prog/$sd_binary" -plugin "$@"
+if [ "X${OOO_DEBUG}" = "X" ]; then
+  exec_cmd=exec
 else
-	exec "$sd_prog/$sd_binary" "$@"
+  exec_cmd="$OOO_DEBUG"
 fi
 
+# execute soffice binary
+$exec_cmd "$sd_prog/$sd_binary" $plugin_flag "$@"
+
