Index: oo_cvs/desktop/scripts/soffice.sh
===================================================================
RCS file: /cvs/framework/desktop/scripts/soffice.sh,v
retrieving revision 1.1.4.2
diff -u -3 -p -r1.1.4.2 soffice.sh
--- oo_cvs/desktop/scripts/soffice.sh	20 Feb 2003 22:05:22 -0000	1.1.4.2
+++ oo_cvs/desktop/scripts/soffice.sh	22 Apr 2003 14:50:22 -0000
@@ -180,7 +180,7 @@ err () {
 # parse command line arguments
 #
 
-plugin_mode=false
+plugin_flag=
 for DUMMY in ${1+"$@"}
 do
 	case $1 in
@@ -195,7 +195,9 @@ do
 		;;
 	?plugin)
 
-		plugin_mode=true
+		plugin_flag=-plugin
+		SAL_IGNOREXERRORS=true
+		export SAL_IGNOREXERRORS
 		shift
 		;;	
 	*)
@@ -227,12 +229,12 @@ fi
 PATH="$sd_prog":$PATH
 export PATH
 
-# execute soffice binary
-if [ "X${plugin_mode}" = "Xtrue" ]; then
-	SAL_IGNOREXERRORS=true
-	export SAL_IGNOREXERRORS
-	exec "$sd_prog/$sd_binary" -plugin "$@"
+if [ "X${OOO_DEBUG}" = "X" ]; then
+	exec_cmd=exec
 else
-	exec "$sd_prog/$sd_binary" "$@"
+	exec_cmd="$OOO_DEBUG"
 fi
+
+# execute soffice binary
+$exec_cmd "$sd_prog/$sd_binary" $plugin_flag "$@"
 
