Index: oo_1_0_1/solenv/inc/unxlngi4.mk
===================================================================
RCS file: /cvs/oo/tools/solenv/inc/unxlngi4.mk,v
retrieving revision 1.13
diff -u -3 -p -r1.13 unxlngi4.mk
--- oo_1_0_1/solenv/inc/unxlngi4.mk	2002/03/04 15:53:59	1.13
+++ oo_1_0_1/solenv/inc/unxlngi4.mk	2002/10/04 08:58:04
@@ -103,7 +103,7 @@ CFLAGSEXCEPTIONS=-fexceptions -fno-enfor
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # -fpermissive should be removed as soon as possible
-CFLAGSCXX= -pipe -mcpu=pentiumpro -fno-for-scope -fpermissive -fno-rtti 
+CFLAGSCXX= -pipe -mcpu=pentiumpro -fno-for-scope -fpermissive
 
 # HACK: enable Hamburg developers to build on glibc-2.2 machines but compile vs. glibc-2.1 headers
 .IF "$(BUILD_SOSL)"==""
Index: oo_1_0_1/solenv/inc/unxlngppc4.mk
===================================================================
RCS file: /cvs/oo/tools/solenv/inc/unxlngppc4.mk,v
retrieving revision 1.1.2.3
diff -u -3 -p -r1.1.2.3 unxlngppc4.mk
--- oo_1_0_1/solenv/inc/unxlngppc4.mk	2002/06/07 14:11:58	1.1.2.3
+++ oo_1_0_1/solenv/inc/unxlngppc4.mk	2002/10/04 08:58:04
@@ -107,7 +107,7 @@ CFLAGSEXCEPTIONS=-fexceptions -fno-enfor
 CFLAGS_NO_EXCEPTIONS=-fno-exceptions
 
 # -fpermissive should be removed as soon as possible
-CFLAGSCXX= -fsigned-char -pipe -fno-for-scope -fpermissive -fno-rtti 
+CFLAGSCXX= -fsigned-char -pipe -fno-for-scope -fpermissive
 
 
 # Compiler flags for compiling static object in single threaded environment with graphical user interface
