--- oo_1.0.1_src/offmgr/source/offapp/dialog/optgdlg.src.old	2002-10-14 13:56:06.000000000 +0200
+++ oo_1.0.1_src/offmgr/source/offapp/dialog/optgdlg.src	2002-10-14 13:59:05.000000000 +0200
@@ -539,154 +539,154 @@
 		{
 			< "Standard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ portuguese ] =
 		{
 			< "Padrão" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ english_us ] =
 		{
 			< "Standard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ portuguese_brazilian ] =
 		{
 			< "Star Division" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ swedish ] =
 		{
 			< "Standard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ danish ] =
 		{
 			< "Standard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ italian ] =
 		{
 			< "Standard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ spanish ] =
 		{
 			< "Predeterminado" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ french ] =
 		{
 			< "Par défaut" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		stringlist [ dutch ] =
 		{
 			< "Standaard" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ chinese_simplified ] =
 		{
 			< "æ ‡å‡†" ; > ;
 			< "Macintosh" ; > ;
-			< "X Windows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ russian ] =
 		{
 			< "Ñòàíäàðò" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ polish ] =
 		{
 			< "Domyœlnie" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ japanese ] =
 		{
 			< "æ¨™æº–" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ chinese_traditional ] =
 		{
 			< "æ¨™æº–" ; > ;
 			< "Macintosh" ; > ;
-			< "X Windows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ arabic ] =
 		{
 			< "ÞíÇÓí" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ greek ] =
 		{
 			< "ÐñïåðéëïãÞ" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 		StringList [ korean ] =
 		{
 			< "í‘œì¤€" ; > ;
 			< "Macintosh" ; > ;
-			< "XWindows" ; > ;
+			< "X Window" ; > ;
 			< "OS/2" ; > ;
 		};
 	StringList [ turkish ] =
 	{
 			< "Standart" ; > ;
 		< "Macintosh" ; > ;
-		< "XWindows" ; > ;
+		< "X Window" ; > ;
 		< "OS/2" ; > ;
 	};
 	StringList [ language_user1 ] =
 	{
 			< "Standard" ; > ;
 		< "Macintosh" ; > ;
-		< "XWindows" ; > ;
+		< "X Window" ; > ;
 		< "OS/2" ; > ;
 	};
 	StringList [ finnish ] =
 	{
 			< "Vakio" ; > ;
 		< "Macintosh" ; > ;
-		< "XWindows" ; > ;
+		< "X Window" ; > ;
 		< "OS/2" ; > ;
 	};
 	StringList [ catalan ] =
 	{
 			< "Estàndard" ; > ;
 		< "Macintosh" ; > ;
-		< "XWindows" ; > ;
+		< "X Window" ; > ;
 		< "OS/2" ; > ;
 	};
 	};
