Index: oo_cvs_641/officecfg/util/makefile.pmk
===================================================================
RCS file: /cvs/util/officecfg/util/makefile.pmk,v
retrieving revision 1.10
diff -u -3 -p -r1.10 makefile.pmk
--- oo_cvs_641/officecfg/util/makefile.pmk	19 Jun 2001 16:55:53 -0000	1.10
+++ oo_cvs_641/officecfg/util/makefile.pmk	30 Jan 2003 15:53:44 -0000
@@ -61,13 +61,13 @@
 #*************************************************************************
 
 $(MISC)$/instance$/%.xml : %.xcd
-	java -Xmx128m -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/jaxp.jar$(PATH_SEPERATOR)$(CLASSDIR)$/cfgimport.jar org.openoffice.configuration.XMLDefaultGenerator $< $(PRJ)$/util $(MISC) $(PRJ)$/data
+	$(JAVA) -Xmx128m -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/jaxp.jar$(PATH_SEPERATOR)$(CLASSDIR)$/cfgimport.jar org.openoffice.configuration.XMLDefaultGenerator $< $(PRJ)$/util $(MISC) $(PRJ)$/data
 
 $(MISC)$/ldap$/%.conf : %.xcd
-	java -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $< $(PRJ)$/util$/ldap-schema.xsl $@
+	$(JAVA) -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $< $(PRJ)$/util$/ldap-schema.xsl $@
 
 $(MISC)$/ldif$/%.ldif : %.xcd
-	java -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar$(PATH_SEPERATOR)$(CLASSDIR)$/cfgimport.jar org.openoffice.configuration.LDIFImport $< $(PRJ)$/util$/ldif.xsl $@ $(PRJ)$/data
+	$(JAVA) -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar$(PATH_SEPERATOR)$(CLASSDIR)$/cfgimport.jar org.openoffice.configuration.LDIFImport $< $(PRJ)$/util$/ldif.xsl $@ $(PRJ)$/data
 
 
 
