Index: oo_cvs_641/instsetoo/util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo/util/makefile.mk,v
retrieving revision 1.8.10.3
diff -u -3 -p -r1.8.10.3 makefile.mk
--- oo_cvs_641/instsetoo/util/makefile.mk	15 May 2002 11:06:44 -0000	1.8.10.3
+++ oo_cvs_641/instsetoo/util/makefile.mk	30 Jan 2003 19:21:50 -0000
@@ -80,9 +80,12 @@ EXTRARMFLAG=-r
 #ALLTAR : pack
 ALLTAR : pack
 
-.IF "$(alllangext)"!=""
+# Create installation sets for languages in instsetext, or in alllangext if insetsetext is not defined
+instsetext*="$(alllangext)"
 
-pack: $(foreach,i,$(alllangext) pack_$i)
+.IF "$(instsetext)"!=""
+
+pack: $(foreach,i,$(instsetext) pack_$i)
 
 pack_%:
 	+-$(RM) $(EXTRARMFLAG) $(OUT)$/$(@:s/pack_//)$/*
@@ -98,9 +101,11 @@ test:
 test_player:
 	+-lzip $(LZIPFLAGS) -l 01 -f player.lst -o -n OpenOfficePlayer 
 
-.ELSE			# "$(alllangext)"!=""
+.ELSE			# "$(instsetext)"!=""
 pack:
-	@+echo cannot pack nothing...
+	@+echo no installation sets selected to be packed.
+	@+echo instsetext: $(instsetext)
+	@+echo alllangext: $(alllangext)
 
-.ENDIF			# "$(alllangext)"!=""
+.ENDIF			# "$(instsetext)"!=""
 
