Index: oo_cvs/instsetoo/util/makefile.mk
===================================================================
RCS file: /cvs/installation/instsetoo/util/makefile.mk,v
retrieving revision 1.11.8.1
diff -u -3 -p -r1.11.8.1 makefile.mk
--- oo_cvs/instsetoo/util/makefile.mk	3 May 2003 22:19:53 -0000	1.11.8.1
+++ oo_cvs/instsetoo/util/makefile.mk	8 May 2003 12:42:02 -0000
@@ -85,9 +85,14 @@ INSTALLDIR=$(OUT)
 
 ALLTAR : pack
 
-.IF "$(alllangext)"!=""
+# Create installation sets for languages in instsetext, or in alllangext if
+# insetsetext is not defined in the format instsetext=<num>,[<num>,]...
+# e.g.: instsetext=01,49,99
+instsetext*=$(alllangext)
+  
+.IF "$(instsetext)"!=""
 
-pack: $(foreach,i,$(alllangext) pack_$i)
+pack: $(foreach,i,$(instsetext) pack_$i)
 
 pack_%:
 	+-$(MKDIRHIER) $(INSTALLDIR)$/$(@:s/pack_//)$/normal
@@ -97,9 +102,11 @@ pack_%:
 test:
 	+-$(LZIP) $(LZIPFLAGS) -l 01 -f openoffice.lst -o -n OfficeOSL 
 
-.ELSE			# "$(alllangext)"!=""
+.ELSE			# "$(instsetext)"!=""
 pack:
-	@+echo cannot pack nothing...
+	@+echo no installation sets selected to be packed.
+	@+echo instsetext: $(instsetext)
+	@+echo alllangext: $(alllangext)
 
-.ENDIF			# "$(alllangext)"!=""
+.ENDIF			# "$(instsetext)"!=""
 
