Index: oo_cvs_641/sal/osl/unx/process.c
===================================================================
RCS file: /cvs/porting/sal/osl/unx/process.c,v
retrieving revision 1.27
diff -u -3 -p -r1.27 process.c
--- oo_cvs_641/sal/osl/unx/process.c	14 Dec 2002 12:57:16 -0000	1.27
+++ oo_cvs_641/sal/osl/unx/process.c	30 Jan 2003 16:14:07 -0000
@@ -466,6 +466,10 @@ oslProcessError SAL_CALL osl_psz_getExec
     char szAbsolutePath[PATH_MAX] = "";
     char szRealPathBuf[PATH_MAX] = "";
 
+    if( pszCmdLine == NULL ) {
+    	return osl_Process_E_None;
+    }
+
     /* if the command line argument #0 starts with a '/', this program has been */
     /* invoked using a full qualified path */
     if( '/' != pszCmdLine[0] )
