Index: vcl/unx/source/app/wmadaptor.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/app/wmadaptor.cxx,v
retrieving revision 1.28.2.3
retrieving revision 1.28.2.4
diff -u -3 -p -r1.28.2.3 -r1.28.2.4
--- oo_1.0.1_src/vcl/unx/source/app/wmadaptor.cxx	18 Jun 2002 11:05:52 -0000	1.28.2.3
+++ oo_1.0.1_src/vcl/unx/source/app/wmadaptor.cxx	30 Jul 2002 07:43:53 -0000	1.28.2.4
@@ -548,7 +548,7 @@ GnomeWMAdaptor::GnomeWMAdaptor( SalDispl
     initAtoms();
 
     // check for GnomeWM
-    if( m_aWMAtoms[ WIN_SUPPORTING_WM_CHECK ] )
+    if( m_aWMAtoms[ WIN_SUPPORTING_WM_CHECK ] && m_aWMAtoms[ WIN_PROTOCOLS ] )
     {
         XLIB_Window			aWMChild	= None;
         if( XGetWindowProperty( m_pDisplay,
