This patch created by Gwenole Beauchesne.  IssueZilla #11005

--- oo_1.0.2_src/unzip/inc/unzip.h.system-zlib	2003-01-27 03:43:24.000000000 -0500
+++ oo_1.0.2_src/unzip/inc/unzip.h	2003-01-27 10:46:21.000000000 -0500
@@ -827,9 +827,9 @@ int    ReadMemoryByte            __((UWO
 
 int    explode                   __((void));                    /* explode.c */
 
-/* [ed] 5/17/02 On OS X, use a different function name to avoid name collisions with
-   indirectly linked system level zip libraries. */
-#ifdef MACOSX
+/* [ed] 5/17/02 On OS X, use a different function name to avoid name
+   collisions with indirectly linked system level zip libraries. */
+#if defined(MACOSX) || defined(LINUX)
 #define inflate OOoinflate
 #endif
 
