--- oo_1.0.1_src/nas/makefile.mk.orig	2002-12-12 17:35:55.000000000 +0100
+++ oo_1.0.1_src/nas/makefile.mk	2002-12-12 17:36:15.000000000 +0100
@@ -99,11 +99,3 @@
 .INCLUDE :	target.mk
 .INCLUDE :	tg_ext.mk
 
-
-TG_DELIVER : $(INPATH)$/misc$/build$/so_predeliver
-		$(DELIVER)
-
-.IF "$(BUILD_SOSL)"!=""
-ALLTAR : TG_DELIVER
-.ENDIF			# "$(BUILD_SOSL)"!=""
-
--- oo_1.0.1_src/solenv/inc/target.mk.orig	2002-06-19 01:59:19.000000000 +0200
+++ oo_1.0.1_src/solenv/inc/target.mk	2002-12-12 17:44:28.000000000 +0100
@@ -543,6 +543,7 @@
 
 .IF "$(JARTARGET)"!=""
 JARCLASSDIRS*=.
+JARTARGETN_DUMMY=JAR_DUMMY
 .IF "$(NEW_JAR_PACK)"!=""
 JARMANIFEST*=META-INF$/MANIFEST.MF
 .ENDIF			# "$(NEW_JAR_PACK)"!=""
@@ -1949,7 +1950,7 @@
 		$(APP1TARGETN)	$(APP2TARGETN)	$(APP3TARGETN)		\
 		$(APP4TARGETN)	$(APP5TARGETN)	$(APP6TARGETN)		\
 		$(APP7TARGETN)	$(APP8TARGETN)	$(APP9TARGETN)		\
-		$(JARTARGETN)	\
+		$(JARTARGETN_DUMMY)	\
 		$(JARTARGETDEPN)	\
 		$(SRCTARGET)	$(SRSTARGET) \
 		$(SRC1TARGET)	$(SRS1TARGET) \
@@ -2075,6 +2076,11 @@
 $(GENJAVAFILES) : $(RDB)
 $(JAVATARGET) : $(GENJAVAFILES)
 .ENDIF			# "$(GENJAVAFILES)"!=""
+.IF "$(JARTARGETN)"!=""
+# This is what I want to do, but JARTARGETN uses .SETDIR so I can't :(
+#$(JARTARGETN) : $(ALL_JAVA_TARGETS)
+JAR_DUMMY .PHONY .SEQUENTIAL : $(ALL_JAVA_TARGETS) $(JARTARGETN)
+.ENDIF
 .ENDIF          # "$(SOLAR_JAVA)"!=""
 
 .INCLUDE : tg_dir.mk
--- oo_1.0.1_src/berkeleydb/makefile.mk.orig	2002-12-12 17:53:29.000000000 +0100
+++ oo_1.0.1_src/berkeleydb/makefile.mk	2002-12-12 17:53:38.000000000 +0100
@@ -118,11 +118,3 @@
 .INCLUDE : set_ext.mk
 .INCLUDE :	target.mk
 .INCLUDE :	tg_ext.mk
-
-TG_DELIVER : $(INPATH)$/misc$/build$/so_predeliver
-		$(DELIVER)
-
-.IF "$(BUILD_SOSL)"!=""
-ALLTAR : TG_DELIVER
-.ENDIF			# "$(BUILD_SOSL)"!=""
-
--- oo_1.0.1_src/zlib/makefile.mk.orig	2002-12-12 17:58:37.000000000 +0100
+++ oo_1.0.1_src/zlib/makefile.mk	2002-12-12 17:56:53.000000000 +0100
@@ -96,8 +97,3 @@
 
 TG_DELIVER : $(INPATH)$/misc$/build$/so_predeliver
 		$(DELIVER)
-
-.IF "$(BUILD_SOSL)"!=""
-ALLTAR : TG_DELIVER
-.ENDIF			# "$(BUILD_SOSL)"!=""
-
Index: oo_cvs_641/sw/source/core/makefile.mk
===================================================================
RCS file: /cvs/sw/sw/source/core/makefile.mk,v
retrieving revision 1.3
diff -u -3 -p -u -r1.3 makefile.mk
--- oo_cvs_641/sw/source/core/makefile.mk	24 Aug 2001 14:27:37 -0000	1.3
+++ oo_cvs_641/sw/source/core/makefile.mk	13 Dec 2002 10:29:41 -0000
@@ -98,7 +98,6 @@ LIBFLAGS=$(LIBFLAGS) /p=256
 # fuer VC++/NT andere Label als die Verzeichnisnamen
 .IF "$(RC_SUBDIRS)" == ""
 SWSUBDIRS= \
-	core_1st \
 	attr \
 	bastyp \
 	crsr \
@@ -167,10 +166,10 @@ SUBLIBS2+= \
 .IF "$(L10N-framework)" == ""
 .IF "$(RC_SUBDIRS)" == ""
 .IF "$(depend)" == ""
-core:						\
-	core_1st				\
-	$(SWSUBDIRS)				\
-	ALLTAR
+#core:						\
+#	core_1st				\
+#	$(SWSUBDIRS)				\
+#	ALLTAR
 .ELSE
 core:
 	@+echo Doing nothing in source\core
@@ -213,107 +212,9 @@ dbutilx="debug=true"
 $(LB)$/%.lib : %
 	@echo @
 
-core_1st .SETDIR=core_1st:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-attr .SETDIR=attr:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-bastyp .SETDIR=bastyp:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-crsr .SETDIR=crsr:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-doc .SETDIR=doc:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-docnode .SETDIR=docnode:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-draw .SETDIR=draw:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-edit .SETDIR=edit:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-except .SETDIR=except:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-fields .SETDIR=fields:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-frmedt .SETDIR=frmedt:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-graphic .SETDIR=graphic:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-layout .SETDIR=layout:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-.IF "$(SOLAR_JAVA)" != ""
-javascrp .SETDIR=javascrp:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-.ENDIF
-
-ole .SETDIR=ole:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-para .SETDIR=para:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-rtlfix .SETDIR=rtlfix:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-sw3io .SETDIR=sw3io:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-swg .SETDIR=swg:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-text .SETDIR=text:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-tox .SETDIR=tox:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-txtnode .SETDIR=txtnode:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-undo .SETDIR=undo:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-unocore .SETDIR=unocore:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-view .SETDIR=view:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
+#SUBDIRS = core_1st $(SWSUBDIRS)
+
+#$(SWSUBDIRS) : core_1st
 
 .ENDIF
 .ELSE
@@ -326,24 +227,3 @@ kill:
 	+-$(RM) $(SLB)$/core1.lib
 	+-$(RM) $(SLB)$/core2.lib
 
-attr  : core_1st
-bastyp  : core_1st
-crsr  : core_1st
-doc  : core_1st
-docnode  : core_1st
-draw  : core_1st
-edit  : core_1st
-fields  : core_1st
-frmedt  : core_1st
-graphic  : core_1st
-layout  : core_1st
-ole  : core_1st
-para  : core_1st
-sw3io  : core_1st
-swg  : core_1st
-text  : core_1st
-tox  : core_1st
-txtnode  : core_1st
-unocore  : core_1st
-undo  : core_1st
-view : core_1st
Index: oo_cvs_641/sw/source/ui/makefile.mk
===================================================================
RCS file: /cvs/sw/sw/source/ui/makefile.mk,v
retrieving revision 1.3
diff -u -3 -p -u -r1.3 makefile.mk
--- oo_cvs_641/sw/source/ui/makefile.mk	24 Aug 2001 14:22:26 -0000	1.3
+++ oo_cvs_641/sw/source/ui/makefile.mk	13 Dec 2002 11:03:17 -0000
@@ -157,8 +157,6 @@ SUBLIBS2= \
 .IF "$(RC_SUBDIRS)" == ""
 .IF "$(depend)" == ""
 ui:						\
-	ui_1st				\
-	$(SWSUBDIRS) \
 	ALLTAR
 .ELSE
 ui:
@@ -204,110 +202,6 @@ dcompact=
 $(LB)$/%.lib : %
 	@echo @
 
-ui_1st .SETDIR=ui_1st:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS)  $(dbutilx) $(CALLMACROS)
-
-#	app $(LB)$/app.lib .SETDIR=app:
-app .SETDIR=app:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-cctrl  .SETDIR=cctrl:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-chrdlg  .SETDIR=chrdlg:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-config  .SETDIR=config:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-dbui  .SETDIR=dbui:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-dialog  .SETDIR=dialog:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-dochdl  .SETDIR=dochdl:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-docvw  .SETDIR=docvw:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-envelp  .SETDIR=envelp:
-		@echo $@
-        @$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-fldui  .SETDIR=fldui:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-fmtui  .SETDIR=fmtui:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-frmdlg  .SETDIR=frmdlg:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-globdoc  .SETDIR=globdoc:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-index  .SETDIR=index:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-lingu  .SETDIR=lingu:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-misc  .SETDIR=misc:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-ribbar  .SETDIR=ribbar:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-shells  .SETDIR=shells:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-table  .SETDIR=table:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-uiview  .SETDIR=uiview:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-utlui  .SETDIR=utlui:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-uno  .SETDIR=uno:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-wizard  .SETDIR=wizard:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-web  .SETDIR=web:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
-
-wrtsh  .SETDIR=wrtsh:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(dcompact) $(CALLMACROS)
 .ENDIF
 .ELSE
 ui_1st:
@@ -319,28 +213,3 @@ kill:
 	@+$(RM) $(SLB)$/ui.lst
 	@+$(RM) $(SLB)$/ui.lib
 
-app      : ui_1st
-cctrl    : ui_1st
-chrdlg   : ui_1st
-config   : ui_1st
-dbui     : ui_1st
-dialog   : ui_1st
-dochdl   : ui_1st
-docvw    : ui_1st
-envelp   : ui_1st
-fldui    : ui_1st
-fmtui    : ui_1st
-frmdlg   : ui_1st
-globdoc  : ui_1st
-index    : ui_1st
-lingu    : ui_1st
-misc     : ui_1st
-ribbar   : ui_1st
-shells   : ui_1st
-table    : ui_1st
-uno	     : ui_1st
-uiview   : ui_1st
-utlui    : ui_1st
-web	     : ui_1st
-wizard	 : ui_1st
-wrtsh    : ui_1st
Index: oo_cvs_641/sw/source/filter/makefile.mk
===================================================================
RCS file: /cvs/sw/sw/source/filter/makefile.mk,v
retrieving revision 1.3
diff -u -3 -p -u -r1.3 makefile.mk
--- oo_cvs_641/sw/source/filter/makefile.mk	24 Aug 2001 14:29:41 -0000	1.3
+++ oo_cvs_641/sw/source/filter/makefile.mk	13 Dec 2002 11:20:04 -0000
@@ -140,8 +140,6 @@ SUBLIBS+= \
 .IF "$(RC_SUBDIRS)" == ""
 .IF "$(depend)" == ""
 filter: 					\
-	filt_1st				\
-	$(SWSUBDIRS)				\
 	ALLTAR
 .ELSE
 filter:
@@ -178,61 +176,6 @@ dbutilx += "debug=true"
 $(SLB)$/%.lib : %
 	@echo @
 
-filt_1st .SETDIR=filt_1st:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-ascii .SETDIR=ascii:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-basflt .SETDIR=basflt:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-debug .SETDIR=debug:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-excel .SETDIR=excel:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-html .SETDIR=html:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-lotus .SETDIR=lotus:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-rtf .SETDIR=rtf:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-sw6 .SETDIR=sw6:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-w4w .SETDIR=w4w:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-writer .SETDIR=writer:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-ww1 .SETDIR=ww1:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-ww8 .SETDIR=ww8:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
-
-xml .SETDIR=xml:
-		@echo $@
-		@$(MAKECMD) -d $(MFLAGS) $(dbutilx) $(CALLMACROS)
 
 .ENDIF
 .ELSE
@@ -244,15 +187,4 @@ filt_1st:
 kill:
 	del $(SLB)$/filter.lib
 
-ascii : filt_1st
-basflt : filt_1st
-excel : filt_1st
-html : filt_1st
-lotus : filt_1st
-rtf : filt_1st
-w4w : filt_1st
-writer : filt_1st
-ww1 : filt_1st
-ww8 : filt_1st
-xml : filt_1st
 
Index: oo_cvs_641/solenv/inc/_tg_def.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/_tg_def.mk,v
retrieving revision 1.13.4.1
diff -u -3 -p -r1.13.4.1 _tg_def.mk
--- oo_cvs_641/solenv/inc/_tg_def.mk	7 Jan 2003 15:21:58 -0000	1.13.4.1
+++ oo_cvs_641/solenv/inc/_tg_def.mk	11 Feb 2003 08:09:11 -0000
@@ -175,10 +175,10 @@ $(DEF1TARGETN) .PHONY : \
 $(DEF1TARGETN): \
 		$(DEF1DEPN) \
 		$(DEF1EXPORTFILE)
-	@echo ------------------------------
-	@echo Making Module-Definitionfile : $@
-	@echo just a dummy for UNIX
-	@echo > $@
+	@+echo ------------------------------
+	@+echo Making Module-Definitionfile : $@
+	@+echo just a dummy for UNIX
+	@+echo > $@
 .ENDIF
 .IF "$(GUI)"=="MAC"
 $(DEF1TARGETN): \
Index: oo_cvs_641/solenv/inc/startup/startup.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/startup/startup.mk,v
retrieving revision 1.2.10.1
diff -u -3 -p -r1.2.10.1 startup.mk
--- oo_cvs_641/solenv/inc/startup/startup.mk	7 Jan 2003 15:22:40 -0000	1.2.10.1
+++ oo_cvs_641/solenv/inc/startup/startup.mk	11 Feb 2003 08:09:11 -0000
@@ -99,7 +99,7 @@ MAXLINELENGTH	= 65530
    RMTARGET      *=  $<
 
 # Default recipe that is used to remove intermediate targets.
-.REMOVE :; +-echo dummy remove #$(RM) $(RMFLAGS) $(RMTARGET)
+.REMOVE :; #+-echo dummy remove #$(RM) $(RMFLAGS) $(RMTARGET)
 #.REMOVE :; echo $(RM) $(RMFLAGS) $(RMTARGET)
 
 dummy_mh:
