This patch was submitted by George Zahopoulos to Issue 6947

Index: oo_cvs_641/solenv/inc/target.mk
===================================================================
RCS file: /cvs/tools/solenv/inc/target.mk,v
retrieving revision 1.91.4.2
diff -u -3 -p -r1.91.4.2 target.mk
--- oo_cvs_641/solenv/inc/target.mk	18 Jun 2002 23:59:19 -0000	1.91.4.2
+++ oo_cvs_641/solenv/inc/target.mk	13 Dec 2002 14:46:20 -0000
@@ -2323,8 +2323,21 @@ $(MISC)$/$(TARGET)_xxl_%.done : %.xxl
 .IF "$(MAKEFILERC)"==""
 .IF "$(CXXFILES)$(CFILES)$(RCFILES)$(SLOFILES)$(OBJFILES)$(DEPOBJFILES)$(PARFILES)" != ""
 .IF "$(DEPFILES)" != ""
+
+.IF "$(YACCTARGET)" != ""
+DEPFILE_YACC=$(YACCFILES:s/.y/.dpcc/)
+DEPFILES_EXPANDED=$(subst,$(MISC),'\n$(MISC)' $(DEPFILES))
+DEPFILE_GEN_CXX=$(shell -echo $(DEPFILES_EXPANDED) | grep $(DEPFILE_YACC))
+GEN_DEPFILE_EXISTS= $(shell -test -f $(DEPFILE_GEN_CXX) && echo true)
+.ENDIF			# "$(YACCTARGET)" != ""
+
+.IF "$(GEN_DEPFILE_EXISTS)" == "true"
 .INCLUDE : $(DEPFILES)
+.ELSE
+.INCLUDE .IGNORE : $(DEPFILES)
+.ENDIF			# "$(GEN_DEPFILE_EXISTS)" == "true"
 .ENDIF			# "$(DEPFILES)" != ""
+
 .INCLUDE : $(MISC)$/$(TARGET).dpc
 .IF "$(GROUP)"=="WRITER"
 .IF "$(debug)"!=""
