Index: oo_cvs/config_office/set_soenv.in
===================================================================
RCS file: /cvs/tools/config_office/set_soenv.in,v
retrieving revision 1.8.2.1
diff -u -3 -p -r1.8.2.1 set_soenv.in
--- oo_cvs/config_office/set_soenv.in	27 Jun 2003 00:46:37 -0000	1.8.2.1
+++ oo_cvs/config_office/set_soenv.in	27 Jun 2003 12:24:46 -0000
@@ -49,8 +49,8 @@ my ( $outfile, $outfile_sh, $outfile_bat
 #-------------------------------------------------
 #
 # Platform independent constant values.
-my @LANGUAGES = ("ARAB", "CAT", "CHINSIM", "CHINTRAD", "CZECH", "DAN", "DTCH", "ENUS", "FREN", "GREEK", "ITAL",
-"JAPN", "KOREAN", "POL", "PORT", "RUSS", "SLOVAK", "SPAN", "SWED", "TURK" ); 
+my @LANGUAGES = ("ARAB", "CAT", "CHINSIM", "CHINTRAD", "CZECH", "DAN", "DTCH", "ENUS", "FINN", "FREN", "GREEK", "ITAL",
+"JAPN", "KOREAN", "POL", "PORT", "PORTBR", "RUSS", "SLOVAK", "SPAN", "SWED", "TURK" ); 
 my @DICTIONARIES = ("DADK", "DEDE", "ENGB", "ENUS", "ITIT", "RURU" );
 my ( $CALL_CDECL, $COMMON_OUTDIR, $NO_SRS_PATCH, $PRODUCT, $PROFULLSWITCH, $BIG_GOODIES, $BMP_WRITES_FLAG, 
      $common_build, $MK_UNROLL, $NO_REC_RES, $PROEXT, $SO3, $SOLAR_JAVA, 
