From faf72664d11deaa570a0f8c7e702ff07bf5914c8 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Sun, 17 Feb 2013 10:49:34 +0000
Subject: Missing include in liborcus when building against Boost 1.53.0

Change-Id: Ie6c944d53d6e2fcdd161f674e32c9fbfd761d844
---
diff --git a/liborcus/UnpackedTarball_orcus.mk b/liborcus/UnpackedTarball_orcus.mk
index ca5232d..de0788a 100644
--- a/liborcus/UnpackedTarball_orcus.mk
+++ b/liborcus/UnpackedTarball_orcus.mk
@@ -18,6 +18,10 @@ orcus_patches :=
 orcus_patches += liborcus_0.1.0-configure.patch
 # disable boost "auto lib" in MSVC build
 orcus_patches += liborcus_0.1.0-boost_disable_auto_lib.patch
+# <https://gitorious.org/orcus/orcus/merge_requests/2#
+# f60d6eecee72349993a392a9a63ddf3383d3b8c8-
+# f60d6eecee72349993a392a9a63ddf3383d3b8c8@2>:
+orcus_patches += liborcus_0.1.0-boost-include.patch.1
 
 $(eval $(call gb_UnpackedTarball_fix_end_of_line,orcus,\
 	vsprojects/liborcus-static-nozip/liborcus-static-nozip.vcproj \
diff --git a/liborcus/liborcus_0.1.0-boost-include.patch.1 b/liborcus/liborcus_0.1.0-boost-include.patch.1
new file mode 100644
index 0000000..07558b0
--- a/dev/null
+++ b/liborcus/liborcus_0.1.0-boost-include.patch.1
@@ -0,0 +1,26 @@
+From f60d6eecee72349993a392a9a63ddf3383d3b8c8 Mon Sep 17 00:00:00 2001
+From: Stephan Bergmann <sbergman@redhat.com>
+Date: Sat, 16 Feb 2013 13:25:44 +0100
+Subject: [PATCH] Missing include <boost/noncopyable.hpp>
+
+(at least when compiling against Boost 1.53.0)
+---
+ src/liborcus/dom_tree.cpp | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/src/liborcus/dom_tree.cpp b/src/liborcus/dom_tree.cpp
+index ecf7beb..14725ad 100644
+--- a/src/liborcus/dom_tree.cpp
++++ b/src/liborcus/dom_tree.cpp
+@@ -34,6 +34,8 @@
+ #include <iostream>
+ #include <sstream>
+ 
++#include <boost/noncopyable.hpp>
++
+ using namespace std;
+ 
+ namespace orcus {
+-- 
+1.8.1.2
+
--
cgit v0.9.0.2-2-gbebe
