diff -urN OOo/idlc/source/preproc/cpp.c OOo-s390/idlc/source/preproc/cpp.c
--- OOo/idlc/source/preproc/cpp.c	2001-03-15 13:48:45.000000000 +0100
+++ OOo-s390/idlc/source/preproc/cpp.c	2002-12-25 11:16:25.000000000 +0100
@@ -170,7 +170,7 @@
 void
     process(Tokenrow * trp)
 {
-    int anymacros = 0;  /* S390: XDBX stürzt ab! */
+    int anymacros = 0;
 
     for (;;)
     {
diff -urN OOo/idlc/source/preproc/include.c OOo-s390/idlc/source/preproc/include.c
--- OOo/idlc/source/preproc/include.c	2001-03-15 13:48:45.000000000 +0100
+++ OOo-s390/idlc/source/preproc/include.c	2002-12-25 11:16:25.000000000 +0100
@@ -63,7 +63,7 @@
 #include <string.h>
 #include <fcntl.h>
 
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #	include <io.h>
 #else
 #	include <unistd.h>
diff -urN OOo/idlc/source/preproc/lex.c OOo-s390/idlc/source/preproc/lex.c
--- OOo/idlc/source/preproc/lex.c	2001-03-15 13:48:45.000000000 +0100
+++ OOo-s390/idlc/source/preproc/lex.c	2002-12-25 11:16:25.000000000 +0100
@@ -61,7 +61,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
@@ -97,11 +97,7 @@
 #define	C_ALPH	2
 #define	C_NUM	3
 #define	C_EOF	4
-#ifdef S390
-#define	C_XX	6
-#else
 #define	C_XX	5
-#endif
 
 enum state
 {
@@ -328,12 +324,8 @@
                     continue;
                 case C_ALPH:
                     for (j = 0; j <= 256; j++)
-#ifdef S390
-						if( isalpha( j ) || (j == '_') )
-#else							
                         if ('a' <= j && j <= 'z' || 'A' <= j && j <= 'Z'
                             || j == '_')
-#endif							
                             bigfsm[j][fp->state] = nstate;
                     continue;
                 case C_NUM:
@@ -411,7 +403,7 @@
             }
     }
     maxp = &trp->bp[trp->max];
-    runelen = 1;  /* S390: XDBX stürzt ab! */
+    runelen = 1;
     for (;;)
     {
 continue2:
@@ -430,7 +422,7 @@
         {
             oldstate = state;
 
-            c = *ip;  /* S390: XDBX stürzt (gelegentlich) ab! */
+            c = *ip;
 						
             if ((state = bigfsm[c][state]) >= 0)
             {
@@ -476,7 +468,7 @@
 
 					if (c == '\n')
 					{
-					    while (s->inp + 1 >= s->inl && fillbuf(s) != EOF); /* S390: XDBX stürzt (manchmal) ab! */
+					    while (s->inp + 1 >= s->inl && fillbuf(s) != EOF);
 					
 						if (s->inp[1] == '\r')
 						{
diff -urN OOo/idlc/source/preproc/tokens.c OOo-s390/idlc/source/preproc/tokens.c
--- OOo/idlc/source/preproc/tokens.c	2001-03-15 13:48:46.000000000 +0100
+++ OOo-s390/idlc/source/preproc/tokens.c	2002-12-25 11:16:25.000000000 +0100
@@ -62,7 +62,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
diff -urN OOo/idlc/source/preproc/unix.c OOo-s390/idlc/source/preproc/unix.c
--- OOo/idlc/source/preproc/unix.c	2001-03-15 13:48:46.000000000 +0100
+++ OOo-s390/idlc/source/preproc/unix.c	2002-12-25 11:16:25.000000000 +0100
@@ -64,7 +64,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <fcntl.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
diff -urN OOo/rsc/source/rscpp/cpp1.c OOo-s390/rsc/source/rscpp/cpp1.c
--- OOo/rsc/source/rscpp/cpp1.c	2001-04-18 12:31:56.000000000 +0200
+++ OOo-s390/rsc/source/rscpp/cpp1.c	2002-12-25 11:16:25.000000000 +0100
@@ -66,13 +66,8 @@
 #include        "cppdef.h"
 #include        "cpp.h"
 
-/*#ifdef S390*/
 FILE *pCppOut = NULL;
 FILE *pCppIn  = NULL;
-/* #else
-FILE *pCppOut = stdout;*/  /* BP */ /* in cpp1.c: file-pointer auf stdout oder file */
-/*FILE *pCppIn  = stdin;
-#endif*/
 
 #ifdef DEBUG
 FILE *pDefOut = stdout;		/* ER  evtl. #define's dump */
diff -urN OOo/rsc/source/rscpp/cppdef.h OOo-s390/rsc/source/rscpp/cppdef.h
--- OOo/rsc/source/rscpp/cppdef.h	2001-06-28 14:18:01.000000000 +0200
+++ OOo-s390/rsc/source/rscpp/cppdef.h	2002-12-25 11:16:25.000000000 +0100
@@ -260,12 +260,8 @@
  * (Or is it the other way around?) -- Warning: BIG_ENDIAN code is untested.
  */
 #ifndef BIG_ENDIAN
-#ifdef S390
-#define BIG_ENDIAN 				TRUE
-#else
 #define BIG_ENDIAN              FALSE
 #endif
-#endif
 
 /*
  * COMMENT_INVISIBLE may be defined to allow "old-style" comment
diff -urN OOo/sal/inc/osl/endian.h OOo-s390/sal/inc/osl/endian.h
--- OOo/sal/inc/osl/endian.h	2001-04-26 17:27:19.000000000 +0200
+++ OOo-s390/sal/inc/osl/endian.h	2002-12-25 11:16:25.000000000 +0100
@@ -158,10 +158,6 @@
 #   endif
 #endif
 
-#ifdef S390
-#	define  _BIG_ENDIAN
-#endif
-
 #ifdef _WIN16
 #	define	_LITTLE_ENDIAN
 #endif
@@ -202,7 +198,7 @@
  */
 #if !defined(_WIN32)  && !defined(_WIN16) && !defined(OS2)   && \
     !defined(LINUX)   && !defined(NETBSD) && !defined(SCO)   && \
-	!defined(AIX)     && !defined(HPUX)   && !defined(S390)  && \
+	!defined(AIX)     && !defined(HPUX)   && \
 	!defined(SOLARIS) && !defined(IRIX)   && !defined(MACOS) && \
 	!defined(MACOSX) && !defined(FREEBSD)
 #	error "Target plattform not specified !"
diff -urN OOo/sal/osl/unx/diagnose.c OOo-s390/sal/osl/unx/diagnose.c
--- OOo/sal/osl/unx/diagnose.c	2001-02-28 10:59:34.000000000 +0100
+++ OOo-s390/sal/osl/unx/diagnose.c	2002-12-25 11:16:25.000000000 +0100
@@ -127,18 +127,6 @@
 	else
 	{
 		/* format message into buffer */
-#ifdef S390		
-		if(pszMessage != 0) 
-		{
-			s390_printf( "Assertion Failed: %s: File %s, Line %d: %s\n",
-						 lpszAppName, pszFileName, nLine, pszMessage);
-		}
-		else
-		{
-			s390_printf( "Assertion Failed: %s: File %s, Line %d:\n",
-						 lpszAppName, pszFileName, nLine);
-		}
-#else	
 		if(pszMessage != 0) 
 		{
 			fprintf(stderr, "Assertion Failed: %s: File %s, Line %lu: %s",
@@ -149,7 +137,6 @@
 			fprintf(stderr, "Assertion Failed: %s: File %s, Line %lu ",
 					lpszAppName, pszFileName, nLine);
 		}
-#endif	
 	}
 	
 	return sal_False;   /* no abort */
@@ -157,11 +144,7 @@
 
 sal_Int32 SAL_CALL osl_reportError(sal_uInt32 nType, const sal_Char* pszMessage)
 {
-#ifdef S390
-	s390_printf( pszMessage );
-#else	
 	fputs(pszMessage, stderr);
-#endif	
 	return 0;
 }
 
diff -urN OOo/sal/osl/unx/security.c OOo-s390/sal/osl/unx/security.c
--- OOo/sal/osl/unx/security.c	2001-05-28 09:18:29.000000000 +0200
+++ OOo-s390/sal/osl/unx/security.c	2002-12-25 11:16:25.000000000 +0100
@@ -661,84 +661,6 @@
 		return osl_Security_E_UserUnknown;
 	}
 
-#elif defined S390
-
-	struct passwd 			*pPasswd;
-
-	if ((pszUserName == NULL) || (pszPasswd == NULL) || (pSecurity == NULL))
-		return osl_SEUNKNOWN;
-	
-	OSL_TRACE("osl_loginUser %s (for now without password validation!)\n", pszUserName);
-
-    pPasswd = getpwnam(pszUserName);
-	if (pPasswd)
-	{	
-		int   iResult = 0;
-/* the function __passwd is not correctly implemented now; we have to wait for the next OS/390 release...
-		sal_Char* _pszUserName = strdup(pszUserName);
-		sal_Char* _pszPasswd = strdup(pszPasswd);
-		asc2ebc(_pszUserName);
-		asc2ebc(_pszPasswd);
-
-		iResult = __passwd(_pszUserName, _pszPasswd, NULL);
-		if(iResult == -1)
-		{
-			OSL_TRACE("__passwd result: %d", errno);
-			perror(NULL);
-		}
-
-		free(_pszUserName);
-		free(_pszPasswd);
-*/
-		if (iResult == 0) 
-		{
-			if (strlen(pPasswd->pw_name) + 
-				strlen(pPasswd->pw_dir) +
-				strlen(pPasswd->pw_shell) + 7 <= PASSWD_BUFFER_SIZE)
-			{
-				sal_Char *p;
-				oslSecurityImpl *pSecImpl = malloc(sizeof(oslSecurityImpl));
-
-				pSecImpl->m_pPasswd = *pPasswd;
-
-				p = pSecImpl->m_buffer;
-				pSecImpl->m_pPasswd.pw_name = p;
-				strcpy(p, pPasswd->pw_name);
-
-				p += strlen(pPasswd->pw_name) + 1;
-				pSecImpl->m_pPasswd.pw_dir = p;
-				strcpy(p, pPasswd->pw_dir);
-
-				p += strlen(pPasswd->pw_dir) + 1;
-				pSecImpl->m_pPasswd.pw_shell = p;			
-				strcpy(p, pPasswd->pw_shell);
-
-				pSecImpl->m_isValid = sal_True;
-				*pSecurity = (oslSecurity)pSecImpl;
-
-				OSL_TRACE("Homedirectory of %s: %s\n", pSecImpl->m_pPasswd.pw_name, pSecImpl->m_pPasswd.pw_dir);
-
-				return osl_Security_E_None;
-			}
-			else
-			{
-				*pSecurity = NULL;
-				return osl_SEUNKNOWN;	
-			}
-		}
-		else
-		{
-			*pSecurity = NULL;
-			return osl_Security_E_WrongPassword;		
-		}
-	}
-	else
-	{
-		*pSecurity = NULL;
-		return osl_Security_E_UserUnknown;
-	}
-
-
 #elif (LINUX && (GLIBC >= 2))
 	
 	struct passwd	*pPasswd; 
diff -urN OOo/sal/osl/unx/system.h OOo-s390/sal/osl/unx/system.h
--- OOo/sal/osl/unx/system.h	2002-08-20 17:54:55.000000000 +0200
+++ OOo-s390/sal/osl/unx/system.h	2002-12-25 11:16:25.000000000 +0100
@@ -341,28 +341,6 @@
 extern char *strdup(const char *);
 #endif
 
-#ifdef S390
-#   define  AF_IPX -1
-#	include <stropts.h>
-#	include <pthread.h>
-#	include <xti.h>
-#	include <sys/time.h>
-#	include <sys/ioctl.h>
-#   include <sys/ps.h>
-#	include <dll.h>
-#	define  _BIG_ENDIAN
-#	define  LIBPATH "LIPPATH"
-# 	define  PTHREAD_NONE_INIT 			{ 0 }
-#   define  sched_yield()		 		pthread_yield(NULL)
-#	define  SLEEP_TIMESPEC(timespec)  	OSL_TRACE("WARNINIG:sleep not implemented (%s:%d)", __FILE__, __LINE__)
-#   define  pthread_testcancel 			pthread_testintr
-#	define  pthread_detach(t)  			pthread_detach(&(t))
-#   define  NO_PTHREAD_SEMAPHORES
-#	define  NO_PTHREAD_PRIORITY
-#   define  NO_DL_FUNCTIONS
-#	define  PATH_MAX 					_POSIX_PATH_MAX
-#endif
-
 #ifdef SOLARIS
 #	include <shadow.h>
 #	include <sys/procfs.h>
@@ -418,7 +396,7 @@
 
 #if !defined(_WIN32)  && !defined(_WIN16) && !defined(OS2)  && \
     !defined(LINUX)   && !defined(NETBSD) && !defined(FREEBSD) && !defined(SCO)  && \
-	!defined(AIX)     && !defined(HPUX)   && !defined(S390) && \
+	!defined(AIX)     && !defined(HPUX)   && \
 	!defined(SOLARIS) && !defined(IRIX)   && !defined(MAC) && \
 	!defined(MACOSX)
 #	error "Target plattform not specified !"
diff -urN OOo/solenv/bin/modules/osarch.pm OOo-s390/solenv/bin/modules/osarch.pm
--- OOo/solenv/bin/modules/osarch.pm	2000-09-20 16:43:11.000000000 +0200
+++ OOo-s390/solenv/bin/modules/osarch.pm	2002-12-25 11:16:25.000000000 +0100
@@ -34,7 +34,6 @@
 %osDefTable	= ("SunOS",		"-DSYSV -DSOLARIS",
 			   "HP-UX",		"-DSYSV -DHPUX",
 			   "AIX",		"-DSYSV -DAIX",
-			   "OS/390",	"-DS390 -DS390_ASCII",
 			   "Linux",		"-DLINUX",
 			   "FreeBSD",	"-DFREEBSD",
 			   "NetBSD",	"-DNETBSD",
diff -urN OOo/solenv/inc/_tg_shl.mk OOo-s390/solenv/inc/_tg_shl.mk
--- OOo/solenv/inc/_tg_shl.mk	2002-11-08 14:35:47.000000000 +0100
+++ OOo-s390/solenv/inc/_tg_shl.mk	2002-12-25 11:16:25.000000000 +0100
@@ -416,9 +416,6 @@
 	$(SHL1STDLIBS) $(SHL1ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL1NOCHECK)"==""
 	+-$(RM) $(SHL1TARGETN:d)check_$(SHL1TARGETN:f)
@@ -878,9 +875,6 @@
 	$(SHL2STDLIBS) $(SHL2ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL2NOCHECK)"==""
 	+-$(RM) $(SHL2TARGETN:d)check_$(SHL2TARGETN:f)
@@ -1340,9 +1334,6 @@
 	$(SHL3STDLIBS) $(SHL3ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL3NOCHECK)"==""
 	+-$(RM) $(SHL3TARGETN:d)check_$(SHL3TARGETN:f)
@@ -1802,9 +1793,6 @@
 	$(SHL4STDLIBS) $(SHL4ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL4NOCHECK)"==""
 	+-$(RM) $(SHL4TARGETN:d)check_$(SHL4TARGETN:f)
@@ -2264,9 +2252,6 @@
 	$(SHL5STDLIBS) $(SHL5ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL5NOCHECK)"==""
 	+-$(RM) $(SHL5TARGETN:d)check_$(SHL5TARGETN:f)
@@ -2726,9 +2711,6 @@
 	$(SHL6STDLIBS) $(SHL6ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL6NOCHECK)"==""
 	+-$(RM) $(SHL6TARGETN:d)check_$(SHL6TARGETN:f)
@@ -3188,9 +3170,6 @@
 	$(SHL7STDLIBS) $(SHL7ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL7NOCHECK)"==""
 	+-$(RM) $(SHL7TARGETN:d)check_$(SHL7TARGETN:f)
@@ -3650,9 +3629,6 @@
 	$(SHL8STDLIBS) $(SHL8ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL8NOCHECK)"==""
 	+-$(RM) $(SHL8TARGETN:d)check_$(SHL8TARGETN:f)
@@ -4112,9 +4088,6 @@
 	$(SHL9STDLIBS) $(SHL9ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL9NOCHECK)"==""
 	+-$(RM) $(SHL9TARGETN:d)check_$(SHL9TARGETN:f)
@@ -4574,9 +4547,6 @@
 	$(SHL10STDLIBS) $(SHL10ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL10NOCHECK)"==""
 	+-$(RM) $(SHL10TARGETN:d)check_$(SHL10TARGETN:f)
diff -urN OOo/solenv/inc/tg_shl.mk OOo-s390/solenv/inc/tg_shl.mk
--- OOo/solenv/inc/tg_shl.mk	2002-11-08 14:35:52.000000000 +0100
+++ OOo-s390/solenv/inc/tg_shl.mk	2002-12-25 11:16:25.000000000 +0100
@@ -516,9 +516,6 @@
 	$(SHL$(TNR)STDLIBS) $(SHL$(TNR)ARCHIVES) $(STDSHL) $(LINKOUTPUT_FILTER) > $(MISC)$/$(@:b).cmd
 	@cat $(MISC)$/$(@:b).cmd
 	@+source $(MISC)$/$(@:b).cmd
-.IF "$(OS)"=="S390"
-	+mv -f ($@:s/$(DLLPOST)/.x/) $(LB)
-.ENDIF
 .IF "$(UPDATER)"=="YES"
 .IF "$(SHL$(TNR)NOCHECK)"==""
 	+-$(RM) $(SHL$(TNR)TARGETN:d)check_$(SHL$(TNR)TARGETN:f)
diff -urN OOo/solenv/inc/unx.mk OOo-s390/solenv/inc/unx.mk
--- OOo/solenv/inc/unx.mk	2002-10-30 14:59:31.000000000 +0100
+++ OOo-s390/solenv/inc/unx.mk	2002-12-25 11:16:25.000000000 +0100
@@ -140,10 +140,6 @@
 .INCLUDE : unxbsda.mk
 .ENDIF
 
-.IF "$(COM)$(OS)$(CPU)" == "ICCS390G"
-.INCLUDE : unxmvsg.mk
-.ENDIF
-
 .IF "$(COM)$(OS)$(CPU)" == "ICCAIXP"
 .INCLUDE : unxaixp.mk
 .ENDIF
diff -urN OOo/solenv/inc/unxmvsg.mk OOo-s390/solenv/inc/unxmvsg.mk
--- OOo/solenv/inc/unxmvsg.mk	2002-11-08 14:36:00.000000000 +0100
+++ OOo-s390/solenv/inc/unxmvsg.mk	1970-01-01 01:00:00.000000000 +0100
@@ -1,149 +0,0 @@
-#
-# cc statisch
-#
-cc  -W "c,LONGNAME,LANGLVL(EXTENDED)" -c  
--o ../../salc32s390/obj/diagnose.o 
--I. -I../inc -I../../inc -I../../inc -I../../unx/inc -I../../salc32s390/inc 
-	-I../../../solar/inc -I../../../solar/sal/res -I../../../solar/res 
-	-I/nw386/dev/s/solenv/res -I/solarserver/inc/stl 
-	-I/solarserver/364/salc32s390/inc -I/solarserver/s390g3/inc 
-	-I/solarserver/s390g3/java/inc -I/solarserver/inc/Xp31 -I/solarserver/inc      
--D_DEBUG -DS390_ASCII -DS390 -DG3 -DC32 -DUNX -DVCL -DSO3 -D_PTHREADS 
-	-D_STD_NO_NAMESPACE -D_VOS_NO_NAMESPACE -D_UNO_NO_NAMESPACE -DNO_AUDIO 
-	-D_ALL_SOURCE -DS390_ASCII -DCSET_31 -DSUPD=364 -DSOLAR_VERSION=3640 
-	-DUPDVER='"SV364"' -DDBG_UTIL          
-_diagnose.c
-
-#
-# cc pic
-#
-cc  -W "c,LONGNAME,LANGLVL(EXTENDED),EXPORTALL,DLL" -c 
--o ../../salc32s390/so/diagnose.o 
--I. -I../inc -I../../inc -I../../inc -I../../unx/inc -I../../salc32s390/inc 
-	-I../../../solar/inc -I../../../solar/sal/res -I../../../solar/res 
-	-I/nw386/dev/s/solenv/res -I/solarserver/inc/stl 
-	-I/solarserver/364/salc32s390/inc -I/solarserver/s390g3/inc 	
-	-I/solarserver/s390g3/java/inc -I/solarserver/inc/Xp31 -I/solarserver/inc         
--D_DEBUG -DSHAREDLIB -DSHARED -DS390_ASCII -DS390 -DG3 -DC32 -DUNX -DVCL -DSO3 
-	-D_PTHREADS -D_STD_NO_NAMESPACE -D_VOS_NO_NAMESPACE -D_UNO_NO_NAMESPACE 
-	-DNO_AUDIO -D_ALL_SOURCE -DS390_ASCII -DCSET_31 -DSUPD=364 -DSOLAR_VERSION=3640 
-	-DUPDVER='"SV364"' -DDBG_UTIL     
-_diagnose.c
-
-#
-# CC statisch
-#
-cxx -W "c,LONGNAME,LANGLVL(EXTENDED)" -c  
--o ../../salc32s390/obj/date.o 
--I. -I../inc -I../../inc -I../../inc -I../../unx/inc -I../../salc32s390/inc 
-	-I../../../solar/inc -I../../../solar/sal/res -I../../../solar/res 
-	-I/nw386/dev/s/solenv/res -I/solarserver/inc/stl 	
-	-I/solarserver/364/salc32s390/inc -I/solarserver/s390g3/inc 
-	-I/solarserver/s390g3/java/inc -I/solarserver/inc/Xp31 -I/solarserver/inc 
--D_DEBUG -DS390_ASCII -DS390 -DG3 -DC32 -DUNX -DVCL -DSO3 -D_PTHREADS 
-	-D_STD_NO_NAMESPACE -D_VOS_NO_NAMESPACE -D_UNO_NO_NAMESPACE -DNO_AUDIO 
-	-D_ALL_SOURCE -DS390_ASCII -DCSET_31 -DSUPD=364 -DSOLAR_VERSION=3640 
-	-DUPDVER='"SV364"' -DDBG_UTIL  
--+ _date.cxx
-
-#
-# CC pic
-#
-cxx -W "c,LONGNAME,LANGLVL(EXTENDED),EXPORTALL" -c  
--o ../../salc32s390/so/date.o 
--I. -I../inc -I../../inc -I../../inc -I../../unx/inc -I../../salc32s390/inc 
-	-I../../../solar/inc -I../../../solar/sal/res -I../../../solar/res 
-	-I/nw386/dev/s/solenv/res -I/solarserver/inc/stl 
-	-I/solarserver/364/salc32s390/inc -I/solarserver/s390g3/inc 	
-	-I/solarserver/s390g3/java/inc -I/solarserver/inc/Xp31 -I/solarserver/inc         
--D_DEBUG -DSHAREDLIB -DSHARED -DS390_ASCII -DS390 -DG3 -DC32 -DUNX -DVCL -DSO3 	
-	-D_PTHREADS -D_STD_NO_NAMESPACE -D_VOS_NO_NAMESPACE -D_UNO_NO_NAMESPACE 
-	-DNO_AUDIO -D_ALL_SOURCE -DS390_ASCII -DCSET_31 -DSUPD=364 -DSOLAR_VERSION=3640 
-	-DUPDVER='"SV364"' -DDBG_UTIL     
--+ _date.cxx
-
-
-# mak file fuer unxmvsg
-
-ASM=
-AFLAGS=
-
-CC=				cc -c
-CXX=			cxx -c
-
-CFLAGSCCOBJ=	-W "c,LONGNAME,LANGLVL(EXTENDED)"
-CFLAGSCCSLO=	-W "c,LONGNAME,LANGLVL(EXTENDED),EXPORTALL,DLL"
-CFLAGSCXXOBJ=	-W "c,LONGNAME,LANGLVL(EXTENDED)"
-CFLAGSCXXSLO=	-W "c,LONGNAME,LANGLVL(EXTENDED),EXPORTALL"
-
-CDEFS+=			-DALL_SOURCE -DS390_ASCII -DCSET_31
-CDEFS+=			-D_STD_NO_NAMESPACE -D_VOS_NO_NAMESPACE -D_UNO_NO_NAMESPACE
-CDEFS+=			-D_PTHREADS
-CDEFS+= 		-DNO_AUDIO -DPRINTER_DUMMY
-
-CFLAGS=			-I. -I$(INC) -I$(INCLOCAL) -I$(INCGUI) -I$(INCCOM) $(SOLARINC)
-CFLAGSCC=
-CFLAGSCXX=
-CFLAGSOBJGUIST=
-CFLAGSOBJCUIST=
-CFLAGSOBJGUIMT=
-CFLAGSOBJCUIMT=
-CFLAGSSLOGUIMT=
-CFLAGSSLOCUIMT=
-CFLAGSPROF=
-CFLAGSDEBUG=	-g
-CFLAGSDBGUTIL=
-CFLAGSOPT=		-O2
-CFLAGSNOOPT=
-CFLAGSOUTOBJ=	-o
-
-
-STATIC=			-Bstatic
-DYNAMIC=		-Bdynamic
-
-THREADLIB=
-LINK=			cxx 
-LINKFLAGS=		-W "l,DLL"
-LINKFLAGSAPPGUI=
-LINKFLAGSSHLGUI=
-LINKFLAGSAPPCUI=
-LINKFLAGSSHLCUI=
-LINKFLAGSTACK=
-LINKFLAGSPROF=
-LINKFLAGSDEBUG=
-LINKFLAGSOPT=
-
-STDOBJGUI=
-STDSLOGUI=
-STDOBJCUI=
-STDSLOCUI=
-STDLIBGUIST=-lrpclib -lm -lc		# Applikationen
-STDLIBCUIST=
-STDLIBGUIMT=
-STDLIBCUIMT=
-STDSHLGUIMT=-lrpclib -lm -lc		# Shared Libraries
-STDSHLCUIMT=
-
-LIBMGR=			ar
-LIBFLAGS=		-r
-LIBEXT=			.a
-
-IMPLIB=
-IMPLIBFLAGS=
-
-MAPSYM=
-MAPSYMFLAGS=
-
-RC=				irc
-RCFLAGS=		-fo$@ $(RCFILES)
-RCLINK=
-RCLINKFLAGS=
-RCSETVERSION=
-
-DLLPOSTFIX=		mg
-DLLPRE=			lib
-DLLPOST=		.so
-
-LDUMP=
-
-
diff -urN OOo/soltools/cpp/_cpp.c OOo-s390/soltools/cpp/_cpp.c
--- OOo/soltools/cpp/_cpp.c	2001-07-10 10:57:52.000000000 +0200
+++ OOo-s390/soltools/cpp/_cpp.c	2002-12-25 11:16:36.000000000 +0100
@@ -51,7 +51,7 @@
 void
     process(Tokenrow * trp)
 {
-    int anymacros = 0;  /* S390: XDBX stürzt ab! */
+    int anymacros = 0;
 
     for (;;)
     {
diff -urN OOo/soltools/cpp/_include.c OOo-s390/soltools/cpp/_include.c
--- OOo/soltools/cpp/_include.c	2001-04-19 10:20:22.000000000 +0200
+++ OOo-s390/soltools/cpp/_include.c	2002-12-25 11:16:36.000000000 +0100
@@ -2,7 +2,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <fcntl.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #	include <io.h>
 #else
 #	include <unistd.h>
@@ -14,9 +14,6 @@
 #ifdef _MSC_VER
 #	define _POSIX_
 #endif
-#ifdef S390
-#   define PATH_MAX _POSIX_PATH_MAX
-#endif
 #ifdef __IBMC__
 #	include <fcntl.h>
 #   define PATH_MAX _MAX_PATH
diff -urN OOo/soltools/cpp/_lex.c OOo-s390/soltools/cpp/_lex.c
--- OOo/soltools/cpp/_lex.c	2001-04-19 10:20:22.000000000 +0200
+++ OOo-s390/soltools/cpp/_lex.c	2002-12-25 11:16:36.000000000 +0100
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
@@ -37,11 +37,7 @@
 #define	C_ALPH	2
 #define	C_NUM	3
 #define	C_EOF	4
-#ifdef S390
-#define	C_XX	6
-#else
 #define	C_XX	5
-#endif
 
 enum state
 {
@@ -268,12 +264,8 @@
                     continue;
                 case C_ALPH:
                     for (j = 0; j <= 256; j++)
-#ifdef S390
-						if( isalpha( j ) || (j == '_') )
-#else							
                         if ('a' <= j && j <= 'z' || 'A' <= j && j <= 'Z'
                             || j == '_')
-#endif							
                             bigfsm[j][fp->state] = nstate;
                     continue;
                 case C_NUM:
@@ -351,7 +343,7 @@
             }
     }
     maxp = &trp->bp[trp->max];
-    runelen = 1;  /* S390: XDBX stürzt ab! */
+    runelen = 1;
     for (;;)
     {
 continue2:
@@ -370,7 +362,7 @@
         {
             oldstate = state;
 
-            c = *ip;  /* S390: XDBX stürzt (gelegentlich) ab! */
+            c = *ip;
 						
             if ((state = bigfsm[c][state]) >= 0)
             {
@@ -416,7 +408,7 @@
 
 					if (c == '\n')
 					{
-					    while (s->inp + 1 >= s->inl && fillbuf(s) != EOF); /* S390: XDBX stürzt (manchmal) ab! */
+					    while (s->inp + 1 >= s->inl && fillbuf(s) != EOF);
 					
 						if (s->inp[1] == '\r')
 						{
diff -urN OOo/soltools/cpp/_macro.c OOo-s390/soltools/cpp/_macro.c
--- OOo/soltools/cpp/_macro.c	2001-10-31 17:33:02.000000000 +0100
+++ OOo-s390/soltools/cpp/_macro.c	2002-12-25 11:16:36.000000000 +0100
@@ -8,9 +8,6 @@
 #ifdef _MSC_VER
 #	define _POSIX_
 #endif
-#ifdef S390
-#   define PATH_MAX _POSIX_PATH_MAX
-#endif
 #ifdef __IBMC__
 #   define PATH_MAX _MAX_PATH
 #endif
diff -urN OOo/soltools/cpp/_tokens.c OOo-s390/soltools/cpp/_tokens.c
--- OOo/soltools/cpp/_tokens.c	2001-04-19 10:20:22.000000000 +0200
+++ OOo-s390/soltools/cpp/_tokens.c	2002-12-25 11:16:36.000000000 +0100
@@ -2,7 +2,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
diff -urN OOo/soltools/cpp/_unix.c OOo-s390/soltools/cpp/_unix.c
--- OOo/soltools/cpp/_unix.c	2001-04-19 10:20:22.000000000 +0200
+++ OOo-s390/soltools/cpp/_unix.c	2002-12-25 11:16:36.000000000 +0100
@@ -4,7 +4,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <fcntl.h>
-#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__)) && !(defined S390)
+#if (defined(_WIN32) || defined(_MSDOS) || defined(__IBMC__))
 #include <io.h>
 #else
 #include <unistd.h>
diff -urN OOo/soltools/cpp/cpp.h OOo-s390/soltools/cpp/cpp.h
--- OOo/soltools/cpp/cpp.h	2001-04-19 10:20:22.000000000 +0200
+++ OOo-s390/soltools/cpp/cpp.h	2002-12-25 11:16:36.000000000 +0100
@@ -47,14 +47,8 @@
 #define	ISARCHITECTURE	0x10            /* architecture */
 #define ISACTIVE        0x80            /* is macro currently expanded */
 
-#ifdef S390
-/* 0xfe und 0xfd fallen auf der S/390 auf Control-Chars */
-#define EOB     0x07
-#define EOFC    0x08
-#else
 #define	EOB		0xFE                    /* sentinel for end of input buffer */
 #define	EOFC	0xFD                    /* sentinel for end of input file */
-#endif
 #define	XPWS	1                       /* token flag: white space to assure token sep. */
 #define XTWS	2
 
diff -urN OOo/svtools/inc/svarray.hxx OOo-s390/svtools/inc/svarray.hxx
--- OOo/svtools/inc/svarray.hxx	2002-05-30 10:57:57.000000000 +0200
+++ OOo-s390/svtools/inc/svarray.hxx	2002-12-25 11:16:36.000000000 +0100
@@ -1042,7 +1042,7 @@
 #define C40_PTR_REPLACE( c, p) Replace( (c const *) p )
 #define C40_GETPOS( c, r) GetPos( (c const *)r )
 #else
-#if defined WTC || defined IRIX || defined ICC || defined S390 || defined HPUX || (defined GCC && __GNUC__ >= 3)
+#if defined WTC || defined IRIX || defined ICC || defined HPUX || (defined GCC && __GNUC__ >= 3)
 #define C40_INSERT( c, p, n ) Insert( (c const *&) p, n )
 #define C40_PUSH( c, p) Push( (c const *&) p )
 #define C40_PTR_INSERT( c, p ) Insert( (c const *&) p )
diff -urN OOo/svtools/source/filter.vcl/filter/sgvspln.cxx OOo-s390/svtools/source/filter.vcl/filter/sgvspln.cxx
--- OOo/svtools/source/filter.vcl/filter/sgvspln.cxx	2002-06-05 15:54:22.000000000 +0200
+++ OOo-s390/svtools/source/filter.vcl/filter/sgvspln.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -92,9 +92,7 @@
 
 /*-----------------------  FILE u_const.h  ---------------------------*/
 
-#ifndef S390
 #define IEEE
-#endif
 
 /* IEEE - Norm fuer die Darstellung von Gleitkommazahlen:
 
@@ -133,12 +131,10 @@
 
 #else               /*------------------ sonst -----------------------*/
 
-#ifndef S390
 double exp  (double);
 double atan (double);
 double pow  (double,double);
 double sqrt (double);
-#endif
 
 double masch()            /* MACH_EPS maschinenunabhaengig bestimmen  */
 {
@@ -167,14 +163,8 @@
    gepasst werden.
    */
 
-#ifndef S390
 #define MAXEXPON  1023.0                   /* groesster Exponent      */
 #define MINEXPON -1024.0                   /* kleinster Exponent      */
-#else
-#include <float.h>
-#define MAXEXPON  75.0
-#define MINEXPON  -79.0
-#endif
 
 
 #define MACH_EPS  masch()
diff -urN OOo/svtools/source/numbers/zformat.cxx OOo-s390/svtools/source/numbers/zformat.cxx
--- OOo/svtools/source/numbers/zformat.cxx	2002-10-31 21:37:30.000000000 +0100
+++ OOo-s390/svtools/source/numbers/zformat.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -109,13 +109,8 @@
 const USHORT _MAX_FRACTION_PREC = 3;
 const double D_EPS = 1.0E-2;
 
-#ifdef S390
-const double _D_MAX_D_BY_100  = 7.2E73;
-const double _D_MIN_M_BY_1000 = 5.4E-76;
-#else
 const double _D_MAX_D_BY_100  = 1.7E306;
 const double _D_MIN_M_BY_1000 = 2.3E-305;
-#endif
 
 static BYTE cCharWidths[ 128-32 ] = {
 	1,1,1,2,2,3,2,1,1,1,1,2,1,1,1,1,
diff -urN OOo/tools/source/fsys/unx.cxx OOo-s390/tools/source/fsys/unx.cxx
--- OOo/tools/source/fsys/unx.cxx	2000-11-16 22:06:23.000000000 +0100
+++ OOo-s390/tools/source/fsys/unx.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -73,9 +73,6 @@
 #include <sys/mntctl.h>
 #include <sys/vmount.h>
 extern "C" int mntctl( int cmd, size_t size, char* buf );
-#elif defined S390
-#include <sys/mntent.h>
-#define mnttab w_mntent
 #elif defined(NETBSD) 
 #include <sys/mount.h>
 #elif defined(FREEBSD) || defined(MACOSX)
@@ -111,9 +108,6 @@
 #define MNTTAB 		 "/etc/mnttab"
 #define MOUNTSPECIAL mt_dev
 #define MOUNTPOINT   mt_filsys
-#elif defined S390
-#define MOUNTSPECIAL mnt_fsname
-#define MOUNTPOINT 	 mnt_mountpoint
 #else
 #define MOUNTSPECIAL mnt_fsname
 #define MOUNTPOINT   mnt_dir
@@ -190,7 +184,7 @@
 		return FALSE;
 	struct mnttab mnt[1];
 	while (fread (&mnt, sizeof mnt, 1, fp) > 0)
-#elif defined DECUNIX || defined AIX || defined S390
+#elif defined DECUNIX || defined AIX
 	FILE *fp = NULL;
 	if (! fp)
 		return FALSE;
diff -urN OOo/tools/source/fsys/unx.hxx OOo-s390/tools/source/fsys/unx.hxx
--- OOo/tools/source/fsys/unx.hxx	2000-09-18 19:03:07.000000000 +0200
+++ OOo-s390/tools/source/fsys/unx.hxx	2002-12-25 11:16:36.000000000 +0100
@@ -65,11 +65,7 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-
-#ifndef S390
 #include <sys/param.h>
-#endif
-
 #include <dirent.h>
 #include <unistd.h>
 /* #include <sysent.h> */
diff -urN OOo/tools/source/rc/resmgr.cxx OOo-s390/tools/source/rc/resmgr.cxx
--- OOo/tools/source/rc/resmgr.cxx	2002-05-30 10:42:25.000000000 +0200
+++ OOo-s390/tools/source/rc/resmgr.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -144,8 +144,6 @@
 
 #if defined( OS2 ) && defined( ICC )
 static int _Optlink Compare( const void * pFirst, const void * pSecond )
-#elif S390
-extern "C" { int Compare( const void * pFirst, const void * pSecond )
 #else
 static int __LOADONCALLAPI Compare( const void * pFirst, const void * pSecond )
 #endif
@@ -158,14 +156,8 @@
         return( 0 );
 }
 
-#ifdef S390
-}
-#endif
-
 #if defined( OS2 ) && defined( ICC )
 static int _Optlink Search( const void * nTypeAndId, const void * pSecond )
-#elif S390
-extern "C" { int Search( const void * nTypeAndId, const void * pSecond )
 #else
 static int __LOADONCALLAPI Search( const void * nTypeAndId, const void * pSecond )
 #endif
@@ -178,10 +170,6 @@
         return( 0 );
 }
 
-#ifdef S390
-}
-#endif
-
 // =======================================================================
 
 static ResHookProc pImplResHookProc = 0;
diff -urN OOo/tools/source/stream/strmunx.cxx OOo-s390/tools/source/stream/strmunx.cxx
--- OOo/tools/source/stream/strmunx.cxx	2001-07-27 10:10:24.000000000 +0200
+++ OOo-s390/tools/source/stream/strmunx.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -244,7 +244,7 @@
         { 0,            SVSTREAM_OK },
         { EACCES,       SVSTREAM_ACCESS_DENIED },
         { EBADF,        SVSTREAM_INVALID_HANDLE },
-#if defined( RS6000 ) || defined( ALPHA ) || defined( HP9000 ) || defined( NETBSD ) || defined( S390 ) || defined(FREEBSD) || defined(MACOSX)
+#if defined( RS6000 ) || defined( ALPHA ) || defined( HP9000 ) || defined( NETBSD ) || defined(FREEBSD) || defined(MACOSX)
         { EDEADLK,      SVSTREAM_LOCKING_VIOLATION },
 #else
         { EDEADLOCK,    SVSTREAM_LOCKING_VIOLATION },
diff -urN OOo/vcl/unx/inc/salpdecl.h OOo-s390/vcl/unx/inc/salpdecl.h
--- OOo/vcl/unx/inc/salpdecl.h	2000-09-18 19:05:41.000000000 +0200
+++ OOo-s390/vcl/unx/inc/salpdecl.h	2002-12-25 11:16:36.000000000 +0100
@@ -71,7 +71,7 @@
 extern "C" {
 #endif
 
-#if defined S390 || defined AIX
+#if defined AIX
 #define NO_SHM_FUNCTIONS
 #endif
 
diff -urN OOo/vcl/unx/source/app/saldisp.cxx OOo-s390/vcl/unx/source/app/saldisp.cxx
--- OOo/vcl/unx/source/app/saldisp.cxx	2002-08-08 04:49:59.000000000 +0200
+++ OOo-s390/vcl/unx/source/app/saldisp.cxx	2002-12-25 11:16:36.000000000 +0100
@@ -219,7 +219,7 @@
 #include "vertcurs_mask.h" 
 
 #include "invert50.h"
-#if !(defined S390 || defined AIX)
+#if !(defined AIX)
 #include <X11/extensions/XShm.h>
 #endif
 #include <X11/keysym.h>
diff -urN OOo/vcl/unx/source/app/salmain.cxx OOo-s390/vcl/unx/source/app/salmain.cxx
--- OOo/vcl/unx/source/app/salmain.cxx	2001-03-21 11:03:44.000000000 +0100
+++ OOo-s390/vcl/unx/source/app/salmain.cxx	2002-12-25 11:16:37.000000000 +0100
@@ -78,11 +78,7 @@
 // -=-= main() -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 BEGIN_C
 
-#ifdef S390
-int sv_main( int argc, char *argv[] )
-#else
 int main( int argc, char *argv[] )
-#endif
 {
     SVMain();
 
