From e5539b0b53ebaf9b792652e4866eeacf8c10b101 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 28 Aug 2015 15:09:31 +0100
Subject: deb#795131 plausible fix

Change-Id: I95c191ed865d6920845b146c3ef4baf30280a734

diff --git a/vcl/unx/gtk3/app/gtk3gtkinst.cxx b/vcl/unx/gtk3/app/gtk3gtkinst.cxx
index 823d741..9d7284f 100644
--- a/vcl/unx/gtk3/app/gtk3gtkinst.cxx
+++ b/vcl/unx/gtk3/app/gtk3gtkinst.cxx
@@ -382,6 +382,8 @@ Reference< css::datatransfer::XTransferable > VclGtkClipboard::getContents() thr
 void VclGtkClipboard::ClipboardGet(GtkClipboard* /*clipboard*/, GtkSelectionData *selection_data,
                                    guint info)
 {
+    if (!m_aContents.is())
+        return;
 
     GdkAtom type(gdk_atom_intern(OUStringToOString(m_aInfoToFlavor[info].MimeType,
                                                    RTL_TEXTENCODING_UTF8).getStr(),
-- 
cgit v0.10.2

