--- oo_1.0_src/product/inc/scp/udk_basefiles.scp.setup-no-stdlibs	2002-03-15 12:58:28.000000000 +0100
+++ oo_1.0_src/product/inc/scp/udk_basefiles.scp	2002-05-06 11:09:13.000000000 +0200
@@ -1,30 +1,5 @@
 // all file objects that are relevant for the udk base
 
-// Temporary solution/hack: at the moment libstdc++.so and libgcc_s.so are needed for unxlngi4 environment
-// (setup is linked against it).
-#if ( defined( _gcc3 ) && defined( C300 ) )
-File gid_File_Lib_gcc
-    BIN_FILE_BODY;
-	Name		= STRING(libgcc_s.so.1);
-    Dir			= GCFG_BINARY_DIR;
-    Styles		= (PACKED, SETUPZIP);
-End
-
-File gid_File_Lib_Stdc
-    BIN_FILE_BODY;
-    Name		= STRING(libstdc++.so.3.0.1);
-    Dir			= GCFG_BINARY_DIR;
-    Styles		= (PACKED, SETUPZIP);
-End
-
-Shortcut gid_Shortcut_Lib_Stdc
-    FileID      = gid_File_Lib_Stdc;
-    Dir         = GCFG_BINARY_DIR;
-    Name        = STRING(libstdc++.so.3);
-    Styles      = (NETWORK,RELATIVE);
-End
-#endif
-
 File gid_File_Lib_Sal
     BIN_FILE_BODY;
     #ifdef UNX
--- oo_1.0_src/product/util/makefile.mk.orig	Tue May 28 15:37:43 2002
+++ oo_1.0_src/product/util/makefile.mk	Tue May 28 15:38:25 2002
@@ -107,13 +107,6 @@
 DESTDIRDLL=$(DESTDIRLIB)
 STLPORTLIBNAME=libstlport_gcc.so
 BINDINGDLL=$(COMNAME)_uno
-.IF "$(COMID)"=="gcc3"
-ADDITIONAL_DLLS= \
-	$(DESTDIRDLL)$/libgcc_s.so.1 		\
-	$(DESTDIRDLL)$/libgcc_s.so 		\
-	$(DESTDIRDLL)$/libstdc++.so.3 		\
-	$(DESTDIRDLL)$/libstdc++.so.3.0.4
-.ENDIF
 .ELIF "$(OS)"=="FREEBSD"
 ###########
 # FREEBSD
