Index: oo_1.1rc_src/vcl/util/makefile.mk
===================================================================
RCS file: /cvs/gsl/vcl/util/makefile.mk,v
retrieving revision 1.45
diff -u -3 -p -u -r1.45 makefile.mk
--- oo_1.1rc_src/vcl/util/makefile.mk	10 Jun 2003 14:31:36 -0000	1.45
+++ oo_1.1rc_src/vcl/util/makefile.mk	24 Jul 2003 09:13:34 -0000
@@ -279,9 +279,11 @@ SHL1STDLIBS += -lXext
 .ENDIF
 
 # search for freetype in systems dir at runtime
-.IF "$(OS)"=="LINUX"
-LINKFLAGSRUNPATH!:="-Wl,-rpath,/usr/lib" $(LINKFLAGSRUNPATH)
-.ENDIF
+# Debian: rpaths cause problems; we solve this by not
+# installing OOo's freetype at all
+#.IF "$(OS)"=="LINUX"
+#LINKFLAGSRUNPATH!:="-Wl,-rpath,/usr/lib" $(LINKFLAGSRUNPATH)
+#.ENDIF
 
 # Solaris
 .IF "$(OS)"=="SOLARIS"
Index: oo_1.1rc_src/icu/icu-2.2.patch
===================================================================
RCS file: /cvs/external/icu/icu-2.2.patch,v
retrieving revision 1.14.2.2
diff -u -3 -p -r1.14.2.2 icu-2.2.patch
--- oo_1.1rc_src/icu/icu-2.2.patch	27 Jun 2003 13:04:32 -0000	1.14.2.2
+++ oo_1.1rc_src/icu/icu-2.2.patch	24 Jul 2003 09:56:20 -0000
@@ -70,22 +70,6 @@
   SHAREDLIBCFLAGS = -fPIC
   SHAREDLIBCXXFLAGS = -fPIC
   SHAREDLIBCPPFLAGS = -DPIC
-***************
-*** 22,27 ****
---- 25,37 ----
-  LD_RPATH=	
-  LD_RPATH_PRE = -Wl,-rpath,
-  
-+ ## Force RPATH=$ORIGIN to locate own dependencies w/o need for LD_LIBRARY_PATH
-+ ENABLE_RPATH=YES
-+ RPATHLDFLAGS=${LD_RPATH_PRE}'$$ORIGIN'
-+ 
-+ #SH#	ENABLE_RPATH=YES
-+ #SH#	RPATHLDFLAGS="${LD_RPATH_PRE}'$$ORIGIN'"
-+ 
-  ## Compiler switch to embed a library name
-  LD_SONAME = -Wl,-soname -Wl,$(MIDDLE_SO_TARGET)
-  
 *** misc/icu/source/config/mh-solaris	2002-07-30 04:46:10.000000000 +0200
 --- misc/build/icu/source/config/mh-solaris	2003-06-27 14:09:29.000000000 +0200
 ***************
