Index: oo_cvs/connectivity/source/drivers/odbc/OFunctions.cxx
===================================================================
RCS file: /cvs/oo/dba/connectivity/source/drivers/odbc/OFunctions.cxx,v
retrieving revision 1.4
diff -u -3 -p -r1.4 OFunctions.cxx
--- oo_cvs/connectivity/source/drivers/odbc/OFunctions.cxx	2001/05/15 08:18:13	1.4
+++ oo_cvs/connectivity/source/drivers/odbc/OFunctions.cxx	2002/06/17 09:09:03
@@ -152,7 +152,7 @@ sal_Bool LoadLibrary_ODBC3(::rtl::OUStri
 	_rPath = ::rtl::OUString::createFromAscii("ODBC32.DLL");
 #endif
 #ifdef UNX
-	_rPath = ::rtl::OUString::createFromAscii("libodbc.so");
+	_rPath = ::rtl::OUString::createFromAscii("libodbc.so.1");
 #endif
 #ifdef OS2
 	_rPath = ::rtl::OUString::createFromAscii("ODBC");
Index: oo_cvs/dbaccess/source/ui/dlg/odbcconfig.cxx
===================================================================
RCS file: /cvs/oo/dba/dbaccess/source/ui/dlg/odbcconfig.cxx,v
retrieving revision 1.6
diff -u -3 -p -r1.6 odbcconfig.cxx
--- oo_cvs/dbaccess/source/ui/dlg/odbcconfig.cxx	2001/03/23 13:29:31	1.6
+++ oo_cvs/dbaccess/source/ui/dlg/odbcconfig.cxx	2002/06/17 09:09:03
@@ -84,8 +84,8 @@
 #define ODBC_UI_LIBRARY	"ODBCCP32.DLL"
 #endif
 #ifdef UNX
-#define ODBC_LIBRARY	"libodbc.so"
-#define ODBC_UI_LIBRARY	"libodbc.so"
+#define ODBC_LIBRARY	"libodbc.so.1"
+#define ODBC_UI_LIBRARY	"libodbc.so.1"
 #endif
 
 // just to go with calling convention of windows
