--- oo_1.0.2_src/sysui/oounix/office/gnome/gnomeint.old	2003-03-24 22:11:57.000000000 +0100
+++ oo_1.0.2_src/sysui/oounix/office/gnome/gnomeint	2003-03-24 22:15:45.000000000 +0100
@@ -212,8 +212,9 @@
         "Type", "Application",
 #        "Exec", "\"<progpath>/program/$prog\"",
         "Exec", "<progpath>/program/$prog",
-		"Icon", "<progpath>/share/icons/$icon.png" ,
+	"Icon", "<progpath>/share/icons/$icon.png" ,
         "MimeType", "$mime_types",
+	"Categories", "Application;Office",
         "Name[$lang]", $name,
 #        "Comment[$lang]", *ref->{ sprintf("<%s_control_label>", $name),
         "MultipleArgs", "false",
@@ -280,14 +281,15 @@
 	local($outpath, $name, $prog, $icon, $lang, *ref) = @_;
 	my($outfile, $infile);
 	my(%replace) = (
-        "Type", "Application",
+         "Type", "Application",
 #        "Exec", "\"<progpath>/program/$prog\"",
         "Exec", "<progpath>/program/$prog",
-		"Icon", "<progpath>/share/icons/$icon.png" ,
-        "Name[$lang]", *ref->{ sprintf("<%s_action_label>", $name)  },
-#        "Comment[$lang]", *ref->{ sprintf("<%s_action_label>", $name)  }
+	"Icon", "<progpath>/share/icons/$icon.png" ,
+	"Categories", "Application;Setup",
+        "Name[$lang]", $name,
+#        "Comment[$lang]", *ref->{ sprintf("<%s_control_label>", $name),
         "MultipleArgs", "false",
-        "Terminal", "false",
+        "Terminal", "false"
     );
 
     # finally write desktop entry file
@@ -431,7 +433,7 @@
     local($out_dir, $lang, *ref) = @_;
 
     # create language specific control files
-    createControl($out_dir, "%PRODUCTNAME Calc", "spreadsheet", "ooo_calc", "scalc", "application/vnd.sun.xml.calc application/vnd.stardivision.calc application/vnd.stardivision.chart", $lang, *ref);
+    createControl($out_dir, "%PRODUCTNAME Calc", "spreadsheet", "ooo_calc", "scalc", "application/vnd.sun.xml.calc application/vnd.stardivision.calc application/vnd.stardivision.chart", $lang,, *ref);
 
     # create data type files
 #    createDataType($out_dir, "vnd.sun.xml.calc.document", "vnd.sun.xml.calc", "*.[sS][xX][cC]", "004_spreadsheet_document", "spreadsheet");
@@ -474,7 +476,6 @@
  #   createDataType($out_dir, "vnd.stardivision.draw.impress", "vnd.stardivision.impress", "*.[sS][dD][dD]", "009_presentation_document", "presentation");
 }
 
-
 #*********************************************************************
 #
 # main
@@ -528,12 +529,12 @@
     %replace = getNLSStrings(sprintf("%s/$zipkind.%s", $res_dir, $locale));
 
     SWITCH: {
-        if ( $zipkind eq "group" )   { createOfficeGroupFiles  ( $target_dir, $locale, \%replace ); last SWITCH; }
-        if ( $zipkind eq "writer" )  { createOfficeWriterFiles ( $target_dir, $locale, \%replace ); last SWITCH; }
-        if ( $zipkind eq "calc" )    { createOfficeCalcFiles   ( $target_dir, $locale, \%replace ); last SWITCH; }
-        if ( $zipkind eq "draw" )    { createOfficeDrawFiles   ( $target_dir, $locale, \%replace ); last SWITCH; }
-        if ( $zipkind eq "impress" ) { createOfficeImpressFiles( $target_dir, $locale, \%replace ); last SWITCH; }
-        if ( $zipkind eq "math" )    { createOfficeMathFiles   ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "group" )          { createOfficeGroupFiles        ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "writer" )         { createOfficeWriterFiles       ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "calc" )           { createOfficeCalcFiles         ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "draw" )           { createOfficeDrawFiles         ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "impress" )        { createOfficeImpressFiles      ( $target_dir, $locale, \%replace ); last SWITCH; }
+        if ( $zipkind eq "math" )           { createOfficeMathFiles         ( $target_dir, $locale, \%replace ); last SWITCH; }
     }
 }
 
