--- oo_1.0.1_src/sysui/unix/office/gnome/gnomeint	2001-11-12 14:42:39.000000000 +0100
+++ oo_1.0.1_src/sysui/unix/office/gnome/gnomeint_new	2002-09-10 23:55:28.000000000 +0200
@@ -208,6 +208,7 @@
         "Exec", "<progpath>/program/$prog",
 		"Icon", "<progpath>/share/icons/$icon.png" ,
         "MimeType", "$mime_types",
+	"Categories", "Application;Office",
         "Name[$lang]", *ref->{ sprintf("<%s_control_label>", $name)  },
 #        "Comment[$lang]", *ref->{ sprintf("<%s_control_label>", $name),
         "MultipleArgs", "false",
