diff --git a/cppuhelper/qa/misc/test_misc.cxx b/cppuhelper/qa/misc/test_misc.cxx
index 0e4fdf9efb42..be98752c9ae9 100644
--- a/cppuhelper/qa/misc/test_misc.cxx
+++ b/cppuhelper/qa/misc/test_misc.cxx
@@ -19,11 +19,15 @@ class Test : public ::CppUnit::TestFixture
 {
 public:
     void testCatchThrow();
+#ifndef __aarch64__
     void testgetCaughtException();
+#endif
 
     CPPUNIT_TEST_SUITE(Test);
     CPPUNIT_TEST(testCatchThrow);
+#ifndef __aarch64__
     CPPUNIT_TEST(testgetCaughtException);
+#endif
     CPPUNIT_TEST_SUITE_END();
 };
 
@@ -53,6 +57,7 @@ void Test::testCatchThrow()
     }
 }
 
+#ifndef __aarch64__
 void Test::testgetCaughtException()
 {
     css::uno::Any aSavedExceptionAny;
@@ -82,6 +87,7 @@ void Test::testgetCaughtException()
     }
     CPPUNIT_ASSERT(aSavedExceptionAny.hasValue());
 }
+#endif
 
 CPPUNIT_TEST_SUITE_REGISTRATION(Test);
 }
