From fe0e97628499a1485b71c87736029cadf9a7669f Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 16 Dec 2011 02:06:28 +0100
Subject: [PATCH] add debian multiarch support

---
 configure.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/configure.in b/configure.in
index 59cb7fb..904003a 100644
--- a/configure.in
+++ b/configure.in
@@ -7868,6 +7868,7 @@ if test "$test_kde4" = "yes" -a "$ENABLE_KDE4" = "TRUE" ; then
         AC_MSG_ERROR([Qt4 headers not found.  Please specify the root of your Qt4 installation by exporting QT4DIR before running "configure".])
     fi
 
+    qt_libdirs="/usr/lib/`dpkg-architecture -qDEB_HOST_MULTIARCH` $qt_libdirs"
     AC_MSG_CHECKING([for Qt4 libraries])
     qt_lib_dir="no"
     for lib_dir in $qt_libdirs ; do
-- 
1.7.5.4

