diff --git a/configure.ac b/configure.ac
index 0ffa7979dc39..888906096e94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13268,6 +13268,7 @@ if test "x$with_theme" != "xno"; then
     for theme in $with_theme; do
         case $theme in
         breeze|breeze_dark|breeze_dark_svg|breeze_svg|colibre|colibre_svg|colibre_dark|colibre_dark_svg|elementary|elementary_svg|karasa_jaga|karasa_jaga_svg|sifr|sifr_svg|sifr_dark|sifr_dark_svg|sukapura|sukapura_dark|sukapura_dark_svg|sukapura_svg) WITH_THEMES="${WITH_THEMES:+$WITH_THEMES }$theme" ;;
+        yaru*) WITH_THEMES="${WITH_THEMES:+$WITH_THEMES }$theme" ;;
         *) AC_MSG_ERROR([Unknown value for --with-theme: $theme]) ;;
         esac
     done
