--- openoffice.org-3.2.1/ooo-build-3-2-1-4/po/ooo-build-de.po	2010-06-16 11:58:10.000000000 +0200
+++ openoffice.org-3.2.1/ooo-build-3-2-1-4/po/ooo-build-de.po	2010-08-31 17:53:31.000000000 +0200
@@ -783,7 +783,7 @@
 
 #: sc/source/ui/src/optdlg.src#RID_SCPAGE_FORMULA.CB_ENGLISH_FUNC_NAME.checkbox.text
 msgid "Use English function names"
-msgstr ""
+msgstr "Englische Funktionsnamen benutzen"
 
 #: sc/source/ui/src/optdlg.src#RID_SCPAGE_FORMULA.BTN_FORMULA_SEP_RESET.pushbutton.text
 msgid "Rese~t"
--- openoffice.org-3.2.1/ooo-build-3-2-1-4/po/ooo-build-de.sdf	2010-06-16 12:00:24.000000000 +0200
+++ openoffice.org-3.2.1/ooo-build-3-2-1-4/po/ooo-build-de.sdf	2010-08-31 17:59:32.000000000 +0200
@@ -262,6 +262,7 @@
 sc	source\ui\src\optdlg.src	0	checkbox	RID_SCPAGE_CALC	BTN_GENERAL_PREC			999	de	Dezimalstellen für allgemeines Zahlenformat beschränken				2002-02-02 02:02:02
 sc	source\ui\src\optdlg.src	0	checkbox	RID_SCPAGE_CALC	BTN_GENERAL_PREC			999	en-US	Limit decimals for general number format				2002-02-02 02:02:02
 sc	source\ui\src\optdlg.src	0	checkbox	RID_SCPAGE_FORMULA	CB_ENGLISH_FUNC_NAME			999	en-US	Use English function names				2002-02-02 02:02:02
+sc	source\ui\src\optdlg.src	0	checkbox	RID_SCPAGE_FORMULA	CB_ENGLISH_FUNC_NAME			999	de	Englische Funktionsnamen benutzen				2002-02-02 02:02:02
 sc	source\ui\src\optdlg.src	0	pushbutton	RID_SCPAGE_FORMULA	BTN_FORMULA_SEP_RESET			999	de	Zurückse~tzen				2002-02-02 02:02:02
 sc	source\ui\src\optdlg.src	0	pushbutton	RID_SCPAGE_FORMULA	BTN_FORMULA_SEP_RESET			999	en-US	Rese~t				2002-02-02 02:02:02
 sc	source\ui\src\optdlg.src	0	stringlist	RID_SCPAGE_FORMULA.LB_FORMULA_SYNTAX	1			999	de	Calc A1				2002-02-02 02:02:02
