--- a/configure.in	2012-01-13 09:42:37.366836952 +0100
+++ b/configure.in	2012-01-13 10:07:59.389705736 +0100
@@ -8006,13 +8006,13 @@
 AC_MSG_CHECKING([which themes to include])
 # if none given, use all available themes
 if test "x$with_theme" = "x" -o "x$with_theme" = "xyes"; then
-    with_theme="default crystal hicontrast oxygen tango"
+    with_theme="default crystal hicontrast oxygen tango human"
 fi
 
 WITH_THEMES=""
 for theme in $with_theme; do
     case $theme in
-        default|crystal|hicontrast|oxygen|tango) : ;;
+        default|crystal|hicontrast|oxygen|tango|human) : ;;
         *) AC_MSG_ERROR([Unknown value for --with-themes: $theme]) ;;
     esac
     WITH_THEMES="$WITH_THEMES $theme"
