diff --git a/odk/CustomTarget_build-examples.mk b/odk/CustomTarget_build-examples.mk
index 89e427348087..958fe7971ff9 100644
--- a/odk/CustomTarget_build-examples.mk
+++ b/odk/CustomTarget_build-examples.mk
@@ -84,7 +84,9 @@ endif
 #    java/NotesAccess \
 #    java/Storage \
 
-
+ifneq ($(gb_SUPPRESS_TESTS),)
+	@true
+else
 .PHONY: $(call gb_CustomTarget_get_target,odk/build-examples)
 
 $(call gb_CustomTarget_get_target,odk/build-examples): \
@@ -116,5 +118,6 @@ $(call gb_CustomTarget_get_workdir,odk/build-examples)/setsdkenv: \
         -e 's!@OO_SDK_JAVA_HOME@!$(JAVA_HOME)!' \
         -e 's!@OO_SDK_OUTPUT_DIR@!$(call gb_CustomTarget_get_workdir,odk/build-examples)/out!' \
         -e 's!@SDK_AUTO_DEPLOYMENT@!YES!' $< > $@
+endif
 
 # vim: set noet sw=4 ts=4:
