diff --git a/odk/CustomTarget_build-examples.mk b/odk/CustomTarget_build-examples.mk
index 89e427348087..ea9172d8ae84 100644
--- a/odk/CustomTarget_build-examples.mk
+++ b/odk/CustomTarget_build-examples.mk
@@ -84,10 +84,12 @@ endif
 #    java/NotesAccess \
 #    java/Storage \
 
-
 .PHONY: $(call gb_CustomTarget_get_target,odk/build-examples)
 
 $(call gb_CustomTarget_get_target,odk/build-examples): \
+ifneq ($(gb_SUPPRESS_TESTS),)
+	@true
+else
         $(call gb_CustomTarget_get_workdir,odk/build-examples)/setsdkenv
 	$(call gb_Output_announce,$(subst $(WORKDIR)/,,$@),$(true),CHK,1)
 	(saved_library_path=$${$(gb_Helper_LIBRARY_PATH_VAR)} && . $< \
@@ -102,6 +104,7 @@ $(call gb_CustomTarget_get_target,odk/build-examples): \
         || (RET=$$? \
             && cat $(call gb_CustomTarget_get_workdir,odk/build-examples)/log \
             && exit $$RET)
+endif
 
 $(call gb_CustomTarget_get_workdir,odk/build-examples)/setsdkenv: \
         $(SRCDIR)/odk/config/setsdkenv_unix.sh.in \
