diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 9c7ec3c2a3f7..cbed1b69d4a2 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -84,6 +84,8 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   #include <abstractions/ibus>
   #include <abstractions/nameservice>
   #include <abstractions/gnome>
+# GnuPG1 only...
+# #include <abstractions/gnupg>
   #include <abstractions/python>
   #include <abstractions/p11-kit>
 
@@ -124,6 +126,8 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   /usr/bin/lpr                          rmPUx,
   /usr/bin/paperconf                    rmix,
   /usr/bin/gpgconf                      rmix,
+  /usr/bin/gpg                          rmCx -> gpg,
+  /usr/bin/gpgsm                        rmCx -> gpg,
 
   /dev/tty                              rw,
 
@@ -156,4 +156,8 @@
   /usr/share/fonts-config/conf.avail/*.conf r,
 
   owner @{HOME}/.mozilla/firefox/** r,
+  # there is abstractions/gnupg but that's just for gpg1...
+  profile gpg {
+    owner @{HOME}/.gnupg/* r,
+  }
 }
