diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 42053db2abef..3a0051215f0f 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -202,6 +212,10 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   owner @{HOME}/.mozilla/firefox/*/cert8.db r,
   # firefox >= 58
   owner @{HOME}/.mozilla/firefox/*/cert9.db r,
+  
+  # same as above, for thunderbird
+  owner @{HOME}/.thunderbird/profiles.ini r,
+  owner @{HOME}/.thunderbird/*/cert9.db r,
 
   owner @{HOME}/.local/share/user-places.xbel r,
 
