Description: Apparmor profile update (allow JVM execution)
Bug: https://bugs.documentfoundation.org/show_bug.cgi?id=115208
Author: Olivier Tilloy <olivier.tilloy@canonical.com>

--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -130,6 +130,7 @@ profile libreoffice-soffice INSTDIR-prog
 
   /usr/lib{,32,64}/jvm/                         r,
   /usr/lib{,32,64}/jvm/**                       r,
+  /usr/lib{,32,64}/jvm/**/jre/bin/java          mix,
   INSTDIR-**                        rw,
   INSTDIR-**.so                     m,
   INSTDIR-program/soffice.bin       mix,
