diff --git a/sysui/desktop/apparmor/program.oosplash b/sysui/desktop/apparmor/program.oosplash
index dd1136bc9b06..0160f7b389d4 100644
--- a/sysui/desktop/apparmor/program.oosplash
+++ b/sysui/desktop/apparmor/program.oosplash
@@ -33,4 +33,7 @@ profile libreoffice-oosplash INSTDIR-program/oosplash {
   owner @{HOME}/.config/libreoffice{,dev}/?/user/uno_packages/cache/log.txt rw,
   unix peer=(addr=@/tmp/.ICE-unix/* label=unconfined),
   unix peer=(addr=@/tmp/.X11-unix/* label=unconfined),
+
+  # Site-specific additions and overrides. See local/README for details.
+  include if exists <local/usr.lib.libreoffice.program.oosplash>
 }
diff --git a/sysui/desktop/apparmor/program.senddoc b/sysui/desktop/apparmor/program.senddoc
index 969130f4ea90..a3e98d87aa3c 100644
--- a/sysui/desktop/apparmor/program.senddoc
+++ b/sysui/desktop/apparmor/program.senddoc
@@ -33,5 +33,8 @@ profile libreoffice-senddoc INSTDIR-program/senddoc {
   INSTDIR-program/uri-encode rmpux,
   /usr/share/libreoffice/share/config/* r,
   owner @{HOME}/.config/libreoffice{,dev}/?/user/uno_packages/cache/log.txt rw,
+
+  # Site-specific additions and overrides. See local/README for details.
+  include if exists <local/usr.lib.libreoffice.program.senddoc>
 }
 
diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 42053db2abef..c832dc504517 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -260,4 +271,7 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   owner @{HOME}/.config/kdeglobals rw,
   owner @{HOME}/.config/kdeglobals.* rwl -> @{HOME}/.config/#[0-9]*,
   owner @{HOME}/.config/kdeglobals.lock rwk,
+
+  # Site-specific additions and overrides. See local/README for details.
+  include if exists <local/usr.lib.libreoffice.program.soffice.bin>
 }
diff --git a/sysui/desktop/apparmor/program.xpdfimport b/sysui/desktop/apparmor/program.xpdfimport
index f8bfbfe8fa49..15008dd7ebe0 100644
--- a/sysui/desktop/apparmor/program.xpdfimport
+++ b/sysui/desktop/apparmor/program.xpdfimport
@@ -27,5 +27,8 @@ profile libreoffice-xpdfimport INSTDIR-program/xpdfimport {
 
   #Uncomment for build testing (should be one directory <- of instdir)
   #/mnt/store/git/libo/**    r,
+
+  # Site-specific additions and overrides. See local/README for details.
+  include if exists <local/usr.lib.libreoffice.program.xpdfimport>
 }
 
