Index: libreoffice-3.4.2/libreoffice-build/Makefile.am
===================================================================
--- libreoffice-3.4.2.orig/libreoffice-build/Makefile.am	2011-07-18 22:17:06.000000000 +0200
+++ libreoffice-3.4.2/libreoffice-build/Makefile.am	2011-08-31 14:11:45.000000000 +0200
@@ -110,6 +110,7 @@
 	mkdir -p $(BUILDDIR)/man
 	mkdir -p $(BUILDDIR)/man/man1
 	mkdir -p $(OOBUILDDIR)
+	cp --archive "$(BUILDDIR)/../ext-human-updates/ooo_custom_images/" "$(OOBUILDDIR)/"
 	touch $@
 
 build.tools : $(STAMP_DIR)/build.tools
Index: libreoffice-3.4.2/libreoffice-build/Makefile.in
===================================================================
--- libreoffice-3.4.2.orig/libreoffice-build/Makefile.in	2011-08-31 14:20:30.013684993 +0200
+++ libreoffice-3.4.2/libreoffice-build/Makefile.in	2011-08-31 14:25:32.843688252 +0200
@@ -907,6 +907,7 @@
 	mkdir -p $(BUILDDIR)/man
 	mkdir -p $(BUILDDIR)/man/man1
 	mkdir -p $(OOBUILDDIR)
+	cp --archive "$(BUILDDIR)/../../ext-human-updates/ooo_custom_images/" "$(OOBUILDDIR)/"
 	touch $@
 
 build.tools : $(STAMP_DIR)/build.tools
