From 33ef1ffbd15994ec71be99d38c0d5171c63344a2 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 06 Mar 2012 17:16:35 +0000
Subject: lp#527938, debian#602953, fdo#33266, i#105408: do not crash on clicking bibliography when base isnt installed

Signed-off-by: Michael Meeks <michael.meeks@suse.com>

From 285b3a4fe0b5b21abe061dfb664800b7e537471e Mon Sep 17 00:00:00 2001
From: Caolán McNamara <caolanm@redhat.com>
Date: Fri, 09 Mar 2012 14:09:30 +0000
Subject: Fix tools->bibliography database, brown paper bag over head commit :-)

(cherry picked from commit bd1088567bafa6293ffecf5331a9016b6d0438e3)

Signed-off-by: Miklos Vajna <vmiklos@suse.cz>
---
diff --git a/extensions/source/bibliography/bibload.cxx b/extensions/source/bibliography/bibload.cxx
index c80c81e..9159124 100644
--- a/extensions/source/bibliography/bibload.cxx
+++ b/extensions/source/bibliography/bibload.cxx
@@ -54,6 +54,7 @@
 #include <com/sun/star/text/BibliographyDataField.hpp>
 #include <com/sun/star/form/XLoadListener.hpp>
 #include <com/sun/star/frame/XLayoutManager.hpp>
+#include <com/sun/star/uno/XAggregation.hpp>
 #include <toolkit/awt/vclxwindow.hxx>
 #include <vcl/window.hxx>
 #include <vcl/edit.hxx>
@@ -243,13 +244,27 @@ void BibliographyLoader::cancel(void) throw (::com::sun::star::uno::RuntimeExcep
 }
 
 // -----------------------------------------------------------------------
+namespace
+{
+    // lp#527938, debian#602953, fdo#33266, i#105408
+    static bool lcl_isBaseAvailable()
+    {
+        Reference< XMultiServiceFactory >  xMgr = comphelper::getProcessServiceFactory();
+        Reference< XAggregation > xAggregate = Reference< XAggregation >( xMgr->createInstance(C2U("com.sun.star.sdb.RowSet")), UNO_QUERY);
+        return xAggregate.is();
+    }
+}
 void BibliographyLoader::load(const Reference< XFrame > & rFrame, const rtl::OUString& rURL,
         const Sequence< PropertyValue >& rArgs,
         const Reference< XLoadEventListener > & rListener) throw (::com::sun::star::uno::RuntimeException)
 {
-    //!
+    // lp#527938, debian#602953, fdo#33266, i#105408
+    // make sure we actually can instanciate services from base first
+    if(!lcl_isBaseAvailable())
+        return;
 
     SolarMutexGuard aGuard;
+    
     m_pBibMod = OpenBibModul();
 
     String aURLStr( rURL );
--
cgit v0.9.0.2-2-gbebe
