--- libreoffice-3.5.0/scripting/java/build.xml	2011-06-28 11:11:12.000000000 +0000
+++ libreoffice-3.5.0/scripting/java/build.xml-old	2011-06-28 11:10:53.000000000 +0000
@@ -52,7 +52,7 @@
     <pathelement location="${solar.jar}/juh.jar"/>
     <pathelement location="${solar.jar}/java_uno.jar"/>
     <pathelement location="${solar.jar}/ridl.jar"/>
-    <pathelement location="${env.BSH_JAR}"/>
+    <pathelement location="/usr/share/java/bsh.jar"/>
     <pathelement location="${jardir}"/>
   </path> 
 
