Index: libreoffice-3.4.1/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.1.orig/libreoffice-build/patches/dev300/apply	2011-07-15 11:39:45.556848313 +0200
+++ libreoffice-3.4.1/libreoffice-build/patches/dev300/apply	2011-07-15 11:40:25.496848861 +0200
@@ -354,7 +354,7 @@
 #FIXME src680: separate-helpcontent.diff, jholesov
 
 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek
 
 [ DebianOnly ]
 seamonkey-is-iceape.diff
