Index: libreoffice-3.4.2/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.2.orig/libreoffice-build/patches/dev300/apply	2011-08-19 11:40:48.779438099 +0200
+++ libreoffice-3.4.2/libreoffice-build/patches/dev300/apply	2011-08-19 11:40:58.649438023 +0200
@@ -356,7 +356,7 @@
 #FIXME src680: separate-helpcontent.diff, jholesov
 
 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek
 
 [ DebianOnly ]
 seamonkey-is-iceape.diff
