--- libreoffice-3.3.0/libreoffice-build/patches/dev300/apply-old	2010-10-18 09:11:18.000000000 +0200
+++ libreoffice-3.3.0/libreoffice-build/patches/dev300/apply	2010-10-18 09:11:30.000000000 +0200
@@ -1878,7 +1878,7 @@
 [ Fixes ]
 fix-ppt-linespacing-import-export.diff, n#355302, rodo
 
-fields-table-formula.diff, n#631912, cbosdo
+#fields-table-formula.diff, n#631912, cbosdo
 
 # Backport from cbosdo06 CWS
 sw-table-frame-uno-fix.diff, i#112564, cbosdo
