Index: libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.0.orig/libreoffice-build/patches/dev300/apply	2011-06-28 16:21:40.672707032 +0200
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply	2011-06-28 16:21:49.912707031 +0200
@@ -148,7 +148,7 @@
 # don't lstat() that much while creating absolute URLs - breaks unopkg
 # run as e.g. used during Mac dmg generation, as parts of the
 # configmgr paths are not yet existing then
-speed-sal-fewer-lstats.diff, i#89730, jholesov
+#speed-sal-fewer-lstats.diff, i#89730, jholesov
 
 #[ CJKTemporaryHacks ]
 # the following patch is disabled temporary.
@@ -361,7 +361,7 @@
 #FIXME src680: separate-helpcontent.diff, jholesov
 
 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek
 
 [ DebianOnly ]
 seamonkey-is-iceape.diff
