diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index e75444f..ec22d92 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -129,7 +129,7 @@ gb_Module_DEVINSTALLHINT=
 debugrun :| build
 endif
 
-all : build unitcheck
+all : build
 	$(info $(gb_Module_BUILDHINT))
 
 build : 
diff --git a/solenv/bin/build.pl b/solenv/bin/build.pl
index 30f7c7b..cec5821 100755
--- a/solenv/bin/build.pl
+++ b/solenv/bin/build.pl
@@ -1837,7 +1837,7 @@ sub run_job {
             return 1 if (! $path =~ /prj$/ );
             mkpath("$workdir/Logs");
             my $gbuild_flags = '-j' . $ENV{GMAKE_MODULE_PARALLELISM};
-            my $gbuild_target = 'all slowcheck';
+            my $gbuild_target = 'all';
             if ($registered_name =~ /tail_build\/prj$/ )
             {
                 $gbuild_flags = '-j' . $ENV{GMAKE_PARALLELISM};
--- a/Makefile.top-old	2012-06-08 20:04:36.405997882 +0200
+++ b/Makefile.top	2012-06-08 20:04:51.626073355 +0200
@@ -244,7 +244,7 @@
 zlib\
 
 gb_TAILBUILDMODULES := $(shell $(GNUMAKE) -r -f $(SRCDIR)/tail_build/Makefile showmodules)
-export gb_TAILBUILDTARGET=all slowcheck
+export gb_TAILBUILDTARGET=all
 
 define gbuild_module_rules
 .PHONY: $(1) $(1).all $(1).clean $(1).deliver
@@ -481,7 +481,7 @@
               | grep -v WPX \
               > unusedcode.easy
 
-check: dev-install subsequentcheck
+check: unitcheck slowcheck dev-install subsequentcheck
 
 subsequentcheck :| $(if $(filter-out subsequentcheck,$(MAKECMDGOALS)),dev-install)
 	$(GNUMAKE) -j $(GMAKE_PARALLELISM) $(GMAKE_OPTIONS) -f Makefile.post $@
