From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Wed Mar 12 17:49:28 CET 2014
Subject: [PATCH] make ConnectivityTools a normal target, so we can pack it

---
diff --git a/connectivity/Module_connectivity.mk b/connectivity/Module_connectivity.mk
index b49a9ad..9a32501 100644
--- a/connectivity/Module_connectivity.mk
+++ b/connectivity/Module_connectivity.mk
@@ -150,14 +150,14 @@ $(eval $(call gb_Module_add_check_targets,connectivity,\
 endif
 endif
 
 ifneq ($(filter QADEVOOO,$(BUILD_TYPE)),)
-$(eval $(call gb_Module_add_subsequentcheck_targets,connectivity,\
+$(eval $(call gb_Module_add_targets,connectivity,\
 	Jar_ConnectivityTools \
 ))
 # FIXME: Does not work. Convert to JUnit.
 # JunitTest_complex \
 
 endif
 
 # general tests
 $(eval $(call gb_Module_add_check_targets,connectivity,\
