diff --git a/patches/dev300/apply b/patches/dev300/apply
index dd51188..bc5cf54 100644
--- libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
@@ -28,7 +28,7 @@
 Experimental: VBAUntested, ArkOnlyExperimental, \
 	      UnUsedButNotYetRemovedFromGit, WebDAV, \
 	      PostgreSQL, RadioButtons, WWInProgress, \
-	      KDE4Experimental, MinGW, NetBook, \
+	      KDE4Experimental, MinGW, Netbook, \
 	      CrossWin32Patches, AutoLayout, \
 	      Speed, GSoC2010, Toolbars, Win32Packaging
 DebianLooseSections: DebianBaseNoHelpContent
@@ -148,7 +148,7 @@
 # don't lstat() that much while creating absolute URLs - breaks unopkg
 # run as e.g. used during Mac dmg generation, as parts of the
 # configmgr paths are not yet existing then
-speed-sal-fewer-lstats.diff, i#89730, jholesov
+#speed-sal-fewer-lstats.diff, i#89730, jholesov
 
 #[ CJKTemporaryHacks ]
 # the following patch is disabled temporary.
@@ -361,7 +361,7 @@ open-url-support-iceanimals.diff
 #FIXME src680: separate-helpcontent.diff, jholesov

 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek

 [ DebianOnly ]
 seamonkey-is-iceape.diff
