diff --git a/patches/dev300/apply b/patches/dev300/apply
index dd51188..bc5cf54 100644
--- libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
@@ -361,7 +361,7 @@ open-url-support-iceanimals.diff
 #FIXME src680: separate-helpcontent.diff, jholesov

 # use -Wl,--as-needed, save 1% of relocations
-link-as-needed.diff, i#89511, pmladek
+#link-as-needed.diff, i#89511, pmladek

 [ DebianOnly ]
 seamonkey-is-iceape.diff
