diff --git a/Makefile.in b/Makefile.in
index 197eb8e..ae0f09b 100644
--- libreoffice-3.5.0/Makefile.in
+++ libreoffice-3.5.0/Makefile.in
@@ -57,7 +57,7 @@ check : allcheck
 	@true
 
 %check:
-	@. ./Env.Host.sh && $$GNUMAKE -f "$$SRC_ROOT"/GNUmakefile.mk -r \
+	@. ./Env.Host.sh && export WITH_LANG=en-US && $$GNUMAKE -f "$$SRC_ROOT"/GNUmakefile.mk -r \
 			$(if @VERBOSE@,,-s) --jobs="$(if \
 			   $(CHECK_PARALLELISM),$(CHECK_PARALLELISM),@GMAKE_PARALLELISM@)" \
 			$(patsubst allcheck,check,$@)
