diff --git a/configure.ac b/configure.ac
index 8fa43fd0c31e..2d5491522cca 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10355,6 +10363,27 @@ elif test "$_os" = "Linux" -o "$_os" = "Darwin" -o "$_os" = "WINNT" ; then
         # gpg installation to run OpenPGP signature verification
         AC_DEFINE([HAVE_FEATURE_GPGVERIFY])
     fi
+    if test "$_os" = "Linux"; then
+      uid=`id -u`
+      AC_MSG_CHECKING([for /run/user/$uid])
+      if test -d /run/user/$uid; then
+        AC_MSG_RESULT([yes])
+        AC_PATH_PROG(GPGCONF, gpgconf)
+        AC_MSG_CHECKING([for gpgconf --create-socketdir... ])
+        if $GPGCONF --dump-options > /dev/null ; then
+          if $GPGCONF --dump-options | grep -q create-socketdir ; then
+	    AC_MSG_RESULT([yes])
+            AC_DEFINE([HAVE_GPGCONF_SOCKETDIR])
+          else
+            AC_MSG_RESULT([no])
+          fi
+        else
+	  AC_MSG_RESULT([no])
+        fi
+      else
+        AC_MSG_RESULT([no. missing or broken gpgconf?])
+     fi
+   fi
 fi
 AC_SUBST(ENABLE_GPGMEPP)
 AC_SUBST(SYSTEM_GPGMEPP)
@@ -10364,6 +10392,8 @@ AC_SUBST(LIBASSUAN_CFLAGS)
 AC_SUBST(LIBASSUAN_LIBS)
 AC_SUBST(GPGMEPP_CFLAGS)
 AC_SUBST(GPGMEPP_LIBS)
+AC_SUBST(GPGCONF)
+AC_SUBST(HAVE_GPGCONF_SOCKETDIR)
 
 AC_MSG_CHECKING([whether to build the Wiki Publisher extension])
 if test "x$enable_ext_wiki_publisher" = "xyes" -a "x$enable_extension_integration" != "xno" -a "$with_java" != "no"; then
diff --git a/xmlsecurity/qa/unit/signing/signing.cxx b/xmlsecurity/qa/unit/signing/signing.cxx
index 6efb6f02e029..04ab762d646f 100644
--- a/xmlsecurity/qa/unit/signing/signing.cxx
+++ b/xmlsecurity/qa/unit/signing/signing.cxx
@@ -107,8 +107,10 @@ public:
     /// Test a typical broken ODF signature where the XML dsig hash is corrupted.
     void testODFBrokenDsigGPG();
+#if HAVE_GPGCONF_SOCKETDIR
     /// Test loading an encrypted ODF document
     void testODFEncryptedGPG();
 #endif
+#endif
     CPPUNIT_TEST_SUITE(SigningTest);
     CPPUNIT_TEST(testDescription);
     CPPUNIT_TEST(testODFGood);
@@ -136,8 +138,10 @@ public:
     CPPUNIT_TEST(testODFUntrustedGoodGPG);
     CPPUNIT_TEST(testODFBrokenStreamGPG);
     CPPUNIT_TEST(testODFBrokenDsigGPG);
+#if HAVE_GPGCONF_SOCKETDIR
     CPPUNIT_TEST(testODFEncryptedGPG);
 #endif
+#endif
     CPPUNIT_TEST_SUITE_END();
 
 private:
@@ -736,6 +740,8 @@ void SigningTest::testODFBrokenDsigGPG()
     CPPUNIT_ASSERT_EQUAL(static_cast<int>(SignatureState::BROKEN), static_cast<int>(pObjectShell->GetDocumentSignatureState()));
 }
 
+#if HAVE_GPGCONF_SOCKETDIR
+
 void SigningTest::testODFEncryptedGPG()
 {
     createDoc(m_directories.getURLFromSrc(DATA_DIRECTORY) + "encryptedGPG.odt");
@@ -747,6 +753,8 @@ void SigningTest::testODFEncryptedGPG()
 
 #endif
 
+#endif
+
 void SigningTest::registerNamespaces(xmlXPathContextPtr& pXmlXpathCtx)
 {
     xmlXPathRegisterNs(pXmlXpathCtx, BAD_CAST("odfds"), BAD_CAST("urn:oasis:names:tc:opendocument:xmlns:digitalsignature:1.0"));
--- a/xmlsecurity/CppunitTest_xmlsecurity_signing.mk	2018-03-18 09:16:48.591850273 +0000
+++ b/xmlsecurity/CppunitTest_xmlsecurity_signing.mk	2018-03-18 09:46:15.243745953 +0000
@@ -59,14 +59,16 @@
     EXTRA_ENV_VARS := \
         LIBO_LD_PATH=$$LD_LIBRARY_PATH
 
+ifneq (,$(HAVE_GPGCONF_SOCKETDIR))
 # create socket dir below /run/user/ instead of in workdir
 .PHONY : gb_CppunitTest_run_gpgconf
 gb_CppunitTest_run_gpgconf:
 	GNUPGHOME=$(WORKDIR)/CppunitTest/xmlsecurity_signing.test.user \
-	gpgconf --create-socketdir 2>/dev/null || true
+	$(GPGCONF) --create-socketdir
 
 $(call gb_CppunitTest_get_target,xmlsecurity_signing): \
     gb_CppunitTest_run_gpgconf
 endif
+endif
 
 # vim: set noet sw=4 ts=4:
--- a/config_host/config_gpgme.h.in	2017-12-09 18:39:03.895484884 +0100
+++ b/config_host/config_gpgme.h.in	2018-03-18 11:05:47.132093590 +0100
@@ -19,6 +19,9 @@
 // Defined if gpg and gpgme signature verification is available
 #define HAVE_FEATURE_GPGVERIFY 0
 
+// Defined if gpgconf --create-socketdir works
+#define HAVE_GPGCONF_SOCKETDIR 0
+
 #if HAVE_FEATURE_GPGME
 # include "config_lgpl.h"
 #endif
diff --git a/config_host.mk.in b/config_host.mk.in
index 8ce71275ca5c..cff1500b3dd2 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -203,6 +203,7 @@ export EPOXY_LIBS=$(gb_SPACE)@EPOXY_LIBS@
 export GLM_CFLAGS=$(gb_SPACE)@GLM_CFLAGS@
 export GPG_ERROR_CFLAGS=$(gb_SPACE)@GPG_ERROR_CFLAGS@
 export GPG_ERROR_LIBS=$(gb_SPACE)@GPG_ERROR_LIBS@
+export GPGCONF=@GPGCONF@
 export GPGMEPP_CFLAGS=$(gb_SPACE)@GPGMEPP_CFLAGS@
 export GPGMEPP_LIBS=$(gb_SPACE)@GPGMEPP_LIBS@
 export GNUTLS_CFLAGS=$(gb_SPACE)@GNUTLS_CFLAGS@
@@ -247,6 +249,7 @@ export HAVE_GCC_FNO_INLINE=@HAVE_GCC_FNO_INLINE@
 export HAVE_GCC_FNO_SIZED_DEALLOCATION=@HAVE_GCC_FNO_SIZED_DEALLOCATION@
 export HAVE_GCC_GGDB2=@HAVE_GCC_GGDB2@
 export HAVE_GNUMAKE_FILE_FUNC=@HAVE_GNUMAKE_FILE_FUNC@
+export HAVE_GPGCONF_SOCKETDIR=@HAVE_GPGCONF_SOCKETDIR@
 export HAVE_LD_BSYMBOLIC_FUNCTIONS=@HAVE_LD_BSYMBOLIC_FUNCTIONS@
 export HAVE_LD_HASH_STYLE=@HAVE_LD_HASH_STYLE@
 export HAVE_POSIX_FALLOCATE=@HAVE_POSIX_FALLOCATE@
