--- libreoffice-5.4.0.0.beta1/vcl/qa/cppunit/complextext.cxx	2017-05-18 12:48:14.000000000 +0200
+++ tmp/libreoffice-5.4.0.0.beta1/vcl/qa/cppunit/complextext.cxx	2017-05-22 15:23:13.668459293 +0200
@@ -19,6 +19,8 @@
 #include <osl/file.hxx>
 #include <osl/process.h>
 
+#include <config_test.h>
+
 std::ostream& operator<<(std::ostream& rStream, const std::vector<long>& rVec)
 {
     rStream << "{ ";
@@ -41,13 +43,16 @@
 #endif
 
     CPPUNIT_TEST_SUITE(VclComplexTextTest);
+#if !TEST_FONTS_MISSING
     CPPUNIT_TEST(testArabic);
+#endif
 #if defined(_WIN32)
     CPPUNIT_TEST(testTdf95650);
 #endif
     CPPUNIT_TEST_SUITE_END();
 };
 
+#if !TEST_FONTS_MISSING
 void VclComplexTextTest::testArabic()
 {
     const unsigned char pOneTwoThreeUTF8[] = {
@@ -121,6 +126,7 @@
     (void)aRect; (void)aRectRot;
 #endif
 }
+#endif
 
 #if defined(_WIN32)
 void VclComplexTextTest::testTdf95650()
--- libreoffice-5.4.0.0.beta1/configure.ac	2017-05-18 12:48:14.000000000 +0200
+++ tmp/libreoffice-5.4.0.0.beta1/configure.ac	2017-05-22 15:51:26.406507002 +0200
@@ -11290,7 +11289,8 @@
     then
         AC_MSG_RESULT([ok])
     else
-        AC_MSG_RESULT([unknown ($FONTFILE)])
+        AC_MSG_WARN([unknown ($FONTFILE)])
+        add_warning "unknown ($FONTFILE)"
         TEST_FONTS_MISSING=1
     fi
 }
@@ -11315,9 +11315,11 @@
             TEST_FONTS_MISSING=1
         else
             test_font_map 'Calibri' 'Carlito'
+            test_font_map 'DejaVuSans' 'DejaVuSans'
             if test ${TEST_FONTS_MISSING} -eq 1
             then
                 AC_MSG_WARN([Unknown font mappings - unit tests disabled.])
+                add_warning "Unknown font mappings - unit tests disabled."
             fi
         fi
     else
