Index: libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.0.orig/libreoffice-build/patches/dev300/apply	2011-06-29 22:44:40.491663979 +0200
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply	2011-06-29 22:46:49.361663005 +0200
@@ -54,13 +54,14 @@
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
-UbuntuBased : DebianBase, UbuntuOnly
-UbuntuHardy : UbuntuBased, UbuntuHardyOnly
-UbuntuJaunty : UbuntuBased, UbuntuJauntyOnly
-UbuntuKarmic : UbuntuBased, UbuntuKarmicOnly
-UbuntuLucid : UbuntuBased, UbuntuLucidOnly
-UbuntuMaverick : UbuntuBased, UbuntuMaverickOnly
-UbuntuNatty : UbuntuBased, UbuntuNattyOnly
+UbuntuBased : DebianBase
+UbuntuHardy : UbuntuBased
+UbuntuJaunty : UbuntuBased
+UbuntuKarmic : UbuntuBased
+UbuntuLucid : UbuntuBased
+UbuntuMaverick : UbuntuBased
+UbuntuNatty : UbuntuBased
+UbuntuOneiric : UbuntuBased
 UbuntuL10n : UbuntuBased, UbuntuL10nOnly
 # DroplineGNOME
 DroplineGNOME : LinuxCommon, Mono, msaccess
@@ -1026,23 +1027,6 @@
 # use -llpsolve55_pic
 system-lpsolve-use-_pic.diff, rengelha
 
-[ UbuntuOnly ]
-SectionOwner => ccheney
-
-ubuntu-palette.diff
-ubuntu-sparc-hack.diff, mklose
-ubuntu-no-stack-protector.diff, doko
-#gccXXXXX.solenv.javaregistration.diff
-ubuntu-mstopdf.diff, ccheney
-# FIXME: update for LO
-#f11-fullscreen.diff, ccheney
-
-#[ UbuntuOnly ]
-#unxlngi6-notune.diff, doko
-
-[ UbuntuL10nOnly ]
-human-icons-i18n.diff, mklose
-
 [ DebianBaseOnly ]
 soffice-opengl-and-locking-config.diff, mklose
 
@@ -1198,40 +1182,6 @@
 [ Fixes ]
 external-apm-header.diff, freuter
 
-[ UbuntuHardyOnly ]
-# Add patch to only show local files needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-local-only.diff
-
-[ UbuntuJauntyOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-
-[ UbuntuKarmicOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-
-[ UbuntuLucidOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-ubuntu-arm-thumb.diff, doko
-
-[ UbuntuMaverickOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-ubuntu-arm-thumb.diff, doko
-
-[ UbuntuNattyOnly ]
-# Add patch to force use of gvfs fuse path needed when gnome-vfs/gio is disabled
-# FIXME: update for LO
-#ubuntu-gnome-fpicker-gfile-fuse.diff, ccheney
-# FIXME: still needed for GCC-4.5?
-#ubuntu-arm-thumb.diff, doko
-
 [ Experimental ]
 experimental_ooapi.diff, flr
 
@@ -1265,9 +1215,6 @@
 [ DebianSidOnly ]
 # link dynamically with liblpsolve55.so
 system-lpsolve-rpath.diff, rengelha
-[ UbuntuOnly ]
-# link dynamically with liblpsolve55.so
-system-lpsolve-rpath.diff, rengelha
 
 [ DebianBaseOnly ]
 # link with -lcolamd, needed because our liblpsolve55{,_pic}.a doesn't include it
