From 0796b3041231063c3594e85a8f2ea2b15602d46d Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@collabora.co.uk>
Date: Tue, 21 Oct 2014 16:27:42 +0200
Subject: dbaccess: make sure fdo84315.odb.lck is removed after the test passed

Change-Id: I6c9a42625f10a3a30e2636a2a2be2b9bcba8f6af

diff --git a/dbaccess/qa/python/fdo84315.py b/dbaccess/qa/python/fdo84315.py
index 0670f66..bf13e1b 100644
--- a/dbaccess/qa/python/fdo84315.py
+++ b/dbaccess/qa/python/fdo84315.py
@@ -66,6 +66,7 @@ class Fdo84315(unittest.TestCase):
         while xResultset.next():
             self.assertEqual(xResultset.getInt(1), expected_values.popleft())
         self.assertEqual(len(expected_values), 0)
+        xCon.dispose()
 
 if __name__ == '__main__':
     unittest.main()
-- 
cgit v0.10.2

