--- a/configure.in-old	2012-08-13 13:19:05.592297194 +0200
+++ b/configure.in	2012-08-13 13:20:34.224736590 +0200
@@ -6189,6 +6189,8 @@
     AC_MSG_RESULT([external])
     SYSTEM_CLUCENE=YES
     PKG_CHECK_MODULES(CLUCENE, libclucene-core)
+    # hack for include in multiarch path...
+    CLUCENE_CFLAGS="$CLUCENE_CFLAGS -I/usr/lib/`dpkg-architecture -qDEB_HOST_MULTIARCH`"
 
     AC_LANG_PUSH([C++])
     save_CXXFLAGS=$CXXFLAGS
