From 410b6f01f07f8f1a84f2118c71735cdbb5331d5f Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Mon, 20 Aug 2012 08:01:47 +0000
Subject: concat-deps.c: add || defined (__FreeBSD_kernel__) for arch-specifics

Change-Id: Ie321ada6d2c078881fdedc61d72c94034297fea3
---
diff --git a/solenv/bin/concat-deps.c b/solenv/bin/concat-deps.c
index 3770755..65fda1d 100644
--- a/solenv/bin/concat-deps.c
+++ b/solenv/bin/concat-deps.c
@@ -39,7 +39,7 @@
 
 #if defined(__linux) || defined(__OpenBSD__) || \
     defined(__FreeBSD__) || defined(__NetBSD__) || \
-    defined(__DragonFly__)
+    defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #if __BYTE_ORDER == __LITTLE_ENDIAN
 #define CORE_BIG_ENDIAN 0
 #define CORE_LITTLE_ENDIAN 1
--
cgit v0.9.0.2-2-gbebe
