diff --git a/RepositoryModule_ooo.mk b/RepositoryModule_ooo.mk
index f59b2e0..5f7fa6f 100644
--- a/RepositoryModule_ooo.mk
+++ b/RepositoryModule_ooo.mk
@@ -67,7 +67,6 @@ $(eval $(call gb_Module_add_moduledirs,ooo,\
     package \
     padmin \
     psprint_config \
-    qadevOOo \
     regexp \
     reportdesign \
     sane \
@@ -108,6 +107,14 @@ $(eval $(call gb_Module_add_moduledirs,ooo,\
     xmlscript \
 ))
 
+ifneq ($(SOLAR_JAVA),)
+ ifneq "$(OOO_JUNIT_JAR)" ""
+$(eval $(call gb_Module_add_moduledirs,ooo,\
+    qadevOOo \
+))
+  endif
+endif
+
 # these have only migrated subsequentcheck for now
 $(eval $(call gb_Module_add_moduledirs,ooo,\
     configmgr \
diff --git a/configure.in b/configure.in
index 0a39ef4..290b294 100644
--- a/configure.in
+++ b/configure.in
@@ -9137,6 +9137,9 @@ provide a full junit jar or use --without-junit])
 location (/usr/share/java), specify its pathname via
 --with-junit=..., or disable it via --without-junit])
     fi
+    if test $OOO_JUNIT_JAR != ""; then
+	BUILD_TYPE="$BUILD_TYPE QADEVOOO"
+    fi
 fi
 AC_SUBST(OOO_JUNIT_JAR)
 
diff --git a/connectivity/prj/build.lst b/connectivity/prj/build.lst
index fd358fa..f4851c5 100644
--- a/connectivity/prj/build.lst
+++ b/connectivity/prj/build.lst
@@ -1,4 +1,4 @@
-cn  connectivity    :    shell  TRANSLATIONS:translations comphelper MOZ:moz POSTGRESQL:postgresql svl UNIXODBC:unixODBC unoil javaunohelper HSQLDB:hsqldb qadevOOo officecfg NSS:nss LIBXSLT:libxslt NULL
+cn  connectivity    :    shell  TRANSLATIONS:translations comphelper MOZ:moz POSTGRESQL:postgresql svl UNIXODBC:unixODBC unoil javaunohelper HSQLDB:hsqldb QADEVOOO:qadevOOo officecfg NSS:nss LIBXSLT:libxslt NULL
 cn  connectivity                                    usr1    -   all cn_mkout NULL
 cn  connectivity\inc                                nmake   -   all cn_inc NULL
 cn  connectivity\com\sun\star\sdbcx\comp\hsqldb     nmake   -   all cn_jhsqldbdb cn_hsqldb cn_inc NULL
diff --git a/dbaccess/prj/build.lst b/dbaccess/prj/build.lst
index 45372a7..ce429ec 100644
--- a/dbaccess/prj/build.lst
+++ b/dbaccess/prj/build.lst
@@ -1,3 +1,3 @@
-ba      dbaccess        :       TRANSLATIONS:translations BOOST:boost connectivity svx stoc qadevOOo xmlscript LIBXSLT:libxslt test NULL
+ba      dbaccess        :       TRANSLATIONS:translations BOOST:boost connectivity svx stoc QADEVOOO:qadevOOo xmlscript LIBXSLT:libxslt test NULL
 ba	dbaccess								usr1	-	all	ba_mkout NULL
 ba	dbaccess\prj							nmake	-	all	ba_prj NULL
diff --git a/forms/prj/build.lst b/forms/prj/build.lst
index 819a57c..131e8e4 100644
--- a/forms/prj/build.lst
+++ b/forms/prj/build.lst
@@ -1,4 +1,4 @@
-fm	forms	:    TRANSLATIONS:translations oovbaapi svx sfx2 qadevOOo LIBXSLT:libxslt NULL
+fm	forms	:    TRANSLATIONS:translations oovbaapi svx sfx2 QADEVOOO:qadevOOo LIBXSLT:libxslt NULL
 fm  forms                                   usr1    -   all fm_mkofrm NULL
 fm  forms\prj				    nmake   -	all fm_prj NULL
 
diff --git a/tail_build/prj/build.lst b/tail_build/prj/build.lst
index 72e90b4..394ac7d 100644
--- a/tail_build/prj/build.lst
+++ b/tail_build/prj/build.lst
@@ -1,2 +1,2 @@
-tb      tail_build : APACHE_COMMONS:apache-commons BSH:beanshell BERKELEYDB:berkeleydb BOOST:boost DESKTOP:l10ntools LIBXML2:libxml2 LIBXMLSEC:libxmlsec LIBXSLT:libxslt LIBWPG:libwpg LIBWPS:libwps LIBWPD:libwpd LIBVISIO:libvisio LPSOLVE:lpsolve MDDS:mdds NEON:neon NSS:nss OPENSSL:openssl PYTHON:python REDLAND:redland SAXON:saxon TRANSLATIONS:translations XPDF:xpdf avmedia basegfx basic bridges canvas comphelper configmgr connectivity cppcanvas cppu cppuhelper DESKTOP:fpicker javaunohelper jurt jvmaccess offapi officecfg oovbaapi qadevOOo DESKTOP:rdbmaker readlicense_oo RHINO:rhino ridljar sal sax sfx2 solenv soltools sot stoc svtools svx sysui test toolkit tools ucbhelper udkapi unoil unotools ure vcl xmlhelp xmloff xmlscript xsltml NULL
+tb      tail_build : APACHE_COMMONS:apache-commons BSH:beanshell BERKELEYDB:berkeleydb BOOST:boost DESKTOP:l10ntools LIBXML2:libxml2 LIBXMLSEC:libxmlsec LIBXSLT:libxslt LIBWPG:libwpg LIBWPS:libwps LIBWPD:libwpd LIBVISIO:libvisio LPSOLVE:lpsolve MDDS:mdds NEON:neon NSS:nss OPENSSL:openssl PYTHON:python REDLAND:redland SAXON:saxon TRANSLATIONS:translations XPDF:xpdf avmedia basegfx basic bridges canvas comphelper configmgr connectivity cppcanvas cppu cppuhelper DESKTOP:fpicker javaunohelper jurt jvmaccess offapi officecfg oovbaapi QADEVOOO:qadevOOo DESKTOP:rdbmaker readlicense_oo RHINO:rhino ridljar sal sax sfx2 solenv soltools sot stoc svtools svx sysui test toolkit tools ucbhelper udkapi unoil unotools ure vcl xmlhelp xmloff xmlscript xsltml NULL
 tb tail_build\prj nmake - all tb_prj   NULL
diff --git a/testgraphical/prj/build.lst b/testgraphical/prj/build.lst
index 75878a6..d7b1a92 100644
--- a/testgraphical/prj/build.lst
+++ b/testgraphical/prj/build.lst
@@ -1,4 +1,4 @@
-gfxcmp    testgraphical     :       instsetoo_native qadevOOo NULL
+gfxcmp    testgraphical     :       instsetoo_native QADEVOOO:qadevOOo NULL
 gfxcmp    testgraphical                             usr1     -  all    gfxcmp_mkout NULL
 #gfxcmp    testgraphical\prechecks                   nmake    -  all    gfxcmp_pre NULL
 gfxcmp    testgraphical\ui\java\ConvwatchGUIProject nmake    -  all    gfxcmp_java_ui NULL
diff --git a/toolkit/prj/build.lst b/toolkit/prj/build.lst
index a0ddbff..e8a7b48 100644
--- a/toolkit/prj/build.lst
+++ b/toolkit/prj/build.lst
@@ -1,2 +1,2 @@
-ti toolkit : LIBXSLT:libxslt vcl qadevOOo test NULL
+ti toolkit : LIBXSLT:libxslt vcl QADEVOOO:qadevOOo test NULL
 ti toolkit\prj					nmake - all ti_prj NULL
diff --git a/unotest/prj/build.lst b/unotest/prj/build.lst
index 1cbe6e3..0a60de3 100644
--- a/unotest/prj/build.lst
+++ b/unotest/prj/build.lst
@@ -1,4 +1,4 @@
-unote unotest : BOOST:boost cppu cppuhelper comphelper CPPUNIT:cppunit javaunohelper offapi ridljar sal solenv unoil qadevOOo NULL
+unote unotest : BOOST:boost cppu cppuhelper comphelper CPPUNIT:cppunit javaunohelper offapi ridljar sal solenv unoil QADEVOOO:qadevOOo NULL
 unote unotest\source\cpp nmake - all source_cpp NULL
 unote unotest\source\cpp\unoexceptionprotector nmake - all source_cpp_unoexceptionprotector NULL
 unote unotest\source\java\org\openoffice\test nmake - all source_java NULL
