From 3d2333d362a7f02eef4c0d826ae17581e84ec479 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Tue, 17 Apr 2012 15:46:03 +0000
Subject: add 5.3 into the db check for-loop

Signed-off-by: Petr Mladek <pmladek@suse.cz>
---
diff --git a/configure.in b/configure.in
index ea9dee1..d01e82f 100644
--- a/configure.in
+++ b/configure.in
@@ -5110,7 +5110,7 @@ if test "$with_system_db" = "yes"; then
     AC_MSG_RESULT([external])
 
     db_header=
-    for dbver in 5.1 5.0 5 4.8 4.7 4; do
+    for dbver in 5.3 5.1 5.0 5 4.8 4.7 4; do
         for dash in - ''; do
             AC_CHECK_HEADER([db$dash$dbver/db.h],
                 [ db_header="db$dash$dbver/db.h"; break 2 ])
--
cgit v0.9.0.2-2-gbebe
