diff --git a/configure.ac b/configure.in
index c0b76f8..d10222f 100644
--- libreoffice-3.5.0/configure.ac
+++ libreoffice-3.5.0/configure.ac
@@ -12091,6 +12108,9 @@ PRODUCTVERSION=AC_PACKAGE_VERSION
 AC_MSG_RESULT([$PRODUCTVERSION])
 AC_SUBST(PRODUCTVERSION)
 
+CDEFAULTOPT=`echo $CFLAGS 2>/dev/null | grep -E -o "\-O[[0-3]]"`
+AC_SUBST(CDEFAULTOPT)
+
 dnl ===================================================================
 dnl Dealing with l10n options
 dnl ===================================================================
--- a/config_host.mk.in-old	2012-04-06 14:08:12.755739649 +0200
+++ b/config_host.mk.in	2012-04-06 14:08:35.351739800 +0200
@@ -74,6 +74,7 @@ export CPPUNIT_CFLAGS=$(gb_SPACE)@CPPUNIT_CFLAGS@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
+export CDEFAULTOPT=@CDEFAULTOPT@
 export CPUNAME_FOR_BUILD=@CPUNAME_FOR_BUILD@
 export CPU_FOR_BUILD=@CPU_FOR_BUILD@
 export CROSS_COMPILING=@CROSS_COMPILING@
