diff --git a/configure.ac b/configure.in
index c0b76f8..d10222f 100644
--- libreoffice-3.5.0/configure.ac
+++ libreoffice-3.5.0/configure.ac
@@ -11762,6 +11779,9 @@ else
 fi
 AC_SUBST(ICECREAM_RUN)
 
+CDEFAULTOPT=`echo $CFLAGS 2>/dev/null | grep -E -o "\-O[[0-3]]"`
+AC_SUBST(CDEFAULTOPT)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
--- a/config_host.mk.in-old	2012-04-06 14:08:12.755739649 +0200
+++ b/config_host.mk.in	2012-04-06 14:08:35.351739800 +0200
@@ -71,6 +71,7 @@ export CPPUNIT_CFLAGS=$(gb_SPACE)@CPPUNIT_CFLAGS@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
+export CDEFAULTOPT=@CDEFAULTOPT@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
