Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:20.846934485 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:20.842934485 +0100
@@ -9611,6 +9611,11 @@
 AC_SUBST(GSTREAMER_0_10_LIBS)
 AC_SUBST(ENABLE_GSTREAMER_0_10)
 
+if test -x /usr/bin/dpkg-buildflags; then
+	HARDEN_CPPFLAGS=`dpkg-buildflags --get CPPFLAGS 2>/dev/null`
+fi
+AC_SUBST(HARDEN_CPPFLAGS)
+
 dnl ===================================================================
 dnl Check whether the OpenGL libraries are available
 dnl ===================================================================
Index: libreoffice-4.1.4~rc2/config_host.mk.in
===================================================================
--- libreoffice-4.1.4~rc2.orig/config_host.mk.in	2013-12-16 14:35:20.846934485 +0100
+++ libreoffice-4.1.4~rc2/config_host.mk.in	2013-12-16 14:35:20.842934485 +0100
@@ -68,6 +68,7 @@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
 export CDEFAULTOPT=@CDEFAULTOPT@
+export HARDEN_CPPFLAGS=@HARDEN_CPPFLAGS@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
Index: libreoffice-4.1.4~rc2/solenv/gbuild/gbuild.mk
===================================================================
--- libreoffice-4.1.4~rc2.orig/solenv/gbuild/gbuild.mk	2013-12-16 14:35:20.846934485 +0100
+++ libreoffice-4.1.4~rc2/solenv/gbuild/gbuild.mk	2013-12-16 14:35:20.842934485 +0100
@@ -265,6 +265,8 @@
 		ENABLE_READONLY_INSTALLSET \
 	)
 
+gb_GLOBALDEFS += $(HARDEN_CPPFLAGS)
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))
 
 ifeq ($(origin gb_RUNNABLE_INSTDIR),undefined)
