Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:03.406935412 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:03.398935412 +0100
@@ -11834,6 +11834,9 @@
 AC_SUBST(MPL_SUBSET)
 
 
+CDEFAULTOPT=`echo $CFLAGS 2>/dev/null | grep -E -o "\-O[[0-3]]"`
+AC_SUBST(CDEFAULTOPT)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
Index: libreoffice-4.1.4~rc2/config_host.mk.in
===================================================================
--- libreoffice-4.1.4~rc2.orig/config_host.mk.in	2013-12-16 14:35:03.406935412 +0100
+++ libreoffice-4.1.4~rc2/config_host.mk.in	2013-12-16 14:35:03.402935412 +0100
@@ -67,6 +67,7 @@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
+export CDEFAULTOPT=@CDEFAULTOPT@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
