--- a/configure.ac-old	2012-03-02 21:48:47.000000000 +0000
+++ b/configure.ac	2012-03-02 21:51:29.000000000 +0000
@@ -5751,17 +5751,6 @@ if test "$GCC" = "yes"; then
         AC_MSG_RESULT([no])
     fi
 
-    AC_MSG_CHECKING([whether $CC supports -fstack-protector-strong])
-    save_CFLAGS=$CFLAGS
-    CFLAGS="$CFLAGS -Werror -fstack-protector-strong"
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[ return 0; ]])],[ HAVE_GCC_STACK_PROTECTOR_STRONG=TRUE ],[])
-    CFLAGS=$save_CFLAGS
-    if test "$HAVE_GCC_STACK_PROTECTOR_STRONG" = "TRUE"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_RESULT([no])
-    fi
-
     AC_MSG_CHECKING([whether $CC supports atomic functions])
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[
     int v = 0;
@@ -10225,6 +10225,11 @@ else
 fi
 AC_SUBST(ENABLE_VLC)
 
+if test -x /usr/bin/dpkg-buildflags; then
+       HARDEN_CPPFLAGS=`dpkg-buildflags --get CPPFLAGS 2>/dev/null`
+fi
+AC_SUBST(HARDEN_CPPFLAGS)
+
 ENABLE_OPENGL_TRANSITIONS=
 ENABLE_OPENGL_CANVAS=
 if test $_os = iOS -o $_os = Android; then
--- a/config_host.mk.in-old	2012-04-06 14:17:55.164239390 +0200
+++ b/config_host.mk.in	2012-04-06 14:18:55.440239065 +0200
@@ -76,6 +76,7 @@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPUNAME=@CPUNAME@
 export CDEFAULTOPT=@CDEFAULTOPT@
+export HARDEN_CPPFLAGS=@HARDEN_CPPFLAGS@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
@@ -240,7 +240,6 @@ export GUIBASE=@GUIBASE@
 export USING_X11=@USING_X11@
 export HAMCREST_JAR=@HAMCREST_JAR@
 export HAVE_GCC_AVX=@HAVE_GCC_AVX@
-export HAVE_GCC_STACK_PROTECTOR_STRONG=@HAVE_GCC_STACK_PROTECTOR_STRONG@
 export HAVE_GCC_BUILTIN_ATOMIC=@HAVE_GCC_BUILTIN_ATOMIC@
 export HAVE_GCC_FINLINE_LIMIT=@HAVE_GCC_FINLINE_LIMIT@
 export HAVE_GCC_FNO_DEFAULT_INLINE=@HAVE_GCC_FNO_DEFAULT_INLINE@
diff --git a/solenv/gbuild/gbuild.mk b/solenv/gbuild/gbuild.mk
index c71cb40..f826d65 100644
--- a/solenv/gbuild/gbuild.mk
+++ b/solenv/gbuild/gbuild.mk
@@ -259,6 +259,8 @@ gb_GLOBALDEFS += \
 		ENABLE_LTO \
 	)
 
+gb_GLOBALDEFS += $(HARDEN_CPPFLAGS)
+
 gb_GLOBALDEFS := $(sort $(gb_GLOBALDEFS))
 
 # This is used to detect whether LibreOffice is being built (as opposed to building
diff --git a/solenv/gbuild/platform/com_GCC_defs.mk b/solenv/gbuild/platform/com_GCC_defs.mk
index 7acd068..b6d9735 100644
--- a/solenv/gbuild/platform/com_GCC_defs.mk
+++ b/solenv/gbuild/platform/com_GCC_defs.mk
@@ -113,12 +113,6 @@ endif
 gb_VISIBILITY_FLAGS_CXX := -fvisibility-inlines-hidden
 gb_CXXFLAGS_COMMON += $(gb_VISIBILITY_FLAGS_CXX)
 
-ifeq ($(HAVE_GCC_STACK_PROTECTOR_STRONG),TRUE)
-gb_CFLAGS_COMMON += -fstack-protector-strong
-gb_CXXFLAGS_COMMON += -fstack-protector-strong
-gb_LinkTarget_LDFLAGS += -fstack-protector-strong
-endif
-
 ifeq ($(ENABLE_PCH),TRUE)
 ifneq ($(COM_IS_CLANG),TRUE)
 gb_CFLAGS_COMMON += -fpch-preprocess -Winvalid-pch
