diff --git a/configure.ac b/configure.in
index c0b76f8..d10222f 100644
--- libreoffice-3.5.0/configure.ac
+++ libreoffice-3.5.0/configure.ac
@@ -11741,6 +11780,9 @@ fi
 AC_SUBST(MPL_SUBSET)
 
 
+CDEFAULTOPT=`echo $CFLAGS 2>/dev/null | grep -E -o "\-O[[0-3]]"`
+AC_SUBST(CDEFAULTOPT)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
diff --git a/config_host.mk.in b/config_host.mk.in
index 57e0574..2ad8cc0 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -70,6 +70,7 @@ export CPPUNIT_CFLAGS=$(gb_SPACE)@CPPUNIT_CFLAGS@
 export CPPUNIT_LIBS=$(gb_SPACE)@CPPUNIT_LIBS@
 export CPU=@CPU@
 export CPUNAME=@CPUNAME@
+export CDEFAULTOPT=@CDEFAULTOPT@
 export CROSS_COMPILING=@CROSS_COMPILING@
 export CURL=@CURL@
 export CURL_CFLAGS=$(gb_SPACE)@CURL_CFLAGS@
