Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:08.246935154 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:08.242935155 +0100
@@ -3353,13 +3353,19 @@
 dnl ===================================================================
 AC_MSG_CHECKING([whether to include symbols while preserve optimization])
 if test -n "$enable_symbols" -a "$enable_symbols" != "no"; then
-    ENABLE_SYMBOLS="TRUE"
+    if test "$enable_symbols" = "SMALL" -o "$enable_symbols" = "small"; then
+        ENABLE_SYMBOLS="SMALL"
+        AC_MSG_RESULT([yes, small ones])
+    else
+        ENABLE_SYMBOLS="TRUE"
+        AC_MSG_RESULT([yes])
+    fi
+
     if test -n "$ENABLE_DBGUTIL"; then
         AC_MSG_ERROR([--enable-dbgutil cannot be used with --enable-symbols])
     elif test -n "$ENABLE_DEBUG"; then
         AC_MSG_ERROR([--enable-debug cannot be used with --enable-symbols])
     fi
-    AC_MSG_RESULT([yes])
 else
     if test -n "$enable_symbols" -a "$enable_symbols" = "no"; then
         ENABLE_SYMBOLS="FALSE"
Index: libreoffice-4.1.4~rc2/solenv/gbuild/platform/com_GCC_defs.mk
===================================================================
--- libreoffice-4.1.4~rc2.orig/solenv/gbuild/platform/com_GCC_defs.mk	2013-12-16 14:35:08.246935154 +0100
+++ libreoffice-4.1.4~rc2/solenv/gbuild/platform/com_GCC_defs.mk	2013-12-16 14:35:08.242935155 +0100
@@ -145,7 +145,11 @@
 FNO_DEFAULT_INLINE=-fno-default-inline
 endif
 
+ifeq ($(ENABLE_SYMBOLS),SMALL)
+gb_DEBUG_CFLAGS := -g1 $(FINLINE_LIMIT0) $(FNO_INLINE)
+else
 gb_DEBUG_CFLAGS := $(GGDB2) $(FINLINE_LIMIT0) $(FNO_INLINE)
+endif
 gb_DEBUG_CXXFLAGS := $(FNO_DEFAULT_INLINE)
 
 
Index: libreoffice-4.1.4~rc2/solenv/gbuild/platform/unxgcc.mk
===================================================================
--- libreoffice-4.1.4~rc2.orig/solenv/gbuild/platform/unxgcc.mk	2013-12-16 14:35:08.246935154 +0100
+++ libreoffice-4.1.4~rc2/solenv/gbuild/platform/unxgcc.mk	2013-12-16 14:35:08.242935155 +0100
@@ -169,9 +169,14 @@
 gb_LinkTarget_CXXFLAGS := $(gb_CXXFLAGS)
 
 ifeq ($(gb_SYMBOL),$(true))
+ifeq ($(ENABLE_SYMBOLS),SMALL)
+gb_LinkTarget_CXXFLAGS += -g1
+gb_LinkTarget_CFLAGS += -g1
+else
 gb_LinkTarget_CXXFLAGS += $(GGDB2)
 gb_LinkTarget_CFLAGS += $(GGDB2)
 endif
+endif
 
 # note that `cat $(extraobjectlist)` is needed to build with older gcc versions, e.g. 4.1.2 on SLED10
 # we want to use @$(extraobjectlist) in the long run
