diff --git a/instsetoo_native/util/openoffice.lst.in b/instsetoo_native/util/openoffice.lst.in
index 8cf38a2..6bda352 100644
--- a/instsetoo_native/util/openoffice.lst.in
+++ b/instsetoo_native/util/openoffice.lst.in
@@ -16,7 +16,7 @@ Globals
             BASISPACKAGEPREFIX libobasis
             UREPACKAGEPREFIX libreoffice
             SOLSUREPACKAGEPREFIX libreoffice
-            PROGRESSBARCOLOR 0,0,0
+            PROGRESSBARCOLOR 215,7,81
             PROGRESSSIZE 444,8
             PROGRESSPOSITION 35,153
             PROGRESSFRAMECOLOR 102,102,102
