diff --git a/solenv/gbuild/platform/com_GCC_defs.mk b/solenv/gbuild/platform/com_GCC_defs.mk
index 9de88a2..9161a4e 100644
--- a/solenv/gbuild/platform/com_GCC_defs.mk
+++ b/solenv/gbuild/platform/com_GCC_defs.mk
@@ -170,6 +170,8 @@ gb_COMPILERNOOPTFLAGS := -O0 -fstrict-aliasing -fstrict-overflow
 ifeq ($(OS),ANDROID)
 gb_DEBUGINFO_FLAGS=-glldb
 # Clang does not know -ggdb2 or some other options
+else ifeq ($(ENABLE_SYMBOLS),SMALL)
+gb_DEBUGINFO_FLAGS=-g1
 else ifeq ($(HAVE_GCC_GGDB2),TRUE)
 gb_DEBUGINFO_FLAGS=-ggdb2
 else
diff --git a/configure.ac b/configure.ac
index a91de78..1c03b1d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4043,15 +4043,20 @@ if test -z "$enable_symbols"; then
     fi
 fi
 if test "$enable_symbols" != no; then
-    ENABLE_SYMBOLS=TRUE
-    AC_MSG_RESULT([yes])
+    if test "$enable_symbols" = "SMALL" -o "$enable_symbols" = "small"; then
+         ENABLE_SYMBOLS="SMALL"
+         AC_MSG_RESULT([yes, small ones])
+    else
+        ENABLE_SYMBOLS=TRUE
+        AC_MSG_RESULT([yes])
+    fi
 else
     ENABLE_SYMBOLS=
     AC_MSG_RESULT([no])
 fi
 AC_SUBST(ENABLE_SYMBOLS)
 
-if test -n "$with_android_ndk" -a \( -n "$ENABLE_SYMBOLS" -o -n "$ENABLE_DEBUG" -o -n "$ENABLE_DBGUTIL" \) -a "$ENABLE_DEBUGINFO_FOR" = "all"; then
+if test -n "$with_android_ndk" -a \( "$ENABLE_SYMBOLS" != "SMALL" -o -n "$ENABLE_DEBUG" -o -n "$ENABLE_DBGUTIL" \) -a "$ENABLE_DEBUGINFO_FOR" = "all"; then
     # Building on Android with full symbols: without enough memory the linker never finishes currently.
     AC_MSG_CHECKING([whether enough memory is available for linking])
     mem_size=$(grep -o 'MemTotal: *.\+ kB' /proc/meminfo | sed 's/MemTotal: *\(.\+\) kB/\1/')
