diff --git a/svtools/source/misc/langhelp.cxx b/svtools/source/misc/langhelp.cxx
index 47cb87e36963..8dc00e09bd3b 100644
--- a/svtools/source/misc/langhelp.cxx
+++ b/svtools/source/misc/langhelp.cxx
@@ -137,8 +137,7 @@ OUString getInstalledLocaleForSystemUILanguage(const css::uno::Sequence<OUString
                     else
                         aPackages.emplace_back("libreoffice-langpack-" + install);
                 }
-                //Debian would be: "The Document Foundation/Debian"
-                else if (sVendor == "The Document Foundation, Debian and Ubuntu")
+                else if (sVendor == "The Document Foundation/Debian" || sVendor == "The Document Foundation, Debian and Ubuntu")
                 {
                     // l10n is the typical Debian/Ubuntu naming convention
                     aPackages.emplace_back("libreoffice-l10n-" + install);
