diff --git a/configure.ac b/configure.in
index 3257668..afd36de 100755
--- libreoffice-3.5.0/configure.ac
+++ libreoffice-3.5.0/configure.ac
@@ -3361,13 +3361,19 @@ dnl Check for enable symbols option
 dnl ===================================================================
 AC_MSG_CHECKING([whether to include symbols while preserve optimization])
 if test -n "$enable_symbols" -a "$enable_symbols" != "no"; then
-    ENABLE_SYMBOLS="TRUE"
+    if test "$enable_symbols" = "SMALL" -o "$enable_symbols" = "small"; then
+        ENABLE_SYMBOLS="SMALL"
+        AC_MSG_RESULT([yes, small ones])
+    else
+        ENABLE_SYMBOLS="TRUE"
+        AC_MSG_RESULT([yes])
+    fi
+
     if test -n "$ENABLE_DBGUTIL"; then
         AC_MSG_ERROR([--enable-dbgutil cannot be used with --enable-symbols])
     elif test -n "$ENABLE_DEBUG"; then
         AC_MSG_ERROR([--enable-debug cannot be used with --enable-symbols])
     fi
-    AC_MSG_RESULT([yes])
 else
     if test -n "$enable_symbols" -a "$enable_symbols" = "no"; then
         ENABLE_SYMBOLS="FALSE"
diff --git a/solenv/gbuild/platform/com_GCC_defs.mk b/solenv/gbuild/platform/com_GCC_defs.mk
index 9de88a2..9161a4e 100644
--- a/solenv/gbuild/platform/com_GCC_defs.mk
+++ b/solenv/gbuild/platform/com_GCC_defs.mk
@@ -153,11 +153,15 @@ gb_PrecompiledHeader_EXCEPTIONFLAGS := $(gb_LinkTarget_EXCEPTIONFLAGS)
 gb_COMPILERNOOPTFLAGS := -O0 -fstrict-aliasing -fstrict-overflow
 
 # Clang does not know -ggdb2 or some other options
+ifeq ($(ENABLE_SYMBOLS),SMALL)
+gb_DEBUGINFO_FLAGS=-g1
+else
 ifeq ($(HAVE_GCC_GGDB2),TRUE)
 gb_DEBUGINFO_FLAGS=-ggdb2
 else
 gb_DEBUGINFO_FLAGS=-g2
 endif
+endif
 
 ifeq ($(HAVE_GCC_FINLINE_LIMIT),TRUE)
 FINLINE_LIMIT0=-finline-limit=0
