--- a/Makefile-old	2012-04-07 02:25:12.933759413 +0200
+++ b/Makefile	2012-04-07 02:46:00.766017382 +0200
@@ -434,10 +434,10 @@
 	@rm -rf $(OUTDIR)/installation
 	@mkdir $(OUTDIR)/installation
 ifeq ($(DISABLE_LINKOO),TRUE)
-	@ooinstall $(OUTDIR)/installation/opt
+	@export WITH_LANG="en-US"; ooinstall $(OUTDIR)/installation/opt
 	@install-gdb-printers -L
 else
-	@ooinstall -l $(OUTDIR)/installation/opt
+	@export WITH_LANG="en-US"; ooinstall -l $(OUTDIR)/installation/opt
 endif
 	@rm -f $(SRCDIR)/install && ln -s $(OUTDIR)/installation/opt/ $(SRCDIR)/install
 
