diff --git a/patches/dev300/apply b/patches/dev300/apply
index dd51188..6e09796 100644
--- libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
@@ -50,7 +50,7 @@ SUSE      : NovellBase, ImpressFontSize, PostgreSQL
 # Slackware
 Slackware : LinuxCommon, Mono, msaccess
 # Debian
-DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL
+DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
