From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 3 Jun 2014 00:59:35 +0200
Subject: [PATCH] disable lotuswordpro test for now

---
 lotuswordpro/Module_lotuswordpro.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lotuswordpro/Module_lotuswordpro.mk b/lotuswordpro/Module_lotuswordpro.mk
--- a/lotuswordpro/Module_lotuswordpro.mk
+++ b/lotuswordpro/Module_lotuswordpro.mk
@@ -15,7 +15,7 @@ $(eval $(call gb_Module_add_targets,lotuswordpro,\
     Library_lwpft \
 ))
 
-$(eval $(call gb_Module_add_check_targets,lotuswordpro,\
+#$(eval $(call gb_Module_add_check_targets,lotuswordpro,\
     CppunitTest_lotuswordpro_test_lotuswordpro \
 ))
 
-- 
1.9.1

