diff --git a/configure.ac b/configure.ac
index 9ee77171..dd2ab7f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9181,21 +9181,9 @@ else
 fi
 AC_SUBST(ENABLE_HARFBUZZ)
 
-dnl ===================================================================
-dnl Check for NPAPI interface to plug browser plugins into LibreOffice documents
-dnl ===================================================================
 AC_MSG_CHECKING([whether to plug browser plugins into LibreOffice documents])
-# Obviously no such thing on iOS or Android. Also not possible when building
-# 64-bit OS X code as the plugin code uses QuickTime and Carbon.
-if test "$_os" != Android -a "$_os" != iOS -a $_os != Darwin -a \
-        "$with_x" != "no" -a "$enable_mpl_subset" != yes
-then
-    AC_MSG_RESULT([yes])
-    ENABLE_NPAPI_FROM_BROWSER=TRUE
-else
-    AC_MSG_RESULT([no])
-    ENABLE_NPAPI_FROM_BROWSER=
-fi
+AC_MSG_RESULT([no])
+ENABLE_NPAPI_FROM_BROWSER=
 AC_SUBST(ENABLE_NPAPI_FROM_BROWSER)
 
 AC_MSG_CHECKING([whether to use X11])
