Index: libreoffice-3.3.3/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.3.3.orig/libreoffice-build/patches/dev300/apply	2011-06-23 01:57:20.510980322 +0200
+++ libreoffice-3.3.3/libreoffice-build/patches/dev300/apply	2011-06-23 01:57:27.020980320 +0200
@@ -1567,8 +1567,8 @@
 # FIXME: update for LO
 #f11-fullscreen.diff, ccheney
 
-[ UbuntuOnly ]
-unxlngi6-notune.diff, doko
+#[ UbuntuOnly ]
+#unxlngi6-notune.diff, doko
 
 [ UbuntuL10nOnly ]
 human-icons-i18n.diff, mklose
