--- libreoffice-3.3.2/libreoffice-build/patches/dev300/apply-old	2011-05-30 00:09:10.000000000 +0200
+++ libreoffice-3.3.2/libreoffice-build/patches/dev300/apply	2011-05-30 00:09:42.000000000 +0200
@@ -1946,7 +1946,7 @@
 # fix problem with calling private and protected members in layout code
 sfx2-privateprotectedpublic.diff
 # UI + core to make OLE editing optionally outplace
-optional-outplace-ole.diff, i#98970, thorsten
+#optional-outplace-ole.diff, i#98970, thorsten
 # let ctrl-<keycode> stuff pass dlg keyboard handling, to permit
 # global application-wide shortcuts also in open dialogs
 vcl-permit-global-shortcuts.diff, i#105676, thorsten
