diff --git a/vcl/Module_vcl.mk b/vcl/Module_vcl.mk
index 9cc9ee0..58f87fd 100644
--- a/vcl/Module_vcl.mk
+++ b/vcl/Module_vcl.mk
@@ -27,7 +27,6 @@ $(eval $(call gb_Module_add_targets,vcl,\
         StaticLibrary_vclmain \
         Executable_ui-previewer \
 		$(if $(filter LINUX MACOSX WNT,$(OS)), \
-			Executable_icontest \
 			Executable_outdevgrind \
 			Executable_vcldemo \
             Executable_mtfdemo )) \
