diff --git a/libreofficekit/Module_libreofficekit.mk b/libreofficekit/Module_libreofficekit.mk
index 000c2fe..6d7e1c6 100644
--- a/libreofficekit/Module_libreofficekit.mk
+++ b/libreofficekit/Module_libreofficekit.mk
@@ -11,9 +11,9 @@ $(eval $(call gb_Module_Module,libreofficekit))
 
 ifeq ($(OS),LINUX)
 
-$(eval $(call gb_Module_add_subsequentcheck_targets,libreofficekit,\
-    CppunitTest_libreofficekit_tiledrendering \
-))
+#$(eval $(call gb_Module_add_subsequentcheck_targets,libreofficekit,\
+#    CppunitTest_libreofficekit_tiledrendering \
+#))
 
 ifneq ($(ENABLE_GTK),)
 $(eval $(call gb_Module_add_targets,libreofficekit,\
