diff --git a/vcl/Module_vcl.mk b/vcl/Module_vcl.mk
index 9cc9ee0..58f87fd 100644
--- a/vcl/Module_vcl.mk
+++ b/vcl/Module_vcl.mk
@@ -31,7 +31,6 @@ $(eval $(call gb_Module_add_targets,vcl,\
 			Executable_outdevgrind \
 			$(if $(ENABLE_HEADLESS),, \
 				Executable_vcldemo \
-				Executable_icontest \
 				Executable_mtfdemo ))) \
     $(if $(filter-out ANDROID IOS WNT,$(OS)), \
         Executable_svdemo \
