diff --git a/sc/Module_sc.mk b/sc/Module_sc.mk
index 9c8696b..385e126 100644
--- a/sc/Module_sc.mk
+++ b/sc/Module_sc.mk
@@ -52,7 +52,6 @@
 
 $(eval $(call gb_Module_add_subsequentcheck_targets,sc,\
     JunitTest_sc_complex \
-    JunitTest_sc_unoapi \
     CppunitTest_sc_cellrangeobj \
     $(if $(filter-out $(OS),IOS), \
 	    CppunitTest_sc_databaserangeobj) \
@@ -63,5 +62,6 @@
     CppunitTest_sc_namedrangesobj \
     CppunitTest_sc_tablesheetsobj \
 ))
+#    JunitTest_sc_unoapi \
 
 # vim: set noet sw=4 ts=4:
--- a/sw/Module_sw.mk-old	2012-02-04 11:25:21.000000000 +0100
+++ b/sw/Module_sw.mk	2012-02-04 11:25:30.000000000 +0100
@@ -48,7 +48,7 @@
 
 $(eval $(call gb_Module_add_subsequentcheck_targets,sw,\
     JunitTest_sw_complex \
-    JunitTest_sw_unoapi \
 ))
+#    JunitTest_sw_unoapi \
 
 # vim: set noet sw=4 ts=4:
