--- a/smoketest/Module_smoketest.mk-old	2014-04-16 07:59:38.995065597 +0200
+++ b/smoketest/Module_smoketest.mk	2014-04-16 07:47:56.579054761 +0200
@@ -30,11 +30,11 @@
 ))
 endif
 
-ifneq (MACOSX/TRUE,$(OS)/$(ENABLE_MACOSX_SANDBOX))
-$(eval $(call gb_Module_add_subsequentcheck_targets,smoketest,\
-	CppunitTest_smoketest \
-))
-endif
+#ifneq (MACOSX/TRUE,$(OS)/$(ENABLE_MACOSX_SANDBOX))
+#$(eval $(call gb_Module_add_subsequentcheck_targets,smoketest,\
+#	CppunitTest_smoketest \
+#))
+#endif
 
 endif
 
