From 23f23a5de14eabd80fd18e57ac03d34208ffb852 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Thu, 22 Dec 2011 23:46:42 +0100
Subject: [PATCH] disable sc unoapi subsequentcheck for now

---
 sc/Module_sc.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sc/Module_sc.mk b/sc/Module_sc.mk
index 9c8696b..385e126 100644
--- a/sc/Module_sc.mk
+++ b/sc/Module_sc.mk
@@ -45,7 +45,7 @@ $(eval $(call gb_Module_add_subsequentcheck_targets,sc,\
     CppunitTest_sc_subsequent_filters_test \
     CppunitTest_sc_tableautoformatfield \
     JunitTest_sc_complex \
-    JunitTest_sc_unoapi \
 ))
+#    JunitTest_sc_unoapi \
 
 # vim: set noet sw=4 ts=4:
-- 
1.7.5.4

