--- a/sd/qa/unit/export-tests.cxx	2015-06-20 11:49:52.000000000 +0200
+++ b/sd/qa/unit/export-tests.cxx	2015-06-20 18:40:25.797612229 +0200
@@ -81,7 +81,7 @@
     void testBnc870233_2();
     void testN828390_4();
     void testN828390_5();
-    void testMediaEmbedding();
+//    void testMediaEmbedding();
     void testFdo71961();
     void testFdo84043();
     void testN828390();
@@ -115,7 +115,7 @@
     CPPUNIT_TEST(testBnc870233_2);
     CPPUNIT_TEST(testN828390_4);
     CPPUNIT_TEST(testN828390_5);
-    CPPUNIT_TEST(testMediaEmbedding);
+//    CPPUNIT_TEST(testMediaEmbedding);
     CPPUNIT_TEST(testFdo71961);
     CPPUNIT_TEST(testFdo84043);
     CPPUNIT_TEST(testN828390);
@@ -336,6 +336,7 @@
     xDocShRef->DoClose();
 }
 
+/*
 void SdExportTest::testMediaEmbedding()
 {
     ::sd::DrawDocShellRef xDocShRef = loadURL(getURLFromSrc("/sd/qa/unit/data/media_embedding.odp"), ODP);
@@ -370,7 +371,7 @@
 
     xDocShRef->DoClose();
 }
-
+*/
 void SdExportTest::testFdo84043()
 {
     ::sd::DrawDocShellRef xDocShRef = loadURL(getURLFromSrc("/sd/qa/unit/data/fdo84043.odp"), ODP);
