From ca847d43de4c706f906f6473b05f4cf53c39feb7 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Sun, 9 Jun 2013 13:15:29 +0200
Subject: [PATCH] disable sw tests for beta

---
 sw/Module_sw.mk | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/sw/Module_sw.mk b/sw/Module_sw.mk
index 202c67c..90d9416 100644
--- a/sw/Module_sw.mk
+++ b/sw/Module_sw.mk
@@ -53,7 +53,6 @@ $(eval $(call gb_Module_add_slowcheck_targets,sw,\
     CppunitTest_sw_ww8export \
     CppunitTest_sw_ww8import \
     CppunitTest_sw_rtfexport \
-    CppunitTest_sw_rtfimport \
     CppunitTest_sw_odfexport \
     CppunitTest_sw_odfimport \
 ))
@@ -66,15 +65,12 @@ endif
 
 ifneq ($(OOO_JUNIT_JAR),)
 $(eval $(call gb_Module_add_subsequentcheck_targets,sw,\
-    JunitTest_sw_complex \
-    JunitTest_sw_unoapi \
 ))
 endif
 
 ifneq ($(DISABLE_PYTHON),TRUE)
 ifneq ($(OS),WNT)
 $(eval $(call gb_Module_add_subsequentcheck_targets,sw,\
-	PythonTest_sw_python \
 ))
 endif
 endif
-- 
1.8.3

