--- a/configure.ac-old	2018-12-28 11:10:35.953066686 +0100
+++ b/configure.ac	2018-12-28 11:11:06.049403639 +0100
@@ -6976,6 +6976,7 @@
 
             # set to limit VM usage for JunitTests
             JAVAIFLAGS=-Xmx64M
+            JAVAIFLAGS="$JAVAIFLAGS -Djdk.net.URLClassPath.disableClassPathURLCheck=true"
             # set to limit VM usage for javac
             JAVAFLAGS=-J-Xmx128M
         fi
