Index: libreoffice-3.3.2/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.3.2.orig/libreoffice-build/patches/dev300/apply	2011-03-23 14:22:33.234216324 +0100
+++ libreoffice-3.3.2/libreoffice-build/patches/dev300/apply	2011-03-23 14:22:59.844216317 +0100
@@ -1560,8 +1560,8 @@
 # FIXME: update for LO
 #f11-fullscreen.diff, ccheney
 
-[ UbuntuOnly ]
-unxlngi6-notune.diff, doko
+#[ UbuntuOnly ]
+#unxlngi6-notune.diff, doko
 
 [ UbuntuL10nOnly ]
 human-icons-i18n.diff, mklose
