Description: Two tests have begun segfaulting on arm64 since v6.3.0
Author: Marcus Tomlinson <marcus.tomlinson@canonical.com>

--- a/sc/Module_sc.mk
+++ b/sc/Module_sc.mk
@@ -182,7 +182,8 @@ $(eval $(call gb_Module_add_subsequentcheck_targets,sc,\
 	CppunitTest_sc_importdescriptorbaseobj \
 	CppunitTest_sc_labelrangeobj \
 	CppunitTest_sc_labelrangesobj \
-	CppunitTest_sc_macros_test \
+	$(if $(filter-out $(CPUNAME),AARCH64), \
+		CppunitTest_sc_macros_test) \
 	CppunitTest_sc_modelobj \
 	CppunitTest_sc_namedrangeobj \
 	CppunitTest_sc_namedrangesobj \
--- a/testtools/Module_testtools.mk
+++ b/testtools/Module_testtools.mk
@@ -35,9 +35,11 @@ $(eval $(call gb_Module_add_targets,testtools,\
 ))
 endif
 
+ifneq ($(CPUNAME),AARCH64)
 $(eval $(call gb_Module_add_check_targets,testtools,\
 	CustomTarget_uno_test \
 ))
+endif
 
 endif

