--- libreoffice-3.3.4.orig/libreoffice-build/patches/apply.pl.in
+++ libreoffice-3.3.4/libreoffice-build/patches/apply.pl.in
@@ -95,7 +95,7 @@ sub do_patch {
     my $cmd = "$cmd_base --dry-run $cmd_output";

     print "$cmd\n" unless $quiet;
-    system ($cmd) && die "Testing patch $patch_file failed.";
+    #system ($cmd) && die "Testing patch $patch_file failed.";

     $cmd = "$cmd_base > /dev/null";
     if($quiet)
