diff --git a/sc/qa/uitest/calc_dialogs/openDialogs.py b/sc/qa/uitest/calc_dialogs/openDialogs.py
index 7d38ba004da1..272dd804afef 100644
--- a/sc/qa/uitest/calc_dialogs/openDialogs.py
+++ b/sc/qa/uitest/calc_dialogs/openDialogs.py
@@ -150,7 +150,7 @@ dialogs = [
     # {"command": ".uno:ToolProtectionDocument", "closeButton": "cancel"},
         # tested in sc/qa/uitest/protect/protectSpreadsheet.py
     {"command": ".uno:RunMacro", "closeButton": "cancel"},
-    {"command": ".uno:ScriptOrganizer", "closeButton": "close"},
+#    {"command": ".uno:ScriptOrganizer", "closeButton": "close"},
     {"command": ".uno:ShowLicense", "closeButton": "close"},
     # {"command": ".uno:About", "closeButton": "close"},
         # tested in sc/qa/uitest/calc_tests7/tdf106667.py
