Index: libreoffice-3.3.2/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.3.2.orig/libreoffice-build/patches/dev300/apply	2011-03-26 22:54:48.278463488 +0100
+++ libreoffice-3.3.2/libreoffice-build/patches/dev300/apply	2011-03-26 22:54:49.578463488 +0100
@@ -1561,8 +1561,8 @@
 #f11-fullscreen.diff, ccheney
 arm-opt-gcc-pre46.diff
 
-[ UbuntuOnly ]
-unxlngi6-notune.diff, doko
+#[ UbuntuOnly ]
+#unxlngi6-notune.diff, doko
 
 [ UbuntuL10nOnly ]
 human-icons-i18n.diff, mklose
