From: Debian LibreOffice Maintainers <debian-openoffice@lists.debian.org>
Date: Tue, 28 Jan 2014 12:12:27 +0100
Subject: [PATCH] disable-ftp-nohost-test.diff

---
 ucb/Module_ucb.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ucb/Module_ucb.mk b/ucb/Module_ucb.mk
--- a/ucb/Module_ucb.mk
+++ b/ucb/Module_ucb.mk
@@ -42,7 +42,6 @@ $(eval $(call gb_Module_add_targets,ucb,\
 endif
 
 $(eval $(call gb_Module_add_subsequentcheck_targets,ucb,\
-	JunitTest_ucb_complex \
 	JunitTest_ucb_unoapi \
 ))
 
-- 
1.8.3.2

