From: Debian LibreOffice Maintainers <debian-openoffice@lists.debian.org>
Date: Fri, 26 Feb 2016 13:05:28 +0100
Subject: [PATCH] disable-npapi-plugin-support.diff

---
 configure.ac | 16 ++--------------
 1 file changed, 2 insertions(+), 14 deletions(-)

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -9219,21 +9219,9 @@ else
 fi
 AC_SUBST(ENABLE_HARFBUZZ)
 
-dnl ===================================================================
-dnl Check for NPAPI interface to plug browser plugins into LibreOffice documents
-dnl ===================================================================
 AC_MSG_CHECKING([whether to plug browser plugins into LibreOffice documents])
-# Obviously no such thing on iOS or Android. Also not possible when building
-# 64-bit OS X code as the plugin code uses QuickTime and Carbon.
-if test "$_os" != Android -a "$_os" != iOS -a $_os != Darwin -a \
-        "$with_x" != "no" -a "$enable_mpl_subset" != yes
-then
-    AC_MSG_RESULT([yes])
-    ENABLE_NPAPI_FROM_BROWSER=TRUE
-else
-    AC_MSG_RESULT([no])
-    ENABLE_NPAPI_FROM_BROWSER=
-fi
+AC_MSG_RESULT([no])
+ENABLE_NPAPI_FROM_BROWSER=
 AC_SUBST(ENABLE_NPAPI_FROM_BROWSER)
 
 AC_MSG_CHECKING([whether to use X11])
-- 
2.5.0

