Index: libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.0.orig/libreoffice-build/patches/dev300/apply	2011-06-28 16:27:43.462707013 +0200
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply	2011-06-28 16:28:05.212707013 +0200
@@ -1037,8 +1037,8 @@
 # FIXME: update for LO
 #f11-fullscreen.diff, ccheney
 
-[ UbuntuOnly ]
-unxlngi6-notune.diff, doko
+#[ UbuntuOnly ]
+#unxlngi6-notune.diff, doko
 
 [ UbuntuL10nOnly ]
 human-icons-i18n.diff, mklose
