diff --git a/basegfx/Module_basegfx.mk b/basegfx/Module_basegfx.mk
index c854d995e876..c1c601a8bdc9 100644
--- a/basegfx/Module_basegfx.mk
+++ b/basegfx/Module_basegfx.mk
@@ -13,8 +13,8 @@ $(eval $(call gb_Module_add_targets,basegfx,\
     Library_basegfx \
 ))
 
-$(eval $(call gb_Module_add_check_targets,basegfx,\
-    CppunitTest_basegfx \
-))
+#$(eval $(call gb_Module_add_check_targets,basegfx,\
+#    CppunitTest_basegfx \
+#))
 
 # vim: set noet sw=4 ts=4:
