--- a/configure.in-old	2012-06-24 17:26:46.094370935 +0200
+++ b/configure.in	2012-06-24 17:27:17.306525701 +0200
@@ -2599,6 +2599,7 @@
         AC_MSG_RESULT([gcc $_gcc_version])
     fi
 fi
+AC_SUBST(GCCVER)
 
 dnl ===================================================================
 dnl  Is GCC actually Clang?
--- a/config_host.mk.in-old	2012-06-24 17:28:01.030742514 +0200
+++ b/config_host.mk.in	2012-06-24 17:28:21.430843669 +0200
@@ -49,6 +49,7 @@
 export CLUCENE_LIBS=@CLUCENE_LIBS@
 export COM=@COM@
 export COM_GCC_IS_CLANG=@COM_GCC_IS_CLANG@
+export GCCVER=@GCCVER@
 export COMEX=@COMEX@
 export COMMONS_CODEC_JAR=@COMMONS_CODEC_JAR@
 export COMMONS_HTTPCLIENT_JAR=@COMMONS_HTTPCLIENT_JAR@
--- a/xmloff/Module_xmloff.mk-old	2012-06-24 17:32:08.911971688 +0200
+++ b/xmloff/Module_xmloff.mk	2012-06-24 17:32:20.352028415 +0200
@@ -35,9 +35,11 @@
     Package_inc \
 ))
 
+ifneq "$(GCCVER)" "040700"
 $(eval $(call gb_Module_add_subsequentcheck_targets,xmloff,\
     JunitTest_xmloff_unoapi \
 ))
+endif
 
 #todo: map file
 #todo: xmlkywd.lib - built, but never used?!
