diff --git a/sc/Module_sc.mk b/sc/Module_sc.mk
index 9c8696b..385e126 100644
--- a/sc/Module_sc.mk
+++ b/sc/Module_sc.mk
@@ -45,7 +45,7 @@ $(eval $(call gb_Module_add_subsequentcheck_targets,sc,\
     CppunitTest_sc_subsequent_filters_test \
     CppunitTest_sc_tableautoformatfield \
     JunitTest_sc_complex \
-    JunitTest_sc_unoapi \
 ))
+#    JunitTest_sc_unoapi \
 
 # vim: set noet sw=4 ts=4:
-- 
1.7.5.4

--- a/sw/Module_sw.mk-old	2012-02-04 11:25:21.000000000 +0100
+++ b/sw/Module_sw.mk	2012-02-04 11:25:30.000000000 +0100
@@ -48,7 +48,7 @@
 
 $(eval $(call gb_Module_add_subsequentcheck_targets,sw,\
     JunitTest_sw_complex \
-    JunitTest_sw_unoapi \
 ))
+#    JunitTest_sw_unoapi \
 
 # vim: set noet sw=4 ts=4:
