diff --git a/patches/dev300/apply b/patches/dev300/apply
index dd51188..6e09796 100644
--- libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply
@@ -66,7 +66,7 @@
 # Slackware
 Slackware : LinuxCommon, SlackwareOnly, NotDebian, Mono, msaccess
 # Debian
-DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL, PowerPC
+DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PowerPC
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only, HPPA, AXP, kFreeBSD
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
