diff --git a/bin/package-ooo b/bin/package-ooo
index 040ea48..389627f 100755
--- libreoffice-3.5.0/libreoffice-3.4.0/libreoffice-build/bin/package-ooo
+++ libreoffice-3.5.0/libreoffice-3.4.0/libreoffice-build/bin/package-ooo
@@ -34,10 +34,6 @@ if test "z$PIECE" != "z"; then
     exit 0;
 fi
 
-echo "Cleaning $OOINSTDIR";
-rm -Rf $OOINSTDIR;
-set -e
-
 mkdir -p $PREFIX/bin
 
 sed -e "s|@OOINSTBASE@|$OOINSTBASE|g" $TOOLSDIR/bin/ootool.in \
@@ -215,28 +211,6 @@ Libs: -L\${libdir} -lprot_uno_uno
 Cflags: -I\${includeddir}" > $LIBDIRBASE/pkgconfig/openoffice$BINSUFFIX.pc
 fi
 
-export DISPLAY=''; # clobber;
-echo "Execute ooinstall ...";
-
-cd $TOOLSDIR/bin
-
-./ooinstall $OOINSTDIR || exit 1;
-##cp -ra $OODESTDIR $OODESTDIR.1 || exit 1
-##rm -r $OODESTDIR
-##cp -ra $OODESTDIR.1 $OODESTDIR || exit 1
-echo "Cleaning up ...";
-
-# No idea what these files are good for (?)
-# they don't appear in the RPM file lists.
-rm -Rf $OOINSTDIR/share/uno_packages/cache/*
-
-# FIXME: we need to packagethe extensions some way
-if test -n "$OODESTDIR" ; then
-    rm -f $OOINSTDIR/share/extension/install/scsolver.uno.oxt
-fi
-
-echo "Done";
-
 remove_help_localization()
 {
     lang=$1
