diff --git a/sw/CppunitTest_sw_core_layout.mk b/sw/CppunitTest_sw_core_layout.mk
index 80dc1530fdb3..2cae001add53 100644
--- a/sw/CppunitTest_sw_core_layout.mk
+++ b/sw/CppunitTest_sw_core_layout.mk
@@ -71,11 +71,6 @@ $(eval $(call gb_CppunitTest_use_uiconfigs,sw_core_layout, \
     svx \
 ))
 
-# assert if font/glyph fallback occurs
-$(call gb_CppunitTest_get_target,sw_core_layout): \
-    EXTRA_ENV_VARS := \
-        SAL_ABORT_ON_NON_APPLICATION_FONT_USE=1
-
 $(eval $(call gb_CppunitTest_use_more_fonts,sw_core_layout))
 
 # vim: set noet sw=4 ts=4:
diff --git a/sw/layoutwriter_setup.mk b/sw/layoutwriter_setup.mk
index 9d10fa9ac2c6..cd0d86b7f98c 100644
--- a/sw/layoutwriter_setup.mk
+++ b/sw/layoutwriter_setup.mk
@@ -82,11 +82,6 @@ $(eval $(call gb_CppunitTest_use_uiconfigs,sw_layoutwriter$(1), \
 $(call gb_CppunitTest_get_target,sw_layoutwriter$(1)): \
     $(call gb_Library_get_target,textconv_dict)
 
-# assert if font/glyph fallback occurs
-$(call gb_CppunitTest_get_target,sw_layoutwriter$(1)): \
-    EXTRA_ENV_VARS := \
-        SAL_ABORT_ON_NON_APPLICATION_FONT_USE=1
-
 $(eval $(call gb_CppunitTest_use_more_fonts,sw_layoutwriter$(1)))
 
 endef
