diff --git a/configure.ac b/configure.ac
index 1b60e74..e03eae0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1933,6 +1933,18 @@ AC_ARG_WITH(
     ],,
     [with_doxygen=yes])
 
+AC_ARG_WITH(
+    [javadoc],
+    AS_HELP_STRING(
+        [--with-javadoc],
+        [Specifies the javadoc executable to use when generating ODK Java
+         documentation. --without-javadoc disables generation of ODK Java
+         documentation. Not relevant in the --disable-odk case.])
+    [
+                          Usage:     --with-javadoc=<absolute path to doxygen executable>
+    ],,
+    [with_javadoc=yes])
+
 AC_ARG_WITH(visual-studio,
     AS_HELP_STRING([--with-visual-studio=<2012/2010/2013>],
         [Specify which Visual Studio version to use in case several are
@@ -3962,13 +3974,19 @@ dnl Check for enable symbols option
 dnl ===================================================================
 AC_MSG_CHECKING([whether to include symbols while preserve optimization])
 if test -n "$enable_symbols" -a "$enable_symbols" != "no"; then
-    ENABLE_SYMBOLS="TRUE"
+    if test "$enable_symbols" = "SMALL" -o "$enable_symbols" = "small"; then
+        ENABLE_SYMBOLS="SMALL"
+        AC_MSG_RESULT([yes, small ones])
+    else
+        ENABLE_SYMBOLS="TRUE"
+        AC_MSG_RESULT([yes])
+    fi
+
     if test -n "$ENABLE_DBGUTIL"; then
         AC_MSG_ERROR([--enable-dbgutil cannot be used with --enable-symbols])
     elif test -n "$ENABLE_DEBUG"; then
         AC_MSG_ERROR([--enable-debug cannot be used with --enable-symbols])
     fi
-    AC_MSG_RESULT([yes])
 else
     if test -n "$enable_symbols" -a "$enable_symbols" = "no"; then
         ENABLE_SYMBOLS="FALSE"
@@ -7014,35 +7032,45 @@ AC_SUBST(JAVACISGCJ)
 dnl ===================================================================
 dnl Checks for javadoc
 dnl ===================================================================
-if test "$ENABLE_JAVA" != ""; then
+if test "$ENABLE_JAVA" != "" -a "$enable_odk" != no; then
     if test -z "$with_jdk_home"; then
         AC_PATH_PROG(JAVADOC, javadoc)
     else
-        _javadoc_path="$with_jdk_home/bin/javadoc"
-        dnl Check if there is a javadoc at all.
-        if test -x "$_javadoc_path"; then
-            JAVADOC=$_javadoc_path
-        else
-            AC_PATH_PROG(JAVADOC, javadoc)
-        fi
-    fi
-    if test -z "$JAVADOC"; then
-        AC_MSG_ERROR([$_javadoc_path not found set with_jdk_home])
-    fi
-    if test "$build_os" = "cygwin"; then
-        if test x`echo "$JAVADOC" | $GREP -i '\.exe$'` = x; then
-            JAVADOC="${JAVADOC}.exe"
-        fi
-        JAVADOC=`cygpath -d "$JAVADOC"`
-        JAVADOC=`cygpath -u "$JAVADOC"`
-    fi
+        if test "$with_javadoc" = "yes"; then
+            _javadoc_path="$with_jdk_home/bin/javadoc"
+            dnl Check if there is a javadoc at all.
+            if test -x "$_javadoc_path"; then
+                JAVADOC=$_javadoc_path
+            else
+                AC_PATH_PROG(JAVADOC, javadoc)
+            fi
+            if test -z "$JAVADOC"; then
+                AC_MSG_ERROR([$_javadoc_path not found set with_jdk_home])
+            fi
+            if test "$build_os" = "cygwin"; then
+                if test x`echo "$JAVADOC" | $GREP -i '\.exe$'` = x; then
+                    JAVADOC="${JAVADOC}.exe"
+                fi
+                JAVADOC=`cygpath -d "$JAVADOC"`
+                JAVADOC=`cygpath -u "$JAVADOC"`
+            fi
 
-    if test `$JAVADOC --version 2>&1 | $GREP -c "gjdoc"` -gt 0; then
-    JAVADOCISGJDOC="yes"
+            if test `$JAVADOC --version 2>&1 | $GREP -c "gjdoc"` -gt 0; then
+            JAVADOCISGJDOC="yes"
+            fi
+        else
+	    unset JAVADOC
+	fi
     fi
+else
+    unset JAVADOC
 fi
 AC_SUBST(JAVADOCISGJDOC)
 
+if test "$JAVADOCISGJDOC" = "yes"; then
+	unset JAVADOC
+fi
+
 if test "$ENABLE_JAVA" != ""; then
     # check if JAVA_HOME was (maybe incorrectly?) set automatically to /usr
     if test "$JAVA_HOME" = "/usr" -a "x$with_jdk_home" = "x"; then
@@ -7222,12 +7250,18 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAINC"; then
     k*bsd*-gnu*)
         JAVAINC="-I$JAVA_HOME/include"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
+	if test "$JDK" = "gcj"; then
+		JAVAINC="$JAVAINC -I`$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`"
+	fi
         ;;
 
     linux-gnu*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
+	if test "$JDK" = "gcj"; then
+		JAVAINC="$JAVAINC -I`$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`"
+	fi
         ;;
 
     mingw*)
@@ -7709,12 +7743,12 @@ libo_CHECK_SYSTEM_MODULE([librevenge],[REVENGE],[librevenge-0.0],["-I${WORKDIR}/
 dnl ===================================================================
 dnl Check for system libe-book
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([libebook],[EBOOK],[libe-book-0.1 >= 0.1.1])
+libo_CHECK_SYSTEM_MODULE([libebook],[EBOOK],[libe-book-0.1])
 
 dnl ===================================================================
 dnl Check for system libetonyek
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([libetonyek],[ETONYEK],[libetonyek-0.1 >= 0.1.1])
+libo_CHECK_SYSTEM_MODULE([libetonyek],[ETONYEK],[libetonyek-0.1])
 
 dnl ===================================================================
 dnl Check for system libfreehand
@@ -7989,7 +8023,7 @@ no|disable)
         # Unset variables set by the above AM_PATH_PYTHON so that
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
-        AM_PATH_PYTHON([3.3],, [:])
+        AM_PATH_PYTHON([2.6],, [:])
         if test "$PYTHON" = :; then
             AC_MSG_RESULT([internal])
             enable_python=internal
@@ -8047,7 +8081,7 @@ if test $enable_python = system; then
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         # This causes an error if no python command is found
-        AM_PATH_PYTHON([3.3])
+        AM_PATH_PYTHON([2.6])
         python_include=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'));"`
         python_version=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('VERSION'));"`
         python_libs=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBS'));"`
@@ -9588,8 +9622,9 @@ if test "$ENABLE_LPSOLVE" = TRUE; then
         # some systems need this. Like Ubuntu....
         AC_CHECK_LIB(m, floor)
         AC_CHECK_LIB(dl, dlopen)
+        AC_CHECK_LIB(colamd, colamd)
         AC_CHECK_LIB([lpsolve55], [make_lp], [:],
-            [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
+            [ AC_MSG_ERROR(lpsolve library not found or too old.)], [-L/usr/lib/lpsolve])
         LIBS=$save_LIBS
         libo_MINGW_CHECK_DLL([lpsolve55])
     else
@@ -9993,11 +10028,14 @@ if test "x$enable_gtk3" = "xyes"; then
     else
         AC_MSG_ERROR([gtk3 libraries of the correct versions, not found])
     fi
+    PKG_CHECK_MODULES(GMODULE, gmodule-2.0,,AC_MSG_ERROR([requirements to build the gtk3-plugin not met. Use --disable-gtk3 or install the missing packages]))
     GTK3_CFLAGS=$(printf '%s' "$GTK3_CFLAGS" | sed -e "s/-I/${ISYSTEM?}/g")
 fi
 AC_SUBST(GTK3_LIBS)
 AC_SUBST(GTK3_CFLAGS)
 AC_SUBST(ENABLE_GTK3)
+AC_SUBST(GMODULE_LIBS)
+AC_SUBST(GMODULE_CFLAGS)
 
 AC_MSG_CHECKING([which VCLplugs shall be built])
 if test "$GUIBASE" != "unx" -o "$enable_headless" = "yes"; then
@@ -10381,6 +10419,11 @@ else
 fi
 AC_SUBST(ENABLE_VLC)
 
+if test -x /usr/bin/dpkg-buildflags; then
+	HARDEN_CPPFLAGS=`dpkg-buildflags --get CPPFLAGS 2>/dev/null`
+fi
+AC_SUBST(HARDEN_CPPFLAGS)
+
 dnl ===================================================================
 dnl Check whether the OpenGL libraries are available
 dnl ===================================================================
@@ -12815,6 +12858,9 @@ fi
 AC_SUBST(MPL_SUBSET)
 
 
+CDEFAULTOPT=`echo $CFLAGS 2>/dev/null | grep -E -o "\-O[[0-3]]"`
+AC_SUBST(CDEFAULTOPT)
+
 dnl ===================================================================
 dnl Setting up the environment.
 dnl ===================================================================
