diff --git a/patches/dev300/apply b/patches/dev300/apply
index 916afff..1d3731c 100644
--- openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/apply
+++ openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/apply
@@ -427,6 +427,10 @@
 # .. and fix 64bit archs
 plugin.long.diff, i#110747, cmc
 
+[ DebianBaseOnly and TemporaryHacks ]
+extensions-mozilla-plugin-pc-if-libxul.diff
+
+[ Fixes ]
 # sd printing fixes
 printing-bugfix-254573.diff, n#254573, rodo
 
--- /dev/null	2010-05-02 17:01:50.016998199 +0200
+++ openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/extensions-mozilla-plugin-pc-if-libxul.diff	2010-05-13 19:02:36.000000000 +0200
@@ -0,0 +1,18 @@
+diff --git a/extensions/source/plugin/util/makefile.pmk b/extensions/source/plugin/util/makefile.pmk
+--- extensions/source/plugin/util/makefile.pmk
++++ extensions/source/plugin/util/makefile.pmk
+@@ -30,8 +30,14 @@
+ 
+ .IF "$(SYSTEM_MOZILLA)" == "YES"
+ CFLAGS+=-DSYSTEM_MOZILLA
++# for libxul (newer xulrunners) the .pc file is mozilla-plugin, libxul-plugin
++# doesn't exist
++.IF "$(MOZ_FLAVOUR)" == "libxul"
++PKGCONFIG_MODULES+=mozilla-plugin
++.ELSE
+ PKGCONFIG_MODULES+=$(MOZ_FLAVOUR)-plugin
+ .ENDIF
++.ENDIF
+ 
+ .IF "$(GUIBASE)" == "unx" && "$(ENABLE_GTK)" == "TRUE"
+ CDEFS+=-DENABLE_GTK
