From f6e5a0ac09eb1b02caf62980e4caaf3f239b9ca7 Mon Sep 17 00:00:00 2001
From: Noel Grandin <noel@peralex.com>
Date: Mon, 8 Jun 2015 12:11:58 +0200
Subject: vclptr fix in ScrollableWrapper

Change-Id: I28b0c9712d3eb35181ccd323e583d947db06094d

diff --git a/toolkit/source/awt/scrollabledialog.cxx b/toolkit/source/awt/scrollabledialog.cxx
index a615bd2..b1d898d 100644
--- a/toolkit/source/awt/scrollabledialog.cxx
+++ b/toolkit/source/awt/scrollabledialog.cxx
@@ -82,7 +82,7 @@ void ScrollableWrapper<T>::setScrollVisibility( ScrollBarVisibility rVisState )
 template< class T>
 ScrollableWrapper<T>::~ScrollableWrapper()
 {
-    dispose();
+    T::disposeOnce();
 }
 
 template< class T>
-- 
cgit v0.10.2

