diff --git a/filter/qa/cppunit/filters-pict-test.cxx b/filter/qa/cppunit/filters-pict-test.cxx
index 36d8d22..d69cf30 100644
--- a/filter/qa/cppunit/filters-pict-test.cxx
+++ b/filter/qa/cppunit/filters-pict-test.cxx
@@ -16,10 +16,6 @@
 #include <osl/file.hxx>
 #include <osl/process.h>
 
-#ifdef LIBO_MERGELIBS
-#define GraphicImport iptGraphicImport
-#endif
-
 extern "C"
 {
     SAL_DLLPUBLIC_EXPORT sal_Bool SAL_CALL
diff --git a/filter/qa/cppunit/filters-tga-test.cxx b/filter/qa/cppunit/filters-tga-test.cxx
index 365adb8..a0e0453 100644
--- a/filter/qa/cppunit/filters-tga-test.cxx
+++ b/filter/qa/cppunit/filters-tga-test.cxx
@@ -16,10 +16,6 @@
 #include <osl/file.hxx>
 #include <osl/process.h>
 
-#ifdef LIBO_MERGELIBS
-#define GraphicImport itgGraphicImport
-#endif
-
 extern "C"
 {
     SAL_DLLPUBLIC_EXPORT sal_Bool SAL_CALL
diff --git a/filter/qa/cppunit/filters-tiff-test.cxx b/filter/qa/cppunit/filters-tiff-test.cxx
index e7594e9..0089674 100644
--- a/filter/qa/cppunit/filters-tiff-test.cxx
+++ b/filter/qa/cppunit/filters-tiff-test.cxx
@@ -16,10 +16,6 @@
 #include <osl/file.hxx>
 #include <osl/process.h>
 
-#ifdef LIBO_MERGELIBS
-#define GraphicImport itiGraphicImport
-#endif
-
 extern "C"
 {
     SAL_DLLPUBLIC_EXPORT sal_Bool SAL_CALL
