--- a/solenv/gbuild/platform/com_GCC_defs.mk
+++ b/solenv/gbuild/platform/com_GCC_defs.mk
@@ -180,10 +180,12 @@ gb_LTOFLAGS := -flto
 gb_LTOPLUGINFLAGS := --plugin LLVMgold.so
 endif
 else
-gb_LTOFLAGS := -flto$(if $(filter-out 0,$(PARALLELISM)),=$(PARALLELISM)) -fuse-linker-plugin -O2
+gb_LTOFLAGS := -flto=auto -ffat-lto-objects
 endif
 endif
 
+gb_FilterOutClangCFLAGS += -flto=auto -ffat-lto-objects
+
 gb_LinkTarget_EXCEPTIONFLAGS := \
 	-DEXCEPTIONS_ON \
 	-fexceptions
