diff --git a/configure.ac b/configure.ac
index 4ee8955..7ce4b94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8096,20 +8126,10 @@ if test "x$enable_firebird_sdbc" = "xyes"; then
     if test "$with_system_firebird" = "yes"; then
         AC_MSG_RESULT([external])
         SYSTEM_FIREBIRD=YES
-        AC_PATH_PROG(FIREBIRDCONFIG, [fb_config])
-        if test -z "$FIREBIRDCONFIG"; then
-            AC_MSG_NOTICE([No fb_config -- using pkg-config])
-            FIREBIRD_VERSION=`pkg-config --modversion fbembed`
-            AC_MSG_CHECKING([for Firebird Client library])
-            FIREBIRD_CFLAGS=`pkg-config --cflags fbembed`
-            FIREBIRD_LIBS=`pkg-config --libs fbembed`
-        else
-            AC_MSG_NOTICE([fb_config found])
-            FIREBIRD_VERSION=`$FIREBIRDCONFIG --version`
-            AC_MSG_CHECKING([for Firebird Client library])
-            FIREBIRD_CFLAGS=`$FIREBIRDCONFIG --cflags`
-            FIREBIRD_LIBS=`$FIREBIRDCONFIG --embedlibs`
-        fi
+        AC_MSG_CHECKING([for Firebird Client library])
+        FIREBIRD_CFLAGS=
+        FIREBIRD_LIBS="-L/usr/lib/`dpkg-architecture -qDEB_HOST_MULTIARCH` -lfbembed"
+        FIREBIRD_VERSION=`dpkg -l firebird-dev | grep ^ii | awk '{ print $3 }' | cut -d. -f1,2,3`
         AC_MSG_CHECKING([Firebird version])
         FIREBIRD_MAJOR=`echo $FIREBIRD_VERSION | cut -d"." -f1`
         FIREBIRD_MINOR=`echo $FIREBIRD_VERSION | cut -d"." -f2`
