diff --git a/avmedia/source/vlc/vlcframegrabber.cxx b/avmedia/source/vlc/vlcframegrabber.cxx
index 4ccb8fa..69e17f8 100644
--- a/avmedia/source/vlc/vlcframegrabber.cxx
+++ b/avmedia/source/vlc/vlcframegrabber.cxx
@@ -44,7 +44,7 @@ SAL_CALL VLCFrameGrabber::VLCFrameGrabber( VLC::Player& player, boost::shared_pt
 
         if ( !mPlayer.play() )
         {
-            std::cerr << "Couldn't play" << std::endl;
+            fprintf(stderr, "Couldn't play");
         }
 
         mPlayer.setTime( ( fMediaTime > 0 ? fMediaTime : 0 ) * MSEC_IN_SEC );
@@ -55,7 +55,7 @@ SAL_CALL VLCFrameGrabber::VLCFrameGrabber( VLC::Player& player, boost::shared_pt
 
         if ( mUrl.isEmpty() || !mPlayer.hasVout() )
         {
-            std::cerr << "Couldn't grab frame" << std::endl;
+            fprintf(stderr, "Couldn't grab frame");
             mPlayer.setMute( false );
 
             manager.onPaused();
