diff --git a/vcl/qt5/Qt5AccessibleWidget.cxx b/vcl/qt5/Qt5AccessibleWidget.cxx
index be3480aa3f77..748dda66f625 100644
--- a/vcl/qt5/Qt5AccessibleWidget.cxx
+++ b/vcl/qt5/Qt5AccessibleWidget.cxx
@@ -116,8 +116,8 @@ void lcl_appendRelation(QVector<QPair<QAccessibleInterface*, QAccessible::Relati
     for (sal_uInt32 i = 0; i < nTargetCount; i++)
     {
         Reference<XAccessible> xAccessible(aRelation.TargetSet[i], uno::UNO_QUERY);
-        relations->append(QPair(
-            QAccessible::queryAccessibleInterface(new Qt5XAccessible(xAccessible)), aQRelation));
+        relations->append(
+            { QAccessible::queryAccessibleInterface(new Qt5XAccessible(xAccessible)), aQRelation });
     }
 }
 }
