--- a/configure.in-old	2012-04-08 23:14:14.294482576 +0200
+++ b/configure.in	2012-04-08 23:16:50.266434228 +0200
@@ -6500,6 +6500,8 @@
                     [AC_MSG_ERROR(mysql_driver.h not found. install MySQL C++ Connectivity)], [])
         AC_CHECK_LIB([mysqlcppconn], [main], [:],
                     [AC_MSG_ERROR(MySQL C++ Connectivity lib not found or functional)], [])
+        save_LDFLAGS=$LDFLAGS
+        LDFLAGS="$LDFLAGS -lmysqlcppconn"
         AC_MSG_CHECKING([version])
         AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <mysql_driver.h>
@@ -6517,7 +6519,7 @@
       ]])],[AC_MSG_RESULT(OK)],[AC_MSG_ERROR([not suitable, we need >= 1.0.6])],[])
 
         AC_LANG_POP([C++])
-
+        LDFLAGS=$save_LDFLAGS
     else
         AC_MSG_RESULT([internal])
         AC_MSG_CHECKING([for mysqlcppconn module])
