diff --git a/bin/package-ooo b/bin/package-ooo
index 9127848..3d6b1a6 100755
--- libreoffice-3.3.0/libreoffice-build/bin/package-ooo
+++ libreoffice-3.3.0/libreoffice-build//bin/package-ooo
@@ -913,13 +913,23 @@ else
 			shift
 		done
 	}
-	move_wrappers pkg/libreoffice-common ooffice oofromtemplate soffice unopkg
-	move_wrappers pkg/libreoffice-base oobase
-	move_wrappers pkg/libreoffice-writer oowriter ooweb
-	move_wrappers pkg/libreoffice-calc oocalc
-	move_wrappers pkg/libreoffice-impress ooimpress
-	move_wrappers pkg/libreoffice-math oomath
-	move_wrappers pkg/libreoffice-draw oodraw
+	move_wrappers pkg/libreoffice-common soffice unopkg
+	if test "$COMPAT_OOWRAPPERS" = "YES" ; then
+		move_wrappers pkg/libreoffice-common ooffice oofromtemplate
+		move_wrappers pkg/libreoffice-base oobase
+		move_wrappers pkg/libreoffice-writer oowriter ooweb
+		move_wrappers pkg/libreoffice-calc oocalc
+		move_wrappers pkg/libreoffice-impress ooimpress
+		move_wrappers pkg/libreoffice-math oomath
+		move_wrappers pkg/libreoffice-draw oodraw
+	fi
+	move_wrappers pkg/libreoffice-common libreoffice lofromtemplate
+	move_wrappers pkg/libreoffice-base lobase
+	move_wrappers pkg/libreoffice-writer lowriter loweb
+	move_wrappers pkg/libreoffice-calc localc
+	move_wrappers pkg/libreoffice-impress loimpress
+	move_wrappers pkg/libreoffice-math lomath
+	move_wrappers pkg/libreoffice-draw lodraw
 	
 	# Move all libraries, binaries, *.rdb from -common to -core
 	for d in $OOINSTBASE/basis$VERSION/program $OOINSTBASE/program; do \
@@ -935,10 +945,14 @@ else
 
 	# install additional ooo-build scripts & misc stuff
 	mkdir -p pkg/libreoffice-common/usr/share/man/man1
-	mv usr/share/man/man1/openoffice$BINSUFFIX.1 \
+	if test "$COMPAT_OOWRAPPERS" = "YES" ; then
+		mv usr/share/man/man1/openoffice$BINSUFFIX.1 \
+			pkg/libreoffice-common/usr/share/man/man1
+	fi
+	mv usr/share/man/man1/libreoffice$BINSUFFIX.1 \
 		pkg/libreoffice-common/usr/share/man/man1
 	mkdir -p pkg/libreoffice-common/etc/bash_completion.d
-	mv etc/bash_completion.d/ooffice$BINSUFFIX.sh \
+	mv etc/bash_completion.d/libreoffice$BINSUFFIX.sh \
 		pkg/libreoffice-common/etc/bash_completion.d
 	mv .$OOINSTBASE/basis$VERSION/program/java-set-classpath \
 		pkg/libreoffice-common/$OOINSTBASE/program
