--- openoffice.org-3.2.0/ooo-build-3-2-0-10/src/sdf/calc-formula-option-tree.sdf-old	2010-04-30 14:03:49.000000000 +0200
+++ openoffice.org-3.2.0/ooo-build-3-2-0-10/src/sdf/calc-formula-option-tree.sdf	2010-04-30 14:35:25.000000000 +0200
@@ -1,6 +1,6 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	af	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ar	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	as-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	as	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	be-BY	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	bg	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	bn	Formula				2009-05-27 21:46:35
@@ -27,7 +27,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ga	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gl	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gu	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gu-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gu	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	he	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	hi-IN	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	hr	Formula				2009-05-27 21:46:35
@@ -45,10 +45,10 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	lt	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mai	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mk	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ml-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ml	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mn	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mni	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mr-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mr	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	my	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	nb	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ne	Formula				2009-05-27 21:46:35
@@ -57,7 +57,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	nr	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ns	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	oc	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	or-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	or	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	pa-IN	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	pl	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	pt	Formula				2009-05-27 21:46:35
@@ -76,8 +76,8 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	sv	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	sw-TZ	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ta	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ta-IN	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	te-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ta	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	te	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	tg	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	th	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	tn	Formula				2009-05-27 21:46:35
@@ -94,7 +94,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	zu	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	af	Sorteer lyste				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ar	فرز القوائم				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	as-IN	তালিকাবোৰ বৰ্গীকৰণ কৰক				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	as	তালিকাবোৰ বৰ্গীকৰণ কৰক				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	be-BY	Спісы парадкавання				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	bg	Списъци за сортиране				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	bn	বাছাইকৃত তালিকা				2009-05-27 21:46:35
@@ -121,7 +121,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ga	Sórtáil Liostaí				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gl	Listas de ordenación				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gu	યાદીને ગોઠવો				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gu-IN	યાદીને ગોઠવો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gu	યાદીને ગોઠવો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	he	מיון רשימות				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	hi-IN	छांटन सूची				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	hr	Razvrstaj popise				2009-05-27 21:46:35
@@ -140,10 +140,10 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	lt	Rikiavimas				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mai	छांटन सूची				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mk	Листи за подредување				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ml-IN	തരംതിരിക്കേണ്ട ലിസ്റ്റ്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ml	തരംതിരിക്കേണ്ട ലിസ്റ്റ്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mn	Жагсаалт эрэмбэлэх				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mni	মমিং পরিংশিং মথং-মনাও শেম্বা				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mr-IN	सूची वर्गीकृत करा				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mr	सूची वर्गीकृत करा				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	my	စာရင်းများ စနစ်တကျ စီထားခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	nb	Sorteringslister				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ne	सूचीहरू क्रमबद्ध गर्नुहोस्				2009-05-27 21:46:35
@@ -151,7 +151,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	nn	Sorteringslister				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	nr	Hlela amarhelo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ns	Lokiša Malokelelo				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	or-IN	ତାଲିକାଗୁଡିକ ସଜାଅ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	or	ତାଲିକାଗୁଡିକ ସଜାଅ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	pa-IN	ਕ੍ਰਮਬੱਧ ਸੂਚੀ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	pl	Listy sortowania				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	pt	Listas de ordenação				2009-05-27 21:46:35
@@ -170,8 +170,8 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	sv	Sorteringslistor				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	sw-TZ	Panga Orodha				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ta	மாற்றங்கள்				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ta-IN	மாற்றங்கள்				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	te-IN	 జాబితాలను చక్కదిద్దు				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ta	மாற்றங்கள்				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	te	 జాబితాలను చక్కదిద్దు				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	tg	Низомноккунии рӯйхатҳо				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	th	เรียงลำดับรายการ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	tn	Tlahola Listi				2009-05-27 21:46:35
@@ -188,7 +188,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	zu	Uhlu Lokuhlela				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	af	Wysigings				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ar	التغييرات				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	as-IN	সলনিবোৰ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	as	সলনিবোৰ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	be-BY	Змены				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	bg	Промени				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	bn	পরিবর্তনসমূহ				2009-05-27 21:46:35
@@ -215,7 +215,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ga	Athruithe				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gl	Cambios				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gu	બદલાવો				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gu-IN	ફેરફારો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gu	ફેરફારો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	he	שינויים				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	hi-IN	परिवर्तन				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	hr	Promjene				2009-05-27 21:46:35
@@ -233,10 +233,10 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	lt	Keitiniai				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mai	बदलाव				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mk	Промени				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ml-IN	മാറ്റങ്ങള്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ml	മാറ്റങ്ങള്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mn	Өөрчлөлт				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mni	অহোংবশিং				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mr-IN	बदल				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mr	बदल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	my	ပြောင်းလဲမှုများ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	nb	Endringer				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ne	परिवर्तनहरू				2009-05-27 21:46:35
@@ -245,7 +245,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	nr	Amatjhuguluko				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ns	Diphetogo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	oc	Modificacions				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	or-IN	ପରିବର୍ତ୍ତନଗୁଡିକ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	or	ପରିବର୍ତ୍ତନଗୁଡିକ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	pa-IN	ਤਬਦੀਲੀਆਂ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	pl	Zmiany				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	pt	Alterações				2009-05-27 21:46:35
@@ -264,8 +264,8 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	sv	Ändringar				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	sw-TZ	Mabadiliko				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ta	மாற்றங்கள்				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ta-IN	Changes				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	te-IN	మార్పులు				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ta	Changes				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	te	మార్పులు				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	tg	Тағирот				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	th	การเปลี่ยนแปลง				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	tr	Değişiklikler				2009-05-27 21:46:35
@@ -281,7 +281,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	zu	Ushintsho				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	af	Rooster				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ar	الشبكة				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	as-IN	জাল				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	as	জাল				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	be-BY	Рашотка				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	bg	Мрежа				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	bn	গরাদ				2009-05-27 21:46:35
@@ -308,7 +308,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ga	Greille				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gl	Grade				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gu	જાળી				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gu-IN	જાળી				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gu	જાળી				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	he	רשת				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	hi-IN	जाल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	hr	Mreža				2009-05-27 21:46:35
@@ -326,10 +326,10 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	lt	Tinklelis				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mai	जाल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mk	Мрежа				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ml-IN	ഗ്രിഡ്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ml	ഗ്രിഡ്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mn	Тор				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mni	গ্রীদ				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mr-IN	ग्रिड				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mr	ग्रिड				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	my	အကွက်ချခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	nb	Rutenett				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ne	ग्रिड				2009-05-27 21:46:35
@@ -338,7 +338,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	nr	Igridi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ns	Kiriti				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	oc	Grasilha				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	or-IN	ଜାଲ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	or	ଜାଲ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	pa-IN	ਗਰਿੱਡ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	pl	Siatka				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	pt	Grelha				2009-05-27 21:46:35
@@ -357,8 +357,8 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	sv	Raster				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	sw-TZ	Gridi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ta	கட்டம்				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ta-IN	Grid				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	te-IN	గ్రిడ్				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ta	Grid				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	te	గ్రిడ్				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	tg	Тӯр				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	th	เส้นตาราง				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	tr	Izgara				2009-05-27 21:46:35
@@ -374,7 +374,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	zu	Izikwedlana				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	af	Druk				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ar	طباعة				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	as-IN	মুদ্ৰণ কৰক				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	as	মুদ্ৰণ কৰক				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	be-BY	Друкаваць				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	bg	Печат				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	bn	মুদ্রণ				2009-05-27 21:46:35
@@ -401,7 +401,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ga	Priontáil				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gl	Imprimir				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gu	છાપો				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gu-IN	છાપો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gu	છાપો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	he	הדפסה				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	hi-IN	छपाई				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	hr	Ispis				2009-05-27 21:46:35
@@ -419,10 +419,10 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	lt	Spausdinimas				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mai	छपाइ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mk	Печатење				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ml-IN	അച്ചടിക്കുക				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ml	അച്ചടിക്കുക				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mn	Хэвлэлт				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mni	নমথোকপা				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mr-IN	छपाई				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mr	छपाई				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	my	ပရင့်ထုတ်ခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	nb	Skriv ut				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ne	मुद्रण गर्नुहोस्				2009-05-27 21:46:35
@@ -431,7 +431,7 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	nr	Phrinta				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ns	Kgatišo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	oc	Estampar				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	or-IN	ମୁଦ୍ରଣ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	or	ମୁଦ୍ରଣ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	pa-IN	ਛਾਪੋ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	pl	Drukowanie				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	pt	Imprimir				2009-05-27 21:46:35
@@ -450,8 +450,8 @@
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	sv	Skriv ut				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	sw-TZ	Print				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ta	அச்சிடு				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ta-IN	Print				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	te-IN	ముద్రణ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ta	Print				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	te	ముద్రణ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	tg	Чопкунӣ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	th	พิมพ์				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	tn	Gatisa				2009-05-27 21:46:35
