Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:09.430935091 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:09.422935092 +0100
@@ -8866,8 +8866,9 @@
     # some systems need this. Like Ubuntu....
     AC_CHECK_LIB(m, floor)
     AC_CHECK_LIB(dl, dlopen)
+    AC_CHECK_LIB(colamd, colamd)
     AC_CHECK_LIB([lpsolve55], [make_lp], [:],
-        [ AC_MSG_ERROR(lpsolve library not found or too old.)], [])
+        [ AC_MSG_ERROR(lpsolve library not found or too old.)], [-L/usr/lib/lpsolve])
     LIBS=$save_LIBS
     libo_MINGW_CHECK_DLL([lpsolve55])
 else
Index: libreoffice-4.1.4~rc2/RepositoryExternal.mk
===================================================================
--- libreoffice-4.1.4~rc2.orig/RepositoryExternal.mk	2013-12-16 14:35:09.430935091 +0100
+++ libreoffice-4.1.4~rc2/RepositoryExternal.mk	2013-12-16 14:35:09.426935092 +0100
@@ -1659,7 +1659,7 @@
 ifeq ($(SYSTEM_LPSOLVE),YES)
 
 define gb_LinkTarget__use_lpsolve
-$(call gb_LinkTarget_add_libs,$(1),-llpsolve55)
+$(call gb_LinkTarget_add_ldflags,$(1),-Wl$(COMMA)--no-as-needed /usr/lib/lp_solve/liblpsolve55.so -Wl$(COMMA)-rpath=/usr/lib/lp_solve)
 $(call gb_LinkTarget_add_defs,$(1),\
 	-DSYSTEM_LPSOLVE \
 )
