From f2a1882be486986a76682ed51f481595e1cf0d5c Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 16 Dec 2011 19:55:37 +0100
Subject: [PATCH 1/2] fix racecondition in sdext/source/pdfimport/makefile.mk

---
 sdext/source/pdfimport/makefile.mk |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/sdext/source/pdfimport/makefile.mk b/sdext/source/pdfimport/makefile.mk
index 641e90f..d33680f 100644
--- a/sdext/source/pdfimport/makefile.mk
+++ b/sdext/source/pdfimport/makefile.mk
@@ -156,11 +156,14 @@ $(COMPONENT_DIALOGS) : dialogs$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
     $(COPY) $< $@
 
-$(COMPONENT_DESCRIPTION) : $(DESCRIPTION)
+$(COMPONENT_DESCRIPTION) : $(MISC)/descriptions
+
+$(MISC)/descriptions : $(DESCRIPTION)
     $(COPY) description-en-US.txt $(EXTENSIONDIR)
 .IF "$(WITH_LANG)" != ""
     $(COPY) $(MISC)/$(EXTENSIONNAME)_in/description-*.txt $(EXTENSIONDIR)
 .ENDIF
+    $(TOUCH) $@
 
 $(COMPONENT_IMAGES) :  $(SOLARSRC)$/$(RSCDEFIMG)$/desktop$/res$/$$(@:f)
     @@-$(MKDIRHIER) $(@:d)
-- 
1.7.5.4

