diff --git a/xmlsecurity/Module_xmlsecurity.mk b/xmlsecurity/Module_xmlsecurity.mk
index d3fafeb7e4ec..771a9ec848d2 100644
--- a/xmlsecurity/Module_xmlsecurity.mk
+++ b/xmlsecurity/Module_xmlsecurity.mk
@@ -47,9 +47,11 @@ $(eval $(call gb_Module_add_screenshot_targets,xmlsecurity,\
 
 ifneq (,$(filter DESKTOP,$(BUILD_TYPE)))
 
+ifneq (,$(filter PDFIUM,$(BUILD_TYPE)))
 $(eval $(call gb_Module_add_targets,xmlsecurity,\
     Executable_pdfverify \
 ))
+endif
 
 endif
 
