diff --git a/configure.ac b/configure.ac
index e333538a2601..d55d4961866f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10716,7 +10716,6 @@ dnl ===================================================================
 dnl Check for building gnutls
 dnl ===================================================================
 AC_MSG_CHECKING([whether to use gnutls])
-if test "$WITH_WEBDAV" = "neon" -a "$with_system_neon" = no -a "$enable_openssl" = "no"; then
     AC_MSG_RESULT([yes])
     AM_PATH_LIBGCRYPT()
     PKG_CHECK_MODULES(GNUTLS, [gnutls],,
@@ -10724,9 +10723,6 @@ if test "$WITH_WEBDAV" = "neon" -a "$with_system_neon" = no -a "$enable_openssl"
                       available in the system to use as replacement.]]))
     FilterLibs "${LIBGCRYPT_LIBS}"
     LIBGCRYPT_LIBS="${filteredlibs}"
-else
-    AC_MSG_RESULT([no])
-fi
 
 AC_SUBST([LIBGCRYPT_CFLAGS])
 AC_SUBST([LIBGCRYPT_LIBS])
diff --git a/external/neon/configs/config.h b/external/neon/configs/config.h
index 3f93d1e68132..bd18d032afbc 100644
--- a/external/neon/configs/config.h
+++ b/external/neon/configs/config.h
@@ -89,6 +89,13 @@
 /* Define to 1 if you have the `gnutls_certificate_verify_peers2' function. */
 #define HAVE_GNUTLS_CERTIFICATE_VERIFY_PEERS2
 
+/* Define to 1 if you have the `gnutls_certificate_set_retrieve_function2'
+   function. */
+#define HAVE_GNUTLS_CERTIFICATE_SET_RETRIEVE_FUNCTION2
+
+/* Define to 1 if you have the `gnutls_privkey_import_ext' function. */
+#define HAVE_GNUTLS_PRIVKEY_IMPORT_EXT
+
 /* Define to 1 if you have the `gnutls_session_get_data2' function. */
 #define HAVE_GNUTLS_SESSION_GET_DATA2
 
