diff --git a/patches/dev300/apply b/patches/dev300/apply
index a6011bb..0ab7d9c 100644
--- openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/apply
+++ openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/apply
@@ -1173,6 +1173,9 @@ solenv-installer-cleaner-paths.diff, pmladek
 # for example, generate gid_Module_Langpack_Basis_es instead of
 # gid_Module_Langpack_Basis_es.es
 solenv-installer-lang-filelist-names.diff, pmladek
+# and fix  DEFAULT_TO_ENGLISH_FOR_PACKING so that it does not
+# create e.g. arrowhd_en-US_as.soe
+DEFAULT_TO_ENGLISH_FOR_PACKING-filenames.diff, i#110901
 
 # allow to get path also from ./file
 # otherwise, it installed the license files into ugly subdirectories
--- /dev/null	2010-04-14 03:44:50.008165136 +0200
+++ openoffice.org-3.2.0/ooo-build-3-2-0-10/patches/dev300/DEFAULT_TO_ENGLISH_FOR_PACKING-filenames.diff	2010-04-18 03:16:17.000000000 +0200
@@ -0,0 +1,48 @@
+
+# HG changeset patch
+# User Ingo Schmidt <is@openoffice.org>
+# Date 1271408858 -7200
+# Node ID bc7556cee857016f7def0f3780a9dfb222cb0dff
+# Parent  19b8efb810452b843cf6242713a002a688c96f2a
+native299 #i110901# improving mechanism for DEFAULT_TO_ENGLISH_FOR_PACKING
+
+diff -r 19b8efb81045 -r bc7556cee857 solenv/bin/modules/installer/archivefiles.pm
+--- solenv/bin/modules/installer/archivefiles.pm	Fri Apr 16 10:40:37 2010 +0200
++++ solenv/bin/modules/installer/archivefiles.pm	Fri Apr 16 11:07:38 2010 +0200
+@@ -50,19 +50,27 @@
+ 	my $filename = "";
+ 	my $extension = "";
+ 	
+-	if ( $oldname =~ /^\s*(.*)(\..*?)\s*$/ )	# files with extension
++	if ( $oldname =~ /en-US/ )	# files, that contain the language in the file name
+ 	{
+-		$filename = $1;
+-		$extension = $2;	
++		$newname = $oldname;
++		$newname =~ s/en-US/$onelanguage/;
+ 	}
+-	else
++	else	# files, that do not contain the language in the file name
+ 	{
+-		$filename = $oldname;
+-		$extension = "";	
++		if ( $oldname =~ /^\s*(.*)(\..*?)\s*$/ )	# files with extension
++		{
++			$filename = $1;
++			$extension = $2;	
++		}
++		else
++		{
++			$filename = $oldname;
++			$extension = "";	
++		}
++	
++		$newname = $1 . "_" . $onelanguage . $2;
+ 	}
+ 	
+-	$newname = $1 . "_" . $onelanguage . $2;
+-	
+ 	return $newname;
+ }
+ 
+
