Index: libreoffice-4.1.4~rc2/bin/lo-xlate-lang
===================================================================
--- libreoffice-4.1.4~rc2.orig/bin/lo-xlate-lang	2013-12-16 14:35:31.586933914 +0100
+++ libreoffice-4.1.4~rc2/bin/lo-xlate-lang	2013-12-16 14:35:31.578933914 +0100
@@ -157,7 +157,7 @@
 :ka:georgian
 :eo:esperanto
 :uk:ukrainian
-:kk:kazahk
+:kk:kazakh
 :dz:dzongkha
 :kn:kannada
 :gl:galician
