diff --git a/sd/Library_sd.mk b/sd/Library_sd.mk
index 9d399e8e3f5a..d70b5216fe05 100644
--- a/sd/Library_sd.mk
+++ b/sd/Library_sd.mk
@@ -99,7 +99,6 @@ $(eval $(call gb_Library_use_libraries,sd,\
 ))
 
 $(eval $(call gb_Library_use_externals,sd,\
-	bluez_bluetooth_headers \
 	boost_headers \
 	libxml2 \
 	dbus \
@@ -575,6 +574,10 @@ $(eval $(call gb_Library_add_defs,sd,\
 	-DENABLE_SDREMOTE_BLUETOOTH \
 ))
 
+$(eval $(call gb_Library_use_externals,sd,\
+	bluez_bluetooth_headers \
+))
+
 endif # ENABLE_SDREMOTE_BLUETOOTH=TRUE
 
 endif # ENABLE_SDREMOTE=TRUE
