diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index 2db4c38..8ea4498 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -3190,10 +3190,12 @@ endef
 else # !SYSTEM_BSH
 
 ifeq ($(ENABLE_JAVA),TRUE)
+ifeq ($(ENABLE_SCRIPTING_BEANSHELL),TRUE)
 $(eval $(call gb_Helper_register_jars_for_install,OOO,extensions_bsh, \
 	bsh \
 ))
 endif
+endif
 
 define gb_Jar__use_bsh
 $(call gb_Jar_use_jar,$(1),bsh)
@@ -3211,10 +3213,12 @@ endef
 else # !SYSTEM_RHINO
 
 ifeq ($(ENABLE_JAVA),TRUE)
+ifeq ($(ENABLE_SCRIPTING_JAVASCRIPT),TRUE)
 $(eval $(call gb_Helper_register_jars_for_install,OOO,extensions_rhino, \
 	js \
 ))
 endif
+endif
 
 define gb_Jar__use_rhino
 $(call gb_Jar_use_jar,$(1),js)
