diff --git a/configure.ac b/configure.ac
index a1bbc7b1c9cc..fc7a6d452b89 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1772,6 +1772,11 @@ libo_FUZZ_ARG_ENABLE(gstreamer-1-0,
         [Disable building with the gstreamer 1.0 avmedia backend.]),
 ,test "${enable_gstreamer_1_0+set}" = set || enable_gstreamer_1_0=yes)
 
+libo_FUZZ_ARG_ENABLE([eot],
+    [AS_HELP_STRING([--enable-eot],
+        [Enable support for Embedded OpenType fonts.])],
+,test "${enable_eot+set}" = set || enable_eot=no)
+
 libo_FUZZ_ARG_ENABLE(cve-tests,
     AS_HELP_STRING([--disable-cve-tests],
         [Prevent CVE tests to be executed]),
@@ -9568,7 +9613,14 @@ dnl Check for system xmlsec
 dnl ===================================================================
 libo_CHECK_SYSTEM_MODULE([xmlsec], [XMLSEC], [xmlsec1-nss >= 1.2.28])
 
-libo_CHECK_SYSTEM_MODULE([eot],[LIBEOT],[libeot >= 0.01],disabled)
+AC_MSG_CHECKING([whether to enable Embedded OpenType support])
+if test "$test_eot" != "no" -a "$enable_eot" = "yes"; then
+    AC_MSG_RESULT([yes])
+    libo_CHECK_SYSTEM_MODULE([libeot],[LIBEOT],[libeot >= 0.01])
+    AC_DEFINE([ENABLE_LIBEOT])
+else
+    AC_MSG_RESULT([no])
+fi
 
 dnl ===================================================================
 dnl Check for DLP libs
