diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index 2db4c38..a56264d 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -3181,6 +3181,8 @@ endef
 endif # SYSTEM_HSQLDB
 
 
+ifeq ($(ENABLE_SCRIPTING_BEANSHELL),TRUE)
+
 ifneq ($(SYSTEM_BSH),)
 
 define gb_Jar__use_bsh
@@ -3201,6 +3203,9 @@ endef
 
 endif # SYSTEM_BSH
 
+endif
+
+ifeq ($(ENABLE_SCRIPTING_JAVASCRIPT),TRUE)
 
 ifneq ($(SYSTEM_RHINO),)
 
@@ -3222,6 +3227,8 @@ endef
 
 endif # SYSTEM_RHINO
 
+endif
+
 ifneq ($(SYSTEM_APACHE_COMMONS),)
 
 define gb_Jar__use_commons-codec
