diff --git a/bin/build-ooo b/bin/build-ooo
index 6a4cd28..4553a94 100755
--- openoffice.org-3.2.1/ooo-build-3-2-1-4/bin/build-ooo
+++ openoffice.org-3.2.1/ooo-build-3-2-1-4/bin/build-ooo
@@ -75,12 +75,11 @@ else
     . $OOBUILDDIR/*.Set.sh
 fi
 
-# Check to see if dmake was not there - if not, this is a from-pristine build
-echo "Build: z$BUILD_DMAKE == zYES -a ! -f $SOLARENV/$OUTPATH/bin/dmake";
-if test "z$BUILD_DMAKE" == "zYES" -a ! -f "$SOLARENV/$OUTPATH/bin/dmake"; then
+# Check to see if makedepend was not there - if not, this is a from-pristine build
+if test ! -f $OOBUILDDIR/soltools/$OUTPATH/bin/makedepend; then 
 	# Accelerate straight-through compiles by not building / using dependencies
 	echo "A very clean straight-through build - deferring dep generation"
-	export nodep=1
+	export nodep=true
 else
     echo "Building with full dependencies"
 fi
