--- a/extensions/source/nsplugin/source/makefile.mk-old	2012-01-25 09:00:55.000000000 +0000
+++ b/extensions/source/nsplugin/source/makefile.mk	2012-01-25 09:01:04.000000000 +0000
@@ -36,8 +36,6 @@
 
 .INCLUDE :  	settings.mk
 
-.IF "$(WITH_MOZILLA)" != "NO"
-
 .IF "$(SYSTEM_MOZILLA_HEADERS)" == "YES"
 INCPRE+=$(MOZILLA_HEADERS_CFLAGS)
 .ELSE
@@ -160,8 +158,6 @@
 DEF2EXPORTFILE=exports_wnt.dxp
 .ENDIF
 
-.ENDIF # $(WITH_MOZILLA) != "NO"
-
 # --- Targets ----------------------------------
 
 .INCLUDE : target.mk
--- a/scp2/source/ooo/common_brand.scp-old	2012-01-25 09:14:08.000000000 +0000
+++ b/scp2/source/ooo/common_brand.scp	2012-01-25 09:14:19.000000000 +0000
@@ -642,7 +642,7 @@
 End
 #endif
 
-#if !defined WITHOUT_MOZILLA && defined ENABLE_NSPLUGIN
+#if defined ENABLE_NSPLUGIN
 File gid_Brand_File_Lib_Npsoplugin
     BIN_FILE_BODY;
     Name = SPECIAL_NAME(npsoplugin);
--- a/scp2/source/ooo/file_ooo.scp-old	2012-01-25 09:12:06.000000000 +0000
+++ b/scp2/source/ooo/file_ooo.scp	2012-01-25 09:12:58.000000000 +0000
@@ -108,8 +108,6 @@
     Name = "basicsrvtutorials.zip";
 End
 
-#ifndef WITHOUT_MOZILLA
-
 #ifdef ENABLE_NSPLUGIN
 File gid_File_Exe_Nsplugin
         BIN_FILE_BODY;
@@ -122,7 +120,6 @@
   #endif
 End
 #endif
-#endif
 
 File gid_File_Bin_Gengal
     BIN_FILE_BODY;
