diff --git a/sd/qa/unit/tiledrendering/tiledrendering.cxx b/sd/qa/unit/tiledrendering/tiledrendering.cxx
index e8b6e4c497eb..cf6e1210256b 100644
--- a/sd/qa/unit/tiledrendering/tiledrendering.cxx
+++ b/sd/qa/unit/tiledrendering/tiledrendering.cxx
@@ -2840,8 +2840,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering)
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     const Color aTransparentColor(ColorAlpha, 0x00000000);
     const tools::Long nLeft = 20;
@@ -2958,8 +2958,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_WithFie
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     const Color aTransparentColor(ColorAlpha, 0x00000000);
     const tools::Long nLeft = 20;
@@ -3168,8 +3168,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_SlideNu
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     // Background Layer
     {
@@ -3381,8 +3381,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
     sal_Int32 nViewWidth = 2000;
     sal_Int32 nViewHeight = 2000;
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -3553,8 +3553,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
     sal_Int32 nViewHeight = 2000;
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -3815,8 +3815,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -3982,8 +3982,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1125, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1125, static_cast<int>(nViewHeight));
 
     {
         std::vector<sal_uInt8> pBuffer(nViewWidth * nViewHeight * 4);
@@ -4087,8 +4087,8 @@ CPPUNIT_TEST_FIXTURE(SdTiledRenderingTest, testSlideshowLayeredRendering_Animati
 
     std::string sHash = GetInterfaceHash(GetXDrawPageForSdrPage(pPage));
     CPPUNIT_ASSERT(pXImpressDocument->createSlideRenderer(sHash.c_str(), 0, nViewWidth, nViewHeight, true, true));
-    CPPUNIT_ASSERT_EQUAL(2000, nViewWidth);
-    CPPUNIT_ASSERT_EQUAL(1500, nViewHeight);
+    CPPUNIT_ASSERT_EQUAL(2000, static_cast<int>(nViewWidth));
+    CPPUNIT_ASSERT_EQUAL(1500, static_cast<int>(nViewHeight));
 
     std::vector<std::string> aBufferHashes;
 
