--- openoffice.org-3.2.0/ooo-build-3-2-0-99-3/src/sdf/calc-formula-option-tree.sdf-old	2010-04-30 14:03:49.000000000 +0200
+++ openoffice.org-3.2.0/ooo-build-3-2-0-99-3/src/sdf/calc-formula-option-tree.sdf	2010-04-30 14:35:25.000000000 +0200
@@ -1,4 +1,4 @@
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	as-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	as	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	br	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	brx	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	dgo	Formula				2009-05-27 21:46:35
@@ -8,24 +8,24 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	eu	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	fa	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ga	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gu-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	gu	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	hi-IN	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	kk	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	kn	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	kok	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ks	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mai	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ml-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ml	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mn	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mni	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mr-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	mr	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	my	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ne	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	nn	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	nr	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ns	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	oc	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	or-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	or	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	pa-IN	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	rw	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	sa-IN	Formula				2009-05-27 21:46:35
@@ -34,14 +34,14 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ss	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	st	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	sw-TZ	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ta-IN	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	te-IN	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ta	Formula				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	te	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	tn	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ts	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ur-IN	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	uz	Formula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	6			0	ve	Formula				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	as-IN	তালিকাবোৰ বৰ্গীকৰণ কৰক				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	as	তালিকাবোৰ বৰ্গীকৰণ কৰক				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	br	Rolloù rummañ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	brx	थाखो फारिलाइफोर				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	dgo	सूचियां छांटो				2009-05-27 21:46:35
@@ -51,7 +51,7 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	eu	Ordenatze-zerrendak				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	fa	مرتب‌سازی فهرست‌ها				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ga	Sórtáil Liostaí				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gu-IN	યાદીને ગોઠવો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	gu	યાદીને ગોઠવો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	hi-IN	छांटन सूची				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	kk	Сұрыптау тiзiмi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	kn	ವಿಂಗಡನಾ ಪಟ್ಟಿಗಳು				2009-05-27 21:46:35
@@ -59,16 +59,16 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ks	فہرست كریو سارٹ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ku	Lîsteya Curetîpan				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mai	छांटन सूची				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ml-IN	തരംതിരിക്കേണ്ട ലിസ്റ്റ്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ml	തരംതിരിക്കേണ്ട ലിസ്റ്റ്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mn	Жагсаалт эрэмбэлэх				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mni	মমিং পরিংশিং মথং-মনাও শেম্বা				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mr-IN	सूची वर्गीकृत करा				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	mr	सूची वर्गीकृत करा				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	my	စာရင်းများ စနစ်တကျ စီထားခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ne	सूचीहरू क्रमबद्ध गर्नुहोस्				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	nn	Sorteringslister				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	nr	Hlela amarhelo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ns	Lokiša Malokelelo				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	or-IN	ତାଲିକାଗୁଡିକ ସଜାଅ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	or	ତାଲିକାଗୁଡିକ ସଜାଅ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	pa-IN	ਕ੍ਰਮਬੱਧ ਸੂਚੀ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	rw	Gushungura Amalisiti				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	sa-IN	सूच्यः क्रमणं कुरु				2009-05-27 21:46:35
@@ -77,14 +77,14 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ss	Lungisa tinhlu				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	st	Hlopha mathathamo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	sw-TZ	Panga Orodha				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ta-IN	மாற்றங்கள்				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	te-IN	 జాబితాలను చక్కదిద్దు				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ta	மாற்றங்கள்				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	te	 జాబితాలను చక్కదిద్దు				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	tn	Tlahola Listi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ts	Minongonoko yo hlawula				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ur-IN	فہرستوں کی اقسام بندی				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	uz	Roʻyxatlarni saralash				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	7			0	ve	Sota Mitevhe				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	as-IN	সলনিবোৰ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	as	সলনিবোৰ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	br	Daskemmadurioù				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	brx	सोलायनायफोर				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	dgo	तब्दीलियां				2009-05-27 21:46:35
@@ -94,24 +94,24 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	eu	Aldaketak				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	fa	تغییرات				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ga	Athruithe				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gu-IN	ફેરફારો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	gu	ફેરફારો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	hi-IN	परिवर्तन				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	kn	ಬದಲಾವಣೆಗಳು				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	kok	बदल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ks	تبدیلی				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ku	Guhertin				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mai	बदलाव				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ml-IN	മാറ്റങ്ങള്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ml	മാറ്റങ്ങള്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mn	Өөрчлөлт				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mni	অহোংবশিং				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mr-IN	बदल				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	mr	बदल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	my	ပြောင်းလဲမှုများ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ne	परिवर्तनहरू				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	nn	Endringar				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	nr	Amatjhuguluko				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ns	Diphetogo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	oc	Modificacions				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	or-IN	ପରିବର୍ତ୍ତନଗୁଡିକ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	or	ପରିବର୍ତ୍ତନଗୁଡିକ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	pa-IN	ਤਬਦੀਲੀਆਂ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	rw	Amahinduka				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	sa-IN	परिणामाः				2009-05-27 21:46:35
@@ -120,13 +120,13 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ss	Tingucuko				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	st	Diphetoho				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	sw-TZ	Mabadiliko				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ta-IN	Changes				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	te-IN	మార్పులు				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ta	Changes				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	te	మార్పులు				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ts	Ku cinca				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ur-IN	 تبدیلیاں				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	uz	Oʻzgarishlar				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	8			0	ve	Dzitshanduko				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	as-IN	জাল				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	as	জাল				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	br	Kael				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	brx	ग्रिड				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	dgo	ग्रिड 				2009-05-27 21:46:35
@@ -136,24 +136,24 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	eu	Sareta				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	fa	توری				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ga	Greille				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gu-IN	જાળી				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	gu	જાળી				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	hi-IN	जाल				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	kn	ಚೌಕಜಾಲ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	kok	ग्रिड				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ks	گریڈ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ku	Rêber				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mai	जाल				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ml-IN	ഗ്രിഡ്				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ml	ഗ്രിഡ്				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mn	Тор				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mni	গ্রীদ				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mr-IN	ग्रिड				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	mr	ग्रिड				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	my	အကွက်ချခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ne	ग्रिड				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	nn	Rutenett				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	nr	Igridi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ns	Kiriti				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	oc	Grasilha				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	or-IN	ଜାଲ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	or	ଜାଲ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	pa-IN	ਗਰਿੱਡ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	rw	Urusobe-tudirishya				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	sa-IN	जालम्				2009-05-27 21:46:35
@@ -162,13 +162,13 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ss	Igridi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	st	Kerite				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	sw-TZ	Gridi				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ta-IN	Grid				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	te-IN	గ్రిడ్				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ta	Grid				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	te	గ్రిడ్				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ts	Giridi				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ur-IN	جال				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	uz	Toʻr				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	9			0	ve	Giridi				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	as-IN	মুদ্ৰণ কৰক				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	as	মুদ্ৰণ কৰক				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	br	Moulladur				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	brx	साफाय				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	dgo	छापा				2009-05-27 21:46:35
@@ -178,24 +178,24 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	eu	Inprimatu				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	fa	چاپ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ga	Priontáil				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gu-IN	છાપો				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	gu	છાપો				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	hi-IN	छपाई				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	kn	ಮುದ್ರಿಸು				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	kok	छापात				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ks	چھپئویو				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ku	Çapkirin				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mai	छपाइ				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ml-IN	അച്ചടിക്കുക				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ml	അച്ചടിക്കുക				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mn	Хэвлэлт				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mni	নমথোকপা				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mr-IN	छपाई				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	mr	छपाई				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	my	ပရင့်ထုတ်ခြင်း				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ne	मुद्रण गर्नुहोस्				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	nn	Skriv ut				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	nr	Phrinta				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ns	Kgatišo				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	oc	Estampar				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	or-IN	ମୁଦ୍ରଣ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	or	ମୁଦ୍ରଣ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	pa-IN	ਛਾਪੋ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	rw	Gucapa				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	sa-IN	मुद्रणम्				2009-05-27 21:46:35
@@ -204,8 +204,8 @@ svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITO
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ss	Kuphrinta				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	st	Printa				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	sw-TZ	Print				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ta-IN	Print				2009-05-27 21:46:35
-svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	te-IN	ముద్రణ				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ta	Print				2009-05-27 21:46:35
+svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	te	ముద్రణ				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	tn	Gatisa				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ts	Printa				2009-05-27 21:46:35
 svx	source\cui\treeopt.src	0	itemlist	RID_OFADLG_OPTIONS_TREE_PAGES.SID_SC_EDITOPTIONS	10			0	ur-IN	چھپائی				2009-05-27 21:46:35
