From 9daa6e293d24694cb6102a15f228a6c008acf79d Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Tue, 2 May 2017 23:07:45 +0200
Subject: fix file name

Change-Id: I43f4f0281d83c402fb7ced9fe0b01ef8a59d7aa5

diff --git a/writerperfect/qa/unit/WpftImpressFilterTest.cxx b/writerperfect/qa/unit/WpftImpressFilterTest.cxx
index ebcc1f1..2da1195 100644
--- a/writerperfect/qa/unit/WpftImpressFilterTest.cxx
+++ b/writerperfect/qa/unit/WpftImpressFilterTest.cxx
@@ -43,7 +43,7 @@ void WpftImpressFilterTest::test()
         {"PowerPoint_Mac_1", REQUIRE_MWAW_VERSION(0, 3, 9)},
         {"PowerPoint_Mac_2", REQUIRE_MWAW_VERSION(0, 3, 9)},
         {"PowerPoint_Mac_3", REQUIRE_MWAW_VERSION(0, 3, 9)},
-        {"PowerPoint_Mac_4", REQUIRE_MWAW_VERSION(0, 3, 10)},
+        {"PowerPoint_Mac_4.ppt", REQUIRE_MWAW_VERSION(0, 3, 10)},
         {"PowerPoint_2.ppt", REQUIRE_MWAW_VERSION(0, 3, 10)},
         {"PowerPoint_3.ppt", REQUIRE_MWAW_VERSION(0, 3, 9)},
         {"PowerPoint_4.ppt", REQUIRE_MWAW_VERSION(0, 3, 10)},
-- 
cgit v0.10.2

