diff --git a/vcl/unx/gtk/a11y/atktextattributes.cxx b/vcl/unx/gtk/a11y/atktextattributes.cxx
index bc286d0..6a2d9c4 100644
--- a/vcl/unx/gtk/a11y/atktextattributes.cxx
+++ b/vcl/unx/gtk/a11y/atktextattributes.cxx
@@ -1235,7 +1235,7 @@ attribute_set_new_from_extended_attributes(
 
         pSet = attribute_set_prepend( pSet,
                                       atk_text_attribute_register( sPropertyName.getStr() ),
-                                      g_strdup_printf( sPropertyValue.getStr() ) );
+                                      g_strdup_printf( "%s", sPropertyValue.getStr() ) );
     }
     while ( nIndex >= 0 && nIndex < sExtendedAttrs.getLength() );
 
