--- a/solenv/gbuild/JavaClassSet.mk-old	2012-01-19 12:30:15.000000000 +0000
+++ b/solenv/gbuild/JavaClassSet.mk	2012-01-19 12:30:29.000000000 +0000
@@ -26,7 +26,7 @@
 #
 #*************************************************************************
 
-gb_JavaClassSet_JAVACCOMMAND := $(JAVACOMPILER)
+gb_JavaClassSet_JAVACCOMMAND := $(JAVACOMPILER) $(JAVAFLAGS)
 gb_JavaClassSet_JAVACDEBUG :=
 
 ifneq ($(gb_DEBUGLEVEL),0)
