diff --git a/configure.ac b/configure.ac
index c4fefea..2ac49bb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3474,6 +3474,11 @@ else
 fi
 AC_SUBST(GCC_VERSION)
 
+if test "$GCC_VERSION" -lr 0409; then
+        CFLAGS=`echo $CFLAGS | sed -e s/-fstack-protector-strong/-fstack-protector/`
+        CXXFLAGS=`echo $CXXFLAGS | sed -e s/-fstack-protector-strong/-fstack-protector/`
+fi
+
 # ===================================================================
 # check various GCC options that Clang does not support now but maybe
 # will somewhen in the future, check them even for GCC, so that the
