diff --git a/configure.ac b/configure.ac
index e6f78a7..0660940 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6331,7 +6337,8 @@ if test "$SOLAR_JAVA" != ""; then
     if test "$JDK" = "gcj"; then
         save_CFLAGS=$CFLAGS
         save_LDFLAGS=$LDFLAGS
-        CFLAGS="$CFLAGS -I$JAVA_HOME/include"
+        GCJ_JNI_INCLUDE=-I`$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`
+        CFLAGS="$CFLAGS -I$JAVA_HOME/include $GCJ_JNI_INCLUDE"
         LDFLAGS="$LDFLAGS -L$JAVA_HOME/lib -lgcj"
         exec 6>/dev/null # no output
         AC_CHECK_HEADER(jni.h, [],
@@ -6658,6 +6665,7 @@ AC_SUBST(JDK)
 AC_SUBST(JDKLIB)
 AC_SUBST(x_JAVALIB)
 AC_SUBST(x_JDKLIB)
+AC_SUBST(GCJ_JNI_INCLUDE)
 
 
 AC_MSG_CHECKING([which C preprocessor to use in idlc])
diff --git a/config_host.mk.in b/config_host.mk.in
index 57e0574..1d169b9 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -469,7 +469,7 @@ export SCPDEFS=@SCPDEFS@
 export SDKDIRNAME=@SDKDIRNAME@
 export SDKDIRNAME_FOR_BUILD=@SDKDIRNAME_FOR_BUILD@
 export SHOWINCLUDES_PREFIX=@SHOWINCLUDES_PREFIX@
-export SOLARINC=@SOLARINC@
+export SOLARINC=@SOLARINC@ @GCJ_JNI_INCLUDE@
 export SOLARLIB=@SOLARLIB@
 export SPLIT_APP_MODULES=@SPLIT_APP_MODULES@
 export SPLIT_OPT_FEATURES=@SPLIT_OPT_FEATURES@
