--- a/configure.ac-old	2015-02-19 15:32:27.000000000 +0100
+++ b/configure.ac	2015-02-19 15:38:58.709444812 +0100
@@ -6670,6 +6670,7 @@ if test "$ENABLE_JAVA" != ""; then
         fi
         JAVAINTERPRETER=`win_short_path_for_make "$JAVAINTERPRETER"`
     elif test "$cross_compiling" != "yes"; then
+      if [ "$JAVAINTERPRETER" != "/usr/lib/jvm/java-gcj/bin/java" ]; then
         case $CPUNAME in
             AARCH64|AXP|X86_64|IA64|POWERPC64|S390X|SPARC64|GODSON64)
                 if test -f "$JAVAINTERPRETER" -a "`$JAVAINTERPRETER -version 2>&1 | $GREP -i 64-bit`" = "" >/dev/null; then
@@ -6684,6 +6685,7 @@ if test "$ENABLE_JAVA" != ""; then
                 fi
                 ;;
         esac
+      fi
     fi
 fi
 
