Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:05.330935309 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:05.326935310 +0100
@@ -6428,7 +6428,8 @@
     if test "$JDK" = "gcj"; then
         save_CFLAGS=$CFLAGS
         save_LDFLAGS=$LDFLAGS
-        CFLAGS="$CFLAGS -I$JAVA_HOME/include"
+        GCJ_JNI_INCLUDE=-I`$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`
+        CFLAGS="$CFLAGS -I$JAVA_HOME/include $GCJ_JNI_INCLUDE"
         LDFLAGS="$LDFLAGS -L$JAVA_HOME/lib -lgcj"
         exec 6>/dev/null # no output
         AC_CHECK_HEADER(jni.h, [],
@@ -6768,6 +6769,7 @@
 AC_SUBST(JDKLIB)
 AC_SUBST(x_JAVALIB)
 AC_SUBST(x_JDKLIB)
+AC_SUBST(GCJ_JNI_INCLUDE)
 
 
 AC_MSG_CHECKING([which C preprocessor to use in idlc])
Index: libreoffice-4.1.4~rc2/config_host.mk.in
===================================================================
--- libreoffice-4.1.4~rc2.orig/config_host.mk.in	2013-12-16 14:35:05.330935309 +0100
+++ libreoffice-4.1.4~rc2/config_host.mk.in	2013-12-16 14:35:05.326935310 +0100
@@ -437,7 +437,7 @@
 export SERVLETAPI_JAR=@SERVLETAPI_JAR@
 export SHOWINCLUDES_PREFIX=@SHOWINCLUDES_PREFIX@
 export SOLARENV=@SRC_ROOT@/solenv
-export SOLARINC=@SOLARINC@
+export SOLARINC=@SOLARINC@ @GCJ_JNI_INCLUDE@
 export SOLARLIB=@SOLARLIB@
 export SOLARVER=@SOLARVER@
 export SOLARVERSION=@SOLARVER@
