Index: libreoffice-3.6.0~alpha0~git20120508/configure.in
===================================================================
--- libreoffice-3.6.0~alpha0~git20120508.orig/configure.in	2012-05-30 10:14:13.994261397 +0200
+++ libreoffice-3.6.0~alpha0~git20120508/configure.in	2012-05-30 10:14:14.862261410 +0200
@@ -5448,7 +5448,8 @@
     if test "$JDK" = "gcj"; then
         save_CFLAGS=$CFLAGS
         save_LDFLAGS=$LDFLAGS
-        CFLAGS="$CFLAGS -I$JAVA_HOME/include"
+        GCJ_JNI_INCLUDE=`$JAVA_HOME/bin/gcj --print-file-name include/jni.h | $SED -e s,\/jni.h,,g`
+        CFLAGS="$CFLAGS -I$JAVA_HOME/include -I$GCJ_JNI_INCLUDE"
         LDFLAGS="$LDFLAGS -L$JAVA_HOME/lib -lgcj"
         exec 6>/dev/null # no output
         AC_CHECK_HEADER(jni.h, [],
@@ -5782,6 +5783,7 @@
 AC_SUBST(XCLASSPATH)
 AC_SUBST(x_JAVALIB)
 AC_SUBST(x_JDKLIB)
+AC_SUBST(GCJ_JNI_INCLUDE)
 
 dnl ===================================================================
 dnl Checks for specific files.
Index: libreoffice-3.6.0~alpha0~git20120508/config_host.mk.in
===================================================================
--- libreoffice-3.6.0~alpha0~git20120508.orig/config_host.mk.in	2012-05-30 10:14:13.994261397 +0200
+++ libreoffice-3.6.0~alpha0~git20120508/config_host.mk.in	2012-05-30 10:14:14.862261410 +0200
@@ -442,8 +442,8 @@
 export SIZEOF_POINTER=@SIZEOF_POINTER@
 export SIZEOF_SHORT=@SIZEOF_SHORT@
 export SOLARENV=@SRC_ROOT@/solenv
-export SOLARINC=@SOLARINC@
-export SOLARINC_FOR_BUILD=@SOLARINC_FOR_BUILD@
+export SOLARINC=@SOLARINC@ @GCJ_JNI_INCLUDE@
+export SOLARINC_FOR_BUILD=@SOLARINC_FOR_BUILD@ @GCJ_JNI_INCLUDE@
 export SOLARLIB=@SOLARLIB@
 export SOLARLIB_FOR_BUILD=@SOLARLIB_FOR_BUILD@
 export SOLARVER=@SOLARVER@
