--- a/configure.ac-old	2015-02-19 15:32:27.000000000 +0100
+++ b/configure.ac	2015-02-19 15:38:58.709444812 +0100
@@ -6584,6 +6584,7 @@
         fi
         JAVAINTERPRETER=`win_short_path_for_make "$JAVAINTERPRETER"`
     elif test "$cross_compiling" != "yes"; then
+      if [ "$JAVAINTERPRETER" != "/usr/lib/jvm/java-gcj/bin/java" ]; then
         # at least 2 reasons to check: officebean needs to link -ljawt,
         # and libjpipe.so needs to be loaded by java to run JunitTests.
         case $CPUNAME in
@@ -6600,6 +6601,7 @@
                 fi
                 ;;
         esac
+      fi
     fi
 fi
 
