diff --git a/config_host.mk.in b/config_host.mk.in
index d7ede5a..8a8f425 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -263,6 +263,7 @@ export JAVACISGCJ=@JAVACISGCJ@
 export JAVACISKAFFE=@JAVACISKAFFE@
 export JAVACOMPILER=@JAVACOMPILER@
 export JAVADOC=@JAVADOC@
+export JAVADOCISGJDOC=@JAVADOCISGJDOC@
 @x_JAVALIB@ export JAVALIB=@JAVALIB@
 export JAVA_HOME=@JAVA_HOME@
 export JAVA_SOURCE_VER=@JAVA_SOURCE_VER@
diff --git a/configure.ac b/configure.ac
index 0cbc7e8..7930ad8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5987,7 +5987,12 @@ if test "$SOLAR_JAVA" != ""; then
         JAVADOC=`cygpath -d "$JAVADOC"`
         JAVADOC=`cygpath -u "$JAVADOC"`
     fi
+
+    if test `$JAVADOC --version 2>&1 | $GREP -c "gjdoc"` -gt 0; then
+	JAVADOCISGJDOC="yes"
+    fi
 fi
+AC_SUBST(JAVADOCISGJDOC)
 
 if test "$SOLAR_JAVA" != ""; then
     # check if JAVA_HOME was (maybe incorrectly?) set automatically to /usr
diff --git a/odk/pack/gendocu/makefile.mk b/odk/pack/gendocu/makefile.mk
index 7345e72..6d3cc03 100644
--- a/odk/pack/gendocu/makefile.mk
+++ b/odk/pack/gendocu/makefile.mk
@@ -59,7 +59,10 @@ JAVA_BEAN_SRC_FILES=\
         $(JAVA_SRC_DIR)$/com$/sun$/star$/beans$/OfficeConnection.java \
         $(JAVA_SRC_DIR)$/com$/sun$/star$/beans$/OfficeWindow.java
 
-JAVADOCPARAMS= -notimestamp -use -splitindex -windowtitle "Java UNO Runtime Reference" -header $(JAVADOCREFNAME) -d $(DESTDIRGENJAVAREF) -sourcepath $(JAVA_SRC_DIR) -classpath $(SOLARBINDIR)$/ridl.jar -linkoffline ../../common/ref ./uno -linkoffline http://java.sun.com/j2se/1.5/docs/api ./java $(JAVA_PACKAGES)
+JAVADOCPARAMS= -use -splitindex -windowtitle "Java UNO Runtime Reference" -header $(JAVADOCREFNAME) -d $(DESTDIRGENJAVAREF) -sourcepath $(JAVA_SRC_DIR) -classpath $(SOLARBINDIR)$/ridl.jar -linkoffline ../../common/ref ./uno -linkoffline http://java.sun.com/j2se/1.5/docs/api ./java $(JAVA_PACKAGES)
+.IF "$(JAVADOCISGJDOC)" != "yes"
+JAVADOCPARAMS+= -notimestamp
+.ENDIF
 
 JAVADOCLOG = $(MISC)$/javadoc_log.txt
 
