From 24c8531ad16540386ca2b6de37de69d83549e96a Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 31 May 2013 17:08:26 +0200
Subject: [PATCH] go for hsqldb 1.8

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c561981..127ce45 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7585,7 +7585,7 @@ if test "$with_system_hsqldb" = "yes"; then
     AC_MSG_RESULT([external])
     SYSTEM_HSQLDB=YES
     if test -z $HSQLDB_JAR; then
-        HSQLDB_JAR=/usr/share/java/hsqldb.jar
+        HSQLDB_JAR=/usr/share/java/hsqldb1.8.0.jar
     fi
     AC_CHECK_FILE($HSQLDB_JAR, [],
                [AC_MSG_ERROR(hsqldb.jar not found.)], [])
-- 
1.8.1.2

