For whatever reason we get undefined referenced to g_module_*.

Index: libreoffice-4.0.0.0-alpha1/vcl/Library_vclplug_gtk3.mk
===================================================================
--- libreoffice-4.0.0.0-alpha1.orig/vcl/Library_vclplug_gtk3.mk	2012-11-20 13:53:34.000000000 +0100
+++ libreoffice-4.0.0.0-alpha1/vcl/Library_vclplug_gtk3.mk	2012-11-21 20:44:24.310157249 +0100
@@ -43,6 +43,7 @@
 $(eval $(call gb_Library_add_libs,vclplug_gtk3,\
 	$(GTK3_LIBS) \
 	$(GTHREAD_LIBS) \
+	$(GMODULE_LIBS) \
 	-lX11 \
 	-lXext \
 	-lSM \
Index: libreoffice-4.0.0.0-alpha1/configure.ac
===================================================================
--- libreoffice-4.0.0.0-alpha1.orig/configure.ac	2012-11-21 20:42:48.000000000 +0100
+++ libreoffice-4.0.0.0-alpha1/configure.ac	2012-11-21 20:44:24.318157248 +0100
@@ -9413,10 +9413,13 @@
     else
         AC_MSG_ERROR([gtk3 libraries of the correct versions, not found])
     fi
+    PKG_CHECK_MODULES(GMODULE, gmodule-2.0,,AC_MSG_ERROR([requirements to build the gtk3-plugin not met. Use --disable-gtk3 or install the missing packages]))
 fi
 AC_SUBST(GTK3_LIBS)
 AC_SUBST(GTK3_CFLAGS)
 AC_SUBST(ENABLE_GTK3)
+AC_SUBST(GMODULE_LIBS)
+AC_SUBST(GMODULE_CFLAGS)
 
 AC_MSG_CHECKING([which VCLplugs shall be built])
 ENABLE_GTK=""
Index: libreoffice-4.0.0.0-alpha1/config_host.mk.in
===================================================================
--- libreoffice-4.0.0.0-alpha1.orig/config_host.mk.in	2012-11-21 20:42:48.000000000 +0100
+++ libreoffice-4.0.0.0-alpha1/config_host.mk.in	2012-11-21 20:44:24.318157248 +0100
@@ -239,6 +239,8 @@
 export GTHREAD_LIBS=$(gb_SPACE)@GTHREAD_LIBS@
 export GTK3_CFLAGS=$(gb_SPACE)@GTK3_CFLAGS@
 export GTK3_LIBS=$(gb_SPACE)@GTK3_LIBS@
+export GMODULE_CFLAGS=$(gb_SPACE)@GMODULE_CFLAGS@
+export GMODULE_LIBS=$(gb_SPACE)@GMODULE_LIBS@
 export GTK_CFLAGS=$(gb_SPACE)@GTK_CFLAGS@
 export GTK_LIBS=$(gb_SPACE)@GTK_LIBS@
 export GTK_PRINT_CFLAGS=$(gb_SPACE)@GTK_PRINT_CFLAGS@
