Index: libreoffice-4.1.4~rc2/sal/osl/unx/asm/interlck_sparc.s
===================================================================
--- libreoffice-4.1.4~rc2.orig/sal/osl/unx/asm/interlck_sparc.s	2013-12-16 14:35:32.474933867 +0100
+++ libreoffice-4.1.4~rc2/sal/osl/unx/asm/interlck_sparc.s	2013-12-16 14:35:32.466933867 +0100
@@ -39,6 +39,14 @@
  * 
  */
 
+/* FIXME: Crude hack to get this compile. Looks like all those three
+get defined now and this we don't get osl_InterlockedCountSetV9 which
+then makes sal fail to link (util.c uses it). Sparc people, better fix
+welcome */
+#undef __sparcv8plus
+#undef __sparcv9
+#undef __sparc_v9__
+
 #if !defined(__sparcv8plus) && !defined(__sparcv9) && !defined(__sparc_v9__)
 
 .section ".data"
