diff --git a/solenv/gbuild/LinkTarget.mk b/solenv/gbuild/LinkTarget.mk
index c094bb4..a0608f4 100644
--- a/solenv/gbuild/LinkTarget.mk
+++ b/solenv/gbuild/LinkTarget.mk
@@ -46,6 +46,8 @@ CXXFLAGS ?= $(gb_COMPILEROPTFLAGS)
 OBJCXXFLAGS ?= $(gb_COMPILEROPTFLAGS)
 endif

+LDFLAGS += "$(shell dpkg-buildflags --get LDFLAGS)"
+
 # if enabled we link all of these libraries into one larger, merged library
 # for which we can do a lot more optimisation, and which is faster to read
 # from disk.
diff --git a/solenv/inc/unxgcc.mk b/solenv/inc/unxgcc.mk
index a3231a6..5a63144 100644
--- a/solenv/inc/unxgcc.mk
+++ b/solenv/inc/unxgcc.mk
@@ -233,6 +233,8 @@ LINKFLAGS += -Wl,--hash-style=$(WITH_LINKER_HASH_STYLE)
 LINKFLAGS += -Wl,-zdynsort
 .ENDIF

+LINKFLAGS += "$(shell dpkg-buildflags --get LDFLAGS)"
+
 # libraries for linking applications
 STDLIBGUIMT+=-Wl,--as-needed $(DL_LIB) $(PTHREAD_LIBS) -lm -Wl,--no-as-needed
 STDLIBCUIMT+=-Wl,--as-needed $(DL_LIB) $(PTHREAD_LIBS) -lm -Wl,--no-as-needed
