diff --git a/sfx2/source/appl/app.src b/sfx2/source/appl/app.src
index 21c5f9175f2e..cbd28e5b538b 100644
--- a/sfx2/source/appl/app.src
+++ b/sfx2/source/appl/app.src
@@ -220,7 +220,7 @@ String RID_STR_NOAUTOSTARTHELPAGENT
 
 String RID_STR_HLPFILENOTEXIST
 {
-    Text [ en-US ] = "The help file for this topic is not installed." ;
+    Text [ en-US ] = "The help file for this topic is not installed. Please install the libreoffice-help-en-us package or the locale specific help package libreoffice-help-<language-code>." ;
 };
 
 String STR_QUICKSTART_EXIT
