Description: Hide startcenter and math from the shell
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1696250
Forwarded: not-needed

--- a/sysui/desktop/menus/math.desktop
+++ b/sysui/desktop/menus/math.desktop
@@ -33,6 +33,7 @@ Keywords=Equation;OpenDocument Formula;F
 InitialPreference=5
 StartupWMClass=libreoffice-math
 X-KDE-Protocols=file,http,ftp,webdav
+NotShowIn=GNOME;
 
 Actions=NewDocument;
 [Desktop Action NewDocument]
--- a/sysui/desktop/menus/startcenter.desktop
+++ b/sysui/desktop/menus/startcenter.desktop
@@ -31,6 +31,7 @@ StartupNotify=true
 X-GIO-NoFuse=true
 StartupWMClass=libreoffice-startcenter
 X-KDE-Protocols=file,http,ftp,webdav
+NotShowIn=GNOME;
 
 ##Define Actions
 Actions=Writer;Calc;Impress;Draw;Base;Math;
