Description: Hide startcenter and math from the shell
Author: Olivier Tilloy <olivier.tilloy@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/1696250
Forwarded: not-needed

--- a/sysui/desktop/menus/startcenter.desktop
+++ b/sysui/desktop/menus/startcenter.desktop
@@ -31,6 +31,7 @@ StartupNotify=true
 StartupWMClass=libreoffice-startcenter
 X-KDE-Protocols=file,http,ftp,webdav,webdavs
 X-AppStream-Ignore=True
+NotShowIn=GNOME;
 
 ##Define Actions
 Actions=Writer;Calc;Impress;Draw;Base;Math;
