--- a/bridges/Library_cpp_uno.mk-old	2014-06-01 12:33:50.610016042 +0200
+++ b/bridges/Library_cpp_uno.mk	2014-06-01 12:34:18.718015211 +0200
@@ -43,8 +43,7 @@
 else ifeq ($(OS)-$(CPUNAME),LINUX-HPPA)
 
 bridges_SELECTED_BRIDGE := gcc3_linux_hppa
-bridge_asm_objects := call
-bridge_noopt_objects := cpp2uno except uno2cpp
+bridge_noopt_objects := call cpp2uno except uno2cpp
 
 else ifeq ($(OS)-$(CPUNAME),LINUX-IA64)
 
