diff --git a/configure.ac b/configure.ac
index 42dc3b4..f5a51c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6587,7 +6587,7 @@ if test "$ENABLE_JAVA" != ""; then
         # at least 2 reasons to check: officebean needs to link -ljawt,
         # and libjpipe.so needs to be loaded by java to run JunitTests.
         case $CPUNAME in
-            AARCH64|AXP|X86_64|HPPA|IA64|POWERPC64|S390X|SPARC64)
+            AARCH64|AXP|X86_64|IA64|POWERPC64|S390X|SPARC64)
                 if test -f "$JAVAINTERPRETER" -a "`$JAVAINTERPRETER -version 2>&1 | $GREP -i 64-bit`" = "" >/dev/null; then
                     AC_MSG_WARN([You are building 64-bit binaries but the JDK $JAVAINTERPRETER is 32-bit])
                     AC_MSG_ERROR([You should pass the --with-jdk-home option pointing to a 64-bit JDK])
