diff --git a/bridges/Library_cpp_uno.mk b/bridges/Library_cpp_uno.mk
index f7153c3..b8cf841 100644
--- a/bridges/Library_cpp_uno.mk
+++ b/bridges/Library_cpp_uno.mk
@@ -43,8 +43,7 @@ bridge_exception_objects := cpp2uno except uno2cpp
 else ifeq ($(OS)$(CPU),LINUXH)
 
 bridges_SELECTED_BRIDGE := gcc3_linux_hppa
-bridge_asm_objects := call
-bridge_noopt_objects := cpp2uno except uno2cpp
+bridge_noopt_objects := call cpp2uno except uno2cpp
 
 else ifeq ($(OS)$(CPU),LINUXA)
 
