diff --git a/src/sdf/fixes-hu.sdf b/src/sdf/fixes-hu.sdf
index 89a6542..36727d6 100644
--- openoffice.org-3.2.1/ooo-build-3-2-1-3/src/sdf/fixes-hu.sdf
+++ openoffice.org-3.2.1/ooo-build-3-2-1-3/src/sdf/fixes-hu.sdf
@@ -1,5 +1,21 @@
-svx	source\cui\chardlg.src	0	tristatebox	RID_SVXPAGE_CHAR_EFFECTS	CB_OUTLINE			82	en-US	Out~line				20100415 16:29:54
-svx	source\cui\chardlg.src	0	tristatebox	RID_SVXPAGE_CHAR_EFFECTS	CB_OUTLINE			82	hu	Kö~rvonalas				2010-03-25 22:00:00
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3147344	14			0	en-US	Series Type				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3147344	14			0	hu	Sorozat típusa				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3148488	16			0	en-US	Linear				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3148488	16			0	hu	Számtani sorozat				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3159238	17			0	en-US	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_ARITHMETIC\"\>Creates a linear number series using the defined increment and end value.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3159238	17			0	hu	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_ARITHMETIC\"\>A megadott növekménnyel, mint különbséggel számtani sorozatot készít a megadott végértékig.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3149210	18			0	en-US	Growth				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3149210	18			0	hu	Mértani sorozat				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3150364	19			0	en-US	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_GEOMETRIC\"\>Creates a growth series using the defined increment and end value.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3150364	19			0	hu	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_GEOMETRIC\"\>A megadott növekménnyel, mint hányadossal mértani sorozatot készít a megadott végértékig.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3149528	20			0	en-US	Date				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	hd_id3149528	20			0	hu	Dátumsorozat				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3153308	26			0	en-US	In this area you can specify the desired unit of time. This area is only active if the \<emph\>Date\</emph\> option has been chosen in the \<emph\>Series type\</emph\> area.				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3153308	26			0	hu	A területen meghatározhatja a kívánt időegységet. A terület csak akkor aktív, ha a \<emph\>Sorozat típusa\</emph\> terület \<emph\>Dátum\</emph\> beállítását választotta.				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3148605	28			0	en-US	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_DAY\"\>Use the \<emph\>Date\</emph\> series type and this option to create a series using seven days.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3148605	28			0	hu	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_DAY\"\>Használja a \<emph\>Dátum\</emph\> sorozattípust és ezt a beállítást egy hét napot használó sorozat létrehozásához.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3150108	30			0	en-US	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_DAY_OF_WEEK\"\>Use the \<emph\>Date\</emph\> series type and this option to create a series of five day sets.\</ahelp\>				20051120 17:50:58
+helpcontent2	source\text\scalc\01\02140600.xhp	0	help	par_id3150108	30			0	hu	\<ahelp hid=\"SC:RADIOBUTTON:RID_SCDLG_FILLSERIES:BTN_DAY_OF_WEEK\"\>Használja a \<emph\>Dátum\</emph\> sorozattípust és ezt a beállítást, ha a sorozat tagjai csak hétköznapokra eshetnek.\</ahelp\>				20051120 17:50:58
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id3153624	3			0	en-US	Use zero (0) or the number sign (#) as placeholders in your number format code to represent numbers. The (#) only displays significant digits, while the (0) displays zeroes if there are fewer digits in the number than in the number format.				20051120 17:50:58
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id3153624	3			0	hu	Használjon nullát (0) vagy a szám jelet (#) helykitöltőnek a számformátumkódban a számok jelölésére. A (#) csak lényeges számjegyeket, míg a (0) nullákat jelenít meg, ha a számban a számformátumban megadottaknál kevesebb számjegy van.				20051120 17:50:58
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id3147088	7			0	en-US	0 (Zero)				20051120 17:50:58
@@ -39,4 +55,12 @@ helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id3154302	103			0	hu
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id1002200811423518				0	en-US	The above listed formatting codes work with your language version of %PRODUCTNAME. However, when you need to switch the locale of %PRODUCTNAME to another locale, you will need to know the formatting codes used in that other locale. 				20051120 17:50:58
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id1002200811423518				0	hu	A fent felsorolt formázási kódok az %PRODUCTNAME jelen nyelvi verziójával működnek. Ha azonban az %PRODUCTNAME területi beállítását módosítania kell, ismernie kell a másik területi beállításnál használt formátumkódokat. 				20051120 17:50:58
 helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id1002200811423556				0	en-US	For example, if your software is set to an English locale, and you want to format a year with four digits, you enter YYYY as a formatting code. When you switch to a German locale, you must use JJJJ instead. The following table lists only the localized differences.				20051120 17:50:58
-helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id1002200811423556				0	hu	Ha például a szoftver angol vagy magyar területi beállítást használ, és egy évszámot négy számjegyen kíván ábrázolni, formátumkódként az YYYY használható. Ha átvált német területi beállításra, a JJJJ kódot kell használnia. A következő táblázat csak a lokalizációból fakadó különbségeket sorolja fel.				20051120 17:50:58
+helpcontent2	source\text\shared\01\05020301.xhp	0	help	par_id1002200811423556				0	hu	Ha például a szoftver magyar területi beállítást használ, és egy évszámot négy számjegyen kíván ábrázolni, formátumkódként az YYYY használható. Ha átvált német területi beállításra, a JJJJ kódot kell használnia. A következő táblázat csak a lokalizációból fakadó különbségeket sorolja fel.				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_ARITHMETIC			69	en-US	Li~near				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_ARITHMETIC			69	hu	Számtani sorozat				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_GEOMETRIC			69	en-US	~Growth				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_GEOMETRIC			69	hu	Mértani sorozat				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_DATE			69	en-US	Da~te				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	radiobutton	RID_SCDLG_FILLSERIES	BTN_DATE			69	hu	Dá~tumsorozat				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	fixedline	RID_SCDLG_FILLSERIES	FL_TYPE			75	en-US	Series type				20051120 17:50:58
+sc	source\ui\src\miscdlgs.src	0	fixedline	RID_SCDLG_FILLSERIES	FL_TYPE			75	hu	Sorozat típusa				20051120 17:50:58
