Index: libreoffice-4.1.4~rc2/sc/CppunitTest_sc_cellrangeobj.mk
===================================================================
--- libreoffice-4.1.4~rc2.orig/sc/CppunitTest_sc_cellrangeobj.mk	2013-12-16 14:35:23.058934367 +0100
+++ libreoffice-4.1.4~rc2/sc/CppunitTest_sc_cellrangeobj.mk	2013-12-16 14:35:23.050934368 +0100
@@ -110,4 +110,6 @@
 
 $(eval $(call gb_CppunitTest_use_unittest_configuration,sc_cellrangesbase))
 
+.IGNORE: $(call gb_CppunitTest_get_target,sc_cellrangesbase)
+
 # vim: set noet sw=4 ts=4:
