--- a/sc/CppunitTest_sc_cellrangeobj.mk-old	2012-05-09 01:02:30.359827642 +0200
+++ b/sc/CppunitTest_sc_cellrangeobj.mk	2012-05-09 01:04:10.416323786 +0200
@@ -125,4 +125,6 @@ $(eval $(call gb_CppunitTest_use_filter_configuration,sc_cellrangesbase))
 
 $(eval $(call gb_CppunitTest_use_unittest_configuration,sc_cellrangesbase))
 
+.IGNORE: $(call gb_CppunitTest_get_target,sc_cellrangesbase)
+
 # vim: set noet sw=4 ts=4:
