diff --git a/solenv/gbuild/extensions/post_AuxTargets.mk b/solenv/gbuild/extensions/post_AuxTargets.mk
index 7622047..9978cda 100644
--- solenv/gbuild/extensions/post_AuxTargets.mk
+++ solenv/gbuild/extensions/post_AuxTargets.mk
@@ -75,7 +75,7 @@ distclean: clean
 	desktop/scripts/soffice.sh ooo.lst post_download post_download.log \
 	set_soenv set_soenv.last set_soenv.stamp src.downloaded warn
 
-install: build
+install:
 	echo "Installing in $(INSTALLDIR)..." && \
 	ooinstall "$(INSTALLDIR)" && \
 	echo "" && \
