From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 23 Oct 2015 00:01:16 +0200
Subject: [PATCH] use different var for inseparatable translations

---
 sysui/CustomTarget_share.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysui/CustomTarget_share.mk b/sysui/CustomTarget_share.mk
--- a/sysui/CustomTarget_share.mk
+++ b/sysui/CustomTarget_share.mk
@@ -224,7 +224,7 @@ $(share_WORKDIR)/%/build.flag: $(share_SRCDIR)/share/brand.pl $(LAUNCHERS) \
 $(eval $(call gb_CustomTarget_ulfex_rule,\
 	$(share_WORKDIR)/%.ulf,\
 	$(share_SRCDIR)/share/%.ulf,\
-	$(foreach lang,$(gb_TRANS_LANGS),\
+	$(foreach lang,$(gb_TRANS_LANGS_INTEGRAL),\
 		$(gb_POLOCATION)/$(lang)/sysui/desktop/share.po)))
 
 # vim: set noet sw=4 ts=4:
-- 
2.7.4

