--- a/download.lst	2020-03-27 16:59:25.809834915 +0100
+++ b/download.lst	2020-03-27 17:00:31.958446096 +0100
@@ -196,7 +196,7 @@
 export OPENLDAP_TARBALL := openldap-2.4.45.tgz
 export OPENSSL_SHA256SUM := 14cb464efe7ac6b54799b34456bd69558a749a4931ecfd9cf9f71d7881cac7bc
 export OPENSSL_TARBALL := openssl-1.0.2t.tar.gz
-export ORCUS_SHA256SUM := 0dd26f3f2e611c51df9ee02d6dbf08887989eaa417b73f6877cd0d94df795fc2
+export ORCUS_SHA256SUM := 12591ab4a9bccd0a956eea7c56ac966c50e1062425fdae4af05627cbe89b8030
 export ORCUS_TARBALL := liborcus-0.15.3.tar.gz
 export OWNCLOUD_ANDROID_LIB_SHA256SUM := b18b3e3ef7fae6a79b62f2bb43cc47a5346b6330f6a383dc4be34439aca5e9fb
 export OWNCLOUD_ANDROID_LIB_TARBALL := owncloud-android-library-0.9.4-no-binary-deps.tar.gz
