diff --git a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
index 943b2ce..d5a2637 100644
--- a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
+++ b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
@@ -29,6 +29,9 @@
   <vendor name="Oracle Corporation">
     <minVersion>1.6.0</minVersion>
   </vendor>
+  <vendor name="Debian">
+    <minVersion>1.6.0</minVersion>
+  </vendor>
   <vendor name="IBM Corporation">
     <minVersion>1.5.0</minVersion>
   </vendor>
diff --git a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
index b2e3aca..2b0e51b 100644
--- a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
@@ -43,6 +43,7 @@ BEGIN_VENDOR_MAP()
 #endif
     VENDOR_MAP_ENTRY("Sun Microsystems Inc.", SunInfo)
     VENDOR_MAP_ENTRY("Oracle Corporation", SunInfo)
+    VENDOR_MAP_ENTRY("Debian", SunInfo)
 #ifndef MACOSX
     VENDOR_MAP_ENTRY("IBM Corporation", OtherInfo)
     VENDOR_MAP_ENTRY("Blackdown Java-Linux Team", OtherInfo)
diff --git a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
index 5d18036..b603387 100644
--- a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
+++ b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
@@ -250,7 +250,7 @@ public class LocalOfficeWindow
         if (getNativeWindowSystemType() == SystemDependent.SYSTEM_XWINDOW )
         {
             String vendor = System.getProperty("java.vendor");
-            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation"))
+            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation") || vendor.equals("Debian"))
                 && Boolean.getBoolean("sun.awt.xembedserver"))
             {
                 xembed = new NamedValue(
