diff --git a/configure.ac b/configure.ac
index 93543ddccf4e..6ee677e222a1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7126,7 +6907,12 @@ then
             my_java_arch=$host_cpu
             ;;
         esac
-        JAWTLIB="-L$JAVA_HOME/jre/lib/$my_java_arch -ljawt"
+        # This is where JDK9 puts the library
+        if test -e "$JAVA_HOME/lib/libjawt.so"; then
+            JAWTLIB="-L$JAVA_HOME/lib/ -ljawt"
+        else
+            JAWTLIB="-L$JAVA_HOME/jre/lib/$my_java_arch -ljawt"
+        fi
         AS_IF([test "$JAVA_ARCH" != ""], [AC_DEFINE_UNQUOTED([JAVA_ARCH], ["$JAVA_ARCH"])])
     fi
     AC_MSG_RESULT([$JAWTLIB])
