diff --git a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
index b5fb7ff1d88a..923c1761bc8b 100644
--- a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
+++ b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
@@ -25,6 +25,9 @@
     <vendor name="Oracle Corporation">
       <minVersion>1.6.0</minVersion>
     </vendor>
+    <vendor name="Debian">
+      <minVersion>1.6.0</minVersion>
+    </vendor>
     <vendor name="IBM Corporation">
       <minVersion>1.6.0</minVersion>
     </vendor>
diff --git a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
index 60911f8a63ca..eaa26f441ad7 100644
--- a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
@@ -42,6 +42,7 @@ VendorSupportMapEntry const gVendorMap[] ={
 #endif
     VENDOR_MAP_ENTRY<SunInfo>("Sun Microsystems Inc."),
     VENDOR_MAP_ENTRY<SunInfo>("Oracle Corporation"),
+    VENDOR_MAP_ENTRY<SunInfo>("Debian"),
     VENDOR_MAP_ENTRY<SunInfo>("AdoptOpenJdk"),
     VENDOR_MAP_ENTRY<SunInfo>("Amazon.com Inc."),
 #ifndef MACOSX
diff --git a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
index 67ace100fb5f..46cdb32fd276 100644
--- a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
+++ b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
@@ -250,7 +250,7 @@ public class LocalOfficeWindow
         if (getNativeWindowSystemType() == SystemDependent.SYSTEM_XWINDOW )
         {
             String vendor = System.getProperty("java.vendor");
-            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation"))
+            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation") || vendor.equals("Debian"))
                 && Boolean.getBoolean("sun.awt.xembedserver"))
             {
                 xembed = new NamedValue(
