diff --git a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
index b008bb1fc0f7..ea0ddc4e7c91 100644
--- a/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
+++ b/jvmfwk/distributions/OpenOfficeorg/javavendors_linux.xml
@@ -25,6 +25,9 @@
     <vendor name="Oracle Corporation">
       <minVersion>1.6.0</minVersion>
     </vendor>
+    <vendor name="Debian">
+      <minVersion>1.6.0</minVersion>
+    </vendor>
     <vendor name="IBM Corporation">
       <minVersion>1.5.0</minVersion>
     </vendor>
diff --git a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
index 5cb85ad39a1c..c6757c6fb5eb 100644
--- a/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
+++ b/jvmfwk/plugins/sunmajor/pluginlib/vendorlist.cxx
@@ -42,6 +42,7 @@ BEGIN_VENDOR_MAP()
 #endif
     VENDOR_MAP_ENTRY("Sun Microsystems Inc.", SunInfo)
     VENDOR_MAP_ENTRY("Oracle Corporation", SunInfo)
+    VENDOR_MAP_ENTRY("Debian", SunInfo)
     VENDOR_MAP_ENTRY("AdoptOpenJdk", SunInfo)
     VENDOR_MAP_ENTRY("Amazon.com Inc.", SunInfo)
 #ifndef MACOSX
diff --git a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
index 67ace100fb5f..46cdb32fd276 100644
--- a/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
+++ b/bean/com/sun/star/comp/beans/LocalOfficeWindow.java
@@ -250,7 +250,7 @@ public class LocalOfficeWindow
         if (getNativeWindowSystemType() == SystemDependent.SYSTEM_XWINDOW )
         {
             String vendor = System.getProperty("java.vendor");
-            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation"))
+            if ((vendor.equals("Sun Microsystems Inc.") || vendor.equals("Oracle Corporation") || vendor.equals("Debian"))
                 && Boolean.getBoolean("sun.awt.xembedserver"))
             {
                 xembed = new NamedValue(
