diff --git a/bean/JunitTest_bean_complex.mk b/bean/JunitTest_bean_complex.mk
index 1d22dcd31f8f..5e7d057e7970 100644
--- a/bean/JunitTest_bean_complex.mk
+++ b/bean/JunitTest_bean_complex.mk
@@ -16,6 +16,7 @@ $(eval $(call gb_JunitTest_use_jars,bean_complex,\
 	test \
 	unoil \
 	officebean \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,bean_complex,\
diff --git a/chart2/JunitTest_chart2_unoapi.mk b/chart2/JunitTest_chart2_unoapi.mk
index d91c6fae3bba..836d2c2117ca 100644
--- a/chart2/JunitTest_chart2_unoapi.mk
+++ b/chart2/JunitTest_chart2_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,chart2_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,chart2_unoapi,\
diff --git a/comphelper/JunitTest_comphelper_complex.mk b/comphelper/JunitTest_comphelper_complex.mk
index af9c3dba88d4..5f02d5b71ae6 100644
--- a/comphelper/JunitTest_comphelper_complex.mk
+++ b/comphelper/JunitTest_comphelper_complex.mk
@@ -30,6 +30,7 @@ $(eval $(call gb_JunitTest_use_jars,comphelper_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,comphelper_complex,\
diff --git a/configmgr/JunitTest_configmgr_unoapi.mk b/configmgr/JunitTest_configmgr_unoapi.mk
index b3b6e49e90ae..18f8b7fb3729 100644
--- a/configmgr/JunitTest_configmgr_unoapi.mk
+++ b/configmgr/JunitTest_configmgr_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,configmgr_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,configmgr_unoapi,\
diff --git a/connectivity/JunitTest_complex.mk b/connectivity/JunitTest_complex.mk
index 78947bfc2076..8f86f526d382 100644
--- a/connectivity/JunitTest_complex.mk
+++ b/connectivity/JunitTest_complex.mk
@@ -20,6 +20,7 @@ $(eval $(call gb_JunitTest_use_jars,connectivity_complex,\
 	ridl \
 	test \
 	unoil \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_use_externals,connectivity_complex,\
diff --git a/dbaccess/JunitTest_dbaccess_unoapi.mk b/dbaccess/JunitTest_dbaccess_unoapi.mk
index 64af95693bb7..a641fc6ad9c9 100644
--- a/dbaccess/JunitTest_dbaccess_unoapi.mk
+++ b/dbaccess/JunitTest_dbaccess_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,dbaccess_unoapi,\
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,dbaccess_unoapi,\
diff --git a/extensions/JunitTest_extensions_unoapi.mk b/extensions/JunitTest_extensions_unoapi.mk
index 0d98b1eab0b7..e4ee015bbb1b 100644
--- a/extensions/JunitTest_extensions_unoapi.mk
+++ b/extensions/JunitTest_extensions_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,extensions_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,extensions_unoapi,\
diff --git a/filter/JunitTest_filter_complex.mk b/filter/JunitTest_filter_complex.mk
index b0c590a1be8c..add5c39d8374 100644
--- a/filter/JunitTest_filter_complex.mk
+++ b/filter/JunitTest_filter_complex.mk
@@ -24,6 +24,7 @@ $(eval $(call gb_JunitTest_use_jars,filter_complex,\
 	test \
 	unoil \
 	jurt \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,filter_complex,\
diff --git a/forms/JunitTest_forms_complex.mk b/forms/JunitTest_forms_complex.mk
index ed4a00af74df..d5386c98ff26 100644
--- a/forms/JunitTest_forms_complex.mk
+++ b/forms/JunitTest_forms_complex.mk
@@ -15,6 +15,7 @@ $(eval $(call gb_JunitTest_use_jars,forms_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,forms_complex,\
diff --git a/forms/JunitTest_forms_unoapi_1.mk b/forms/JunitTest_forms_unoapi_1.mk
index 766496355b63..05934c158eeb 100644
--- a/forms/JunitTest_forms_unoapi_1.mk
+++ b/forms/JunitTest_forms_unoapi_1.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,forms_unoapi_1,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,forms_unoapi_1,\
diff --git a/forms/JunitTest_forms_unoapi_2.mk b/forms/JunitTest_forms_unoapi_2.mk
index c309cc75ebce..504aafa9b064 100644
--- a/forms/JunitTest_forms_unoapi_2.mk
+++ b/forms/JunitTest_forms_unoapi_2.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,forms_unoapi_2,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,forms_unoapi_2,\
diff --git a/forms/JunitTest_forms_unoapi_3.mk b/forms/JunitTest_forms_unoapi_3.mk
index 4ba8829fddfa..ac87612ec0be 100644
--- a/forms/JunitTest_forms_unoapi_3.mk
+++ b/forms/JunitTest_forms_unoapi_3.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,forms_unoapi_3,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,forms_unoapi_3,\
diff --git a/forms/JunitTest_forms_unoapi_4.mk b/forms/JunitTest_forms_unoapi_4.mk
index e63cc3ff069c..5280085e2b3b 100644
--- a/forms/JunitTest_forms_unoapi_4.mk
+++ b/forms/JunitTest_forms_unoapi_4.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,forms_unoapi_4,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,forms_unoapi_4,\
diff --git a/framework/JunitTest_framework_complex.mk b/framework/JunitTest_framework_complex.mk
index 8ed558899a44..89b701162029 100644
--- a/framework/JunitTest_framework_complex.mk
+++ b/framework/JunitTest_framework_complex.mk
@@ -30,6 +30,7 @@ $(eval $(call gb_JunitTest_use_jars,framework_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,framework_complex,\
diff --git a/framework/JunitTest_framework_unoapi.mk b/framework/JunitTest_framework_unoapi.mk
index 35e919235241..5943eed84a2b 100644
--- a/framework/JunitTest_framework_unoapi.mk
+++ b/framework/JunitTest_framework_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,framework_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,framework_unoapi,\
diff --git a/linguistic/JunitTest_linguistic_complex.mk b/linguistic/JunitTest_linguistic_complex.mk
index 26c425c53ab6..926a30b64f1a 100644
--- a/linguistic/JunitTest_linguistic_complex.mk
+++ b/linguistic/JunitTest_linguistic_complex.mk
@@ -31,6 +31,7 @@ $(eval $(call gb_JunitTest_use_jars,linguistic_complex,\
 	test-tools \
 	unoil \
 	jurt \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,linguistic_complex,\
diff --git a/linguistic/JunitTest_linguistic_unoapi.mk b/linguistic/JunitTest_linguistic_unoapi.mk
index 6047d75cb607..a72111e07113 100644
--- a/linguistic/JunitTest_linguistic_unoapi.mk
+++ b/linguistic/JunitTest_linguistic_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,linguistic_unoapi,\
 	test \
 	unoil \
 	jurt \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,linguistic_unoapi,\
diff --git a/reportdesign/JunitTest_reportdesign_complex.mk b/reportdesign/JunitTest_reportdesign_complex.mk
index 8a3fb2efa650..c90853d559f3 100644
--- a/reportdesign/JunitTest_reportdesign_complex.mk
+++ b/reportdesign/JunitTest_reportdesign_complex.mk
@@ -36,6 +36,7 @@ $(eval $(call gb_JunitTest_use_jars,reportdesign_complex,\
 	test \
 	unoil \
 	jurt \
+	juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,reportdesign_complex,\
diff --git a/sc/JunitTest_sc_complex.mk b/sc/JunitTest_sc_complex.mk
index 77dc636fe984..7ddc4a731c71 100644
--- a/sc/JunitTest_sc_complex.mk
+++ b/sc/JunitTest_sc_complex.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_complex, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,sc_complex, \
diff --git a/sc/JunitTest_sc_unoapi_1.mk b/sc/JunitTest_sc_unoapi_1.mk
index 0fa058385b9a..c59ca9ecec3c 100644
--- a/sc/JunitTest_sc_unoapi_1.mk
+++ b/sc/JunitTest_sc_unoapi_1.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_1, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_1,\
diff --git a/sc/JunitTest_sc_unoapi_2.mk b/sc/JunitTest_sc_unoapi_2.mk
index aeee2b3663d2..713a3afd3594 100644
--- a/sc/JunitTest_sc_unoapi_2.mk
+++ b/sc/JunitTest_sc_unoapi_2.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_2, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_2,\
diff --git a/sc/JunitTest_sc_unoapi_3.mk b/sc/JunitTest_sc_unoapi_3.mk
index 18112778fd40..33d8dc494397 100644
--- a/sc/JunitTest_sc_unoapi_3.mk
+++ b/sc/JunitTest_sc_unoapi_3.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_3, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_3,\
diff --git a/sc/JunitTest_sc_unoapi_4.mk b/sc/JunitTest_sc_unoapi_4.mk
index 7cc0629df35b..9aedb9f0eca2 100644
--- a/sc/JunitTest_sc_unoapi_4.mk
+++ b/sc/JunitTest_sc_unoapi_4.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_4, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_4,\
diff --git a/sc/JunitTest_sc_unoapi_5.mk b/sc/JunitTest_sc_unoapi_5.mk
index b9811e5cd6f3..03a8991ebcf4 100644
--- a/sc/JunitTest_sc_unoapi_5.mk
+++ b/sc/JunitTest_sc_unoapi_5.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_5, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_5,\
diff --git a/sc/JunitTest_sc_unoapi_6.mk b/sc/JunitTest_sc_unoapi_6.mk
index 7e877560cb65..13e74a2e7faf 100644
--- a/sc/JunitTest_sc_unoapi_6.mk
+++ b/sc/JunitTest_sc_unoapi_6.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_6, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_6,\
diff --git a/sc/JunitTest_sc_unoapi_7.mk b/sc/JunitTest_sc_unoapi_7.mk
index 58c2ced0f4b7..d57202eb7dd8 100644
--- a/sc/JunitTest_sc_unoapi_7.mk
+++ b/sc/JunitTest_sc_unoapi_7.mk
@@ -19,6 +19,7 @@ $(eval $(call gb_JunitTest_use_jars,sc_unoapi_7, \
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,sc_unoapi_7,\
diff --git a/sd/JunitTest_sd_unoapi.mk b/sd/JunitTest_sd_unoapi.mk
index 62e0906f8df2..a565e1e6a3ca 100644
--- a/sd/JunitTest_sd_unoapi.mk
+++ b/sd/JunitTest_sd_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,sd_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sd_unoapi,\
diff --git a/sfx2/JunitTest_sfx2_complex.mk b/sfx2/JunitTest_sfx2_complex.mk
index ee1029e70abb..60e0138ad7b9 100644
--- a/sfx2/JunitTest_sfx2_complex.mk
+++ b/sfx2/JunitTest_sfx2_complex.mk
@@ -31,6 +31,7 @@ $(eval $(call gb_JunitTest_use_jars,sfx2_complex,\
     test-tools \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,sfx2_complex,\
diff --git a/sfx2/JunitTest_sfx2_unoapi.mk b/sfx2/JunitTest_sfx2_unoapi.mk
index 14da689380e0..5c0a3db24ce6 100644
--- a/sfx2/JunitTest_sfx2_unoapi.mk
+++ b/sfx2/JunitTest_sfx2_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sfx2_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sfx2_unoapi,\
diff --git a/sot/JunitTest_sot_complex.mk b/sot/JunitTest_sot_complex.mk
index 96be0e9d5ea4..bcac0b82baaa 100644
--- a/sot/JunitTest_sot_complex.mk
+++ b/sot/JunitTest_sot_complex.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sot_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sot_complex,\
diff --git a/starmath/JunitTest_starmath_unoapi.mk b/starmath/JunitTest_starmath_unoapi.mk
index d53b20d0c17b..bd91b3879ec2 100644
--- a/starmath/JunitTest_starmath_unoapi.mk
+++ b/starmath/JunitTest_starmath_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,starmath_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,starmath_unoapi,\
diff --git a/svtools/JunitTest_svtools_unoapi.mk b/svtools/JunitTest_svtools_unoapi.mk
index 575add5363ff..3fce0ebe137a 100644
--- a/svtools/JunitTest_svtools_unoapi.mk
+++ b/svtools/JunitTest_svtools_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,svtools_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,svtools_unoapi,\
diff --git a/svx/JunitTest_svx_unoapi.mk b/svx/JunitTest_svx_unoapi.mk
index 3ecb86cde46f..c06b57a86b6c 100644
--- a/svx/JunitTest_svx_unoapi.mk
+++ b/svx/JunitTest_svx_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,svx_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,svx_unoapi,\
diff --git a/sw/JunitTest_sw_complex.mk b/sw/JunitTest_sw_complex.mk
index a4d7a08ec02b..3cf59309a4b8 100644
--- a/sw/JunitTest_sw_complex.mk
+++ b/sw/JunitTest_sw_complex.mk
@@ -29,6 +29,7 @@ $(eval $(call gb_JunitTest_use_jars,sw_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sw_complex,\
diff --git a/sw/JunitTest_sw_unoapi_1.mk b/sw/JunitTest_sw_unoapi_1.mk
index 8861a5ea2810..f02724041edb 100644
--- a/sw/JunitTest_sw_unoapi_1.mk
+++ b/sw/JunitTest_sw_unoapi_1.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sw_unoapi_1,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sw_unoapi_1,\
diff --git a/sw/JunitTest_sw_unoapi_2.mk b/sw/JunitTest_sw_unoapi_2.mk
index 502c2e841ff7..ab6ab2f20c3a 100644
--- a/sw/JunitTest_sw_unoapi_2.mk
+++ b/sw/JunitTest_sw_unoapi_2.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sw_unoapi_2,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sw_unoapi_2,\
diff --git a/sw/JunitTest_sw_unoapi_3.mk b/sw/JunitTest_sw_unoapi_3.mk
index f7bb8ba07089..808e5a2da8c3 100644
--- a/sw/JunitTest_sw_unoapi_3.mk
+++ b/sw/JunitTest_sw_unoapi_3.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sw_unoapi_3,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sw_unoapi_3,\
diff --git a/sw/JunitTest_sw_unoapi_4.mk b/sw/JunitTest_sw_unoapi_4.mk
index 91a3f12aa9ed..57188964b9ec 100644
--- a/sw/JunitTest_sw_unoapi_4.mk
+++ b/sw/JunitTest_sw_unoapi_4.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,sw_unoapi_4,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,sw_unoapi_4,\
diff --git a/toolkit/JunitTest_toolkit_complex.mk b/toolkit/JunitTest_toolkit_complex.mk
index 6f9b069902ec..1d52e9a81213 100644
--- a/toolkit/JunitTest_toolkit_complex.mk
+++ b/toolkit/JunitTest_toolkit_complex.mk
@@ -23,6 +23,7 @@ $(eval $(call gb_JunitTest_use_jars,toolkit_complex,\
     ridl \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,toolkit_complex,\
diff --git a/toolkit/JunitTest_toolkit_unoapi_1.mk b/toolkit/JunitTest_toolkit_unoapi_1.mk
index 5feec1b8f6ca..a45f77226f7d 100644
--- a/toolkit/JunitTest_toolkit_unoapi_1.mk
+++ b/toolkit/JunitTest_toolkit_unoapi_1.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,toolkit_unoapi_1,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,toolkit_unoapi_1,\
diff --git a/toolkit/JunitTest_toolkit_unoapi_2.mk b/toolkit/JunitTest_toolkit_unoapi_2.mk
index 4838cb656055..1df9154408a3 100644
--- a/toolkit/JunitTest_toolkit_unoapi_2.mk
+++ b/toolkit/JunitTest_toolkit_unoapi_2.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,toolkit_unoapi_2,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,toolkit_unoapi_2,\
diff --git a/toolkit/JunitTest_toolkit_unoapi_3.mk b/toolkit/JunitTest_toolkit_unoapi_3.mk
index 0489f6f6931e..26708725a7b0 100644
--- a/toolkit/JunitTest_toolkit_unoapi_3.mk
+++ b/toolkit/JunitTest_toolkit_unoapi_3.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,toolkit_unoapi_3,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,toolkit_unoapi_3,\
diff --git a/toolkit/JunitTest_toolkit_unoapi_4.mk b/toolkit/JunitTest_toolkit_unoapi_4.mk
index 35af1fb5de17..8fe4b38ea89b 100644
--- a/toolkit/JunitTest_toolkit_unoapi_4.mk
+++ b/toolkit/JunitTest_toolkit_unoapi_4.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,toolkit_unoapi_4,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,toolkit_unoapi_4,\
diff --git a/ucb/JunitTest_ucb_complex.mk b/ucb/JunitTest_ucb_complex.mk
index 9cfa504f2368..d913e7d4679a 100644
--- a/ucb/JunitTest_ucb_complex.mk
+++ b/ucb/JunitTest_ucb_complex.mk
@@ -20,6 +20,7 @@ $(eval $(call gb_JunitTest_use_jars,ucb_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,ucb_complex,\
diff --git a/ucb/JunitTest_ucb_unoapi.mk b/ucb/JunitTest_ucb_unoapi.mk
index cae2dc53c594..2c9016160fbc 100644
--- a/ucb/JunitTest_ucb_unoapi.mk
+++ b/ucb/JunitTest_ucb_unoapi.mk
@@ -22,6 +22,7 @@ $(eval $(call gb_JunitTest_use_jars,ucb_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,ucb_unoapi,\
diff --git a/unotools/JunitTest_unotools_complex.mk b/unotools/JunitTest_unotools_complex.mk
index b7a02f90544c..a70c5ac36b3d 100644
--- a/unotools/JunitTest_unotools_complex.mk
+++ b/unotools/JunitTest_unotools_complex.mk
@@ -16,6 +16,7 @@ $(eval $(call gb_JunitTest_use_jars,unotools_complex,\
     test-tools \
     test \
     unoil \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_set_defs,unotools_complex,\
diff --git a/unoxml/JunitTest_unordf_complex.mk b/unoxml/JunitTest_unordf_complex.mk
index a6c40156abc4..ddb86a5c3f39 100644
--- a/unoxml/JunitTest_unordf_complex.mk
+++ b/unoxml/JunitTest_unordf_complex.mk
@@ -30,6 +30,7 @@ $(eval $(call gb_JunitTest_use_jars,unordf_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,unordf_complex,\
diff --git a/unoxml/JunitTest_unoxml_complex.mk b/unoxml/JunitTest_unoxml_complex.mk
index 737e1cb73068..1487e841a16c 100644
--- a/unoxml/JunitTest_unoxml_complex.mk
+++ b/unoxml/JunitTest_unoxml_complex.mk
@@ -30,6 +30,7 @@ $(eval $(call gb_JunitTest_use_jars,unoxml_complex,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_sourcefiles,unoxml_complex,\
diff --git a/xmloff/JunitTest_xmloff_unoapi.mk b/xmloff/JunitTest_xmloff_unoapi.mk
index 650d86b7277a..94ca9f3e5db8 100644
--- a/xmloff/JunitTest_xmloff_unoapi.mk
+++ b/xmloff/JunitTest_xmloff_unoapi.mk
@@ -32,6 +32,7 @@ $(eval $(call gb_JunitTest_use_jars,xmloff_unoapi,\
     test \
     unoil \
     jurt \
+    juh \
 ))
 
 $(eval $(call gb_JunitTest_add_classes,xmloff_unoapi,\
