diff --git a/vcl/unx/kde5/KDE5FilePicker2.cxx b/vcl/unx/kde5/KDE5FilePicker2.cxx
index 5d40a7de39ab..d8d35b2f90b5 100644
--- a/vcl/unx/kde5/KDE5FilePicker2.cxx
+++ b/vcl/unx/kde5/KDE5FilePicker2.cxx
@@ -661,7 +661,7 @@ uno::Any KDE5FilePicker::handleGetListValue(QComboBox* pQComboBox, sal_Int16 nAc
         case ControlActions::GET_SELECTED_ITEM_INDEX:
         {
             int nCurrent = pQComboBox->currentIndex();
-            aAny <<= nCurrent;
+            aAny <<= static_cast<sal_Int32>(nCurrent);
         }
         break;
         default:
