From ed8615a44e98c24e7e8b09ad02bd9d18cb74fc03 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Mon, 10 Dec 2012 19:06:48 +0100
Subject: [PATCH] no need for wget/curl when not fetching anything

Change-Id: I097675450a0eed87eccf90356c81657485609d13
---
 configure.ac |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index ad5e9f3..5d48f02 100644
--- a/configure.ac
+++ b/configure.ac
@@ -11493,6 +11493,10 @@ AC_SUBST(SCPDEFS)
 #
 WGET=
 MD5SUM=
+CURL=
+
+if test "$enable_fetch_external" != "no"; then
+
 CURL=`which curl 2>/dev/null`
 
 for i in wget /usr/bin/wget /usr/local/bin/wget /usr/sfw/bin/wget /opt/sfw/bin/wget /opt/local/bin/wget; do
@@ -11534,6 +11538,8 @@ if test -z "$MD5SUM"; then
     add_warning "no md5sum: found!"
 fi
 
+fi
+
 AC_SUBST(WGET)
 AC_SUBST(CURL)
 AC_SUBST(MD5SUM)
-- 
1.7.10.4

