From ee62fb9b05f736710d674c3946c9ec042a94878a Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Sat, 21 Apr 2012 03:13:12 +0200
Subject: [PATCH] lp#527938: triple paperbag -- tweak bibloader again

---
 extensions/source/bibliography/bibload.cxx |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/extensions/source/bibliography/bibload.cxx b/extensions/source/bibliography/bibload.cxx
index 9c84cdb..c5d3b13 100644
--- a/extensions/source/bibliography/bibload.cxx
+++ b/extensions/source/bibliography/bibload.cxx
@@ -248,9 +248,16 @@ namespace
     // lp#527938, debian#602953, fdo#33266, i#105408
     static bool lcl_isBaseAvailable()
     {
-        Reference< XMultiServiceFactory >  xMgr = comphelper::getProcessServiceFactory();
-        Reference< XAggregation > xAggregate = Reference< XAggregation >( xMgr->createInstance(C2U("com.sun.star.sdbc.RowSet")), UNO_QUERY);
-        return xAggregate.is();
+        try
+        {
+            Reference< XMultiServiceFactory >  xMgr = comphelper::getProcessServiceFactory();
+            Reference< XAggregation > xAggregate = Reference< XAggregation >( xMgr->createInstance(C2U("com.sun.star.sdb.RowSet")), UNO_QUERY);
+            return xAggregate.is();
+        }
+        catch(...)
+        {
+            return false;
+        }
     }
 }
 void BibliographyLoader::load(const Reference< XFrame > & rFrame, const rtl::OUString& rURL,
-- 
1.7.9.5

