From 394f2410f7f2e137f1801d255c99d366eb38adfa Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Sat, 21 Apr 2012 03:13:12 +0200
Subject: [PATCH] lp#527938: triple paperbag -- fix service name (again)

---
 extensions/source/bibliography/bibload.cxx |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extensions/source/bibliography/bibload.cxx b/extensions/source/bibliography/bibload.cxx
index 8b581fd..301e908 100644
--- a/extensions/source/bibliography/bibload.cxx
+++ b/extensions/source/bibliography/bibload.cxx
@@ -250,7 +250,7 @@ namespace
     static bool lcl_isBaseAvailable()
     {
         Reference< XMultiServiceFactory >  xMgr = comphelper::getProcessServiceFactory();
-        Reference< XAggregation > xAggregate = Reference< XAggregation >( xMgr->createInstance(C2U("com.sun.star.sdbc.RowSet")), UNO_QUERY);
+        Reference< XAggregation > xAggregate = Reference< XAggregation >( xMgr->createInstance(C2U("com.sun.star.sdb.RowSet")), UNO_QUERY);
         return xAggregate.is();
     }
 }
-- 
1.7.9.5

