From c57a2bcb471abee33365b410caa7c413fc3a0c00 Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Fri, 20 Apr 2012 09:27:39 +0200
Subject: [PATCH] lp#753627: readd updated Ubuntu brand palette colors

---
 extras/source/palettes/standard.soc |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/extras/source/palettes/standard.soc b/extras/source/palettes/standard.soc
index f48577a..54e1ef8 100644
--- a/extras/source/palettes/standard.soc
+++ b/extras/source/palettes/standard.soc
@@ -172,4 +172,13 @@
   <draw:color draw:name="Chart 10" draw:color="#ff950e"/>
   <draw:color draw:name="Chart 11" draw:color="#c5000b"/>
   <draw:color draw:name="Chart 12" draw:color="#0084d1"/>
+
+  <!-- add Ubuntu colors -->
+  <draw:color draw:name="Ubuntu Orange" draw:color="#dd4814"/>
+  <draw:color draw:name="Ubuntu Warm Grey" draw:color="#aea79f"/>
+  <draw:color draw:name="Ubuntu Cool Grey" draw:color="#333333"/>
+  <draw:color draw:name="Ubuntu Light Aubergine" draw:color="#77216f"/>
+  <draw:color draw:name="Ubuntu Mid Aubergine" draw:color="#5e2750"/>
+  <draw:color draw:name="Ubuntu Dark Aubergine" draw:color="#2c001e"/>
+  <draw:color draw:name="Canonical Aubergine" draw:color="#772953"/>
 </office:color-table>
-- 
1.7.9.5

