Index: libreoffice-3.3.2/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.3.2.orig/libreoffice-build/patches/dev300/apply	2011-05-29 03:23:59.804063165 +0200
+++ libreoffice-3.3.2/libreoffice-build/patches/dev300/apply	2011-05-29 03:24:10.514063163 +0200
@@ -1940,8 +1940,6 @@
 
 # fix problem with calling private and protected members in layout code
 sfx2-privateprotectedpublic.diff
-# UI + core to make OLE editing optionally outplace
-optional-outplace-ole.diff, i#98970, thorsten
 # let ctrl-<keycode> stuff pass dlg keyboard handling, to permit
 # global application-wide shortcuts also in open dialogs
 vcl-permit-global-shortcuts.diff, i#105676, thorsten
