From f35d29c8388744be1f95ec4acfca12eec706911a Mon Sep 17 00:00:00 2001
From: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Date: Sun, 27 Nov 2016 19:22:59 +0100
Subject: Fix typo in configure.ac (mk68 -> m68k)

Change-Id: Ibbedf9444e2c43b3dff886d9513be7e3d2a99095

diff --git a/configure.ac b/configure.ac
index 731e903..4480ed0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7121,7 +7121,7 @@ then
             my_java_arch=i386
             ;;
         m68k)
-            my_java_arch=mk68
+            my_java_arch=m68k
             ;;
         powerpc)
             my_java_arch=ppc
-- 
cgit v0.10.2

