Description: Fix incorrect parameter type to std::min() on m68k
Author: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Last-Update: 2018-12-28

--- libreoffice-6.1.4.orig/cppu/source/typelib/typelib.cxx
+++ libreoffice-6.1.4/cppu/source/typelib/typelib.cxx
@@ -1660,7 +1660,7 @@ extern "C" sal_Int32 typelib_typedescrip
 #ifdef __m68k__
                 // Anything that is at least 16 bits wide is aligned on a 16-bit
                 // boundary on the m68k default abi
-                sal_Int32 nMaxAlign = std::min(rMaxIntegralTypeSize, 2);
+                sal_Int32 nMaxAlign = std::min(rMaxIntegralTypeSize, sal_Int32( 2 ));
                 nStructSize = (nStructSize + nMaxAlign -1) / nMaxAlign * nMaxAlign;
 #else
                 // Example: A { double; int; } structure has a size of 16 instead of 10. The
