diff --git a/xmloff/Library_xo.mk b/xmloff/Library_xo.mk
index 2feb3cf95c69..d8cf94879670 100644
--- a/xmloff/Library_xo.mk
+++ b/xmloff/Library_xo.mk
@@ -368,4 +368,10 @@ $(eval $(call gb_Library_add_exception_objects,xo,\
     xmloff/source/xforms/xformsimport \
 ))
 
+ifeq ($(CPUNAME),M68K)
+$(eval $(call gb_Library_add_cxxflags,xo,\
+    -mlong-jump-table-offsets \
+))
+endif
+
 # vim: set noet sw=4 ts=4:
diff --git a/configure.ac b/configure.ac
index 079e856..58d1672 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3792,6 +3792,19 @@ if test "$GCC" = "yes"; then
     else
         AC_MSG_RESULT([no])
     fi
+
+    if test "$host_cpu" = "m68k"; then
+        AC_MSG_CHECKING([whether $CC supports -mlong-jump-table-offsets])
+        save_CFLAGS=$CFLAGS
+        CFLAGS="$CFLAGS -Werror -mlong-jump-table-offsets"
+        AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[ return 0; ]])],[ HAVE_GCC_LONG_JUMP_TABLE_OFFSETS=TRUE ],[])
+        CFLAGS=$save_CFLAGS
+        if test "$HAVE_GCC_LONG_JUMP_TABLE_OFFSETS" = "TRUE"; then
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_ERROR([no])
+        fi
+    fi
 fi
 AC_SUBST(HAVE_GCC_GGDB2)
 AC_SUBST(HAVE_GCC_FINLINE_LIMIT)
