From 22779b136058934f1af0dcfa50ba23ea6220db87 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Sat, 7 Jan 2017 16:46:14 +0100
Subject: use (new) -mlong-jump-table-offsets on m68k for libxo

Change-Id: I6e2886e2de4f3502c74f1e367216066352994865

diff --git a/configure.ac b/configure.ac
index e808f14..dce6892 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3778,6 +3778,19 @@ if test "$GCC" = "yes"; then
     else
         AC_MSG_RESULT([no])
     fi
+
+    if test "$host_cpu" = "m68k"; then
+        AC_MSG_CHECKING([whether $CC supports -mlong-jump-table-offsets])
+        save_CFLAGS=$CFLAGS
+        CFLAGS="$CFLAGS -Werror -mlong-jump-table-offsets"
+        AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[ return 0; ]])],[ HAVE_GCC_LONG_JUMP_TABLE_OFFSETS=TRUE ],[])
+        CFLAGS=$save_CFLAGS
+        if test "$HAVE_GCC_LONG_JUMP_TABLE_OFFSETS" = "TRUE"; then
+            AC_MSG_RESULT([yes])
+        else
+            AC_MSG_ERROR([no])
+        fi
+    fi
 fi
 AC_SUBST(HAVE_GCC_GGDB2)
 AC_SUBST(HAVE_GCC_FINLINE_LIMIT)
diff --git a/xmloff/Library_xo.mk b/xmloff/Library_xo.mk
index 2feb3cf..d8cf948 100644
--- a/xmloff/Library_xo.mk
+++ b/xmloff/Library_xo.mk
@@ -368,4 +368,10 @@ $(eval $(call gb_Library_add_exception_objects,xo,\
     xmloff/source/xforms/xformsimport \
 ))
 
+ifeq ($(CPUNAME),M68K)
+$(eval $(call gb_Library_add_cxxflags,xo,\
+    -mlong-jump-table-offsets \
+))
+endif
+
 # vim: set noet sw=4 ts=4:
-- 
cgit v0.10.2

From 8f15fa63ce726d0ef1f91a76d640981c16d301cd Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Thu, 12 Jan 2017 19:00:28 +0100
Subject: add -mlong-jump-table-offsets for m68k also for oox

Change-Id: Ibfba6c7c808ca5212884032c17546e95fa16353a

diff --git a/oox/Library_oox.mk b/oox/Library_oox.mk
index fcd7215..e6765f4 100644
--- a/oox/Library_oox.mk
+++ b/oox/Library_oox.mk
@@ -314,4 +314,10 @@ $(eval $(call gb_Library_add_exception_objects,oox,\
 ))
 endif
 
+ifeq ($(CPUNAME),M68K)
+$(eval $(call gb_Library_add_cxxflags,oox,\
+    -mlong-jump-table-offsets \
+))
+endif
+
 # vim: set noet sw=4 ts=4:
-- 
cgit v0.10.2

