diff --git a/scp2/source/base/module_base.scp b/scp2/source/base/module_base.scp
index 57f8b81..7274823 100644
--- a/scp2/source/base/module_base.scp
+++ b/scp2/source/base/module_base.scp
@@ -27,6 +27,8 @@
 
 #include "macros.inc"
 
+#include "AutoInstallLibs/base"
+
 Module gid_Module_Prg_Base
     MOD_NAME_DESC(MODULE_PRG_BASE);
     ParentID = gid_Module_Prg;
@@ -51,6 +51,14 @@ Module gid_Module_Prg_Base_Bin
         ,gid_File_Lib_Abp
         ,gid_File_Lib_Dbp
         ,gid_File_Lib_Dbu
+        ,auto_Base_Lib_ALL
+        gid_File_Lib_Dbpool
+        ,gid_File_Lib_Sdbc_2
+#ifdef SOLAR_JAVA
+        ,gid_File_Lib_Jdbc
+        ,gid_File_Lib_Hsqldb_2
+#endif
+        ,gid_File_Lib_Calc
     );
 End
 
diff --git a/scp2/source/calc/module_calc.scp b/scp2/source/calc/module_calc.scp
index 0e5e790..e5ba644 100644
--- a/scp2/source/calc/module_calc.scp
+++ b/scp2/source/calc/module_calc.scp
@@ -42,7 +42,7 @@
     Minimal = YES;
     Default = YES;
     Styles = (HIDDEN_ROOT);
-    Files = (gid_File_Share_Registry_Calc_Xcd,gid_File_Lib_Calc,gid_File_Lib_Sc,gid_File_Lib_Scui,gid_File_Lib_Scfilt,gid_File_Lib_Scd,gid_File_Extra_Urldesktop_Calc,gid_File_Extra_Urlnew_Calc,gid_File_Extra_Urlstart_Calc,gid_File_Extra_Urltasks_Calc,gid_File_Oo_Office_Filter_Calc_Pkg,
+    Files = (gid_File_Share_Registry_Calc_Xcd,gid_File_Lib_Sc,gid_File_Lib_Scui,gid_File_Lib_Scfilt,gid_File_Lib_Scd,gid_File_Extra_Urldesktop_Calc,gid_File_Extra_Urlnew_Calc,gid_File_Extra_Urlstart_Calc,gid_File_Extra_Urltasks_Calc,gid_File_Oo_Office_Filter_Calc_Pkg,
     gid_File_Lib_Solver,gid_File_Lib_Lpsolve,
     gid_File_Lib_Vbaobj,
     gid_File_Tmp_Userinstall_Calc_Inf);
diff --git a/scp2/source/draw/module_draw.scp b/scp2/source/draw/module_draw.scp
index b5bec12..920fc4e 100644
--- a/scp2/source/draw/module_draw.scp
+++ b/scp2/source/draw/module_draw.scp
@@ -27,6 +27,8 @@

 #include "macros.inc"

+#include "AutoInstallLibs/draw"
+
 Module gid_Module_Prg_Draw
     MOD_NAME_DESC ( MODULE_PRG_DRAW );
     ParentID = gid_Module_Prg;
@@ -43,6 +43,7 @@ Module gid_Module_Prg_Draw_Bin
     Default = YES;
     Styles = (HIDDEN_ROOT);
     Files = (gid_File_Extra_Urldesktop_Draw,gid_File_Extra_Urlnew_Draw,gid_File_Extra_Urlstart_Draw,gid_File_Extra_Urltasks_Draw,
+    auto_Draw_Lib_ALL
     gid_File_Share_Registry_Draw_Xcd,
     gid_File_Tmp_Userinstall_Draw_Inf);
 End
diff --git a/scp2/source/impress/module_impress.scp b/scp2/source/impress/module_impress.scp
index 29164ac..030d004 100644
--- a/scp2/source/impress/module_impress.scp
+++ b/scp2/source/impress/module_impress.scp
@@ -27,6 +27,8 @@
 
 #include "macros.inc"
 
+#include "AutoInstallLibs/impress"
+
 Module gid_Module_Prg_Impress
     MOD_NAME_DESC ( MODULE_PRG_IMPRESS );
     ParentID = gid_Module_Prg;
@@ -43,6 +46,7 @@ Module gid_Module_Prg_Impress_Bin
     Default = YES;
     Styles = (HIDDEN_ROOT);
     Files = (gid_File_Extra_Urldesktop_Impress,gid_File_Extra_Urlnew_Impress,gid_File_Extra_Urlstart_Impress,gid_File_Extra_Urltasks_Impress,gid_File_Lib_Placeware,
+    auto_Impress_Lib_ALL
     gid_File_Lib_PresenterScreen,
     gid_File_Share_Registry_Impress_Xcd,
     gid_File_Lib_Animcore,
diff --git a/scp2/source/kde/module_kde.scp b/scp2/source/kde/module_kde.scp
index f066710..08a1c1c 100644
--- a/scp2/source/kde/module_kde.scp
+++ b/scp2/source/kde/module_kde.scp
@@ -36,5 +36,11 @@ Module gid_Module_Optional_Kde
     MOD_NAME_DESC(MODULE_OPTIONAL_KDE);
     Styles = ();
     Files = (auto_Kde_Lib_ALL
+#ifdef ENABLE_KDE
+    gid_File_Lib_Vclplug_Kde
+#endif
+#ifdef ENABLE_KDE4
+    gid_File_Lib_Vclplug_Kde4
+#endif
     );
 End
diff --git a/scp2/source/ooo/module_hidden_ooo.scp b/scp2/source/ooo/module_hidden_ooo.scp
index ed7729e..5a03e3d 100644
--- a/scp2/source/ooo/module_hidden_ooo.scp
+++ b/scp2/source/ooo/module_hidden_ooo.scp
@@ -166,8 +166,6 @@ Module gid_Module_Root_Files_4
 	gid_File_Lib_BaseBmp,
 	gid_File_Lib_Zip,
 	gid_File_Lib_Vclplug_Tde,
-	gid_File_Lib_Vclplug_Kde,
-	gid_File_Lib_Vclplug_Kde4,
 	gid_File_Lib_Vclplug_Svp);
 End
 
@@ -197,7 +195,6 @@ Module gid_Module_Root_Files_5
 #endif
 	gid_File_Lib_Curl,
 	gid_Unixlink_File_Lib_Curl,
-	gid_File_Lib_Dbpool,
 	gid_File_Lib_Embobj,
 	gid_File_Lib_Emboleobj,
 	gid_File_Lib_Macab_1,
diff --git a/Repository.mk b/Repository.mk
index 4b2c1f0..9423f87 100644
--- a/Repository.mk
+++ b/Repository.mk
@@ -229,11 +229,8 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,OOO, \
     configmgr \
     ctl \
     cui \
-    dba \
-    dbase \
     dbmm \
     $(if $(DISABLE_DBCONNECTIVITY),,dbtools) \
-    dbaxml \
     deploymentmisc \
     $(if $(filter-out MACOSX WNT,$(OS)),desktopbe1) \
     $(if $(filter unx,$(GUIBASE)),desktop_detector) \
@@ -260,7 +260,6 @@
     eti \
     evtatt \
     exp \
-    flat \
 	file \
 	filterconfig \
     for \
@@ -292,9 +291,6 @@
     lnth \
     $(if $(MERGELIBS),merged) \
     msfilter \
-    mysql \
-    odbc \
-    odbcbase \
     odfflatxml \
     offacc \
     pcr \
@@ -294,10 +291,6 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,OOO, \
     sb \
     sdbt \
     scn \
-    sd \
-    sdd \
-    sdfilt \
-    sdui \
     sfx \
     sot \
     $(if $(ENABLE_HEADLESS),,spl) \
@@ -327,7 +327,6 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,OOO, \
     svt \
     svx \
     svxcore \
-    sw \
     $(if $(ENABLE_TDEAB),tdeab1) \
     $(if $(ENABLE_TDEAB),tdeabdrv1) \
     textconversiondlgs \
@@ -325,10 +317,32 @@ $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,OOO, \
     xsltfilter \
 ))
 
+$(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,BASE, \
+    dba \
+    dbase \
+    dbaxml \
+    flat \
+    mysql \
+    odbc \
+    odbcbase \
+))
+
+$(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,DRAW, \
+   sd \
+   sdd \
+   sdfilt \
+   sdui \
+))
+
+$(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,IMPRESS, \
+   slideshow \
+))
+
 $(eval $(call gb_Helper_register_libraries_for_install,OOOLIBS,WRITER, \
 	hwp \
 	$(if $(ENABLE_LWP),lwpft) \
 	msword \
+        sw \
 	swd \
 	swui \
 	t602filter \
@@ -628,7 +628,6 @@ $(eval $(call gb_Helper_register_libraries,OOOLIBS, \
     postgresql-sdbc-impl \
     pythonloader \
     simplecanvas \
-    slideshow \
     stringresource \
     syssh \
     ucpcmis1 \
diff --git a/scp2/InstallModule_base.mk b/scp2/InstallModule_base.mk
index 225cf94..53ad321 100644
--- a/scp2/InstallModule_base.mk
+++ b/scp2/InstallModule_base.mk
@@ -27,6 +27,8 @@
 
 $(eval $(call gb_InstallModule_InstallModule,scp2/base))
 
+$(eval $(call gb_InstallModule_use_auto_install_libs,scp2/base,base))
+
 # Not sure about that?
 #$(eval $(call gb_InstallModule_define_if_set,scp2/base,\
 #	ENABLE_REPORTBUILDER \
diff --git a/scp2/InstallModule_draw.mk b/scp2/InstallModule_draw.mk
index 63a29f8..d5bbf6c 100644
--- a/scp2/InstallModule_draw.mk
+++ b/scp2/InstallModule_draw.mk
@@ -9,6 +9,8 @@
 
 $(eval $(call gb_InstallModule_InstallModule,scp2/draw))
 
+$(eval $(call gb_InstallModule_use_auto_install_libs,scp2/draw,draw))
+
 $(eval $(call gb_InstallModule_add_templates,scp2/draw,\
     scp2/source/templates/module_langpack_draw \
 ))
diff --git a/scp2/InstallModule_impress.mk b/scp2/InstallModule_impress.mk
index 0deac66..48b786d 100644
--- a/scp2/InstallModule_impress.mk
+++ b/scp2/InstallModule_impress.mk
@@ -9,6 +9,8 @@
 
 $(eval $(call gb_InstallModule_InstallModule,scp2/impress))
 
+$(eval $(call gb_InstallModule_use_auto_install_libs,scp2/impress,impress))
+
 $(eval $(call gb_InstallModule_define_if_set,scp2/impress,\
 	ENABLE_OPENGL \
 ))
--- a/scp2/InstallModule_kde.mk	2013-05-04 18:30:06.777713287 +0200
+++ b/scp2/InstallModule_kde.mk	2013-05-07 01:20:44.616196512 +0200
@@ -11,6 +11,12 @@
 
 $(eval $(call gb_InstallModule_use_auto_install_libs,scp2/kde,kde))
 
+$(eval $(call gb_InstallModule_define_if_set,scp2/kde,\
+        ENABLE_KAB \
+        ENABLE_KDE \
+        ENABLE_KDE4 \
+))
+
 $(eval $(call gb_InstallModule_add_localized_scpfiles,scp2/kde,\
     scp2/source/kde/module_kde \
 ))
diff --git a/solenv/gbuild/Helper.mk b/solenv/gbuild/Helper.mk
index ded46b9..447c331 100644
--- a/solenv/gbuild/Helper.mk
+++ b/solenv/gbuild/Helper.mk
@@ -118,7 +118,7 @@ endef
 define gb_Helper_init_registries
 gb_Executable_VALIDGROUPS := UREBIN SDK OOO NONE
 gb_Library_VALIDGROUPS := OOOLIBS PLAINLIBS_NONE PLAINLIBS_URE PLAINLIBS_OOO RTVERLIBS UNOLIBS_URE UNOVERLIBS EXTENSIONLIBS
-gb_Library_VALIDINSTALLMODULES := GRAPHICFILTER GNOME TDE KDE OOO URE WRITER
+gb_Library_VALIDINSTALLMODULES := GRAPHICFILTER GNOME TDE KDE OOO URE WRITER BASE DRAW IMPRESS
 gb_StaticLibrary_VALIDGROUPS := PLAINLIBS
 gb_Jar_VALIDGROUPS := URE OOO OXT NONE
 
--- /dev/null	2012-07-03 16:36:49.000000000 +0200
+++ b/scp2/AutoInstallLibs_base.mk	2013-04-15 18:21:21.695428073 +0200
@@ -0,0 +1,12 @@
+# -*- Mode: makefile-gmake; tab-width: 4; indent-tabs-mode: t -*-
+#
+# This file is part of the LibreOffice project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+$(eval $(call gb_AutoInstallLibs_AutoInstallLibs,base,BASE,LIBO_LIB_FILE,auto_Base_Lib))
+
+# vim: set noet sw=4 ts=4:
--- /dev/null	2012-07-03 16:36:49.000000000 +0200
+++ b/scp2/AutoInstallLibs_draw.mk	2013-04-15 18:21:21.695428073 +0200
@@ -0,0 +1,12 @@
+# -*- Mode: makefile-gmake; tab-width: 4; indent-tabs-mode: t -*-
+#
+# This file is part of the LibreOffice project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+$(eval $(call gb_AutoInstallLibs_AutoInstallLibs,draw,DRAW,LIBO_LIB_FILE,auto_Draw_Lib))
+
+# vim: set noet sw=4 ts=4:
--- /dev/null	2012-07-03 16:36:49.000000000 +0200
+++ b/scp2/AutoInstallLibs_impress.mk	2013-04-15 18:21:21.695428073 +0200
@@ -0,0 +1,12 @@
+# -*- Mode: makefile-gmake; tab-width: 4; indent-tabs-mode: t -*-
+#
+# This file is part of the LibreOffice project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+$(eval $(call gb_AutoInstallLibs_AutoInstallLibs,impress,IMPRESS,LIBO_LIB_FILE,auto_Impress_Lib))
+
+# vim: set noet sw=4 ts=4:
diff --git a/scp2/Module_scp2.mk b/scp2/Module_scp2.mk
index 9d856d7..dad7605 100644
--- a/scp2/Module_scp2.mk
+++ b/scp2/Module_scp2.mk
@@ -13,6 +13,9 @@ $(eval $(call gb_Module_add_targets,scp2,\
 	AutoInstallLibs_kde \
 	AutoInstallLibs_ooo \
 	AutoInstallLibs_ure \
+        AutoInstallLibs_base \
+        AutoInstallLibs_draw \
+        AutoInstallLibs_impress \
 	AutoInstallLibs_writer \
 	CustomTarget_langmacros \
 	InstallModule_accessories \
