--- /dev/null	2011-06-27 20:29:01.643585831 +0200
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/mips-openjdk-jre-paths.diff	2011-06-28 16:06:31.000000000 +0200
@@ -0,0 +1,18 @@
+--- set_soenv.in-old	2011-06-28 16:05:08.000000000 +0200
++++ set_soenv.in	2011-06-28 16:06:07.000000000 +0200
+@@ -651,11 +651,11 @@
+ 
+-       $JRELIBDIR      = '$JAVA_HOME'.$ds."lib".$ds.$ARCH;
++       $JRELIBDIR      = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds.$ARCH;
+       if ( $JDK =~ m/sun/ && -d "@JAVA_HOME@".$ds."jre".$ds."lib".$ds."mips32" ) {
+-      	$JRETOOLKITDIR  = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."client"; # Lemote
++      	$JRETOOLKITDIR  = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds.$ARCH.$ds."client"; # Lemote
+       } else {
+-      	$JRETOOLKITDIR  = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."server";
++      	$JRETOOLKITDIR  = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds.$ARCH.$ds."server";
+       }
+-      $JRETHREADDIR   = '$JAVA_HOME'.$ds."lib".$ds.$ARCH.$ds."native_threads";
++      $JRETHREADDIR   = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds.$ARCH.$ds."native_threads";
+    }
+ 
+ 
--- libreoffice-3.4.0/libreoffice-build/patches/dev300/apply-old	2011-06-28 16:06:54.000000000 +0200
+++ libreoffice-3.4.0/libreoffice-build/patches/dev300/apply	2011-06-28 16:07:30.000000000 +0200
@@ -34,7 +34,7 @@
 	      KDE4Experimental, MinGW, CalcExperimental, \
 	      OOXMLExportExperimental, CrossWin32Patches, AutoLayout, \
 	      WriterNavigation \
-	      HPPA, AXP, PowerPC, Speed, GSoC2010, Toolbars, Win32Packaging
+	      HPPA, AXP, PowerPC, MIPS, Speed, GSoC2010, Toolbars, Win32Packaging
 DebianLooseSections: DebianBaseNoHelpContent
 # Optional sections
 Optional : LiberationFontsOnWindows, NovellOnlyExtensionFixes
@@ -66,7 +66,7 @@
 # Slackware
 Slackware : LinuxCommon, SlackwareOnly, NotDebian, Mono, msaccess
 # Debian
-DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL, PowerPC
+DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL, PowerPC, MIPS
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only, HPPA, AXP, kFreeBSD
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
@@ -2393,3 +2393,6 @@
 [ TemporaryHacks ]
 disable-rhino.diff
 bsh-hardcode-path.diff
+
+[ MIPS ]
+mips-openjdk-jre-paths.diff
