--- jvmfwk/plugins/sunmajor/pluginlib/vendorbase.hxx-old       2011-08-19 17:16:05.000000000 +0200
+++ jvmfwk/plugins/sunmajor/pluginlib/vendorbase.hxx   2011-08-19 17:14:42.000000000 +0200
@@ -56,7 +56,7 @@
 #ifdef OSL_BIGENDIAN
 #  define JFW_PLUGIN_ARCH "mips"
 #else
-#  define JFW_PLUGIN_ARCH "mips32"
+#  define JFW_PLUGIN_ARCH "mipsel"
 #endif
 #elif defined S390X
 #define JFW_PLUGIN_ARCH "s390x"
