diff --git a/emfio/qa/cppunit/emf/EmfImportTest.cxx b/emfio/qa/cppunit/emf/EmfImportTest.cxx
index 30add1aec73f..5eca543a532d 100644
--- a/emfio/qa/cppunit/emf/EmfImportTest.cxx
+++ b/emfio/qa/cppunit/emf/EmfImportTest.cxx
@@ -24,6 +24,8 @@
 
 #include <memory>
 
+#include <config_features.h>
+
 namespace
 {
 
@@ -47,8 +47,10 @@ class Test : public test::BootstrapFixture, public XmlTestTools
 public:
     CPPUNIT_TEST_SUITE(Test);
     CPPUNIT_TEST(testWorking);
+#if HAVE_MORE_FONTS
     CPPUNIT_TEST(TestDrawString);
     CPPUNIT_TEST(TestDrawStringTransparent);
+#endif
     CPPUNIT_TEST(TestDrawLine);
     CPPUNIT_TEST_SUITE_END();
 };
@@ -93,6 +95,7 @@ void Test::testWorking()
     checkRectPrimitive(aSequenceRect);
 }
 
+#if HAVE_MORE_FONTS
 void Test::TestDrawString()
 {
     // This unit checks for a correct import of an EMF+ file with only one DrawString Record
@@ -134,6 +137,7 @@ void Test::TestDrawStringTransparent()
     assertXPath(pDocument, "/primitive2D/metafile/transform/mask/transform/unifiedtransparence/textsimpleportion", "fontcolor", "#0000ff");
     assertXPath(pDocument, "/primitive2D/metafile/transform/mask/transform/unifiedtransparence/textsimpleportion", "familyname", "ARIAL");
 }
+#endif
 
 void Test::TestDrawLine()
 {
diff --git a/sw/qa/extras/uiwriter/uiwriter.cxx b/sw/qa/extras/uiwriter/uiwriter.cxx
index e403adb33e96..933c022374b2 100644
--- a/sw/qa/extras/uiwriter/uiwriter.cxx
+++ b/sw/qa/extras/uiwriter/uiwriter.cxx
@@ -5903,8 +5903,10 @@ void SwUiWriterTest::testTdf114306_2()
     // load regression document without writer crash
     load(DATA_DIRECTORY, "fdo114306_2.odt");
 
+#if HAVE_MORE_FONTS
     // correct number of pages
     CPPUNIT_ASSERT_EQUAL(4, getPages());
+#endif
 }
 
 
