diff -ru a/mysqlc/source/mysqlc_preparedstatement.cxx b/mysqlc/source/mysqlc_preparedstatement.cxx
--- a/mysqlc/source/mysqlc_preparedstatement.cxx	2016-05-26 18:55:21.000000000 +0200
+++ b/mysqlc/source/mysqlc_preparedstatement.cxx	2016-06-04 15:49:03.320192122 +0200
@@ -564,7 +564,7 @@
             setDouble( _parameterIndex, nValue );
             break;
         }
-        BOOST_FALLTHROUGH;
+        //BOOST_FALLTHROUGH;
     }
 
     case DataType::CHAR:
@@ -591,7 +591,7 @@
             setFloat(_parameterIndex,nValue);
             break;
         }
-        BOOST_FALLTHROUGH;
+        //BOOST_FALLTHROUGH;
     }
 
     case DataType::DOUBLE:
