From: Debian LibreOffice Maintainers <debian-openoffice@lists.debian.org>
Date: Mon, 8 Aug 2016 16:42:28 +0200
Subject: [PATCH] mysqlc-boost-fallthrough.diff

---
 mysqlc/source/mysqlc_preparedstatement.cxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mysqlc/source/mysqlc_preparedstatement.cxx b/mysqlc/source/mysqlc_preparedstatement.cxx
--- a/mysqlc/source/mysqlc_preparedstatement.cxx
+++ b/mysqlc/source/mysqlc_preparedstatement.cxx
@@ -564,7 +564,7 @@ void SAL_CALL OPreparedStatement::setObjectWithInfo(sal_Int32 _parameterIndex, c
             setDouble( _parameterIndex, nValue );
             break;
         }
-        BOOST_FALLTHROUGH;
+        //BOOST_FALLTHROUGH;
     }
 
     case DataType::CHAR:
@@ -591,7 +591,7 @@ void SAL_CALL OPreparedStatement::setObjectWithInfo(sal_Int32 _parameterIndex, c
             setFloat(_parameterIndex,nValue);
             break;
         }
-        BOOST_FALLTHROUGH;
+        //BOOST_FALLTHROUGH;
     }
 
     case DataType::DOUBLE:
-- 
2.7.4

