Description: fix build against MySQL 8.0
Author: Olivier Tilloy <olivier.tilloy@canonical.com>

--- a/connectivity/source/drivers/mysqlc/mysqlc_preparedstatement.hxx
+++ b/connectivity/source/drivers/mysqlc/mysqlc_preparedstatement.hxx
@@ -28,6 +28,8 @@
 #include <com/sun/star/sdbc/XPreparedBatchExecution.hpp>
 #include <com/sun/star/io/XInputStream.hpp>
 
+#include <mysql.h>
+
 namespace connectivity
 {
 namespace mysqlc
@@ -41,9 +43,17 @@ using ::com::sun::star::uno::Type;
 
 struct BindMetaData
 {
+#if LIBMYSQL_VERSION_ID >= 80000
+    bool is_null = false;
+#else
     char is_null = 0;
+#endif
     unsigned long length = 0;
+#if LIBMYSQL_VERSION_ID >= 80000
+    bool error = false;
+#else
     char error = 0;
+#endif
 };
 
 typedef ::cppu::ImplHelper5<css::sdbc::XPreparedStatement, css::sdbc::XParameters,
