Description: Japanese new era "Reiwa(令和)" support
Author: Ikuya Awashiro <ikuya@fruitsbasket.info>
Bug: https://launchpad.net/bugs/1827451

Index: libreoffice-5.1.6~rc2/i18npool/source/calendar/calendar_gregorian.cxx
===================================================================
--- libreoffice-5.1.6~rc2.orig/i18npool/source/calendar/calendar_gregorian.cxx
+++ libreoffice-5.1.6~rc2/i18npool/source/calendar/calendar_gregorian.cxx
@@ -206,10 +206,11 @@ Calendar_hanja::loadCalendar( const OUSt
 }
 
 static const Era gengou_eraArray[] = {
-    {1868,  1,  1, 0},
-    {1912,  7, 30, 0},
-    {1926, 12, 25, 0},
-    {1989,  1,  8, 0},
+    {1868,  1,  1, 0},  // Meiji
+    {1912,  7, 30, 0},  // Taisho
+    {1926, 12, 25, 0},  // Showa
+    {1989,  1,  8, 0},  // Heisei
+    {2019,  5,  1, 0},  // Reiwa
     {0, 0, 0, 0}
 };
 Calendar_gengou::Calendar_gengou() : Calendar_gregorian(gengou_eraArray)
Index: libreoffice-5.1.6~rc2/i18npool/source/localedata/data/ja_JP.xml
===================================================================
--- libreoffice-5.1.6~rc2.orig/i18npool/source/localedata/data/ja_JP.xml
+++ libreoffice-5.1.6~rc2/i18npool/source/localedata/data/ja_JP.xml
@@ -480,6 +480,11 @@
           <DefaultAbbrvName>平</DefaultAbbrvName>
           <DefaultFullName>平成</DefaultFullName>
         </Era>
+        <Era>
+          <EraID>Reiwa</EraID>
+          <DefaultAbbrvName>令</DefaultAbbrvName>
+          <DefaultFullName>令和</DefaultFullName>
+        </Era>
       </Eras>
       <StartDayOfWeek>
         <DayID>sun</DayID>
Index: libreoffice-5.1.6~rc2/svl/source/numbers/zformat.cxx
===================================================================
--- libreoffice-5.1.6~rc2.orig/svl/source/numbers/zformat.cxx
+++ libreoffice-5.1.6~rc2/svl/source/numbers/zformat.cxx
@@ -3247,6 +3247,9 @@ void SvNumberformat::ImpAppendEraG( OUSt
         case 4:
             cEra = 'H';
             break;
+        case 5:
+            cEra = 'R';
+            break;
         default:
             cEra = '?';
             break;
