--- a/Makefile.in	2017-07-28 06:09:37.072183489 +0000
+++ b/Makefile.in	2017-10-18 12:43:59.215567467 +0000
@@ -9,7 +9,7 @@
 
 gb_Top_MODULE_CHECK_TARGETS := slowcheck unitcheck subsequentcheck perfcheck uicheck screenshot
 
-.PHONY : all check-if-root bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install distro-pack-install-strip docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
+.PHONY : all bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install distro-pack-install-strip docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
 
 MAKECMDGOALS?=all
 build_goal:=$(if $(filter build check,$(MAKECMDGOALS)),all)\
@@ -57,14 +57,6 @@ else # MAKE_RESTARTS
 
 all: build
 
-check-if-root:
-	@if test `id -u` = 0 && ! grep -q 'lxc\|docker' /proc/self/cgroup; then \
-		echo; \
-		echo 'No. You make ME a sandwich.'; \
-		echo; \
-		exit 1; \
-	fi
-
 gb_Side ?= host
 
 include $(BUILDDIR)/config_$(gb_Side).mk
@@ -252,7 +244,7 @@ endif
 #
 # Bootstrap
 #
-bootstrap: check-if-root compilerplugins
+bootstrap: compilerplugins
 
 #
 # Build
