--- libreoffice-3.4.3/libreoffice-build/bin/package-ooo.old	2011-08-31 19:09:00.000000000 +0200
+++ libreoffice-3.4.3/libreoffice-build/bin/package-ooo	2011-08-31 19:10:24.000000000 +0200
@@ -77,9 +77,9 @@
 
 # startup wrappers
 for app in calc draw impress math web writer base; do
-    create_qstart_wrapper "$PREFIX/bin/lo${app}${BINSUFFIX}" "-${app}" || exit 1;
+    create_qstart_wrapper "$PREFIX/bin/lo${app}${BINSUFFIX}" "--${app}" || exit 1;
     if test "$COMPAT_OOWRAPPERS" = "YES" ; then
-        create_qstart_wrapper "$PREFIX/bin/oo${app}${BINSUFFIX}" "-${app}" || exit 1;
+        create_qstart_wrapper "$PREFIX/bin/oo${app}${BINSUFFIX}" "--${app}" || exit 1;
     fi
 done
 create_qstart_wrapper "$PREFIX/bin/lofromtemplate${BINSUFFIX}" ".uno:NewDoc" || exit 1;
