--- a/libreoffice-build/bin/unpack
+++ b/libreoffice-build/bin/unpack
@@ -934,15 +934,4 @@
     cp $SRCDIR/$OOOBLOGGER_EXTENSION $OOBUILDDIR/extras/source/extensions || exit 1
 fi
 
-if echo $PROPAGATED_ARGS | grep -q enable-mysql-connector; then
-    if ! echo $PROPAGATED_ARGS | grep -q system-mysql-cppconn; then
-	if [ ! -f $OOBUILDDIR/mysqlcppconn/download/mysql-connector-cpp.zip ]; then
-            check_tarball $SRCDIR/mysql-connector-cpp.zip
-            echo "Copying mysql-connector-cpp.zip into the tree"
-            mkdir -p $OOBUILDDIR/mysqlcppconn/download
-            cp -a $SRCDIR/mysql-connector-cpp.zip $OOBUILDDIR/mysqlcppconn/download || exit 1
-	fi
-    fi
-fi
-
 fi # PIECES hack
