<<<<<<< HEAD
Index: libreoffice-3.4.1/libreoffice-build/patches/dev300/apply
===================================================================
--- libreoffice-3.4.1.orig/libreoffice-build/patches/dev300/apply	2011-07-15 11:40:25.496848861 +0200
+++ libreoffice-3.4.1/libreoffice-build/patches/dev300/apply	2011-07-15 11:40:42.896849100 +0200
@@ -30,7 +30,7 @@
 	      PostgreSQL, RadioButtons, WWInProgress, \
 	      KDE4Experimental, MinGW, Netbook, \
 	      CrossWin32Patches, AutoLayout, \
-	      Speed, GSoC2010, Toolbars
+	      Speed, GSoC2010, Toolbars, PowerPC
 DebianLooseSections: DebianBaseNoHelpContent
 # Optional sections
 Optional : LiberationFontsOnWindows, NovellOnlyExtensionFixes
@@ -50,7 +50,7 @@
 # Slackware
 Slackware : LinuxCommon, Mono, msaccess
 # Debian
-DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL
+DebianBase : LinuxCommon, DebianBaseOnly, GCCSTL, BerkeleyDB43, Split, PostgreSQL, PowerPC
 Debian : DebianBase, DebianOnly, DebianSidOnly, Debian32Only
 DebianEtch : DebianBase, DebianOnly, DebianEtchOnly, Debian32Only
 # Ubuntu
@@ -1467,3 +1467,7 @@
 sd_effects_styles.diff, pixie
 symbols-20-august.diff, i#11167, jopsen
 vEdit-13-August.diff, jopsen
+
+[ PowerPC ]
+no-minimaltoc.diff
+
Index: libreoffice-3.4.1/libreoffice-build/patches/dev300/no-minimaltoc.diff
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libreoffice-3.4.1/libreoffice-build/patches/dev300/no-minimaltoc.diff	2011-07-15 11:40:42.896849100 +0200
=======
Index: libreoffice-3.3.1/libreoffice-build/patches/dev300/no-minimaltoc.diff
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libreoffice-3.3.1/libreoffice-build/patches/hotfixes/no-minimaltoc.diff	2011-03-01 14:37:20.344672014 +0100
>>>>>>> debian-experimental-3.4.0
@@ -0,0 +1,9 @@
+--- solenv/inc/unxlngppc64.mk
++++ solenv/inc/unxlngppc64.mk
+@@ -30,5 +30,4 @@
+ .INCLUDE : unxlngppc.mk
+ 
+ CDEFS+=-DPOWERPC64
+-CFLAGSCXX+=-mminimal-toc
+ BUILD64=1
+
