diff --git a/solenv/gbuild/UITest.mk b/solenv/gbuild/UITest.mk
index e428dbd03c48..f90a1abe1b21 100644
--- a/solenv/gbuild/UITest.mk
+++ b/solenv/gbuild/UITest.mk
@@ -73,7 +73,6 @@ else
 		PYTHONPATH="$(PYPATH)" \
 		TestUserDir="$(call gb_Helper_make_url,$(dir $(call gb_UITest_get_target,$*)))" \
 		PYTHONDONTWRITEBYTECODE=0 \
-		PYTHONWARNINGS=error \
 		$(gb_TEST_ENV_VARS) \
 		$(gb_UITest_COMMAND) \
 		--soffice="$(gb_UITest_SOFFICEARG)" \
