From bcdadc2e9f3f6a400438cbbefffc60bf02baa0f1 Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Tue, 14 Feb 2012 09:32:59 +0000
Subject: Resolves rhbz#789622: Adapt SDK to changed paths in LO installation

---
diff --git a/odk/settings/std.mk b/odk/settings/std.mk
index 0726631..705b528 100644
--- a/odk/settings/std.mk
+++ b/odk/settings/std.mk
@@ -27,8 +27,8 @@ SDKTYPEFLAG=$(OUT_MISC)/oosdk_cpp_types.flag
 URE_TYPES="$(subst \,/,$(URE_MISC)$(PS)types.rdb)"
 URE_SERVICES=$(subst \\,\,$(URE_MISC)$(PS)services.rdb)
 
-OFFICE_TYPES="$(subst \,/,$(OFFICE_PROGRAM_PATH)$(PS)offapi.rdb)"
-OFFICE_SERVICES=$(subst \\,\,$(OFFICE_PROGRAM_PATH)$(PS)services.rdb)
+OFFICE_TYPES="$(subst \,/,$(OFFICE_PROGRAM_PATH)$(PS)types$(PS)offapi.rdb)"
+OFFICE_SERVICES=$(subst \\,\,$(OFFICE_PROGRAM_PATH)$(PS)services$(PS)services.rdb)
 
 OFFICE_TYPE_LIBRARY="$(OFFICE_TYPES)"
 
--
cgit v0.9.0.2-2-gbebe
