diff --git a/bin/package-ooo b/bin/package-ooo
index 50ceaf1..7c4b082 100755
--- openoffice.org-3.2.0/ooo-build-3-2-0-9/bin/package-ooo
+++ openoffice.org-3.2.0/ooo-build-3-2-0-9/bin/package-ooo
@@ -83,7 +83,7 @@ mkdir -p $MANDIR/man1
 for app in calc draw impress math web writer base; do
     create_qstart_wrapper "$PREFIX/bin/oo${app}${BINSUFFIX}" "-${app}" || exit 1;
 done
-create_qstart_wrapper "$PREFIX/bin/oofromtemplate${BINSUFFIX}" "slot:5500" || exit 1;
+create_qstart_wrapper "$PREFIX/bin/oofromtemplate${BINSUFFIX}" ".uno:NewDoc" || exit 1;
 create_qstart_wrapper "$PREFIX/bin/ooffice${BINSUFFIX}" "" || exit 1;
 create_unopkg_wrapper "$PREFIX/bin/unopkg${BINSUFFIX}" "" || exit 1;
 if test "z$VENDORNAME" = "zNovell" -o "z$VENDORNAME" = "zPLD" \
