From e3ac30fe6287b4cd0c9ef103d011a109a359a0be Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 22 Oct 2014 09:50:25 +0200
Subject: fix oox build on big endian

Change-Id: I580266d908e30ef076de0517f41a600f4c9372c1

diff --git a/oox/source/ole/olehelper.cxx b/oox/source/ole/olehelper.cxx
index 409eb4a..77e8c4a 100644
--- a/oox/source/ole/olehelper.cxx
+++ b/oox/source/ole/olehelper.cxx
@@ -262,7 +262,7 @@ void OleHelper::exportGuid( BinaryOutputStream& rOStr, const SvGlobalName& rId )
     rOStr << rId.GetCLSID().Data1;
     rOStr << rId.GetCLSID().Data2;
     rOStr << rId.GetCLSID().Data3;
-    rOStr.writeArray( &rId.GetCLSID().Data4, 8 );
+    rOStr.writeArray( rId.GetCLSID().Data4, 8 );
 }
 
 OUString OleHelper::importGuid( BinaryInputStream& rInStrm )
-- 
cgit v0.10.2

