diff --git a/configure.ac b/configure.ac
index 69479c3..2a60133 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9841,7 +9841,7 @@ else
             OPENCL_LIBS="-LIBPATH:$formatted_path/lib/x86 OpenCL.lib"
         else
             OPENCL_CFLAGS="-I$with_opencl_sdk/include"
-            OPENCL_LIBS="-L$with_opencl_sdk/lib/x86 -lOpenCL"
+            OPENCL_LIBS="-L$with_opencl_sdk/lib/`dpkg-architecture -qDEB_HOST_MULTIARCH` -lOpenCL"
         fi
         AC_MSG_RESULT([found at path $with_opencl_sdk])
         AC_DEFINE(HAVE_FEATURE_OPENCL)
