--- libreoffice-3.5.0/postprocess/packcomponents/makefile.mk-old	2011-06-13 23:35:26.000000000 +0000
+++ libreoffice-3.5.0/postprocess/packcomponents/makefile.mk	2011-06-13 23:35:58.000000000 +0000
@@ -229,10 +229,6 @@
 my_components += ucpgvfs
 .END
 
-.IF "$(ENABLE_KAB)" != ""
-my_components += kab1
-.END
-
 .IF "$(ENABLE_KDE)" != ""
 my_components += kdebe1
 .END
@@ -269,12 +265,6 @@
     web
 .END
 
-.IF "$(WITH_BINFILTER)" != "NO"
-my_components += \
-    bf_migratefilter \
-    bindet
-.END
-
 .IF "$(WITH_LDAP)" == "YES"
 my_components += ldapbe2
 .END
@@ -355,10 +345,6 @@
 my_components += cmdmail
 .END
 
-.IF "$(OS)" != "WNT" && "$(ENABLE_EVOAB2)" != ""
-my_components += evoab
-.END
-
 .IF "$(OS)" != "WNT" && "$(ENABLE_GSTREAMER)" != ""
 my_components += avmediagstreamer
 .END
