--- libreoffice-3.3.0/libreoffice-build-3-2-99-2/bin/package-ooo.old	2010-10-11 22:29:06.000000000 +0200
+++ libreoffice-3.3.0/libreoffice-build-3-2-99-2/bin/package-ooo	2010-10-11 22:29:47.000000000 +0200
@@ -820,71 +820,71 @@
 
 	create_package_directory gid_Module_Root_Ure_Hidden		pkg/ure
 
-        create_package_directory gid_Module_Root                        pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Brand                  pkg/openoffice.org-common
-	create_package_directory gid_Module_Root_Files_Images           pkg/openoffice.org-common
-	create_package_directory gid_Module_Oo_Linguistic               pkg/openoffice.org-common
-        create_package_directory gid_Module_Optional_Xsltfiltersamples  pkg/openoffice.org-common
-	create_package_directory gid_Module_Optional_Javafilter		pkg/openoffice.org-common
+        create_package_directory gid_Module_Root                        pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Brand                  pkg/libreoffice-common
+	create_package_directory gid_Module_Root_Files_Images           pkg/libreoffice-common
+	create_package_directory gid_Module_Oo_Linguistic               pkg/libreoffice-common
+        create_package_directory gid_Module_Optional_Xsltfiltersamples  pkg/libreoffice-common
+	create_package_directory gid_Module_Optional_Javafilter		pkg/libreoffice-common
 	if [ -f gid_Module_Optional_Binfilter ]; then 
-        	create_package_directory gid_Module_Optional_Binfilter          pkg/openoffice.org-filter-binfilter
+        	create_package_directory gid_Module_Optional_Binfilter          pkg/libreoffice-filter-binfilter
         fi
-	create_package_directory gid_Module_Optional_Grfflt             pkg/openoffice.org-draw
-        create_package_directory gid_Module_Prg_Calc_Bin                pkg/openoffice.org-calc
-        create_package_directory gid_Module_Prg_Math_Bin                pkg/openoffice.org-math	
-        create_package_directory gid_Module_Prg_Draw_Bin                pkg/openoffice.org-draw
-        create_package_directory gid_Module_Prg_Wrt_Bin                 pkg/openoffice.org-writer
-        create_package_directory gid_Module_Prg_Impress_Bin             pkg/openoffice.org-impress
-        create_package_directory gid_Module_Prg_Base_Bin                pkg/openoffice.org-base
-        create_package_directory gid_Module_Brand_Prg_Calc              pkg/openoffice.org-calc
-        create_package_directory gid_Module_Brand_Prg_Math              pkg/openoffice.org-math	
-        create_package_directory gid_Module_Brand_Prg_Draw              pkg/openoffice.org-draw
-        create_package_directory gid_Module_Brand_Prg_Wrt               pkg/openoffice.org-writer
-        create_package_directory gid_Module_Brand_Prg_Impress           pkg/openoffice.org-impress
-        create_package_directory gid_Module_Brand_Prg_Base              pkg/openoffice.org-base
+	create_package_directory gid_Module_Optional_Grfflt             pkg/libreoffice-draw
+        create_package_directory gid_Module_Prg_Calc_Bin                pkg/libreoffice-calc
+        create_package_directory gid_Module_Prg_Math_Bin                pkg/libreoffice-math	
+        create_package_directory gid_Module_Prg_Draw_Bin                pkg/libreoffice-draw
+        create_package_directory gid_Module_Prg_Wrt_Bin                 pkg/libreoffice-writer
+        create_package_directory gid_Module_Prg_Impress_Bin             pkg/libreoffice-impress
+        create_package_directory gid_Module_Prg_Base_Bin                pkg/libreoffice-base
+        create_package_directory gid_Module_Brand_Prg_Calc              pkg/libreoffice-calc
+        create_package_directory gid_Module_Brand_Prg_Math              pkg/libreoffice-math	
+        create_package_directory gid_Module_Brand_Prg_Draw              pkg/libreoffice-draw
+        create_package_directory gid_Module_Brand_Prg_Wrt               pkg/libreoffice-writer
+        create_package_directory gid_Module_Brand_Prg_Impress           pkg/libreoffice-impress
+        create_package_directory gid_Module_Brand_Prg_Base              pkg/libreoffice-base
         create_package_directory gid_Module_Optional_Pyuno		pkg/python-uno
-        create_package_directory gid_Module_Optional_Gnome		pkg/openoffice.org-gnome
-        create_package_directory gid_Module_Optional_Kde                pkg/openoffice.org-kde
+        create_package_directory gid_Module_Optional_Gnome		pkg/libreoffice-gnome
+        create_package_directory gid_Module_Optional_Kde                pkg/libreoffice-kde
 
-        create_package_directory gid_Module_Root_Files_2                pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Files_3                pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Files_4                pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Files_5                pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Files_6                pkg/openoffice.org-common
-        create_package_directory gid_Module_Root_Files_7                pkg/openoffice.org-common
-        create_package_directory gid_Module_Optional_Testtool           pkg/openoffice.org-qa-tools
+        create_package_directory gid_Module_Root_Files_2                pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Files_3                pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Files_4                pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Files_5                pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Files_6                pkg/libreoffice-common
+        create_package_directory gid_Module_Root_Files_7                pkg/libreoffice-common
+        create_package_directory gid_Module_Optional_Testtool           pkg/libreoffice-qa-tools
         if [ -e gid_Module_Optional_Pymailmerge ]; then
-		create_package_directory gid_Module_Optional_Pymailmerge	pkg/openoffice.org-emailmerge
+		create_package_directory gid_Module_Optional_Pymailmerge	pkg/libreoffice-emailmerge
 	else # post m26
-		mkdir -p pkg/openoffice.org-emailmerge/$OOINSTBASE/basis$VERSION/program
-		mv pkg/openoffice.org-common/$OOINSTBASE/basis$VERSION/program/mailmerge.py \
-			pkg/openoffice.org-emailmerge/$OOINSTBASE/basis$VERSION/program/mailmerge.py
+		mkdir -p pkg/libreoffice-emailmerge/$OOINSTBASE/basis$VERSION/program
+		mv pkg/libreoffice-common/$OOINSTBASE/basis$VERSION/program/mailmerge.py \
+			pkg/libreoffice-emailmerge/$OOINSTBASE/basis$VERSION/program/mailmerge.py
 	fi
-	create_package_directory gid_Module_Optional_OGLTrans		pkg/openoffice.org-ogltrans
+	create_package_directory gid_Module_Optional_OGLTrans		pkg/libreoffice-ogltrans
 
-	create_package_directory gid_Module_Root_SDK                    pkg/openoffice.org-dev
+	create_package_directory gid_Module_Root_SDK                    pkg/libreoffice-dev
 
 	for l in `echo $OOO_LANGS_LIST`; do
 		for p in Impress Draw Math Calc Base Writer; do
-			create_package_directory  gid_Module_Langpack_${p}_`echo $l | sed -e s/-/_/g`   pkg/openoffice.org-l10n-$l
+			create_package_directory  gid_Module_Langpack_${p}_`echo $l | sed -e s/-/_/g`   pkg/libreoffice-l10n-$l
 		done
-		create_package_directory gid_Module_Langpack_Basis_`echo $l | sed -e s/-/_/g`	pkg/openoffice.org-l10n-$l
-		create_package_directory gid_Module_Langpack_Brand_`echo $l | sed -e s/-/_/g`	pkg/openoffice.org-l10n-$l
-		create_package_directory gid_Module_Langpack_Resource_`echo $l | sed -e s/-/_/g`        pkg/openoffice.org-l10n-$l
-		create_package_directory gid_Module_Langpack_Help_`echo $l | sed -e s/-/_/g`    pkg/openoffice.org-help-$l
+		create_package_directory gid_Module_Langpack_Basis_`echo $l | sed -e s/-/_/g`	pkg/libreoffice-l10n-$l
+		create_package_directory gid_Module_Langpack_Brand_`echo $l | sed -e s/-/_/g`	pkg/libreoffice-l10n-$l
+		create_package_directory gid_Module_Langpack_Resource_`echo $l | sed -e s/-/_/g`        pkg/libreoffice-l10n-$l
+		create_package_directory gid_Module_Langpack_Help_`echo $l | sed -e s/-/_/g`    pkg/libreoffice-help-$l
 		if [ -f gid_Module_Optional_Binfilter ]; then
 			if [ "$l" = "en-US" ]; then
-				create_package_directory gid_Module_Langpack_Binfilter_en_US    pkg/openoffice.org-filter-binfilter
+				create_package_directory gid_Module_Langpack_Binfilter_en_US    pkg/libreoffice-filter-binfilter
 			else
-				create_package_directory gid_Module_Langpack_Binfilter_`echo $l | sed -e s/-/_/g`    pkg/openoffice.org-l10n-$l
+				create_package_directory gid_Module_Langpack_Binfilter_`echo $l | sed -e s/-/_/g`    pkg/libreoffice-l10n-$l
 			fi
 		fi
 		# some help files are in _Langpack_{Writer,Impress,...}_<lang>
 		# move them from -l10n to -help
 		if [ "$l" = "en-US" ]; then d=en; else d=$l; fi
-		mv pkg/openoffice.org-l10n-$l/$OOINSTBASE/basis$VERSION/help/$d/* \
-			pkg/openoffice.org-help-$l/$OOINSTBASE/basis$VERSION/help/$d && \
-		rmdir pkg/openoffice.org-l10n-$l/$OOINSTBASE/basis$VERSION/help/$d
+		mv pkg/libreoffice-l10n-$l/$OOINSTBASE/basis$VERSION/help/$d/* \
+			pkg/libreoffice-help-$l/$OOINSTBASE/basis$VERSION/help/$d && \
+		rmdir pkg/libreoffice-l10n-$l/$OOINSTBASE/basis$VERSION/help/$d
 	done
 
 	# move_wrappers <directory_name> <name> [...]
@@ -898,40 +898,40 @@
 			shift
 		done
 	}
-	move_wrappers pkg/openoffice.org-common ooffice oofromtemplate soffice unopkg
-	move_wrappers pkg/openoffice.org-base oobase
-	move_wrappers pkg/openoffice.org-writer oowriter ooweb
-	move_wrappers pkg/openoffice.org-calc oocalc
-	move_wrappers pkg/openoffice.org-impress ooimpress
-	move_wrappers pkg/openoffice.org-math oomath
-	move_wrappers pkg/openoffice.org-draw oodraw
+	move_wrappers pkg/libreoffice-common ooffice oofromtemplate soffice unopkg
+	move_wrappers pkg/libreoffice-base oobase
+	move_wrappers pkg/libreoffice-writer oowriter ooweb
+	move_wrappers pkg/libreoffice-calc oocalc
+	move_wrappers pkg/libreoffice-impress ooimpress
+	move_wrappers pkg/libreoffice-math oomath
+	move_wrappers pkg/libreoffice-draw oodraw
 	
 	# Move all libraries, binaries, *.rdb from -common to -core
 	for d in $OOINSTBASE/basis$VERSION/program $OOINSTBASE/program; do \
-	  if [ ! -d $OODESTDIR/pkg/openoffice.org-core/$d ]; then \
-	  mkdir -p $OODESTDIR/pkg/openoffice.org-core/$d; \
+	  if [ ! -d $OODESTDIR/pkg/libreoffice-core/$d ]; then \
+	  mkdir -p $OODESTDIR/pkg/libreoffice-core/$d; \
 	  fi &&
-	  ( cd pkg/openoffice.org-common/$d
+	  ( cd pkg/libreoffice-common/$d
 	    find -maxdepth 1 \
 	       -regex '\./\(.*\.so.*\|.*\.bin\|pagein\|nsplugin\|kdefilepicker\|msfontextract\|.*\.rdb\|javaldx\|uri-encode\)' \
-		   -exec mv {} $OODESTDIR/pkg/openoffice.org-core/$d \;
+		   -exec mv {} $OODESTDIR/pkg/libreoffice-core/$d \;
 	  ); \
 	done
 
 	# install additional ooo-build scripts & misc stuff
-	mkdir -p pkg/openoffice.org-common/usr/share/man/man1
+	mkdir -p pkg/libreoffice-common/usr/share/man/man1
 	mv usr/share/man/man1/openoffice$BINSUFFIX.1 \
-		pkg/openoffice.org-common/usr/share/man/man1
-	mkdir -p pkg/openoffice.org-common/etc/bash_completion.d
+		pkg/libreoffice-common/usr/share/man/man1
+	mkdir -p pkg/libreoffice-common/etc/bash_completion.d
 	mv etc/bash_completion.d/ooffice$BINSUFFIX.sh \
-		pkg/openoffice.org-common/etc/bash_completion.d
+		pkg/libreoffice-common/etc/bash_completion.d
 	mv .$OOINSTBASE/basis$VERSION/program/java-set-classpath \
-		pkg/openoffice.org-common/$OOINSTBASE/program
+		pkg/libreoffice-common/$OOINSTBASE/program
 	if echo $OOO_LANGS_LIST | grep -q en-US; then
 		for i in forms/resume.ott officorr/project-proposal.ott; do \
-			mkdir -p pkg/openoffice.org-common/$OOINSTBASE/basis$VERSION/share/template/en-US/`dirname $i`; \
+			mkdir -p pkg/libreoffice-common/$OOINSTBASE/basis$VERSION/share/template/en-US/`dirname $i`; \
 			mv .$OOINSTBASE/basis$VERSION/share/template/en-US/$i \
-				pkg/openoffice.org-common/$OOINSTBASE/basis$VERSION/share/template/en-US/$i; \
+				pkg/libreoffice-common/$OOINSTBASE/basis$VERSION/share/template/en-US/$i; \
 		done; \
 	fi
 	# Warn for any remaining files
