--- a/external/pdfium/Library_pdfium.mk-old	2021-11-12 19:28:08.723360535 +0100
+++ b/external/pdfium/Library_pdfium.mk	2021-11-12 19:31:42.430984472 +0100
@@ -19,7 +19,7 @@
     -I$(call gb_UnpackedTarball_get_dir,pdfium) \
     -I$(call gb_UnpackedTarball_get_dir,pdfium)/third_party \
     -I$(call gb_UnpackedTarball_get_dir,pdfium)/third_party/agg23 \
-    -I$(call gb_UnpackedTarball_get_dir,pdfium)/third_party/abseil-cpp \
+    -Isystem/usr/include/absl \
     -isystem/usr/include/openjpeg-2.3 \
     -isystem/usr/include/openjpeg-2.4 \
     $$(INCLUDE) \
@@ -521,11 +521,6 @@
     UnpackedTarball/pdfium/third_party/agg23/agg_vcgen_stroke \
 ))
 
-# third_party/abseil-cpp
-$(eval $(call gb_Library_add_generated_exception_objects,pdfium,\
-    UnpackedTarball/pdfium/third_party/abseil-cpp/absl/types/bad_optional_access \
-))
-
 # pdfium_base
 $(eval $(call gb_Library_add_generated_exception_objects,pdfium,\
     UnpackedTarball/pdfium/third_party/base/allocator/partition_allocator/address_space_randomization \
@@ -554,6 +554,7 @@
 $(eval $(call gb_Library_add_libs,pdfium,\
     -ldl \
     -lrt \
+    -labsl_bad_optional_access \
     -lopenjp2 \
 ))
 
