Index: libreoffice-3.5.0/configure.in
===================================================================
--- libreoffice-3.5.0/configure.in	2011-12-15 18:38:18.961310083 +0100
+++ libreoffice-3.5.0/configure.in	2011-12-15 18:39:06.181310723 +0100
@@ -6105,6 +6105,9 @@
     if test "x$LOCATED" != "xyes"; then
         PKG_CHECK_MODULES(MOZILLA_HEADERS, libxul, [LOCATED=yes], [LOCATED=no])
     fi
+    if test "x$LOCATED" != "xyes"; then
+        PKG_CHECK_MODULES(MOZILLA_HEADERS, mozilla-plugin, [LOCATED=yes], [LOCATED=no])
+    fi
     # if still not found bail out
     if test "x$LOCATED" != "xyes"; then
         AC_MSG_ERROR([npapi.h header file not found])
