From f5b96148cb9e489b2100e6a2fd6805bff637e113 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 11 Jan 2012 15:54:35 +0000
Subject: we need a var. assignment here, not a value

From 2e30b7ca1e89282a2d2d4b5682ee82f13923d715 Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Wed, 11 Jan 2012 16:32:02 +0000
Subject: bah, i forgot *export*

---
diff --git a/configure.in b/configure.in
index 204065b..ab12cea 100644
--- a/configure.in
+++ b/configure.in
@@ -4313,8 +4313,9 @@ if test "$_os" = "Linux" && test "$host_cpu" = "powerpc"; then
     # IBMs JDK needs this...
     JITC_PROCESSOR_TYPE=6
     export JITC_PROCESSOR_TYPE
+    JITC_PROCESSOR_TYPE_EXPORT="export JITC_PROCESSOR_TYPE=$JITC_PROCESSOR_TYPE"
 fi
-AC_SUBST([JITC_PROCESSOR_TYPE])
+AC_SUBST([JITC_PROCESSOR_TYPE_EXPORT])
 
 if test $_os = "WINNT"; then
     WITH_VC_REDIST="TRUE"
diff --git a/desktop/scripts/soffice.sh.in b/desktop/scripts/soffice.sh.in
index defd7b2..2014ab9 100755
--- a/desktop/scripts/soffice.sh.in
+++ b/desktop/scripts/soffice.sh.in
@@ -45,7 +45,7 @@ export SAL_ENABLE_FILE_LOCKING
 # uncomment line below if you encounter problems starting soffice on your system
 # SAL_NO_XINITTHREADS=true; export SAL_NO_XINITTHREADS
 
-@JITC_PROCESSOR_TYPE@
+@JITC_PROCESSOR_TYPE_EXPORT@
 
 # resolve installation directory
 sd_cwd=`pwd`
--
cgit v0.9.0.2-2-gbebe
