--- a/desktop/scripts/soffice.sh.in-old	2012-01-11 15:24:01.000000000 +0000
+++ b/desktop/scripts/soffice.sh.in	2012-01-11 15:24:36.000000000 +0000
@@ -45,7 +45,7 @@
 # uncomment line below if you encounter problems starting soffice on your system
 # SAL_NO_XINITTHREADS=true; export SAL_NO_XINITTHREADS
 
-@JITC_PROCESSOR_TYPE@
+export JITC_PROCESSOR_TYPE=@JITC_PROCESSOR_TYPE@
 
 # resolve installation directory
 sd_cwd=`pwd`
