--- a/configure.ac	2013-10-29 04:11:43.221043311 +0000
+++ b/configure.ac	2013-10-29 04:20:55.317050847 +0000
@@ -8055,7 +8055,10 @@
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         # This causes an error if no python command is found
-        AM_PATH_PYTHON([2.6])
+        AM_PATH_PYTHON([3.3], ,[:])
+        if test "$PYTHON" == ":"; then
+            AM_PATH_PYTHON([2.6])
+        fi
         python_include=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'));"`
         python_version=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('VERSION'));"`
         python_libs=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBS'));"`
