From 4572e00ec358f507e64cbab9d9f3430ea6f97d51 Mon Sep 17 00:00:00 2001
From: Matúš Kukan <matus.kukan@gmail.com>
Date: Wed, 04 Jan 2012 22:06:57 +0000
Subject: sax: this had to be accidentally added when merging

---
diff --git a/sax/prj/build.lst b/sax/prj/build.lst
index 940ef70..34e520f 100644
--- a/sax/prj/build.lst
+++ b/sax/prj/build.lst
@@ -1,4 +1,2 @@
 ax	sax	:	offapi cppuhelper EXPAT:expat comphelper LIBXSLT:libxslt NULL
 ax	sax\prj									nmake	-	all	ax_prj NULL
-
-ax  sax\test							nmake	-	all ax_test NULL
--
cgit v0.9.0.2-2-gbebe
