diff --git a/Repository.mk b/Repository.mk
index 16ef7ba..ea257b1 100755
--- a/Repository.mk
+++ b/Repository.mk
@@ -563,7 +563,9 @@ $(eval $(call gb_Helper_register_libraries_for_install,PLAINLIBS_OOO,ooo, \
 ))
 $(eval $(call gb_Helper_register_libraries,PLAINLIBS_OOO, \
 	pyuno \
+	py2uno \
 	pyuno_wrapper \
+	py2uno_wrapper \
 	recentfile \
 	ucpdav1 \
 	xsec_xmlsec \
@@ -590,6 +592,7 @@ $(eval $(call gb_Helper_register_libraries,OOOLIBS, \
 	postgresql-sdbc \
 	postgresql-sdbc-impl \
 	pythonloader \
+	python2loader \
 ))
 
 $(eval $(call gb_Helper_register_libraries,UNOVERLIBS, \
diff --git a/RepositoryFixes.mk b/RepositoryFixes.mk
index 6beb818..1a570e4 100644
--- a/RepositoryFixes.mk
+++ b/RepositoryFixes.mk
@@ -49,6 +49,7 @@ gb_Library_FILENAMES := $(patsubst pyuno_wrapper:libpyuno_wrapper.dylib,pyuno_wr
 else
 # libpyuno_wrapper.so => pyuno.so
 gb_Library_FILENAMES := $(patsubst pyuno_wrapper:libpyuno_wrapper.so,pyuno_wrapper:pyuno.so,$(gb_Library_FILENAMES))
+gb_Library_FILENAMES := $(patsubst py2uno_wrapper:libpy2uno_wrapper.so,py2uno_wrapper:py2uno.so,$(gb_Library_FILENAMES))
 endif
 
 ifneq ($(OS),ANDROID)
