--- a/Makefile.top-old	2012-04-06 14:26:46.828238642 +0200
+++ b/Makefile.top	2012-04-06 14:26:58.308239187 +0200
@@ -111,6 +111,7 @@ packimages\
 padmin\
 psprint_config\
 pyuno\
+py3uno\
 qadevOOo\
 regexp\
 registry\
diff --git a/Repository.mk b/Repository.mk
index 3609d09..c0b11c2 100644
--- a/Repository.mk
+++ b/Repository.mk
@@ -445,7 +445,9 @@ $(eval $(call gb_Helper_register_libraries,PLAINLIBS_OOO, \
     officebean \
     package2 \
     pyuno \
+    py3uno \
     pyuno_wrapper \
+    py3uno_wrapper \
     recentfile \
     scriptframe \
     sdbc2 \
@@ -555,6 +557,7 @@ $(eval $(call gb_Helper_register_libraries,UNOLIBS_OOO, \
     OGLTrans \
     passwordcontainer \
     pythonloader \
+    python3loader \
     postgresql-sdbc \
     postgresql-sdbc-impl \
     simplecanvas \
