diff --git a/pyuno/source/module/uno.py b/pyuno/source/module/uno.py
index d99884a..febedf2 100644
--- a/pyuno/source/module/uno.py
+++ b/pyuno/source/module/uno.py
@@ -329,7 +329,7 @@ def _uno_import( name, *optargs, **kwargs ):
 
               uno_import_exc = ImportError(
                   "%s (or '%s.%s' is unknown)" % (py_import_exc, name, x)
-                  ).with_traceback(py_import_exc.__traceback__)
+                  )
               uno_import_exc._uno_import_failed = True
               raise uno_import_exc
 
