diff --git a/patches/dev300/apply b/patches/dev300/apply
index 6d6f28f..3661292 100755
--- libreoffice-3.3.0/libreoffice-build/patches/dev300/apply
+++ libreoffice-3.3.0/libreoffice-build/patches/dev300/apply
@@ -837,6 +837,8 @@ system-python-ure-bootstrap.diff, deb#501028, i#90701
 pyuno-ooodir.diff, i#90701
 system-python-uno-path.diff, i#97629
 
+pyuno-demo-fix-PYTHONPATH.diff
+
 [ LiberationFontsOnWindows ]
 
 # Use Liberation fonts on Windows, too
--- /dev/null	2011-01-09 13:55:42.933962422 +0100
+++ libreoffice-3.3.0/libreoffice-build/patches/dev300/pyuno-demo-fix-PYTHONPATH.diff	2011-01-09 18:00:27.000000000 +0100
@@ -0,0 +1,18 @@
+diff --git a/pyuno/demo/pyunoenv.tcsh b/pyuno/demo/pyunoenv.tcsh
+index 1a83199..1c86ca7 100644
+--- pyuno/demo/pyunoenv.tcsh
++++ pyuno/demo/pyunoenv.tcsh
+@@ -17,7 +17,12 @@ if( ! $?LD_LIBRARY_PATH ) then
+ setenv LD_LIBRARY_PATH
+ endif
+ 
+-setenv PYTHONPATH .:$OOOHOME/program:$OOOHOME/program/pydemo:$OOOHOME/program/python/lib:$PYTHONPATH
++if ( $?PYTHOHOME )
++	setenv PYTHONPATH $OOOHOME/program:$OOOHOME/program/pydemo:$OOOHOME/program/python/lib:$PYTHONPATH
++else
++	setenv PYTHONPATH $OOOHOME/program:$OOOHOME/program/pydemo:$OOOHOME/program/python/lib
++endif
++	
+ setenv LD_LIBRARY_PATH $OOOHOME/program:$LD_LIBRARY_PATH
+ 
+ if( $?PYTHONHOME ) then
