Index: libreoffice-4.1.4~rc2/qadevOOo/runner/lib/TestParameters.java
===================================================================
--- libreoffice-4.1.4~rc2.orig/qadevOOo/runner/lib/TestParameters.java	2013-12-16 14:35:33.362933820 +0100
+++ libreoffice-4.1.4~rc2/qadevOOo/runner/lib/TestParameters.java	2013-12-16 14:35:33.354933820 +0100
@@ -314,7 +314,7 @@
         String operatingSystem = "";
         if (osname.indexOf ("windows")>-1) {
             operatingSystem = PropertyName.WNTMSCI;
-        } else if (osname.indexOf ("linux")>-1) {
+        } else if (osname.indexOf ("linux")>-1 || osname.indexOf ("kfreebsd")>-1) {
             operatingSystem = PropertyName.UNXLNGI;
         } else if (osname.indexOf ("sunos")>-1) {
             if (osarch.equals ("x86")) {
