diff --git a/patches/dev300/apply b/patches/dev300/apply
index 4283ba1..6dc9d1e 100644
--- openoffice.org-3.2.1/ooo-build-3-2-1-4/patches/dev300/apply
+++ openoffice.org-3.2.1/ooo-build-3-2-1-4/patches/dev300/apply
@@ -3070,6 +3070,9 @@ lockfile-dont-do-dns-lookup.diff, n#389257, jholesov
 # TODO file up-stream
 lockfile-less-symbols.diff, jholesov
 
+# succesful -> successful
+regcomp-fix-spelling.diff, i#113958, rengelha
+
 [ InternalMesaHeaders ]
 internal-mesa-headers-config_office.diff, fridrich
 internal-mesa-headers-slideshow.diff, fridrich
--- /dev/null	2010-08-16 13:41:41.277183513 +0200
+++ openoffice.org-3.2.1/ooo-build-3-2-1-4/patches/dev300/regcomp-fix-spelling.diff	2010-08-18 13:36:21.000000000 +0200
@@ -0,0 +1,22 @@
+diff --git a/cpputools/source/registercomponent/registercomponent.cxx
+b/cpputools/source/registercomponent/registercomponent.cxx
+--- cpputools/source/registercomponent/registercomponent.cxx
++++ cpputools/source/registercomponent/registercomponent.cxx
+@@ -584,7 +584,7 @@
+                     
+             if ( ! _bSilent )
+             {
+-                fprintf(stderr, "register component '%s' in registry '%s' succesful!\n", sUrl.getStr(), _sRegName.getStr());
++                fprintf(stderr, "register component '%s' in registry '%s' successful!\n", sUrl.getStr(), _sRegName.getStr());
+             }
+ 
+ 		}
+@@ -613,7 +613,7 @@
+             if (bRet)
+             {
+                 if ( ! _bSilent )
+-                    fprintf(stderr, "revoke component '%s' from registry '%s' succesful!\n", sUrl.getStr(), _sRegName.getStr());
++                    fprintf(stderr, "revoke component '%s' from registry '%s' successful!\n", sUrl.getStr(), _sRegName.getStr());
+             }
+             else
+             {
