From d4390846ea8eb22a2257841052e49b286c8ac18a Mon Sep 17 00:00:00 2001
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Thu, 17 May 2012 22:21:45 +0200
Subject: [PATCH] remove broken mysqlcon version check

Change-Id: I7d19a5c6b31c0066a8eca8993b57aab9d477e892
---
 configure.in |   14 --------------
 1 file changed, 14 deletions(-)

diff --git a/configure.in b/configure.in
index 6072782..93763e6 100644
--- a/configure.in
+++ b/configure.in
@@ -6770,20 +6770,6 @@ if test "$ENABLE_MYSQLC" = "YES"; then
         save_LDFLAGS=$LDFLAGS
         LDFLAGS="$LDFLAGS -lmysqlcppconn"
         AC_MSG_CHECKING([version])
-        AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <mysql_driver.h>
-
-int main(int argc, char **argv) {
-    sql::Driver *driver;
-    driver = get_driver_instance();
-    if (driver->getMajorVersion() > 1 || \
-       (driver->getMajorVersion() == 1 && driver->getMinorVersion() > 0) || \
-       (driver->getMajorVersion() == 1 && driver->getMinorVersion() == 0 && driver->getPatchVersion() >= 6))
-        return 0;
-      else
-        return 1;
-}
-      ]])],[AC_MSG_RESULT(OK)],[AC_MSG_ERROR([not suitable, we need >= 1.0.6])],[])
 
         AC_LANG_POP([C++])
         LDFLAGS=$save_LDFLAGS
-- 
1.7.9.5

