Index: libreoffice-4.1.4~rc2/dbaccess/source/ui/misc/dbumiscres.src
===================================================================
--- libreoffice-4.1.4~rc2.orig/dbaccess/source/ui/misc/dbumiscres.src	2013-12-16 14:35:13.022934901 +0100
+++ libreoffice-4.1.4~rc2/dbaccess/source/ui/misc/dbumiscres.src	2013-12-16 14:35:13.014934901 +0100
@@ -86,7 +86,7 @@
 String RID_STR_EXTENSION_NOT_PRESENT
 {
     // #i96130# use hard coded name
-    Text [ en-US ] = "The report, \"$file$\", requires the Oracle Report Builder feature.";
+    Text [ en-US ] = "The report, \"$file$\", requires the Oracle Report Builder feature. For this, install the package 'libreoffice-report-builder'.";
 };
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
