diff --git a/sal/qa/rtl/math/test-rtl-math.cxx b/sal/qa/rtl/math/test-rtl-math.cxx
index bab2774a414c..d434a95326fa 100644
--- a/sal/qa/rtl/math/test-rtl-math.cxx
+++ b/sal/qa/rtl/math/test-rtl-math.cxx
@@ -685,7 +687,9 @@ public:
     CPPUNIT_TEST(test_acosh);
     CPPUNIT_TEST(test_asinh);
     CPPUNIT_TEST(test_atanh);
+#ifndef __riscv
     CPPUNIT_TEST(test_payloadNaN);
+#endif
     CPPUNIT_TEST_SUITE_END();
 };
 
