diff --git a/rsc/source/parser/erscerr.cxx b/rsc/source/parser/erscerr.cxx
index c5a9fd7..c105f72 100644
--- a/rsc/source/parser/erscerr.cxx
+++ b/rsc/source/parser/erscerr.cxx
@@ -163,9 +163,9 @@ void RscError::WriteError( const ERRTYPE& rError, const char * pMessage )
             break;
 
         case ERR_UNKNOWNSW:
-            StdLstErr( "Unknown switch <" );
-            StdLstErr( pMessage );
-            StdLstErr( ">." );
+            StdLstOut( "Unknown switch <" );
+            StdLstOut( pMessage );
+            StdLstOut( ">." );
             break;
 
         case ERR_REFTODEEP:
