From: Debian LibreOffice Maintainers <debian-openoffice@lists.debian.org>
Date: Fri, 26 Feb 2016 13:05:28 +0100
Subject: [PATCH] rsc-no-error-about-unknown-switch.diff

---
 rsc/source/parser/erscerr.cxx | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/rsc/source/parser/erscerr.cxx b/rsc/source/parser/erscerr.cxx
--- a/rsc/source/parser/erscerr.cxx
+++ b/rsc/source/parser/erscerr.cxx
@@ -389,7 +389,7 @@ void RscError::Error( const ERRTYPE& rError, RscTop * pClass,
 void RscError::FatalError( const ERRTYPE& rError, const RscId &aId,
                            const char * pMessage )
 {
-    if( ERR_USAGE != rError ){
+    if( ERR_USAGE != rError && ERR_UNKNOWNSW != rError ){
         nErrors++;
         ErrorFormat( rError, nullptr, aId );
         WriteError( rError, pMessage );
@@ -398,7 +398,8 @@ void RscError::FatalError( const ERRTYPE& rError, const RscId &aId,
     else
         WriteError( rError, pMessage );
 
-    exit( 1 );
+    if( ERR_UNKNOWNSW != rError )
+ 	   exit( 1 );
 }
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
-- 
2.5.0

