diff --git a/sc/source/core/data/global.cxx b/sc/source/core/data/global.cxx
index c657bd81defc..d5678c089f41 100644
--- a/sc/source/core/data/global.cxx
+++ b/sc/source/core/data/global.cxx
@@ -71,6 +71,8 @@
 #include <editutil.hxx>
 #include <docsh.hxx>
 
+#include <config_features.h>
+
 tools::SvRef<ScDocShell>  ScGlobal::xDrawClipDocShellRef;
 std::unique_ptr<SvxSearchItem> ScGlobal::xSearchItem;
 std::unique_ptr<ScAutoFormat> ScGlobal::xAutoFormat;
@@ -472,7 +474,11 @@ void ScGlobal::InitPPT()
 {
     OutputDevice* pDev = Application::GetDefaultDevice();
 
-    if (comphelper::LibreOfficeKit::isActive() || OpenGLWrapper::isVCLOpenGLEnabled())
+    if (comphelper::LibreOfficeKit::isActive()
+#if defined HAVE_FEATURE_OPENGL && defined HAVE_FEATURE_UI
+|| OpenGLWrapper::isVCLOpenGLEnabled()
+#endif
+)
     {
         // LOK: the below limited precision is not enough for RowColumnHeader.
         // OpenGL: limited precision breaks AA of text in charts.
