diff --git a/bin/distro-install-sdk b/bin/distro-install-sdk
index a92c3b5d650c..76eadfd9d020 100755
--- a/bin/distro-install-sdk
+++ b/bin/distro-install-sdk
@@ -71,8 +71,8 @@ if test -d $DESTDIR$INSTALLDIR/sdk ; then
         -e "s,@OO_SDK_MAKE_HOME@,/usr/bin," \
         -e "s,@OO_SDK_ZIP_HOME@,/usr/bin," \
         -e "s,@OO_SDK_CPP_HOME@,/usr/bin," \
-        -e "s,@OO_SDK_SED_HOME@,/usr/bin," \
-        -e "s,@OO_SDK_CAT_HOME@,/usr/bin," \
+        -e "s,@OO_SDK_SED_HOME@,/bin," \
+        -e "s,@OO_SDK_CAT_HOME@,/bin," \
         -e "s,@OO_SDK_JAVA_HOME@,$JAVA_HOME," \
         -e "s,@OO_SDK_OUTPUT_DIR@,\$HOME," \
         -e "s,@SDK_AUTO_DEPLOYMENT@,NO," \
