diff -u b/sd/source/ui/dlg/RemoteDialog.cxx b/sd/source/ui/dlg/RemoteDialog.cxx
--- b/sd/source/ui/dlg/RemoteDialog.cxx
+++ b/sd/source/ui/dlg/RemoteDialog.cxx
@@ -25,9 +25,12 @@
     mClientBox(         this, NULL, SdResId( LB_SERVERS ) ),
     mPreviouslyDiscoverable()
 {
-#ifdef ENABLE_SDREMOTE
+    (void) mPreviouslyDiscoverable; // avoid warnings about unused member
+
     FreeResource();
 
+#ifdef ENABLE_SDREMOTE
+
 #ifdef ENABLE_SDREMOTE_BLUETOOTH
     mPreviouslyDiscoverable = RemoteServer::isBluetoothDiscoverable();
     if ( !mPreviouslyDiscoverable )
@@ -43,13 +46,11 @@
     {
         mClientBox.addEntry( *aIt );
     }
+#endif
 
     mButtonConnect.SetClickHdl( LINK( this, RemoteDialog, HandleConnectButton ) );
     SetCloseHdl( LINK( this, RemoteDialog, CloseHdl ) );
     mButtonCancel.SetClickHdl( LINK( this, RemoteDialog, CloseHdl ) );
-#else
-    (void) mPreviouslyDiscoverable; // avoid warnings about unused member
-#endif
 }
 
 RemoteDialog::~RemoteDialog()
@@ -84,8 +85,8 @@
     {
         RemoteServer::setBluetoothDiscoverable( false );
     }
-    Close();
 #endif
+    Close();
     return 0;
 }
 
