Index: libreoffice-4.1.4~rc2/shell/source/unix/misc/gnome-open-url.sh
===================================================================
--- libreoffice-4.1.4~rc2.orig/shell/source/unix/misc/gnome-open-url.sh	2013-12-16 14:35:12.270934941 +0100
+++ libreoffice-4.1.4~rc2/shell/source/unix/misc/gnome-open-url.sh	2013-12-16 14:35:12.262934941 +0100
@@ -1,6 +1,6 @@
 #!/bin/sh
 
 # use xdg-open or gnome-open if available
-xdg-open "$1" 2>/dev/null || gnome-open "$1" 2>/dev/null || "$0.bin" $1
+xdg-open "$1" 2>/dev/null || gnome-open "$1" 2>/dev/null || sensible-browser "$1" || "$0.bin" $1
 
 exit 0
Index: libreoffice-4.1.4~rc2/shell/source/unix/misc/kde-open-url.sh
===================================================================
--- libreoffice-4.1.4~rc2.orig/shell/source/unix/misc/kde-open-url.sh	2013-12-16 14:35:12.270934941 +0100
+++ libreoffice-4.1.4~rc2/shell/source/unix/misc/kde-open-url.sh	2013-12-16 14:35:12.262934941 +0100
@@ -19,9 +19,13 @@
 
 # special handling for mailto: uris
 if echo $1 | grep '^mailto:' > /dev/null; then
-  kmailservice "$1" &
+  if which kde-open; then
+    kde-open "$1" &
+  else
+    mailservice "$1" &
+  fi
 else
-  kfmclient openURL "$1" &
+  sensible-browser "$1" &
 fi
 
 exit 0
