diff --git a/shell/source/unix/misc/kde-open-url.sh b/shell/source/unix/misc/kde-open-url.sh
index b0eac27..27bf6de 100755
--- a/shell/source/unix/misc/kde-open-url.sh
+++ b/shell/source/unix/misc/kde-open-url.sh
@@ -19,7 +19,11 @@
 
 # special handling for mailto: uris
 if echo $1 | grep '^mailto:' > /dev/null; then
-  kmailservice "$1" &
+  if which kde-open; then
+    kde-open "$1" &
+  else
+    kmailservice "$1" &
+  fi
 else
-  kfmclient openURL "$1" &
+  sensible-browser "$1" &
 fi
