--- libreoffice-3.5.0/scp2/source/ooo/file_library_ooo.scp-old	2011-12-20 05:11:44.000000000 +0100
+++ libreoffice-3.5.0/scp2/source/ooo/file_library_ooo.scp	2011-12-20 05:12:07.000000000 +0100
@@ -446,6 +446,13 @@
 
 STD_LIB_FILE( gid_File_Lib_Evoab , evoab)
 
+File gid_File_Share_Registry_Evoab_Xcd
+    TXT_FILE_BODY;
+    Styles = (PACKED);
+    Dir = gid_Brand_Dir_Share_Registry;
+    Name = "evoab.xcd";
+End
+
 #endif  // #ifdef ENABLE_EVOAB2
 
 #endif  // #ifdef UNX
--- libreoffice-3.5.0/scp2/source/gnome/module_gnome.scp-old	2011-12-20 05:15:16.000000000 +0100
+++ libreoffice-3.5.0/scp2/source/gnome/module_gnome.scp	2011-12-20 05:15:30.000000000 +0100
@@ -43,6 +43,7 @@ Module gid_Module_Optional_Gnome
     Files = (
 #ifdef ENABLE_EVOAB2
       gid_File_Lib_Evoab,
+      gid_File_Share_Registry_Evoab_Xcd,
 #endif
 #ifdef ENABLE_GTK
       gid_File_Lib_Fps_Gnome,
diff --git a/postprocess/CustomTarget_registry.mk b/postprocess/CustomTarget_registry.mk
index 181c782..33f4640 100644
--- a/postprocess/CustomTarget_registry.mk
+++ b/postprocess/CustomTarget_registry.mk
@@ -25,6 +25,7 @@ postprocess_XCDS := \
 	ctl.xcd \
 	ctlseqcheck.xcd \
 	draw.xcd \
+	evoab.xcd \
 	graphicfilter.xcd \
 	impress.xcd \
 	korea.xcd \
@@ -327,8 +328,8 @@ ifeq ($(DISABLE_NEON),$(false))
 postprocess_FILES_main += $(postprocess_MOD)/org/openoffice/ucb/Configuration-neon.xcu
 endif
 ifeq ($(ENABLE_EVOAB2),TRUE)
-postprocess_FILES_main += $(postprocess_MOD)/org/openoffice/Office/DataAccess/Drivers-evoab2.xcu
-postprocess_FILES_main += $(postprocess_MOD)/org/openoffice/Office/DataAccess-evoab2.xcu
+postprocess_FILES_evoab += $(postprocess_MOD)/org/openoffice/Office/DataAccess/Drivers-evoab2.xcu
+postprocess_FILES_evoab += $(postprocess_MOD)/org/openoffice/Office/DataAccess-evoab2.xcu
 postprocess_DRIVERS += evoab
 endif
 ifeq ($(SOLAR_JAVA),TRUE)
