diff --git a/Repository.mk b/Repository.mk
old mode 100644
new mode 100755
index e8412a2..1f04a6a
--- a/Repository.mk
+++ b/Repository.mk
@@ -660,7 +670,7 @@ $(eval $(call gb_Helper_register_jars_for_install,OOO,ooo, \
 	xmerge \
 ))
 
-$(eval $(call gb_Helper_register_jars_for_install,OOO,extensions_rhino, \
+$(eval $(call gb_Helper_register_jars_for_install,OOO,reportbuilder, \
 	reportbuilder \
 	reportbuilderwizard \
 ))
