diff --git a/bin/distro-install-desktop-integration b/bin/distro-install-desktop-integration
index 1d2484b..3b6e502 100755
--- a/bin/distro-install-desktop-integration
+++ b/bin/distro-install-desktop-integration
@@ -141,7 +141,6 @@ cd -
 # desktop files will be added by the corresponding add_wrapper command
 if test -f $DESTDIR/gid_Module_Root_Brand ; then
     for dir in $PREFIXDIR/share/application-registry \
-               $PREFIXDIR/share/mimelnk/application \
                $PREFIXDIR/share/mime/packages \
                $PREFIXDIR/share/mime-info \
                $PREFIXDIR/share/icons ; do
diff --git a/sysui/CustomTarget_share.mk b/sysui/CustomTarget_share.mk
index aeb94c0..a99e1d8 100644
--- a/sysui/CustomTarget_share.mk
+++ b/sysui/CustomTarget_share.mk
@@ -127,7 +126,6 @@ $(eval $(call gb_CustomTarget_register_targets,sysui/share,\
 		$(product)/openoffice.keys \
 		$(product)/openoffice.sh \
 		$(product)/create_tree.sh \
-		$(product)/mimelnklist \
 		$(product)/launcherlist) \
 ))
 
@@ -147,16 +145,6 @@ $(share_WORKDIR)/%/openoffice.keys:  \
 		--ext "keys" --key "description" $(share_WORKDIR)/documents.ulf
 	cat $(MIMEKEYS) > $@
 
-$(share_WORKDIR)/%/mimelnklist: $(MIMEDESKTOPS) $(share_SRCDIR)/share/brand.pl \
-	$(share_TRANSLATE) $(share_WORKDIR)/documents.ulf
-	mkdir -p $(dir $@)
-	$(call gb_Output_announce,$(subst $(WORKDIR)/,,$@),$(true),PRL,1)
-	$(PERL) $(share_SRCDIR)/share/brand.pl -p $* -u $(UNIXFILENAME.$*) \
-		--iconprefix $(UNIXFILENAME.$*)- $^ $(share_WORKDIR)/$*
-	$(PERL) $(share_TRANSLATE) -p $* -d $(share_WORKDIR)/$* \
-		--ext "desktop" --key "Comment" $(share_WORKDIR)/documents.ulf
-	echo "$(MIMEDESKTOPS)" > $@
-
 $(share_WORKDIR)/%/openoffice.mime: $(share_SRCDIR)/mimetypes/openoffice.mime
 	mkdir -p $(dir $@)
 	$(call gb_Output_announce,$(subst $(WORKDIR)/,,$@),$(true),CAT,1)
@@ -167,7 +155,7 @@ $(share_WORKDIR)/%/openoffice.sh: $(share_SRCDIR)/share/openoffice.sh
 	$(call gb_Output_announce,$(subst $(WORKDIR)/,,$@),$(true),CAT,1)
 	cat $< | tr -d "\015" | sed -e "s/%PREFIX/$(UNIXFILENAME.$*)/g" > $@
 
-$(share_WORKDIR)/%/create_tree.sh: $(share_SRCDIR)/share/create_tree.sh $(share_WORKDIR)/%/mimelnklist \
+$(share_WORKDIR)/%/create_tree.sh: $(share_SRCDIR)/share/create_tree.sh \
 	$(share_WORKDIR)/%/openoffice.org.xml $(share_WORKDIR)/%/openoffice.applications $(share_WORKDIR)/%/openoffice.mime \
 	$(share_WORKDIR)/%/openoffice.keys $(share_WORKDIR)/%/launcherlist $(if $(INTROSPECTION_SCANNER),$(call gb_Library_get_target,libreofficekitgtk))
 	mkdir -p $(dir $@)
diff --git a/sysui/desktop/freedesktop/freedesktop-menus.spec b/sysui/desktop/freedesktop/freedesktop-menus.spec
index 4a0f7e7..c3e154f 100755
--- a/sysui/desktop/freedesktop/freedesktop-menus.spec
+++ b/sysui/desktop/freedesktop/freedesktop-menus.spec
@@ -66,7 +66,6 @@ cd $RPM_BUILD_ROOT
 rm -rf usr/share/application-registry
 rm -rf usr/share/applications.flag
 rm -rf usr/share/mime-info
-rm -rf usr/share/mimelnk
 rm -rf usr/share/applnk-redhat
 #find usr/share/icons -name '*.png' -exec chmod g+w {} \;
 
diff --git a/sysui/desktop/share/brand.pl b/sysui/desktop/share/brand.pl
index ae2911b..2be723b3 100644
--- a/sysui/desktop/share/brand.pl
+++ b/sysui/desktop/share/brand.pl
@@ -22,7 +22,7 @@ eval 'exec perl -wS $0 ${1+"$@"}'
 
 #
 # This tool is used to re-write and substitute variables
-# into Unix .desktop, mimelnk, .keys files etc.
+# into Unix .desktop, .keys files etc.
 #
 
 $destdir = pop @ARGV;
diff --git a/sysui/desktop/share/create_tree.sh b/sysui/desktop/share/create_tree.sh
index 19a5ce3..3f24b5f 100755
--- a/sysui/desktop/share/create_tree.sh
+++ b/sysui/desktop/share/create_tree.sh
@@ -29,11 +29,6 @@ if [ "${KDEMAINDIR}" ]; then
   done
   unset targetdir destfile
 
-  mkdir -p "${DESTDIR}/${KDEMAINDIR}/share/mimelnk/application"
-  for i in `cat mimelnklist`; do
-    cp "${i}" "${DESTDIR}/${KDEMAINDIR}/share/mimelnk/application/${PREFIX}-`basename ${i}`"
-  done
-  chmod 0644 "${DESTDIR}/${KDEMAINDIR}/share/mimelnk/application/"*
 fi
 
 if [ "${GNOMEDIR}" ]; then
