From 8840fd3226f0f23ce41273e2abebb9a7ac1eb08b Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Mon, 13 Apr 2015 15:31:41 +0200
Subject: tdf#82695 allow to build with system collada2gltf

Change-Id: I1179e20fd5fbdecd198633d5440621ed08a0465b

diff --git a/RepositoryExternal.mk b/RepositoryExternal.mk
index e3dc039..a0a4d91 100644
--- a/RepositoryExternal.mk
+++ b/RepositoryExternal.mk
@@ -3477,6 +3477,20 @@ endef
 
 endif # SYSTEM_OPENCOLLADA
 
+ifeq ($(SYSTEM_COLLADA2GLTF),TRUE)
+
+define gb_LinkTarget__use_collada2gltf
+$(call gb_LinkTarget_set_include,$(1),\
+	$$(INCLUDE) \
+	$(COLLADA2GLTF_CFLAGS) \
+)
+
+$(call gb_LinkTarget_add_libs,$(1),$(COLLADA2GLTF_LIBS))
+
+endef
+
+else # !SYSTEM_COLLADA2GLTF
+
 define gb_LinkTarget__use_collada2gltf
 $(call gb_LinkTarget_set_include,$(1),\
 	-I$(call gb_UnpackedTarball_get_dir,collada2gltf) \
@@ -3493,6 +3507,8 @@ $(call gb_LinkTarget_use_static_libraries,$(1),\
 )
 endef
 
+endif # SYSTEM_COLLADA2GLTF
+
 endif # ENABLE_COLLADA
 
 endif # ENABLE_GLTF
diff --git a/config_host.mk.in b/config_host.mk.in
index a75560c..f1d2226 100644
--- a/config_host.mk.in
+++ b/config_host.mk.in
@@ -59,6 +59,8 @@ export CLUCENE_CFLAGS=$(gb_SPACE)@CLUCENE_CFLAGS@
 export CLUCENE_LIBS=$(gb_SPACE)@CLUCENE_LIBS@
 export CMIS_CFLAGS=$(gb_SPACE)@CMIS_CFLAGS@
 export CMIS_LIBS=$(gb_SPACE)@CMIS_LIBS@
+export COLLADA2GLTF_CFLAGS=$(gb_SPACE)@COLLADA2GLTF_CFLAGS@
+export COLLADA2GLTF_LIBS=$(gb_SPACE)@COLLADA2GLTF_LIBS@
 export COM=@COM@
 export COMMONS_CODEC_JAR=@COMMONS_CODEC_JAR@
 export COMMONS_HTTPCLIENT_JAR=@COMMONS_HTTPCLIENT_JAR@
@@ -512,6 +514,7 @@ export SYSTEM_BOOST=@SYSTEM_BOOST@
 export SYSTEM_BSH=@SYSTEM_BSH@
 export SYSTEM_CAIRO=@SYSTEM_CAIRO@
 export SYSTEM_CLUCENE=@SYSTEM_CLUCENE@
+export SYSTEM_COLLADA2GLTF=@SYSTEM_COLLADA2GLTF@
 export SYSTEM_CPPUNIT=@SYSTEM_CPPUNIT@
 export SYSTEM_CURL=@SYSTEM_CURL@
 export SYSTEM_DICTS=@SYSTEM_DICTS@
diff --git a/configure.ac b/configure.ac
index ec50613..4801597 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1629,6 +1629,11 @@ AC_ARG_WITH(system-opencollada,
         [Use openCOLLADA already on system.]),,
     [with_system_opencollada=no])
 
+AC_ARG_WITH(system-collada2gltf,
+    AS_HELP_STRING([--with-system-collada2gltf],
+        [Use collada2gltf already on system.]),,
+    [with_system_collada2gltf=no])
+
 AC_ARG_WITH(system-openldap,
     AS_HELP_STRING([--with-system-openldap],
         [Use the OpenLDAP LDAP SDK already on system.]),,
@@ -10342,6 +10347,7 @@ if test "$enable_collada" != "no" -a "$ENABLE_GLTF" = "TRUE"; then
     AC_MSG_RESULT([yes])
     ENABLE_COLLADA=TRUE
     AC_DEFINE(HAVE_FEATURE_COLLADA,1)
+
     AC_MSG_CHECKING([which OPENCOLLADA to use])
     if test "$with_system_opencollada" = "yes"; then
         AC_MSG_RESULT([external])
@@ -10369,7 +10375,35 @@ if test "$enable_collada" != "no" -a "$ENABLE_GLTF" = "TRUE"; then
         AC_MSG_RESULT([internal])
         BUILD_TYPE="$BUILD_TYPE OPENCOLLADA"
     fi
-    BUILD_TYPE="$BUILD_TYPE COLLADA2GLTF"
+
+    AC_MSG_CHECKING([which collada2gltf to use])
+    if test "$with_system_collada2gltf" = "yes"; then
+        if test "$with_system_opencollada" = "no"; then
+            AC_MSG_ERROR([the combination of system collada2gltf and internal openCOLLADA is not allowed])
+        fi
+        AC_MSG_RESULT([external])
+        SYSTEM_COLLADA2GLTF=TRUE
+        AS_IF([test -n "$COLLADA2GLTF_CFLAGS"],[],[AC_MSG_ERROR([export COLLADA2GLTF_CFLAGS])])
+        AS_IF([test -n "$COLLADA2GLTF_LIBS"],[],[AC_MSG_ERROR([export COLLADA2GLTF_LIBS])])
+        AC_LANG_PUSH([C++])
+        save_CXXFLAGS=$CXXFLAGS
+        save_CPPFLAGS=$CPPFLAGS
+        CXXFLAGS="$CXXFLAGS $COLLADA2GLTF_CFLAGS $OPENCOLLADA_CFLAGS $CXXFLAGS_CXX11"
+        CPPFLAGS="$CPPFLAGS $COLLADA2GLTF_CFLAGS $OPENCOLLADA_CFLAGS $CXXFLAGS_CXX11"
+        AC_CHECK_HEADERS(
+                GLTF.h \
+                encodingHelpers.h,
+            [],
+            [AC_MSG_ERROR([collada2gltf headers not found. Install collada2gltf])],
+            [])
+        CXXFLAGS=$save_CXXFLAGS
+        CPPFLAGS=$save_CPPFLAGS
+        AC_LANG_POP([C++])
+        COLLADA2GLTF_CFLAGS=$(printf '%s' "$COLLADA2GLTF_CFLAGS" | sed -e "s/-I/${ISYSTEM?}/g")
+    else
+        AC_MSG_RESULT([internal])
+        BUILD_TYPE="$BUILD_TYPE COLLADA2GLTF"
+    fi
 else
     AC_MSG_RESULT([no])
 fi
@@ -10378,6 +10412,10 @@ AC_SUBST([OPENCOLLADA_CFLAGS])
 AC_SUBST([OPENCOLLADA_LIBS])
 AC_SUBST([SYSTEM_OPENCOLLADA])
 
+AC_SUBST([COLLADA2GLTF_CFLAGS])
+AC_SUBST([COLLADA2GLTF_LIBS])
+AC_SUBST([SYSTEM_COLLADA2GLTF])
+
 # pdf import?
 AC_MSG_CHECKING([whether to build the PDF import feature])
 ENABLE_PDFIMPORT=
