Author: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>

Origin: upstream
Bug: tdf#86449
Forwarded: not-needed

--- libreoffice-3.5.7.orig/writerfilter/source/rtftok/rtfdocumentimpl.cxx
+++ libreoffice-3.5.7/writerfilter/source/rtftok/rtfdocumentimpl.cxx
@@ -3605,9 +3605,12 @@ int RTFDocumentImpl::popState()
         m_aStates.top().aPicture = aState.aPicture;
     else if (bPopShapeProperties)
     {
-        m_aStates.top().aShape = aShape;
-        m_aStates.top().aPicture = aPicture;
-        m_aStates.top().aCharacterAttributes = aAttributes;
+        if(!m_aStates.empty())
+        {
+            m_aStates.top().aShape = aShape;
+            m_aStates.top().aPicture = aPicture;
+            m_aStates.top().aCharacterAttributes = aAttributes;
+        }
     }
     else if (bFaltEnd)
         m_aStates.top().aTableSprms = aSprms;
