diff --git a/sc/qa/uitest/calc_tests4/trackedChanges.py b/sc/qa/uitest/calc_tests4/trackedChanges.py
index 72bfa89cf40c..06dae85b7e66 100644
--- a/sc/qa/uitest/calc_tests4/trackedChanges.py
+++ b/sc/qa/uitest/calc_tests4/trackedChanges.py
@@ -29,7 +29,8 @@ class CalcTrackedChanges(UITestCase):
         xChangesList = xTrackDlg.getChild("calcchanges")
         self.assertEqual(1, len(xChangesList.getChildren()))
 
-        text = "Row inserted \tSheet1.1:1\t \t04/05/2020 17:01:10\t (Row 1:1 inserted)"
+        # time is stored in UTC!
+        text = "Row inserted \tSheet1.1:1\t \t04/05/2020 15:01:10\t (Row 1:1 inserted)"
         self.assertEqual(get_state_as_dict(xChangesList.getChild('0'))["Text"], text)
 
         #it would crash here
diff --git a/solenv/gbuild/UITest.mk b/solenv/gbuild/UITest.mk
index 92eaaf9faefb..73524f58426d 100644
--- a/solenv/gbuild/UITest.mk
+++ b/solenv/gbuild/UITest.mk
@@ -40,6 +40,7 @@ endif
 gb_UITest_COMMAND = $(ICECREAM_RUN) $(gb_CppunitTest_RR) $(gb_UITest_EXECUTABLE) $(SRCDIR)/uitest/test_main.py
 
 gb_TEST_ENV_VARS += LIBO_LANG=C
+gb_TEST_ENV_VARS += TZ=UTC
 
 .PHONY : $(call gb_UITest_get_clean_target,%)
 $(call gb_UITest_get_clean_target,%) :
