diff --git a/patches/dev300/apply b/patches/dev300/apply
index 4775581..cb830f8 100644
--- openoffice.org-3.3.0~beta/ooo-build-28d2a4e7af25256f2919b81d6fe03426ab2c7454/patches/dev300/apply
+++ openoffice.org-3.3.0~beta/ooo-build-28d2a4e7af25256f2919b81d6fe03426ab2c7454/patches/dev300/apply
@@ -3617,7 +3617,7 @@ slideshow-sound.diff, n#515553, rodo
 vcl-gtk-process-events-on-exit.diff, rodo
 sfx2-add-to-recently-used-after-save.diff, rodo
 
-svtools-wmf-clipperf.diff, n#535304, thorsten
+#svtools-wmf-clipperf.diff, n#535304, thorsten
 
 [ Fixes ]
 # Fix for UTF-8 encoded hyphen dictionaries (e.g. Hungarian)
