--- a/solenv/gbuild/UITest.mk-old	2020-04-27 23:05:00.540846895 +0200
+++ b/solenv/gbuild/UITest.mk	2020-04-27 23:03:55.136226712 +0200
@@ -78,7 +78,7 @@
 		PYTHONDONTWRITEBYTECODE=0 \
 		$(if $(ENABLE_WERROR),PYTHONWARNINGS=error) \
 		$(gb_TEST_ENV_VARS) \
-		$(gb_UITest_COMMAND) \
+		timeout --verbose --preserve-status -k 120 60 $(gb_UITest_COMMAND) \
 		--soffice="$(gb_UITest_SOFFICEARG)" \
 		--userdir=$(call gb_Helper_make_url,$(dir $(call gb_UITest_get_target,$*))user) \
 		--dir=$(strip $(MODULES)) \
