From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Date: Tue, 20 Oct 2015 23:10:36 +0200
Subject: [PATCH] on unity, default to human if possible

---
 vcl/source/app/IconThemeSelector.cxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/vcl/source/app/IconThemeSelector.cxx b/vcl/source/app/IconThemeSelector.cxx
--- a/vcl/source/app/IconThemeSelector.cxx
+++ b/vcl/source/app/IconThemeSelector.cxx
@@ -65,6 +65,9 @@ IconThemeSelector::GetIconThemeForDesktopEnvironment(const OUString& desktopEnvi
     else if ( desktopEnvironment.equalsIgnoreAsciiCase("MacOSX") ) {
         r = "breeze";
     }
+    else if ( desktopEnvironment.equalsIgnoreAsciiCase("unity") ) {
+        r = "human";
+    }
     else {
         r = FALLBACK_ICON_THEME_ID;
     }
-- 
2.1.4

