diff -urN cpputools.old/Executable_regcomp.mk cpputools/Executable_regcomp.mk
--- a/cpputools.old/Executable_regcomp.mk	2013-04-30 14:54:23.000000000 +0200
+++ b/cpputools/Executable_regcomp.mk	2013-05-17 21:41:13.665548985 +0200
@@ -34,6 +34,7 @@
     cppu \
     cppuhelper \
     sal \
+    xmlreader \
 	$(gb_UWINAPI) \
 ))
 
diff -urN cpputools.old/Executable_regsingleton.mk cpputools/Executable_regsingleton.mk
--- a/cpputools.old/Executable_regsingleton.mk	2013-04-30 14:54:23.000000000 +0200
+++ b/cpputools/Executable_regsingleton.mk	2013-05-17 21:41:22.977549255 +0200
@@ -34,6 +34,7 @@
     cppu \
     cppuhelper \
     sal \
+    xmlreader \
 	$(gb_UWINAPI) \
 ))
 
diff -urN cpputools.old/Executable_sp2bv.mk cpputools/Executable_sp2bv.mk
--- a/cpputools.old/Executable_sp2bv.mk	2013-04-30 14:54:23.000000000 +0200
+++ b/cpputools/Executable_sp2bv.mk	2013-05-17 21:41:33.373549555 +0200
@@ -34,6 +34,7 @@
     cppu \
     cppuhelper \
     sal \
+    xmlreader \
 ))
 
 $(eval $(call gb_Executable_add_exception_objects,sp2bv,\
--- a/unodevtools.old/Executable_skeletonmaker.mk	2013-05-18 11:31:49.705153823 +0200
+++ b/unodevtools/Executable_skeletonmaker.mk	2013-05-18 11:33:07.993156082 +0200
@@ -41,6 +41,7 @@
     salhelper \
     cppu \
     cppuhelper \
+    xmlreader \
 ))
 
 $(eval $(call gb_Executable_use_static_libraries,uno-skeletonmaker,\
