diff --git a/desktop/scripts/unopkg.sh b/desktop/scripts/unopkg.sh
index 779a137..afabef2 100644
--- a/desktop/scripts/unopkg.sh
+++ b/desktop/scripts/unopkg.sh
@@ -66,7 +66,7 @@ do
 done
 
 #make sure shared extensions will be readable by users
-[ $SHARED = true ] && umask 0022
+[ "$SHARED" = true ] && umask 0022
 
 # extend the ld_library_path for java: javaldx checks the sofficerc for us
 if [ -x "$sd_prog/../ure-link/bin/javaldx" ] ; then
