diff --git a/odk/CustomTarget_unowinreg.mk b/odk/CustomTarget_unowinreg.mk
index 3a46ddfd6860..a847e1d6bb38 100644
--- a/odk/CustomTarget_unowinreg.mk
+++ b/odk/CustomTarget_unowinreg.mk
@@ -19,7 +19,7 @@ $(call gb_CustomTarget_get_workdir,odk/unowinreg)/unowinreg.dll : \
 		$(if $(filter FREEBSD,$(OS)),-I$(JAVA_HOME)/include/freebsd \
 		-I$(JAVA_HOME)/include/bsd -I$(JAVA_HOME)/include/linux,\
 		$(if $(filter NETBSD,$(OS)),-I$(JAVA_HOME)/include/netbsd))) \
-		-shared -o $@ $< \
+		-shared -static-libgcc -o $@ $< \
 		-Wl,--kill-at -lkernel32 -ladvapi32 && \
 	$(MINGWSTRIP) $@
 
