From 17c1f97f55183719b05d91580c937079454b22d3 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Fri, 16 Dec 2011 16:51:33 +0000
Subject: actually add gid_File_Lib_VisioImport to gid_Module_Optional_Grfflt

---
diff --git a/scp2/source/graphicfilter/module_graphicfilter.scp b/scp2/source/graphicfilter/module_graphicfilter.scp
index 5b6dffb..984b247 100644
--- a/scp2/source/graphicfilter/module_graphicfilter.scp
+++ b/scp2/source/graphicfilter/module_graphicfilter.scp
@@ -34,5 +34,5 @@ Module gid_Module_Optional_Grfflt
     ParentID = gid_Module_Optional;
     Default = YES;
     Files = (gid_File_Lib_Svg, gid_File_Lib_Flash, gid_File_Lib_WPGImport,
-    		 gid_File_Share_Registry_Graphicfilter_Xcd);
+    		 gid_File_Lib_VisioImport, gid_File_Share_Registry_Graphicfilter_Xcd);
 End
--
cgit v0.9.0.2-2-gbebe
