--- a/configure.ac-old	2013-02-14 22:36:09.220868022 +0100
+++ b/configure.ac	2013-02-14 22:36:40.228867104 +0100
@@ -7254,7 +7254,7 @@
             PYTHON_LIBS="-framework Python"
         else
             # This causes an error if no python command is found
-            AM_PATH_PYTHON([3.3])
+            AM_PATH_PYTHON([2.6])
         fi
     elif test $enable_python = auto; then
         # We should not come here in a MSVC build, I think?
@@ -7264,7 +7264,7 @@
 
         # This allows a lack of system Python 3.3 with no error, we check below for a system Python of
         # earlier version.
-        AM_PATH_PYTHON([3.3],, [:])
+        AM_PATH_PYTHON([2.6],, [:])
 
         if test "$PYTHON" = :; then
             enable_python=internal
