Index: libreoffice-4.1.4~rc2/configure.ac
===================================================================
--- libreoffice-4.1.4~rc2.orig/configure.ac	2013-12-16 14:35:27.386934137 +0100
+++ libreoffice-4.1.4~rc2/configure.ac	2013-12-16 14:35:27.382934137 +0100
@@ -7445,7 +7445,7 @@
         # Unset variables set by the above AM_PATH_PYTHON so that
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
-        AM_PATH_PYTHON([3.3],, [:])
+        AM_PATH_PYTHON([2.6],, [:])
         if test "$PYTHON" = :; then
             AC_MSG_RESULT([internal])
             enable_python=internal
@@ -7499,7 +7499,7 @@
         # we actually do check anew.
         unset PYTHON am_cv_pathless_PYTHON ac_cv_path_PYTHON am_cv_python_version am_cv_python_platform am_cv_python_pythondir am_cv_python_pyexecdir
         # This causes an error if no python command is found
-        AM_PATH_PYTHON([3.3])
+        AM_PATH_PYTHON([2.6])
         python_include=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('INCLUDEPY'));"`
         python_version=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('VERSION'));"`
         python_libs=`$PYTHON -c "import distutils.sysconfig; print(distutils.sysconfig.get_config_var('LIBS'));"`
Index: libreoffice-4.1.4~rc2/unotest/source/python/org/libreoffice/unotest.py
===================================================================
--- libreoffice-4.1.4~rc2.orig/unotest/source/python/org/libreoffice/unotest.py	2013-12-16 14:35:27.386934137 +0100
+++ libreoffice-4.1.4~rc2/unotest/source/python/org/libreoffice/unotest.py	2013-12-16 14:35:27.382934137 +0100
@@ -7,6 +7,7 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 #
 
+from __future__ import print_function
 import subprocess
 import time
 import uuid
