--- libreoffice-5.2.1.orig/vcl/unx/generic/plugadapt/salplug.cxx
+++ libreoffice-5.2.1/vcl/unx/generic/plugadapt/salplug.cxx
@@ -187,6 +187,11 @@ static SalInstance* autodetect_plugin()
         "gtk3", "gtk", "gen", nullptr
     };
 
+    static const char* const pXFCEFallbackList[] =
+    {
+        "gtk", "gen", nullptr
+    };
+
     static const char* const pHeadlessFallbackList[] =
     {
         "svp", nullptr
@@ -201,10 +206,11 @@ static SalInstance* autodetect_plugin()
         pList = pHeadlessFallbackList;
     else if ( desktop == DESKTOP_GNOME ||
               desktop == DESKTOP_UNITY ||
-              desktop == DESKTOP_XFCE  ||
               desktop == DESKTOP_KDE3 ||
               desktop == DESKTOP_MATE )
         pList = pStandardFallbackList;
+    else if ( desktop == DESKTOP_XFCE)
+	pList = pXFCEFallbackList;
     else if( desktop == DESKTOP_TDE )
         pList = pTDEFallbackList;
     else if( desktop == DESKTOP_KDE4 || desktop == DESKTOP_KDE5 )
