diff --git a/sdext/source/pdfimport/wrapper/wrapper.cxx b/sdext/source/pdfimport/wrapper/wrapper.cxx
index a31e78cb1100..efc544d9b332 100644
--- a/sdext/source/pdfimport/wrapper/wrapper.cxx
+++ b/sdext/source/pdfimport/wrapper/wrapper.cxx
@@ -1046,11 +1046,11 @@ bool xpdf_ImportFromFile( const OUString&                             rURL,
     }
 
     // Determine xpdfimport executable URL:
-    OUString converterURL("$BRAND_BASE_DIR/" LIBO_BIN_FOLDER "/xpdfimport");
+    OUString converterURL("/usr/lib/libreoffice/program/xpdfimport");
     rtl::Bootstrap::expandMacros(converterURL); //TODO: detect failure
 
     // Determine pathname of xpdfimport_err.pdf:
-    OUString errPathname("$BRAND_BASE_DIR/" LIBO_SHARE_FOLDER "/xpdfimport/xpdfimport_err.pdf");
+    OUString errPathname("/usr/lib/libreoffice/share/xpdfimport/xpdfimport_err.pdf");
     rtl::Bootstrap::expandMacros(errPathname); //TODO: detect failure
     if (osl::FileBase::getSystemPathFromFileURL(errPathname, errPathname)
         != osl::FileBase::E_None)
